<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>classify-consensus-vsearch: VSEARCH-based consensus taxonomy classifier &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/language_data.js"></script>
    <script src="../../../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="index" title="Index" href="../../../../genindex/" />
    <link rel="search" title="Search" href="../../../../search/" />
    <link rel="next" title="classify-sklearn: Pre-fitted sklearn-based taxonomy classifier" href="../classify-sklearn/" />
    <link rel="prev" title="classify-consensus-blast: BLAST+ consensus taxonomy classifier" href="../classify-consensus-blast/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="classify-consensus-vsearch-vsearch-based-consensus-taxonomy-classifier">
<h1>classify-consensus-vsearch: VSEARCH-based consensus taxonomy classifier<a class="headerlink" href="#classify-consensus-vsearch-vsearch-based-consensus-taxonomy-classifier" title="Permalink to this headline">¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><ul class="simple" id="bibtex-bibliography-plugins/available/feature-classifier/classify-consensus-vsearch-0">
<li><p>Torbjørn Rognes, Tomáš Flouri, Ben Nichols, Christopher Quince, and Frédéric Mahé. Vsearch: a versatile open source tool for metagenomics. <em>PeerJ</em>, 4:e2584, 2016. <a class="reference external" href="https://doi.org/10.7717/peerj.2584">doi:10.7717/peerj.2584</a>.</p></li>
</ul>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime feature-classifier classify-consensus-vsearch
           [OPTIONS]

  Assign taxonomy to query sequences using VSEARCH. Performs VSEARCH global
  alignment between query and reference_reads, then assigns consensus
  taxonomy to each query sequence from among maxaccepts top hits,
  min_consensus of which share that taxonomic assignment. Unlike classify-
  consensus-blast, this method searches the entire reference database before
  choosing the top N hits, not the first N hits.

Inputs:
  --i-query ARTIFACT FeatureData[Sequence]
                          Sequences to classify taxonomically.      [required]
  --i-reference-reads ARTIFACT FeatureData[Sequence]
                          reference sequences.                      [required]
  --i-reference-taxonomy ARTIFACT FeatureData[Taxonomy]
                          reference taxonomy labels.                [required]
Parameters:
  --p-maxaccepts VALUE Int % Range(1, None) | Str % Choices('all')
                          Maximum number of hits to keep for each query. Set
                          to "all" to keep all hits > perc-identity
                          similarity. Note that if strand=both, maxaccepts
                          will keep N hits for each direction (if searches in
                          the opposite direction yield results that exceed the
                          minimum perc-identity). In those cases use maxhits
                          to control the total number of hits returned. This
                          option works in pair with maxrejects. The search
                          process sorts target sequences by decreasing number
                          of k-mers they have in common with the query
                          sequence, using that information as a proxy for
                          sequence similarity. After pairwise alignments, if
                          the first target sequence passes the acceptation
                          criteria, it is accepted as best hit and the search
                          process stops for that query. If maxaccepts is set
                          to a higher value, more hits are accepted. If
                          maxaccepts and maxrejects are both set to "all", the
                          complete database is searched.         [default: 10]
  --p-perc-identity PROPORTION Range(0.0, 1.0, inclusive_end=True)
                          Reject match if percent identity to query is lower.
                                                                [default: 0.8]
  --p-query-cov PROPORTION Range(0.0, 1.0, inclusive_end=True)
                          Reject match if query alignment coverage per
                          high-scoring pair is lower.           [default: 0.8]
  --p-strand TEXT Choices('both', 'plus')
                          Align against reference sequences in forward
                          ("plus") or both directions ("both").
                                                             [default: 'both']
  --p-min-consensus NUMBER Range(0.5, 1.0, inclusive_start=False,
    inclusive_end=True)   Minimum fraction of assignments must match top hit
                          to be accepted as consensus assignment.
                                                               [default: 0.51]
  --p-unassignable-label TEXT
                                                       [default: 'Unassigned']
  --p-search-exact / --p-no-search-exact
                          Search for exact full-length matches to the query
                          sequences. Only 100% exact matches are reported and
                          this command is much faster than the default. If
                          True, the perc-identity and query-cov settings are
                          ignored. Note: query and reference reads must be
                          trimmed to the exact same DNA locus (e.g., primer
                          site) because only exact matches will be reported.
                                                              [default: False]
  --p-top-hits-only / --p-no-top-hits-only
                          Only the top hits between the query and reference
                          sequence sets are reported. For each query, the top
                          hit is the one presenting the highest percentage of
                          identity. Multiple equally scored top hits will be
                          used for consensus taxonomic assignment if
                          maxaccepts is greater than 1.       [default: False]
  --p-maxhits VALUE Int % Range(1, None) | Str % Choices('all')
                          Maximum number of hits to show once the search is
                          terminated.                         [default: 'all']
  --p-maxrejects VALUE Int % Range(1, None) | Str % Choices('all')
                          Maximum number of non-matching target sequences to
                          consider before stopping the search. This option
                          works in pair with maxaccepts (see maxaccepts
                          description for details).           [default: 'all']
  --p-output-no-hits / --p-no-output-no-hits
                          Report both matching and non-matching queries.
                          WARNING: always use the default setting for this
                          option unless if you know what you are doing! If you
                          set this option to False, your sequences and feature
                          table will need to be filtered to exclude
                          unclassified sequences, otherwise you may run into
                          errors downstream from missing feature IDs.
                                                               [default: True]
  --p-weak-id PROPORTION Range(0.0, 1.0, inclusive_end=True)
                          Show hits with percentage of identity of at least
                          N, without terminating the search. A normal search
                          stops as soon as enough hits are found (as defined
                          by maxaccepts, maxrejects, and perc-identity). As
                          weak-id reports weak hits that are not deduced from
                          maxaccepts, high perc-identity values can be used,
                          hence preserving both speed and sensitivity.
                          Logically, weak-id must be smaller than the value
                          indicated by perc-identity, otherwise this option
                          will be ignored.                      [default: 0.0]
  --p-threads INTEGER     Number of threads to use for job parallelization.
    Range(1, None)                                                [default: 1]
Outputs:
  --o-classification ARTIFACT FeatureData[Taxonomy]
                          The resulting taxonomy classifications.   [required]
Miscellaneous:
  --output-dir PATH       Output unspecified results to a directory
  --verbose / --quiet     Display verbose output to stdout and/or stderr
                          during execution of this action. Or silence output
                          if execution is successful (silence is golden).
  --citations             Show citations and exit.
  --help                  Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>from qiime2.plugins.feature_classifier.methods import classify_consensus_vsearch
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>VSEARCH-based consensus taxonomy classifier

Assign taxonomy to query sequences using VSEARCH. Performs VSEARCH global
alignment between query and reference_reads, then assigns consensus
taxonomy to each query sequence from among maxaccepts top hits,
min_consensus of which share that taxonomic assignment. Unlike classify-
consensus-blast, this method searches the entire reference database before
choosing the top N hits, not the first N hits.

Parameters
----------
query : FeatureData[Sequence]
    Sequences to classify taxonomically.
reference_reads : FeatureData[Sequence]
    reference sequences.
reference_taxonomy : FeatureData[Taxonomy]
    reference taxonomy labels.
maxaccepts : Int % Range(1, None) | Str % Choices('all'), optional
    Maximum number of hits to keep for each query. Set to "all" to keep all
    hits > perc_identity similarity. Note that if strand=both, maxaccepts
    will keep N hits for each direction (if searches in the opposite
    direction yield results that exceed the minimum perc_identity). In
    those cases use maxhits to control the total number of hits returned.
    This option works in pair with maxrejects. The search process sorts
    target sequences by decreasing number of k-mers they have in common
    with the query sequence, using that information as a proxy for sequence
    similarity. After pairwise alignments, if the first target sequence
    passes the acceptation criteria, it is accepted as best hit and the
    search process stops for that query. If maxaccepts is set to a higher
    value, more hits are accepted. If maxaccepts and maxrejects are both
    set to "all", the complete database is searched.
perc_identity : Float % Range(0.0, 1.0, inclusive_end=True), optional
    Reject match if percent identity to query is lower.
query_cov : Float % Range(0.0, 1.0, inclusive_end=True), optional
    Reject match if query alignment coverage per high-scoring pair is
    lower.
strand : Str % Choices('both', 'plus'), optional
    Align against reference sequences in forward ("plus") or both
    directions ("both").
min_consensus : Float % Range(0.5, 1.0, inclusive_start=False, inclusive_end=True), optional
    Minimum fraction of assignments must match top hit to be accepted as
    consensus assignment.
unassignable_label : Str, optional
search_exact : Bool, optional
    Search for exact full-length matches to the query sequences. Only 100%
    exact matches are reported and this command is much faster than the
    default. If True, the perc_identity and query_cov settings are ignored.
    Note: query and reference reads must be trimmed to the exact same DNA
    locus (e.g., primer site) because only exact matches will be reported.
top_hits_only : Bool, optional
    Only the top hits between the query and reference sequence sets are
    reported. For each query, the top hit is the one presenting the highest
    percentage of identity. Multiple equally scored top hits will be used
    for consensus taxonomic assignment if maxaccepts is greater than 1.
maxhits : Int % Range(1, None) | Str % Choices('all'), optional
    Maximum number of hits to show once the search is terminated.
maxrejects : Int % Range(1, None) | Str % Choices('all'), optional
    Maximum number of non-matching target sequences to consider before
    stopping the search. This option works in pair with maxaccepts (see
    maxaccepts description for details).
output_no_hits : Bool, optional
    Report both matching and non-matching queries. WARNING: always use the
    default setting for this option unless if you know what you are doing!
    If you set this option to False, your sequences and feature table will
    need to be filtered to exclude unclassified sequences, otherwise you
    may run into errors downstream from missing feature IDs.
weak_id : Float % Range(0.0, 1.0, inclusive_end=True), optional
    Show hits with percentage of identity of at least N, without
    terminating the search. A normal search stops as soon as enough hits
    are found (as defined by maxaccepts, maxrejects, and perc_identity). As
    weak_id reports weak hits that are not deduced from maxaccepts, high
    perc_identity values can be used, hence preserving both speed and
    sensitivity. Logically, weak_id must be smaller than the value
    indicated by perc_identity, otherwise this option will be ignored.
threads : Int % Range(1, None), optional
    Number of threads to use for job parallelization.

Returns
-------
classification : FeatureData[Taxonomy]
    The resulting taxonomy classifications.</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>