<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>“Atacama soil microbiome” tutorial &#8212; QIIME 2 2017.6.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2017.6.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2017.6.0 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Importing data" href="../importing/" />
    <link rel="prev" title="Fecal microbiota transplant (FMT) study: an exercise" href="../fmt/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="atacama-soil-microbiome-tutorial">
<h1>&#8220;Atacama soil microbiome&#8221; tutorial<a class="headerlink" href="#atacama-soil-microbiome-tutorial" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a> and completed the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial</span></a>.</p>
</div>
<p>This tutorial is designed to serve two purposes. First, it illustrates the initial processing steps of paired-end read analysis, up to the point where the analysis steps are identical to single-end read analysis. This includes the importing, demultiplexing, and denoising steps, and results in a feature table and the associated feature sequences. Second, this is intended to be a self-guided exercise that could be run after <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a> to gain more experience with QIIME 2. For this exercise, we provide some questions that can be used to guide your analysis, but do not provide commands that will allow you to address each. Instead, you should apply the commands that you learned in <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a>.</p>
<p>In this tutorial you&#8217;ll use QIIME 2 to perform an analysis of soil samples from the Atacama Desert in northern Chile. The Atacama Desert is one of the most arid locations on Earth, with some areas receiving less than a millimeter of rain per decade. Despite this extreme aridity, there are microbes living in the soil. The soil microbiomes profiled in this study follow two east-west transects, <em>Baquedano</em> and <em>Yungay</em>, across which average soil relative humidity is positively correlated with elevation (higher elevations are less arid and thus have higher average soil relative humidity). Along these transects, pits were dug at each site and soil samples were collected from three depths in each pit.</p>
<div class="section" id="obtain-the-data">
<h2>Obtain the data<a class="headerlink" href="#obtain-the-data" title="Permalink to this headline">¶</a></h2>
<p>Start by creating a directory to work in.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir qiime2-atacama-tutorial
<span class="nb">cd</span> qiime2-atacama-tutorial
</pre></div>
</div>
<p>Before starting the analysis, explore the sample metadata to familiarize yourself with the samples used in this study. The <a class="reference external" href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/sample_metadata">sample metadata</a> is available as a Google Sheet. This <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code> file is used throughout the rest of the tutorial.</p>
<div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/sample_metadata.tsv">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/sample_metadata.tsv
          </a>
        </p>
        <p>
          <strong>Save as:</strong> sample-metadata.tsv
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "sample-metadata.tsv" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/sample_metadata.tsv"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/sample_metadata.tsv" > "sample-metadata.tsv"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>Next, you&#8217;ll download the multiplexed reads. You will download three <code class="docutils literal"><span class="pre">fastq.gz</span></code> files, corresponding to the forward, reverse, and barcode (i.e., index) reads. These files contain a subset of the reads in the full data set generated for this study, which allows for the following commands to be run relatively quickly. If you are only planning to run through the commands presented here to get experience with the first steps of paired-end read analysis, you can use the 1% subsample data set so that the commands will run quickly. If you&#8217;re planning to work through the questions presented at the end of this document to gain more experience with QIIME analysis and data interpretation, you should use the 10% subsample data set so that the analysis results will be supported by more sequence data.</p>
<div class="section" id="subsample-data">
<h3>1% subsample data<a class="headerlink" href="#subsample-data" title="Permalink to this headline">¶</a></h3>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir emp-paired-end-sequences
</pre></div>
</div>
<div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/forward.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/forward.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/forward.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/forward.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/forward.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/forward.fastq.gz" > "emp-paired-end-sequences/forward.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-2" aria-controls="browser-2" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-2" aria-controls="wget-2" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-2" aria-controls="messages-2" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-2">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/reverse.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/reverse.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/reverse.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-2">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/reverse.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/reverse.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-2">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/reverse.fastq.gz" > "emp-paired-end-sequences/reverse.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-3" aria-controls="browser-3" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-3" aria-controls="wget-3" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-3" aria-controls="messages-3" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-3">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/barcodes.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/barcodes.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/barcodes.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-3">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/barcodes.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-3">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/1p/barcodes.fastq.gz" > "emp-paired-end-sequences/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div></div>
<div class="section" id="id1">
<h3>10% subsample data<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir emp-paired-end-sequences
</pre></div>
</div>
<div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-4" aria-controls="browser-4" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-4" aria-controls="wget-4" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-4" aria-controls="messages-4" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-4">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/forward.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/forward.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/forward.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-4">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/forward.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/forward.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-4">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/forward.fastq.gz" > "emp-paired-end-sequences/forward.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-5" aria-controls="browser-5" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-5" aria-controls="wget-5" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-5" aria-controls="messages-5" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-5">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/reverse.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/reverse.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/reverse.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-5">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/reverse.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/reverse.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-5">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/reverse.fastq.gz" > "emp-paired-end-sequences/reverse.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="download panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-6" aria-controls="browser-6" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-6" aria-controls="wget-6" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-6" aria-controls="messages-6" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-6">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/barcodes.fastq.gz">
            https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/barcodes.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-paired-end-sequences/barcodes.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-6">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-paired-end-sequences/barcodes.fastq.gz" "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-6">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.6/tutorials/atacama-soils/10p/barcodes.fastq.gz" > "emp-paired-end-sequences/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div></div>
</div>
<div class="section" id="paired-end-read-analysis-commands">
<h2>Paired-end read analysis commands<a class="headerlink" href="#paired-end-read-analysis-commands" title="Permalink to this headline">¶</a></h2>
<p>To analyze these data, the sequences that you just downloaded must first be imported into an artifact of type <code class="docutils literal"><span class="pre">EMPPairedEndSequences</span></code>.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
   --type EMPPairedEndSequences <span class="se">\</span>
   --input-path emp-paired-end-sequences <span class="se">\</span>
   --output-path emp-paired-end-sequences.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">emp-paired-end-sequences.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Femp-paired-end-sequences.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/emp-paired-end-sequences.qza">download</a></li>
</ul>
</div>
<p>You next can demultiplex the sequence reads. This requires the sample metadata file, and you must indicate which column in that file contains the per-sample barcodes. In this case, that column name is <code class="docutils literal"><span class="pre">BarcodeSequence</span></code>. In this data set, the barcode reads are the reverse complement of those included in the sample metadata file, so we additionally include the <code class="docutils literal"><span class="pre">--p-rev-comp-mapping-barcodes</span></code> parameter. After demultiplexing, we can generate and view a summary of how many sequences were obtained per sample.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime demux emp-paired <span class="se">\</span>
  --m-barcodes-file sample-metadata.tsv <span class="se">\</span>
  --m-barcodes-category BarcodeSequence <span class="se">\</span>
  --i-seqs emp-paired-end-sequences.qza <span class="se">\</span>
  --o-per-sample-sequences demux <span class="se">\</span>
  --p-rev-comp-mapping-barcodes

qiime demux summarize <span class="se">\</span>
  --i-data demux.qza <span class="se">\</span>
  --o-visualization demux.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">demux.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Fdemux.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/demux.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Fdemux.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/demux.qzv">download</a></li>
</ul>
</div>
<p>After demultiplexing reads, we&#8217;ll look at the sequence quality based on ten randomly selected samples, and then denoise the data. When you view the quality plots, note that in contrast to the corresponding plots in <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a>, there are now two plots per sample. The plot on the left presents the quality scores for the forward reads, and the plot on the right presents the quality scores for the reverse reads. We&#8217;ll use these plots to determine what trimming parameters we want to use for denoising with DADA2, and then denoise the reads using <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise-paired</span></code>.</p>
<p>In this example we have 150-base forward and reverse reads. Since we need the reads to be long enough to overlap when joining paired ends, the first thirteen bases of the forward and reverse reads are being trimmed, but no trimming is being applied to the ends of the sequences to avoid reducing the read length by too much. In this example, the same values are being provided for <code class="docutils literal"><span class="pre">--p-trim-left-f</span></code> and <code class="docutils literal"><span class="pre">--p-trim-left-r</span></code> and for <code class="docutils literal"><span class="pre">--p-trunc-len-f</span></code> and <code class="docutils literal"><span class="pre">--p-trunc-len-r</span></code>, but that is not a requirement.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 denoise-paired <span class="se">\</span>
  --i-demultiplexed-seqs demux.qza <span class="se">\</span>
  --o-table table <span class="se">\</span>
  --o-representative-sequences rep-seqs <span class="se">\</span>
  --p-trim-left-f <span class="m">13</span> <span class="se">\</span>
  --p-trim-left-r <span class="m">13</span> <span class="se">\</span>
  --p-trunc-len-f <span class="m">150</span> <span class="se">\</span>
  --p-trunc-len-r <span class="m">150</span>
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/rep-seqs.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Ftable.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/table.qza">download</a></li>
</ul>
</div>
<p>At this stage, you will have artifacts containing the feature table and corresponding feature sequences. You can generated summaries of those as follows.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-table summarize <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-visualization table.qzv <span class="se">\</span>
  --m-sample-metadata-file sample-metadata.tsv

qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data rep-seqs.qza <span class="se">\</span>
  --o-visualization rep-seqs.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Frep-seqs.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/rep-seqs.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.6%2Fdata%2Ftutorials%2Fatacama-soils%2Ftable.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.6/data/tutorials/atacama-soils/table.qzv">download</a></li>
</ul>
</div>
<p>From this point, analysis of paired-end read data progresses in the same way as analysis of single-end read data. You can therefore continue your analyses of these data following the steps that you ran in <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a>.</p>
</div>
<div class="section" id="questions-to-guide-data-analysis">
<h2>Questions to guide data analysis<a class="headerlink" href="#questions-to-guide-data-analysis" title="Permalink to this headline">¶</a></h2>
<p>Use the following questions to guide your further analyses of these data data.</p>
<ol class="arabic simple">
<li>What value would you choose to pass for <code class="docutils literal"><span class="pre">--p-sampling-depth</span></code>? How many samples will be excluded from your analysis based on this choice? Approximately how many total sequences will you be analyzing in the <code class="docutils literal"><span class="pre">core-metrics</span></code> command?</li>
<li>What sample metadata or combinations of sample metadata are most strongly associated with the differences in microbial composition of the samples? Are these associations stronger with unweighted UniFrac or with Bray-Curtis? Based on what you know about these metrics, what does that difference suggest? For exploring associations between continuous metadata and sample composition, the commands <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">beta-correlation</span></code> and <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">bioenv</span></code> will be useful. These were not covered in the Moving Pictures tutorial, but you can learn about them by running them with the <code class="docutils literal"><span class="pre">--help</span></code> parameter.</li>
<li>What do you conclude about the associations between continuous sample metadata and the richness and evenness of these samples? For exploring associations between continuous metadata and richness or evenness, the command <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">alpha-correlation</span></code> will be useful. This was not covered in the Moving Pictures tutorial, but you can learn about it by running it with the <code class="docutils literal"><span class="pre">--help</span></code> parameter.</li>
<li>What discrete sample metadata categories are most strongly associated with the differences in microbial community richness or evenness? Are these differences statistically significant?</li>
<li>In taxonomic composition bar plots, sort the samples by their average soil relative humidity, and visualize them at the phylum level. What are the dominant phyla in these samples? Which phyla increase and which decrease with increasing average soil relative humidity?</li>
<li>What phyla differ in abundance across vegetated and unvegetated sites?</li>
</ol>
</div>
<div class="section" id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this headline">¶</a></h2>
<p>The data used in this tutorial is presented in: <em>Arid Soil Microbiome: Significant Impacts of Increasing Aridity. Neilson, Califf, Cardona, Copeland, van Treuren, Josephson, Knight, Gilbert, Quade, Caporaso, and Maier. mSystems (under review).</em></p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../upgrade/">Upgrading QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">&#8220;Atacama soil microbiome&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2017, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>