<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>beta-phylogenetic: Beta diversity (phylogenetic) &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/language_data.js"></script>
    <script src="../../../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="index" title="Index" href="../../../../genindex/" />
    <link rel="search" title="Search" href="../../../../search/" />
    <link rel="next" title="filter-distance-matrix: Filter samples from a distance matrix." href="../filter-distance-matrix/" />
    <link rel="prev" title="beta: Beta diversity" href="../beta/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="beta-phylogenetic-beta-diversity-phylogenetic">
<h1>beta-phylogenetic: Beta diversity (phylogenetic)<a class="headerlink" href="#beta-phylogenetic-beta-diversity-phylogenetic" title="Permalink to this headline">¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><ul class="simple" id="bibtex-bibliography-plugins/available/diversity/beta-phylogenetic-0">
<li><p>Qin Chang, Yihui Luan, and Fengzhu Sun. Variance adjusted weighted unifrac: a powerful beta diversity measure for comparing communities based on phylogeny. <em>BMC bioinformatics</em>, 12(1):118, 2011. <a class="reference external" href="https://doi.org/https://doi.org/10.1186/1471-2105-12-118">doi:https://doi.org/10.1186/1471-2105-12-118</a>.</p></li>
<li><p>Jun Chen, Kyle Bittinger, Emily S Charlson, Christian Hoffmann, James Lewis, Gary D Wu, Ronald G Collman, Frederic D Bushman, and Hongzhe Li. Associating microbiome composition with environmental covariates using generalized unifrac distances. <em>Bioinformatics</em>, 28(16):2106–2113, 2012. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts342">doi:10.1093/bioinformatics/bts342</a>.</p></li>
<li><p>Catherine Lozupone and Rob Knight. Unifrac: a new phylogenetic method for comparing microbial communities. <em>Applied and environmental microbiology</em>, 71(12):8228–8235, 2005. <a class="reference external" href="https://doi.org/10.1128/AEM.71.12.8228-8235.2005">doi:10.1128/AEM.71.12.8228-8235.2005</a>.</p></li>
<li><p>Catherine A Lozupone, Micah Hamady, Scott T Kelley, and Rob Knight. Quantitative and qualitative $β$ diversity measures lead to different insights into factors that structure microbial communities. <em>Applied and environmental microbiology</em>, 73(5):1576–1585, 2007. <a class="reference external" href="https://doi.org/10.1128/AEM.01996-06">doi:10.1128/AEM.01996-06</a>.</p></li>
<li><p>Daniel McDonald, Yoshiki Vázquez-Baeza, David Koslicki, Jason McClelland, Nicolai Reeve, Zhenjiang Xu, Antonio Gonzalez, and Rob Knight. Striped unifrac: enabling microbiome analysis at unprecedented scale. <em>Nature Methods</em>, 2018. <a class="reference external" href="https://doi.org/10.1038/s41592-018-0187-8">doi:10.1038/s41592-018-0187-8</a>.</p></li>
</ul>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime diversity beta-phylogenetic [OPTIONS]

  Computes a user-specified phylogenetic beta diversity metric for all pairs
  of samples in a feature table.

Inputs:
  --i-table ARTIFACT FeatureTable[Frequency]
                          The feature table containing the samples over which
                          beta diversity should be computed.        [required]
  --i-phylogeny ARTIFACT  Phylogenetic tree containing tip identifiers that
    Phylogeny[Rooted]     correspond to the feature identifiers in the table.
                          This tree can contain tip ids that are not present
                          in the table, but all feature ids in the table must
                          be present in this tree.                  [required]
Parameters:
  --p-metric TEXT Choices('weighted_unifrac', 'unweighted_unifrac',
    'generalized_unifrac', 'weighted_normalized_unifrac')
                          The beta diversity metric to be computed. [required]
  --p-n-jobs INTEGER      The number of workers to use.           [default: 1]
  --p-variance-adjusted / --p-no-variance-adjusted
                          Perform variance adjustment based on Chang et al.
                          BMC Bioinformatics 2011. Weights distances based on
                          the proportion of the relative abundance represented
                          between the samples at a given node under
                          evaluation.                         [default: False]
  --p-alpha PROPORTION Range(0, 1, inclusive_end=True)
                          This parameter is only used when the choice of
                          metric is generalized_unifrac. The value of alpha
                          controls importance of sample proportions. 1.0 is
                          weighted normalized UniFrac. 0.0 is close to
                          unweighted UniFrac, but only if the sample
                          proportions are dichotomized.             [optional]
  --p-bypass-tips / --p-no-bypass-tips
                          In a bifurcating tree, the tips make up about 50%
                          of the nodes in a tree. By ignoring them,
                          specificity can be traded for reduced compute time.
                          This has the effect of collapsing the phylogeny, and
                          is analogous (in concept) to moving from 99% to 97%
                          OTUs                                [default: False]
Outputs:
  --o-distance-matrix ARTIFACT DistanceMatrix % Properties('phylogenetic')
                          The resulting distance matrix.            [required]
Miscellaneous:
  --output-dir PATH       Output unspecified results to a directory
  --verbose / --quiet     Display verbose output to stdout and/or stderr
                          during execution of this action. Or silence output
                          if execution is successful (silence is golden).
  --citations             Show citations and exit.
  --help                  Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>from qiime2.plugins.diversity.methods import beta_phylogenetic
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Beta diversity (phylogenetic)

Computes a user-specified phylogenetic beta diversity metric for all pairs
of samples in a feature table.

Parameters
----------
table : FeatureTable[Frequency]
    The feature table containing the samples over which beta diversity
    should be computed.
phylogeny : Phylogeny[Rooted]
    Phylogenetic tree containing tip identifiers that correspond to the
    feature identifiers in the table. This tree can contain tip ids that
    are not present in the table, but all feature ids in the table must be
    present in this tree.
metric : Str % Choices('weighted_normalized_unifrac', 'weighted_unifrac', 'generalized_unifrac', 'unweighted_unifrac')
    The beta diversity metric to be computed.
n_jobs : Int, optional
    The number of workers to use.
variance_adjusted : Bool, optional
    Perform variance adjustment based on Chang et al. BMC Bioinformatics
    2011. Weights distances based on the proportion of the relative
    abundance represented between the samples at a given node under
    evaluation.
alpha : Float % Range(0, 1, inclusive_end=True), optional
    This parameter is only used when the choice of metric is
    generalized_unifrac. The value of alpha controls importance of sample
    proportions. 1.0 is weighted normalized UniFrac. 0.0 is close to
    unweighted UniFrac, but only if the sample proportions are
    dichotomized.
bypass_tips : Bool, optional
    In a bifurcating tree, the tips make up about 50% of the nodes in a
    tree. By ignoring them, specificity can be traded for reduced compute
    time. This has the effect of collapsing the phylogeny, and is analogous
    (in concept) to moving from 99% to 97% OTUs

Returns
-------
distance_matrix : DistanceMatrix % Properties('phylogenetic')
    The resulting distance matrix.</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>