<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>extract-reads: Extract reads from reference &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/language_data.js"></script>
    <script src="../../../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="index" title="Index" href="../../../../genindex/" />
    <link rel="search" title="Search" href="../../../../search/" />
    <link rel="next" title="fit-classifier-naive-bayes: Train the naive_bayes classifier" href="../fit-classifier-naive-bayes/" />
    <link rel="prev" title="classify-sklearn: Pre-fitted sklearn-based taxonomy classifier" href="../classify-sklearn/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="extract-reads-extract-reads-from-reference">
<h1>extract-reads: Extract reads from reference<a class="headerlink" href="#extract-reads-extract-reads-from-reference" title="Permalink to this headline">Â¶</a></h1>
<div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime feature-classifier extract-reads [OPTIONS]

  Extract sequencing-like reads from a reference database.

Inputs:
  --i-sequences ARTIFACT FeatureData[Sequence]
                                                                    [required]
Parameters:
  --p-f-primer TEXT       forward primer sequence                   [required]
  --p-r-primer TEXT       reverse primer sequence                   [required]
  --p-trunc-len INTEGER   read is cut to trunc-len if trunc-len is positive.
                          Applied before trim-left.               [default: 0]
  --p-trim-left INTEGER   trim-left nucleotides are removed from the 5' end
                          if trim-left is positive. Applied after trunc-len.
                                                                  [default: 0]
  --p-identity NUMBER     minimum combined primer match identity threshold.
                                                                [default: 0.8]
  --p-min-length INTEGER  Minimum amplicon length. Shorter amplicons are
    Range(0, None)        discarded. Applied after trimming and truncation, so
                          be aware that trimming may impact sequence
                          retention. Set to zero to disable min length
                          filtering.                             [default: 50]
  --p-max-length INTEGER  Maximum amplicon length. Longer amplicons are
    Range(0, None)        discarded. Applied before trimming and truncation,
                          so plan accordingly. Set to zero (default) to
                          disable max length filtering.           [default: 0]
  --p-n-jobs INTEGER      Number of seperate processes to run.
    Range(1, None)                                                [default: 1]
  --p-batch-size VALUE Int % Range(1, None) | Str % Choices('auto')
                          Number of sequences to process in a batch. The
                          `auto` option is calculated from the number of
                          sequences and number of jobs specified.
                                                             [default: 'auto']
  --p-read-orientation TEXT Choices('both', 'forward', 'reverse')
                          Orientation of primers relative to the sequences:
                          "forward" searches for primer hits in the forward
                          direction, "reverse" searches the
                          reverse-complement, and "both" searches both
                          directions.                        [default: 'both']
Outputs:
  --o-reads ARTIFACT FeatureData[Sequence]
                                                                    [required]
Miscellaneous:
  --output-dir PATH       Output unspecified results to a directory
  --verbose / --quiet     Display verbose output to stdout and/or stderr
                          during execution of this action. Or silence output
                          if execution is successful (silence is golden).
  --citations             Show citations and exit.
  --help                  Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>from qiime2.plugins.feature_classifier.methods import extract_reads
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Extract reads from reference

Extract sequencing-like reads from a reference database.

Parameters
----------
sequences : FeatureData[Sequence]
f_primer : Str
    forward primer sequence
r_primer : Str
    reverse primer sequence
trunc_len : Int, optional
    read is cut to trunc_len if trunc_len is positive. Applied before
    trim_left.
trim_left : Int, optional
    trim_left nucleotides are removed from the 5' end if trim_left is
    positive. Applied after trunc_len.
identity : Float, optional
    minimum combined primer match identity threshold.
min_length : Int % Range(0, None), optional
    Minimum amplicon length. Shorter amplicons are discarded. Applied after
    trimming and truncation, so be aware that trimming may impact sequence
    retention. Set to zero to disable min length filtering.
max_length : Int % Range(0, None), optional
    Maximum amplicon length. Longer amplicons are discarded. Applied before
    trimming and truncation, so plan accordingly. Set to zero (default) to
    disable max length filtering.
n_jobs : Int % Range(1, None), optional
    Number of seperate processes to run.
batch_size : Int % Range(1, None) | Str % Choices('auto'), optional
    Number of sequences to process in a batch. The `auto` option is
    calculated from the number of sequences and number of jobs specified.
read_orientation : Str % Choices('both', 'forward', 'reverse'), optional
    Orientation of primers relative to the sequences: "forward" searches
    for primer hits in the forward direction, "reverse" searches the
    reverse-complement, and "both" searches both directions.

Returns
-------
reads : FeatureData[Sequence]</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>