<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Semantic types &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../about/" />
    <link rel="index" title="Index" href="../genindex/" />
    <link rel="search" title="Search" href="../search/" />
    <link rel="next" title="Community" href="../community/" />
    <link rel="prev" title="Developing a QIIME 2 plugin" href="../plugins/developing/" />
  <script src="../_static/bootstrap.min.js"></script>
  <script src="../_static/clipboard.min.js"></script>
  <link rel="icon" href="../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="semantic-types">
<h1>Semantic types<a class="headerlink" href="#semantic-types" title="Permalink to this headline">¶</a></h1>
<p>All QIIME 2 artifacts have defined semantic types. This document is a placeholder for full documentation of how to use these types and how to define new semantic types in QIIME 2 plugins.</p>
<div class="section" id="why-define-semantic-types">
<h2>Why define semantic types?<a class="headerlink" href="#why-define-semantic-types" title="Permalink to this headline">¶</a></h2>
<p>Defining semantic types allows us to ensure that the data that is passed to an <em>action</em> is meaningful for the operation that will be performed. For example, a <code class="docutils literal notranslate"><span class="pre">biom.Table</span></code> object could contain presence/absence data (i.e., a 1 to indicate that an OTU was observed at least one time in a given sample, and a 0 to indicate than an OTU was not observed at least one time in a given sample). However, if that were provided to a function computing a quantitative diversity metric where OTU abundances are included in the calculation (e.g., weighted UniFrac), the function would complete successfully, but the result would not be meaningful.</p>
<p>QIIME 2 defines semantic types to help users avoid using functionality incorrectly, and to allow the system to reason about what QIIME 2 artifacts can be used with which <em>methods</em> and <em>visualizers</em>.</p>
</div>
<div class="section" id="common-semantic-types">
<h2>Common semantic types<a class="headerlink" href="#common-semantic-types" title="Permalink to this headline">¶</a></h2>
<p>Unless otherwise noted the following semantic types are defined by, and importable from, the <a class="reference external" href="https://github.com/qiime2/q2-types">q2-types</a> plugin. It is also possible to define semantic types in any plugin, so the available semantic types are not limited to those defined in <a class="reference external" href="https://github.com/qiime2/q2-types">q2-types</a>. Instructions will be added soon for how to accomplish this. In the meantime, you can refer to the <a class="reference external" href="https://github.com/qiime2/q2-dummy-types">q2-dummy-types</a> repository for annotated examples.</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureTable[Frequency]</span></code>: A feature table (e.g., samples by OTUs) where each value indicates the frequency of an OTU in the corresponding sample expressed as raw counts.</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureTable[RelativeFrequency]</span></code>: A feature table (e.g., samples by OTUs) where each value indicates the relative abundance of an OTU in the corresponding sample such that the values for each sample will sum to 1.0.</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureTable[PresenceAbsence]</span></code>: a feature table (e.g., samples by OTUs) where each value indicates whether an OTU is present or absent in the corresponding sample.</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureTable[Composition]</span></code>: A feature table (e.g., samples by OTUs) where each value indicates the frequency of an OTU in the corresponding sample, and all frequencies are greater than zero.</p>
<p><code class="docutils literal notranslate"><span class="pre">Phylogeny[Rooted]</span></code>: A rooted phylogenetic tree.</p>
<p><code class="docutils literal notranslate"><span class="pre">Phylogeny[Unrooted]</span></code>: An unrooted phylogenetic tree.</p>
<p><code class="docutils literal notranslate"><span class="pre">DistanceMatrix</span></code>: A distance matrix.</p>
<p><code class="docutils literal notranslate"><span class="pre">PCoAResults</span></code>: The results of running principal coordinate analysis (PCoA).</p>
<p><code class="docutils literal notranslate"><span class="pre">SampleData[AlphaDiversity]</span></code>: Alpha diversity values, each associated with a single sample identifier.</p>
<p><code class="docutils literal notranslate"><span class="pre">SampleData[SequencesWithQuality]</span></code>: Sequences with quality scores, where each set of sequences is associated with a sample identifier (i.e. demultiplexed sequences).</p>
<p><code class="docutils literal notranslate"><span class="pre">SampleData[PairedEndSequencesWithQuality]</span></code>: Paired-end sequences (forward and reverse) with quality scores, where each set of paired-end sequences is associated with a sample identifier (i.e. demultiplexed paired-end sequences).</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureData[Taxonomy]</span></code>: Taxonomic information associated with a feature identifier.</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureData[Sequence]</span></code>: A single unaligned sequence associated with a feature identifier (e.g. a representative sequence).</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureData[AlignedSequence]</span></code>: A single aligned sequence associated with a feature identifier, where the alignment is with respect to the aligned sequences associated with all of the other feature identifiers (i.e., when more than one feature identifier is present this represents a multiple sequence alignment).</p>
<p><code class="docutils literal notranslate"><span class="pre">FeatureData[PairedEndSequence]</span></code>: Paired-end sequences (forward and reverse) associated with a feature identifier.</p>
<p><code class="docutils literal notranslate"><span class="pre">EMPSingleEndSequences</span></code>: Multiplexed single-end read sequence data that has been generated according to the <a class="reference external" href="http://www.earthmicrobiome.org/protocols-and-standards/">Earth Microbiome Project sequencing protocol</a>. See <a class="reference internal" href="../tutorials/importing/"><span class="doc">Importing data</span></a> for details about importing this data type. This type was previously called <code class="docutils literal notranslate"><span class="pre">RawSequences</span></code>. The <code class="docutils literal notranslate"><span class="pre">RawSequences</span></code> type name was deprecated, but is now an alias for <code class="docutils literal notranslate"><span class="pre">EMPSingleEndSequences</span></code>. You should always use the name <code class="docutils literal notranslate"><span class="pre">EMPSingleEndSequences</span></code> in favor of <code class="docutils literal notranslate"><span class="pre">RawSequences</span></code>. This type is currently defined in the <a class="reference external" href="https://github.com/qiime2/q2-demux">q2-demux</a> plugin, but in the future will likely be moved to <code class="docutils literal notranslate"><span class="pre">q2-types</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">EMPPairedEndSequences</span></code>: Multiplexed paired-end read sequence data that has been generated according to the <a class="reference external" href="http://www.earthmicrobiome.org/protocols-and-standards/">Earth Microbiome Project sequencing protocol</a>. See <a class="reference internal" href="../tutorials/importing/"><span class="doc">Importing data</span></a> for details about importing this data type. This type is currently defined in the <a class="reference external" href="https://github.com/qiime2/q2-demux">q2-demux</a> plugin, but in the future will likely be moved to <code class="docutils literal notranslate"><span class="pre">q2-types</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">TaxonomicClassifier</span></code>: A trained classifier that can be used for performing taxonomic assignment on sequences. This type is defined in the <a class="reference external" href="https://github.com/qiime2/q2-feature-classifier">q2-feature-classifier</a> plugin.</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../plugins/">Plugins</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Semantic types</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#why-define-semantic-types">Why define semantic types?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#common-semantic-types">Common semantic types</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>