<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>“88 soils” tutorial &#8212; QIIME 2 2.0.6 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2.0.6',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2.0.6 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Importing data" href="../import/" />
    <link rel="prev" title="Fecal microbiota transplant (FMT) study: an exercise" href="../fmt/" />
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/logo400.png" alt="QIIME 2"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="soils-tutorial">
<h1>&#8220;88 soils&#8221; tutorial<a class="headerlink" href="#soils-tutorial" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a>.</p>
</div>
<p>This tutorial is designed to be a self-guided exercise that could be run after <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a> and <a class="reference internal" href="../fmt/"><span class="doc">the fecal microbiota transplant tutorial</span></a> to gain more experience with QIIME 2.</p>
<p>In this tutorial you&#8217;ll use QIIME 2 to perform an analysis of soil samples from around the globe. A study based on these samples was originally published in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19502440">Lauber et al. (2009)</a>. In that study, these samples had the hypervariable region 2 (V2) of the 16S rRNA sequenced on a Roche 454 instrument. The data used in this tutorial are based on a re-sequencing of those same samples on an Illumina HiSeq as part of the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a>. In this instance, the hypervariable region 4 (V4) of the 16S rRNA was sequenced.</p>
<div class="section" id="prepare-for-the-analysis">
<h2>Prepare for the analysis<a class="headerlink" href="#prepare-for-the-analysis" title="Permalink to this headline">¶</a></h2>
<p>You should first look through the sample metadata to familiarize yourself with the samples used in this study. The <a class="reference external" href="https://docs.google.com/spreadsheets/d/1CTOCiyKWLlZiTFkmHjJcTkhHW0OkpEniNoSpscuZapk/edit?usp=sharing">sample metadata</a> is available as a Google Spreadsheet. You should download this file as tab-separated text by selecting <code class="docutils literal"><span class="pre">File</span></code> &gt; <code class="docutils literal"><span class="pre">Download</span> <span class="pre">as</span></code> &gt; <code class="docutils literal"><span class="pre">Tab-separated</span> <span class="pre">values</span></code>. Save the file as <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>.</p>
<p>You can next download the <em>demultiplexed sequences</em> that we&#8217;ll use in this analysis. In this tutorial we&#8217;ll work with a small subset (1%) of the complete sequence data so that the commands will run quickly.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>curl -sLO https://data.qiime2.org/2.0.6/tutorials/88soils/88soils-tutorial-demux-1p.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">88soils-tutorial-demux-1p.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2F88soils%2F88soils-tutorial-demux-1p.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/88soils/88soils-tutorial-demux-1p.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="sequence-processing-and-diversity-analyses">
<h2>Sequence processing and diversity analyses<a class="headerlink" href="#sequence-processing-and-diversity-analyses" title="Permalink to this headline">¶</a></h2>
<p>Use the following questions to guide your analyses of the data. (Note: if you&#8217;re new to QIIME 2, you should work through the <a class="reference internal" href="../moving-pictures/"><span class="doc">the moving pictures tutorial</span></a> and then <a class="reference internal" href="../fmt/"><span class="doc">the fecal microbiota transplant tutorial</span></a>, which will guide you through answering similar questions.)</p>
<ol class="arabic simple">
<li>What value would you choose to pass for the <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code>? How many samples will be excluded from your analysis based on this choice? Approximately how many total sequences will you be analyzing in the <code class="docutils literal"><span class="pre">core-metrics</span></code> command?</li>
<li>What sample metadata or combinations of sample metadata are most strongly associated with the differences in microbial composition of the samples? Are these associations stronger with unweighted UniFrac or with Bray-Curtis? Based on what you know about these metrics, what does that difference suggest?</li>
<li>What do you conclude about the associations between continuous sample metadata and the richness and evenness of these samples? How does this compare to the results presented in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19502440">Lauber et al. (2009)</a>? (Hint: Our findings here differ from what was present in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19502440">Lauber et al. (2009)</a>. Start thinking about why that might be.)</li>
<li>Are the associations between biomes and differences in microbial composition statistically significant? How about pH groups? What biomes appear to be most different from each other? What pH groups appear to be most different from each other?</li>
<li>What discrete sample metadata categories are most strongly associated with the differences in microbial community richness or evenness? Are these differences statistically significant?</li>
<li>What differences do you observe between the unweighted UniFrac and Bray-Curtis PCoA plots?</li>
<li>In taxonomic composition bar plots, sort the samples by their pH, and visualize them at the phylum level. What are the dominant phyla in these samples? Which phyla increase and which decrease with increasing pH?</li>
<li>Compare the taxonomic composition bar plots of these samples with those in Figure 2 of <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19502440">Lauber et al. (2009)</a>. Are the changes you noted in response to the last question consistent with what you see in this plot? There is one major difference between the plots in Figure 2 of <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19502440">Lauber et al. (2009)</a> and those generated here. What is it? (Hint: After spending some time to answer that question, take a look at <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/22267877">Bergmann et al. (2011)</a>. How do the findings presented there relate to the analysis we&#8217;re performing?)</li>
<li>One sample in this analysis is primarily dominated by the bacterial phylum <code class="docutils literal"><span class="pre">Actinobacteria</span></code>. Why do you think this is? (Hint: Refer to the summary of the <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact.)</li>
<li>What features differ in abundance across pH groups? What groups are they most and least abundant in? What are some the taxonomies of some of these features?</li>
</ol>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu dropdown-menu-right"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">&#8220;88 soils&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../import/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../import-sequence-data/">Importing sequence data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../table-filtering/">Filtering feature tables</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016, QIIME 2 Development Team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>