<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Core concepts &#8212; QIIME 2 2017.12.0 documentation</title>
    
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '2017.12.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../about/" />
    <link rel="top" title="QIIME 2 2017.12.0 documentation" href="../" />
    <link rel="next" title="Installing QIIME 2" href="../install/" />
    <link rel="prev" title="What is QIIME 2?" href="../about/" />
  <script src="../_static/bootstrap.min.js"></script>
  <link rel="icon" href="../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="core-concepts">
<h1>Core concepts<a class="headerlink" href="#core-concepts" title="Permalink to this headline">¶</a></h1>
<p>This page describes several core concepts in QIIME 2 that are important to understand before starting to use the software. The <a class="reference internal" href="../glossary/"><span class="doc">glossary</span></a> may be helpful to refer to as you read through this page and other documentation on the site.</p>
<div class="section" id="data-files-qiime-2-artifacts">
<h2>Data files: QIIME 2 artifacts<a class="headerlink" href="#data-files-qiime-2-artifacts" title="Permalink to this headline">¶</a></h2>
<p>Data produced by QIIME 2 exist as <em>QIIME 2 artifacts</em>. A QIIME 2 artifact contains data and metadata. The metadata describes things about the data, such as its type, format, and how it was generated (provenance). A QIIME 2 artifact typically has the <code class="docutils literal"><span class="pre">.qza</span></code> file extension when stored in a file.</p>
<p>Since QIIME 2 works with artifacts instead of data files (e.g. FASTA files), you must create a QIIME 2 artifact by importing data. You can import data at any step in an analysis, though typically you will start by importing raw sequence data. QIIME 2 also has tools to export data from an artifact. See the <a class="reference internal" href="../tutorials/importing/"><span class="doc">importing guide</span></a> for details.</p>
<p>By using QIIME 2 artifacts instead of simple data files, QIIME 2 can automatically track the type, format, and provenance of data for researchers. Using artifacts instead of data files enables researchers to focus on the analyses they want to perform, instead of the particular format the data needs to be in for an analysis.</p>
<p>Artifacts enable QIIME 2 to track, in addition to the data itself, the provenance of how the data came to be. With an artifact&#8217;s provenance, you can trace back to all previous analyses that were run to produce the artifact, including the input data used at each step. This automatic, integrated, and decentralized provenance tracking of data enables a researcher to archive artifacts, or for example, send an artifact to a collaborator, with the ability to understand exactly how the artifact was created. This enables replicability and reproducibility of analyses, as well as generation of diagrams and text that can be used in the methods section of a paper. Provenance also supports and encourages the proper attribution to underlying tools (e.g. FastTree to build a phylogenetic tree) used to generate the artifact.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It has been brought to our attention that the term <em>artifact</em> may be confusing, as it is frequently used by biologists to refer to a source of experimental bias. We use the term artifact here to mean an object that is made by some process (similar, for example, to an archaeological artifact). Throughout our documentation and other educational materials, we try to clarify that we are talking about <em>QIIME 2 artifacts</em> as they are defined in this section.</p>
</div>
</div>
<div class="section" id="data-files-visualizations">
<h2>Data files: visualizations<a class="headerlink" href="#data-files-visualizations" title="Permalink to this headline">¶</a></h2>
<p><em>Visualizations</em> are another type of data generated by QIIME 2. When written to disk, visualization files typically have the <code class="docutils literal"><span class="pre">.qzv</span></code> file extension. Visualizations contain similar types of metadata as QIIME 2 artifacts, including provenance information. Similar to QIIME 2 artifacts, visualizations are standalone information that can be archived or shared with collaborators.</p>
<p>In contrast to QIIME 2 artifacts, visualizations are <em>terminal outputs</em> of an analysis, and can represent, for example, a statistical results table, an interactive visualization, static images, or really any combination of visual data representations. Since visualizations are terminal outputs, they cannot be used as input to other analyses in QIIME 2.</p>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">Use <a class="reference external" href="https://view.qiime2.org">https://view.qiime2.org</a> to easily view QIIME 2 artifacts and visualizations files (generally <code class="docutils literal"><span class="pre">.qza</span></code> and <code class="docutils literal"><span class="pre">.qzv</span></code> files) without requiring a QIIME installation. This is helpful for sharing QIIME 2 data with collaborators who may not have QIIME 2 installed. <a class="reference external" href="https://view.qiime2.org">https://view.qiime2.org</a> also supports viewing data provenance.</p>
</div>
</div>
<div class="section" id="semantic-types">
<h2>Semantic types<a class="headerlink" href="#semantic-types" title="Permalink to this headline">¶</a></h2>
<p>Every artifact generated by QIIME 2 has a semantic type associated with it. Semantic types enable QIIME 2 to identify artifacts that are suitable inputs to an analysis. For example, if an analysis expects a distance matrix as input, QIIME 2 can determine which artifacts have a distance matrix semantic type and prevent incompatible artifacts from being used in the analysis (e.g. an artifact representing a phylogenetic tree).</p>
<p>Semantic types also help users avoid semantically incorrect analyses. For example, a feature table could contain presence/absence data (i.e., a 1 to indicate that an OTU was observed at least one time in a given sample, and a 0 to indicate than an OTU was not observed at least one time in a given sample). However, if that feature table were provided to an analysis computing a quantitative diversity metric where OTU abundances are included in the calculation (e.g., weighted UniFrac), the analysis would complete successfully, but the result would not be meaningful.</p>
<p>Check out the <a class="reference internal" href="../semantic-types/"><span class="doc">semantic types page</span></a> for more information about semantic types and what types are currently available.</p>
</div>
<div class="section" id="plugins">
<h2>Plugins<a class="headerlink" href="#plugins" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2 microbiome analyses are made available to users via <em>plugins</em>. To perform analyses with QIIME 2, you will install one or more plugins that provide the specific analyses you are interested in. For example, if you want to demultiplex your raw sequence data, you might use the <code class="docutils literal"><span class="pre">q2-demux</span></code> QIIME 2 plugin, or if you&#8217;re wanting to perform alpha- or beta-diversity analyses, you could use the <code class="docutils literal"><span class="pre">q2-diversity</span></code> plugin.</p>
<p>Plugins are software packages that can be developed by anyone. The QIIME 2 team has developed several plugins for an initial end-to-end microbiome analysis pipeline, but third-party developers are encouraged to create their own plugins to provide additional analyses. Third-party developers will define these plugins in the same way that the QIIME 2 team has defined the &#8220;official&#8221; plugins. This decentralized development of microbiome analysis functionality means that many more analyses and tools will be accessible to QIIME 2 users, including the latest techniques and protocols. Plugins also allow users to choose and customize analysis pipelines for their specific needs.</p>
<p>Check out the <a class="reference internal" href="../plugins/available/"><span class="doc">plugin availability</span></a> page to see what plugins are currently available and the <a class="reference internal" href="../plugins/future/"><span class="doc">future plugins</span></a> page for those that are being developed.</p>
</div>
<div class="section" id="methods-and-visualizers">
<h2>Methods and visualizers<a class="headerlink" href="#methods-and-visualizers" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2 plugins define <em>methods</em> and <em>visualizers</em> that are used to perform analyses.</p>
<p>A <em>method</em> accepts some combination of QIIME 2 artifacts and parameters as input, and produces one or more QIIME 2 artifacts as output. These output artifacts could subsequently be used as input to other QIIME 2 methods or visualizers. Methods can produce intermediate or terminal outputs in a QIIME 2 analysis. For example, the <code class="docutils literal"><span class="pre">rarefy</span></code> method defined in the <code class="docutils literal"><span class="pre">q2-feature-table</span></code> plugin accepts a feature table artifact and sampling depth as input and produces a rarefied feature table artifact as output. This rarefied feature table artifact could then be used in another analysis, such as alpha diversity calculations provided by the <code class="docutils literal"><span class="pre">alpha</span></code> method in <code class="docutils literal"><span class="pre">q2-diversity</span></code>.</p>
<p>A <em>visualizer</em> is similar to a <em>method</em> in that it accepts some combination of QIIME 2 artifacts and parameters as input. In contrast to a method, a visualizer produces exactly one <em>visualization</em> as output. Visualizations, by definition, cannot be used as input to other QIIME 2 methods or visualizers. Thus, visualizers can only produce terminal output in a QIIME 2 analysis.</p>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>Now that you&#8217;re familiar with the core concepts in QIIME 2, you are ready to <a class="reference internal" href="../install/"><span class="doc">install QIIME 2</span></a> and work through the <a class="reference internal" href="../tutorials/"><span class="doc">tutorials</span></a>.</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about/">What is QIIME 2?</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Core concepts</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#data-files-qiime-2-artifacts">Data files: QIIME 2 artifacts</a></li>
<li class="toctree-l2"><a class="reference internal" href="#data-files-visualizations">Data files: visualizations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#semantic-types">Semantic types</a></li>
<li class="toctree-l2"><a class="reference internal" href="#plugins">Plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="#methods-and-visualizers">Methods and visualizers</a></li>
<li class="toctree-l2"><a class="reference internal" href="#next-steps">Next steps</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2017, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>