<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>“Moving Pictures” tutorial &#8212; QIIME 2 2.0.6 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2.0.6',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2.0.6 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Fecal microbiota transplant (FMT) study: an exercise" href="../fmt/" />
    <link rel="prev" title="Tutorials" href="../" />
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/logo400.png" alt="QIIME 2"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="moving-pictures-tutorial">
<h1>&#8220;Moving Pictures&#8221; tutorial<a class="headerlink" href="#moving-pictures-tutorial" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a>.</p>
</div>
<p>In this tutorial you&#8217;ll use QIIME 2 to perform an analysis of human microbiome samples from two individuals at four body sites at five timepoints, the first of which immediately followed antibiotic usage. A study based on these samples was originally published in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/21624126">Caporaso et al. (2011)</a>. The data used in this tutorial were sequenced on an Illumina HiSeq using the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> hypervariable region 4 (V4) 16S rRNA sequencing protocol.</p>
<div class="qiime1 admonition" id="qiime1users-0">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">These are the same data that are used in the QIIME 1 <a class="reference external" href="http://nbviewer.jupyter.org/github/biocore/qiime/blob/1.9.1/examples/ipynb/illumina_overview_tutorial.ipynb">Illumina Overview Tutorial</a>.</p>
</div>
<p>Before beginning this tutorial, it is recommended to create a new directory that all tutorial commands will be executed in, so that all the tutorial output will be stored in a single place. For example:</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir qiime2-moving-pictures-tutorial
<span class="nb">cd</span> qiime2-moving-pictures-tutorial
</pre></div>
</div>
<div class="section" id="sample-metadata">
<h2>Sample metadata<a class="headerlink" href="#sample-metadata" title="Permalink to this headline">¶</a></h2>
<p>Before starting the analysis, explore the sample metadata to familiarize yourself with the samples used in this study. The <a class="reference external" href="https://docs.google.com/spreadsheets/d/1_3ZbqCtAYx-9BJYHoWlICkVJ4W_QGMfJRPLedt_0hws/edit?usp=sharing">sample metadata</a> is available as a Google Sheet. You can download this file as tab-separated text by selecting <code class="docutils literal"><span class="pre">File</span></code> &gt; <code class="docutils literal"><span class="pre">Download</span> <span class="pre">as</span></code> &gt; <code class="docutils literal"><span class="pre">Tab-separated</span> <span class="pre">values</span></code>. Alternatively, the following command will download the sample metadata as tab-separated text and save it in the file <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>. This <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code> file is used throughout the rest of the tutorial.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>curl -sL <span class="s2">&quot;https://docs.google.com/spreadsheets/d/1_3ZbqCtAYx-9BJYHoWlICkVJ4W_QGMfJRPLedt_0hws/export?gid=0&amp;format=tsv&quot;</span> &gt; sample-metadata.tsv
</pre></div>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last"><a class="reference external" href="http://keemei.qiime.org">Keemei</a> is a Google Sheets add-on for validating sample metadata. Validation of sample metadata is important before beginning any analysis. Try installing Keemei following the instructions on its website, and then validate the sample metadata spreadsheet linked above. The spreadsheet also includes a sheet with some invalid data to try out with Keemei.</p>
</div>
</div>
<div class="section" id="obtaining-and-importing-raw-data">
<h2>Obtaining and importing raw data<a class="headerlink" href="#obtaining-and-importing-raw-data" title="Permalink to this headline">¶</a></h2>
<p>Download the raw sequences that we&#8217;ll use in this analysis. In this tutorial we&#8217;ll work with a small subset of the complete sequence data so that the commands will run quickly.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir raw-sequences
curl -sL https://data.qiime2.org/2.0.6/tutorials/moving-pictures/raw-sequences/barcodes.fastq.gz &gt; raw-sequences/barcodes.fastq.gz
curl -sL https://data.qiime2.org/2.0.6/tutorials/moving-pictures/raw-sequences/sequences.fastq.gz &gt; raw-sequences/sequences.fastq.gz
</pre></div>
</div>
<p>All data that is used as input to QIIME 2 is in form of QIIME 2 artifacts, which contain information about the type of data and the source of the data. So, the first thing we need to do is import these raw data files into a QIIME 2 artifact. The semantic type of this artifact is <code class="docutils literal"><span class="pre">RawSequences</span></code>.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --type RawSequences <span class="se">\</span>
  --input-path raw-sequences <span class="se">\</span>
  --output-path raw-sequences.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">raw-sequences.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fraw-sequences.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/raw-sequences.qza">download</a></li>
</ul>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">Links are included to view and download precomputed QIIME 2 artifacts and visualizations created by commands in the documentation. For example, the command above created a single <code class="docutils literal"><span class="pre">raw-sequences.qza</span></code> file, and a corresponding precomputed file is linked above. You can view precomputed artifacts and visualizations without needing to install additional software (e.g. QIIME 2).</p>
</div>
<p><code class="docutils literal"><span class="pre">RawSequences</span></code> artifacts contain sequences that are multiplexed, meaning that the sequences have not yet been assigned to samples (hence the inclusion of both <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code> and <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> files, where the <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> contains the barcode (i.e., index) read associated with each sequence in <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code>.)</p>
<div class="qiime1 admonition" id="qiime1users-1">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">In QIIME 1, we generally suggested performing demultiplexing through QIIME (e.g., with <code class="docutils literal"><span class="pre">split_libraries.py</span></code> or <code class="docutils literal"><span class="pre">split_libraries_fastq.py</span></code>) as this step also performed quality control of sequences. We now separate the demultiplexing and quality control steps, so you can begin QIIME 2 with either multiplexed sequences (as we&#8217;re doing here) or demultiplexed sequences.</p>
</div>
</div>
<div class="section" id="demultiplexing-sequences">
<h2>Demultiplexing sequences<a class="headerlink" href="#demultiplexing-sequences" title="Permalink to this headline">¶</a></h2>
<p>To demultiplex sequences we need to know which barcode sequence is associated with each sample. This information is contained in the <a class="reference external" href="https://docs.google.com/spreadsheets/d/1_3ZbqCtAYx-9BJYHoWlICkVJ4W_QGMfJRPLedt_0hws/edit?usp=sharing">sample metadata</a> file, so you should download that file if you haven&#8217;t already. You can then run the following command to demultiplex the sequences (the <code class="docutils literal"><span class="pre">demux</span> <span class="pre">emp</span></code> command refers to the fact that these sequences are barcoded according to the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> protocol). This will result in a new <code class="docutils literal"><span class="pre">SampleData[SequencesWithQuality]</span></code> artifact, where each sequence read is associated with its source sample.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime demux emp <span class="se">\</span>
  --i-seqs raw-sequences.qza <span class="se">\</span>
  --m-barcodes-file sample-metadata.tsv <span class="se">\</span>
  --m-barcodes-category BarcodeSequence <span class="se">\</span>
  --o-per-sample-sequences demux.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/demux.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="sequence-quality-control">
<h2>Sequence quality control<a class="headerlink" href="#sequence-quality-control" title="Permalink to this headline">¶</a></h2>
<p>We&#8217;ll next perform quality control on the demultiplexed sequences using <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a>. DADA2 is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the <code class="docutils literal"><span class="pre">q2-dada2</span></code> plugin, this quality control process will additionally filter any phiX reads (a common experimental artifact) that are identified in the sequencing data, and will filter chimeric sequences. The result of this step will be a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact, which contains counts (frequencies) of each unique sequence in each sample in the dataset, and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact, which maps feature identifiers in the <code class="docutils literal"><span class="pre">FeatureTable</span></code> to the sequences they represent.</p>
<div class="qiime1 admonition" id="qiime1users-2">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">The <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact is the equivalent of the QIIME 1 OTU or BIOM table, and the <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact is the equivalent of the QIIME 1 <em>representative sequences</em> file. Because the &#8220;OTUs&#8221; resulting from DADA2 are creating by grouping unique sequences, these are the equivalent of 100% OTUs from QIIME 1. In DADA2, these 100% OTUs are referred to as <em>denoised sequence variants</em>. In QIIME 2, these OTUs are higher resolution than the QIIME 1 default of 97% OTUs, and they&#8217;re higher quality due to the DADA2 denoising process. This should therefore result in more accurate estimates of diversity and taxonomic composition of samples than was achieved with QIIME 1.</p>
</div>
<p>The <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise</span></code> method requires two parameters that are used in quality filtering: <code class="docutils literal"><span class="pre">--p-trim-left</span> <span class="pre">m</span></code>, which trims off the first <code class="docutils literal"><span class="pre">m</span></code> bases of each sequence, and <code class="docutils literal"><span class="pre">--p-trunc-len</span> <span class="pre">n</span></code> which truncates each sequence at position <code class="docutils literal"><span class="pre">n</span></code>. This allows the user to remove low quality regions of the sequences. To determine what values to pass for these two parameters, you should first run the <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">plot-qualities</span></code> visualizer, which will generate plots of the quality scores by position for a randomly selected set of samples. In the following command, we&#8217;ll generate a quality plot using 10 randomly selected samples (specified by passing <code class="docutils literal"><span class="pre">--p-n</span> <span class="pre">10</span></code>).</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 plot-qualities <span class="se">\</span>
  --i-demultiplexed-seqs demux.qza <span class="se">\</span>
  --p-n <span class="m">10</span> <span class="se">\</span>
  --o-visualization demux-qual-plots.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux-qual-plots.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux-qual-plots.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/demux-qual-plots.qzv">download</a></li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>All QIIME 2 visualizers (i.e., commands that take a <code class="docutils literal"><span class="pre">--o-visualization</span></code> parameter) will generate a <code class="docutils literal"><span class="pre">.qzv</span></code> file. You can view these files with <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code>. We provide the command to view this first visualization, but for the remainder of this tutorial we&#8217;ll tell you to <em>view the resulting visualization</em> after running a visualizer, which means that you should run <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code> on the .qzv file that was generated.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools view demux-qual-plots.qzv
</pre></div>
</div>
<p class="last">Alternatively, you can view artifacts and visualizations at <a class="reference external" href="https://view.qiime2.org">view.qiime2.org</a> by uploading files or providing URLs. There are also precomputed results linked above that can be viewed or downloaded.</p>
</div>
<div class="question admonition" id="question-0">
<p class="first admonition-title">Question</p>
<p class="last">Based on the plots you see in <code class="docutils literal"><span class="pre">demux-qual-plots.qzv</span></code>, what values would you choose for <code class="docutils literal"><span class="pre">--p-trunc-len</span></code> and <code class="docutils literal"><span class="pre">--p-trim-left</span></code> in this case?</p>
</div>
<p>In these plots, the quality of the initial bases seems to be high, so we won&#8217;t trim any bases from the beginning of the sequences. The quality seems to drop off around position 100, so we&#8217;ll truncate our sequences at 100 bases. This next command may take up to 10 minutes to run, and is the slowest step in this tutorial.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 denoise <span class="se">\</span>
  --i-demultiplexed-seqs demux.qza <span class="se">\</span>
  --p-trim-left <span class="m">0</span> <span class="se">\</span>
  --p-trunc-len <span class="m">100</span> <span class="se">\</span>
  --o-representative-sequences rep-seqs.qza <span class="se">\</span>
  --o-table table.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/table.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/rep-seqs.qza">download</a></li>
</ul>
</div>
<p>After the <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise</span></code> step completes, you&#8217;ll want to explore the resulting data. You can do this using the following two commands, which will create visual summaries of the data. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">summarize</span></code> command will give you information on how many sequences are associated with each sample and with each feature, histograms of those distributions, and some related summary statistics. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">tabulate-seqs</span></code> command will provide a mapping of feature IDs to sequences, and provide links to easily BLAST each sequence against the NCBI nt database. The latter visualization will be very useful later in the tutorial, when you want to learn more about specific features that are important in the data set.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-table summarize <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-visualization table.qzv
qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data rep-seqs.qza <span class="se">\</span>
  --o-visualization rep-seqs.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/rep-seqs.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/table.qzv">download</a></li>
</ul>
</div>
</div>
<div class="section" id="generate-a-tree-for-phylogenetic-diversity-analyses">
<h2>Generate a tree for phylogenetic diversity analyses<a class="headerlink" href="#generate-a-tree-for-phylogenetic-diversity-analyses" title="Permalink to this headline">¶</a></h2>
<p>QIIME supports several phylogenetic diversity metrics, including Faith&#8217;s Phylogenetic Diversity and weighted and unweighted UniFrac. In addition to counts of features per sample (i.e., the data in the <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact), these metrics require a rooted phylogenetic tree relating the features to one another. This information will be stored in a <code class="docutils literal"><span class="pre">Phylogeny[Rooted]</span></code> artifact. The following steps will generate this artifact.</p>
<p>First, we perform a multiple sequence alignment of the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> to create a <code class="docutils literal"><span class="pre">FeatureData[AlignedSequence]</span></code> artifact. Here we do this with the <cite>mafft</cite> program.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mafft <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --o-alignment aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Faligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/aligned-rep-seqs.qza">download</a></li>
</ul>
</div>
<p>Next, we mask (or filter) the alignment to remove positions that are highly variable. These positions are generally considered to add noise to a resulting phylogenetic tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mask <span class="se">\</span>
  --i-alignment aligned-rep-seqs.qza <span class="se">\</span>
  --o-masked-alignment masked-aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">masked-aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fmasked-aligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/masked-aligned-rep-seqs.qza">download</a></li>
</ul>
</div>
<p>Next, we&#8217;ll apply FastTree to generate a phylogenetic tree from the masked alignment.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny fasttree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree unrooted-tree.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">unrooted-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Funrooted-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/unrooted-tree.qza">download</a></li>
</ul>
</div>
<p>The FastTree program creates an unrooted tree, so in the final step in this section we apply midpoint rooting to place the root of the tree at the midpoint of the longest tip-to-tip distance in the unrooted tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny midpoint-root <span class="se">\</span>
  --i-tree unrooted-tree.qza <span class="se">\</span>
  --o-rooted-tree rooted-tree.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rooted-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Frooted-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/rooted-tree.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="alpha-and-beta-diversity-analysis">
<h2>Alpha and beta diversity analysis<a class="headerlink" href="#alpha-and-beta-diversity-analysis" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2&#8217;s diversity analyses are available through the <code class="docutils literal"><span class="pre">q2-diversity</span></code> plugin, which supports computing alpha and beta diversity metrics, applying related statistical tests, and generating interactive visualizations. We&#8217;ll first apply the <code class="docutils literal"><span class="pre">core-metrics</span></code> method, which rarefies a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> to a user-specified depth, and then computes a series of alpha and beta diversity metrics. The metrics computed by default are:</p>
<ul class="simple">
<li>Alpha diversity<ul>
<li>Shannon&#8217;s diversity index (a quantitative measure of community richness)</li>
<li>Observed OTUs (a qualitative measure of community richness)</li>
<li>Faith&#8217;s Phylogenetic Diversity (a qualitiative measure of community richness that incorporates phylogenetic relationships between the features)</li>
<li>Evenness (or Pielou&#8217;s Evenness; a measure of community evenness)</li>
</ul>
</li>
<li>Beta diversity<ul>
<li>Jaccard distance (a qualitative measure of community dissimilarity)</li>
<li>Bray-Curtis distance (a quantitative measure of community dissimilarity)</li>
<li>unweighted UniFrac distance (a qualitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
<li>weighted UniFrac distance (a quantitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
</ul>
</li>
</ul>
<p>The only parameter that needs to be provided to this script is <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code>, which is the even sampling (i.e. rarefaction) depth. Because most diversity metrics are sensitive to different sampling depths across different samples, this script will randomly subsample the counts from each sample to the value provided for this parameter. For example, if you provide <code class="docutils literal"><span class="pre">--p-counts-per-sample</span> <span class="pre">500</span></code>, this step will subsample the counts in each sample without replacement so that each sample in the resulting table has a total count of 500. If the total count for any sample(s) are smaller than this value, those samples will be dropped from the diversity analysis. Choosing this value is tricky. We recommend making your choice by reviewing the information presented in the <code class="docutils literal"><span class="pre">table.qzv</span></code> file that was created above and choosing a value that is as high as possible (so you retain more sequences per sample) while excluding as few samples as possible.</p>
<div class="question admonition" id="question-1">
<p class="first admonition-title">Question</p>
<p class="last">View the <code class="docutils literal"><span class="pre">table.qzv</span></code> artifact. What value would you choose to pass for the <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code>? How many samples will be excluded from your analysis based on this choice? Approximately how many total sequences will you be analyzing in the <code class="docutils literal"><span class="pre">core-metrics</span></code> command?</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity core-metrics <span class="se">\</span>
  --i-phylogeny rooted-tree.qza <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --p-sampling-depth <span class="m">1441</span> <span class="se">\</span>
  --output-dir cm1441
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/jaccard_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fjaccard_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/jaccard_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/weighted_unifrac_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fweighted_unifrac_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/weighted_unifrac_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/bray_curtis_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fbray_curtis_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/bray_curtis_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/jaccard_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fjaccard_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/jaccard_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/shannon_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fshannon_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/shannon_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/evenness_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fevenness_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/evenness_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted_unifrac_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted_unifrac_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted_unifrac_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted_unifrac_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted_unifrac_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted_unifrac_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/weighted_unifrac_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fweighted_unifrac_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/weighted_unifrac_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/bray_curtis_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fbray_curtis_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/bray_curtis_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/faith_pd_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Ffaith_pd_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/faith_pd_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/observed_otus_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fobserved_otus_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/observed_otus_vector.qza">download</a></li>
</ul>
</div>
<p>Here we set the <code class="docutils literal"><span class="pre">--p-counts-per-sample</span></code> parameter to 1441. This value was chosen here because it&#8217;s nearly the same number of sequences as the next few samples, and because it is the lowest value it will allow us to retain all of our samples. In many Illumina runs however you&#8217;ll observe a few samples that have much lower sequence counts (on the order of tens or a couple of hundred samples) - you will typically want to exclude those from the analysis by choosing a larger value.</p>
<p>After computing diversity metrics, we can begin to explore the microbial composition of the samples in the context of the sample metadata. This information is present in the <a class="reference external" href="https://docs.google.com/spreadsheets/d/1_3ZbqCtAYx-9BJYHoWlICkVJ4W_QGMfJRPLedt_0hws/edit?usp=sharing">sample metadata</a> file that was downloaded earlier (<code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>).</p>
<p>We&#8217;ll first test for associations between discrete metadata categories and alpha diversity data. We&#8217;ll do that here for the Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-group-significance <span class="se">\</span>
  --i-alpha-diversity cm1441/faith_pd_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/faith-pd-group-significance.qzv

qiime diversity alpha-group-significance <span class="se">\</span>
  --i-alpha-diversity cm1441/evenness_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/evenness-group-significance.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/evenness-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fevenness-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/evenness-group-significance.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/faith-pd-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Ffaith-pd-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/faith-pd-group-significance.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-2">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>richness</strong>? Are these differences statistically significant?</p>
</div>
<div class="question admonition" id="question-3">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>evenness</strong>? Are these differences statistically significant?</p>
</div>
<p>Next, we&#8217;ll test for associations between alpha diversity metrics and continuous sample metadata (such as pH or elevation). We can do this running the following two commands, which will support analysis of Faith&#8217;s Phylogenetic Diversity metric and evenness in the context of our continuous metadata. Run these commands and view the resulting artifacts.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-correlation <span class="se">\</span>
  --i-alpha-diversity cm1441/faith_pd_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/faith-pd-correlation.qzv

qiime diversity alpha-correlation <span class="se">\</span>
  --i-alpha-diversity cm1441/evenness_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/evenness-correlation.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/evenness-correlation.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fevenness-correlation.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/evenness-correlation.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/faith-pd-correlation.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Ffaith-pd-correlation.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/faith-pd-correlation.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-4">
<p class="first admonition-title">Question</p>
<p class="last">What do you conclude about the associations between continuous sample metadata and the richness and evenness of these samples?</p>
</div>
<p>Next we&#8217;ll analyze sample composition in the context of discrete metadata using PERMANOVA (first described in <a class="reference external" href="http://onlinelibrary.wiley.com/doi/10.1111/j.1442-9993.2001.01070.pp.x/full">Anderson (2001)</a>) using the <code class="docutils literal"><span class="pre">beta-group-significance</span></code> command. The following commands will test whether distances between samples within a group, such as samples from the same body site (e.g., skin or gut), are more similar to each other then they are to samples from a different group. This command can be slow to run since it is based on permutation tests, so unlike the previous commands we&#8217;ll run this on specific categories of metadata that we&#8217;re interested in exploring, rather than all metadata categories that it&#8217;s applicable to. Here we&#8217;ll apply this to our unweighted UniFrac distances, using two sample metadata categories, as follows.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity beta-group-significance <span class="se">\</span>
  --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category BodySite <span class="se">\</span>
  --o-visualization cm1441/unweighted-unifrac-body-site-significance.qzv

qiime diversity beta-group-significance <span class="se">\</span>
  --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category Subject <span class="se">\</span>
  --o-visualization cm1441/unweighted-unifrac-subject-group-significance.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted-unifrac-body-site-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted-unifrac-body-site-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted-unifrac-body-site-significance.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted-unifrac-subject-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted-unifrac-subject-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted-unifrac-subject-group-significance.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-5">
<p class="first admonition-title">Question</p>
<p class="last">Are the associations between subjects and differences in microbial composition statistically significant? How about sample types? What sample types appear to be most different from each other?</p>
</div>
<p>Finally, we&#8217;ll explore associations between the microbial composition of the samples and continuous sample metadata using bioenv (originally described in <a class="reference external" href="http://www.int-res.com/articles/meps/92/m092p205.pdf">Clarke and Ainsworth (1993)</a>). This approach tests for associations of pairwise distances between sample microbial composition (a measure of beta diversity) and sample metadata (for example, the matrix of Bray-Curtis distances between samples and the matrix of absolute differences in pH between samples). A powerful feature of this method is that it explores combinations of sample metadata to see which groups of metadata differences are most strongly associated with the observed microbial differences between samples. You can apply bioenv to the unweighted UniFrac distances and Bray-Curtis distances between the samples, respectively, as follows. After running these commands, open the resulting visualizations.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity bioenv <span class="se">\</span>
  --i-distance-matrix cm1441/unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/unweighted-unifrac-bioenv.qzv

qiime diversity bioenv <span class="se">\</span>
  --i-distance-matrix cm1441/bray_curtis_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization cm1441/bray-curtis-bioenv.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted-unifrac-bioenv.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted-unifrac-bioenv.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted-unifrac-bioenv.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/bray-curtis-bioenv.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fbray-curtis-bioenv.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/bray-curtis-bioenv.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-6">
<p class="first admonition-title">Question</p>
<p class="last">What sample metadata or combinations of sample metadata are most strongly associated with the differences in microbial composition of the samples? How strong are these correlations?</p>
</div>
<p>Finally, ordination is a popular approach for exploring microbial community composition in the context of sample metadata. We can use the <a class="reference external" href="http://emperor.microbio.me">Emperor</a> tool to explore principal coordinates (PCoA) plots in the context of sample metadata. PCoA is run as part of the <code class="docutils literal"><span class="pre">core-metrics</span></code> command, so we can generate these plots for unweighted UniFrac and Bray-Curtis as follows. The <code class="docutils literal"><span class="pre">--p-custom-axis</span></code> parameter that we pass here is very useful for exploring temporal data. The resulting plot will contain axes for principal coordinate 1 (labelled <code class="docutils literal"><span class="pre">0</span></code>), principal coordinate 1 (labelled <code class="docutils literal"><span class="pre">1</span></code>), and days since the experiment start. This is useful for exploring how the samples change over time.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime emperor plot <span class="se">\</span>
  --i-pcoa cm1441/unweighted_unifrac_pcoa_results.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --p-custom-axis DaysSinceExperimentStart <span class="se">\</span>
  --o-visualization cm1441/unweighted-unifrac-emperor.qzv

qiime emperor plot <span class="se">\</span>
  --i-pcoa cm1441/bray_curtis_pcoa_results.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --p-custom-axis DaysSinceExperimentStart <span class="se">\</span>
  --o-visualization cm1441/bray-curtis-emperor.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">cm1441/unweighted-unifrac-emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Funweighted-unifrac-emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/unweighted-unifrac-emperor.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">cm1441/bray-curtis-emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fbray-curtis-emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/cm1441/bray-curtis-emperor.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-7">
<p class="first admonition-title">Question</p>
<p class="last">Do the Emperor plots support the other beta diversity analyses we&#8217;ve performed here? (Hint: Experiment with coloring points by different metadata.)</p>
</div>
<div class="question admonition" id="question-8">
<p class="first admonition-title">Question</p>
<p class="last">What differences do you observe between the unweighted UniFrac and Bray-Curtis PCoA plots?</p>
</div>
</div>
<div class="section" id="taxonomic-analysis">
<h2>Taxonomic analysis<a class="headerlink" href="#taxonomic-analysis" title="Permalink to this headline">¶</a></h2>
<p>In the next sections we&#8217;ll begin to explore the taxonomic composition of the samples, and again relate that to sample metadata. The first step in this process is to assign taxonomy to the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact. We&#8217;ll do that using a pre-trained Naive Bayes classifier and the <code class="docutils literal"><span class="pre">q2-feature-classifier</span></code> plugin. This classifier was trained on the Greengenes 13_8 99% OTUs, where the sequences have been trimmed to only include 250 bases from the region of the 16S that was sequenced in this analysis (the V4 region, bound by the 515F/806R primer pair). We&#8217;ll apply this classifier to our sequences, and we can generate a visualization of the resulting mapping from sequence to taxonomy.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Taxonomic classifiers perform best when they are trained based on your specific sample preparation and sequencing parameters, including the primers that were used for amplification and the length of your sequence reads. Therefore in general you should follow the instructions in <a class="reference internal" href="../feature-classifier/"><span class="doc">Training feature classifiers with q2-feature-classifier</span></a> to train your own taxonomic classifiers. We provide some common classifiers on our <a class="reference internal" href="../../data-resources/"><span class="doc">data resources page</span></a>, including Silva-based 16S classifiers, as training classifiers can require a large amount of memory.</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>curl -sLO https://data.qiime2.org/2.0.6/common/gg-13-8-99-515-806-nb-classifier.qza

qiime feature-classifier classify <span class="se">\</span>
  --i-classifier gg-13-8-99-515-806-nb-classifier.qza <span class="se">\</span>
  --i-reads rep-seqs.qza <span class="se">\</span>
  --o-classification taxonomy.qza

qiime taxa tabulate <span class="se">\</span>
  --i-data taxonomy.qza <span class="se">\</span>
  --o-visualization taxonomy.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">gg-13-8-99-515-806-nb-classifier.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fgg-13-8-99-515-806-nb-classifier.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/gg-13-8-99-515-806-nb-classifier.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">taxonomy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxonomy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/taxonomy.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">taxonomy.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxonomy.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/taxonomy.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-9">
<p class="first admonition-title">Question</p>
<p class="last">Recall that our <code class="docutils literal"><span class="pre">rep-seqs.qzv</span></code> visualization allows you to easily BLAST the sequence associated with each feature against the NCBI nt database. Using that visualization and the <code class="docutils literal"><span class="pre">taxonomy.qzv</span></code> visualization created here, compare the taxonomic assignments with the taxonomy of the best BLAST hit for a few features. How similar are the assignments? If they&#8217;re dissimilar, at what <em>taxonomic level</em> do they begin to differ (e.g., species, genus, family, ...)?</p>
</div>
<p>Next, we can view the taxonomic composition of our samples with interactive bar plots. Generate those plots with the following command and then open the visualization.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa barplot <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-taxonomy taxonomy.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization taxa-bar-plots.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">taxa-bar-plots.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxa-bar-plots.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/taxa-bar-plots.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-10">
<p class="first admonition-title">Question</p>
<p class="last">Visualize the samples at <em>Level 2</em> (which corresponds to the phylum level in this analysis), and then sort the samples by BodySite, then by Subject, and then by DaysSinceExperimentStart. What are the dominant phyla in each in BodySite? Do you observe any consistent change across the two subjects between DaysSinceExperimentStart <code class="docutils literal"><span class="pre">0</span></code> and the later timepoints?</p>
</div>
</div>
<div class="section" id="differential-abundance-analysis">
<h2>Differential abundance analysis<a class="headerlink" href="#differential-abundance-analysis" title="Permalink to this headline">¶</a></h2>
<p>Finally, we can quantify the process of identifying taxa that are differentially abundance (or present in different abundances) across sample groups. We do that using ANCOM (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/26028277">Mandal et al. (2015)</a>), which is implemented in the <code class="docutils literal"><span class="pre">q2-composition</span></code> plugin. ANCOM operates on a <code class="docutils literal"><span class="pre">FeatureTable[Composition]</span></code> artifact, which is based on relative frequencies of features on a per-sample basis, but cannot tolerate frequencies of zero. We work around this by adding a pseudocount of 1 to every count in our <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> table. We can run this on the <code class="docutils literal"><span class="pre">BodySite</span></code> category to determine what features differ in abundance across our sample types. This step may take about 5 minutes to complete.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime composition add-pseudocount <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-composition-table comp-table.qza

qiime composition ancom <span class="se">\</span>
  --i-table comp-table.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category BodySite <span class="se">\</span>
  --o-visualization ancom-BodySite.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">comp-table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcomp-table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/comp-table.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">ancom-BodySite.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fancom-BodySite.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/ancom-BodySite.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-11">
<p class="first admonition-title">Question</p>
<p class="last">What features differ in abundance across BodySite? What groups are they most and least abundant in? What are some the taxonomies of some of these features? (To answer that last question you&#8217;ll need to refer to a visualization that we generated earlier in this tutorial.)</p>
</div>
<p>We&#8217;re also often interested in performing a differential abundance test at a specific taxonomic level. To do this, we can collapse the features in our <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> at the taxonomic level of interest, and then re-run the above steps.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa collapse <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-taxonomy taxonomy.qza <span class="se">\</span>
  --p-level <span class="m">2</span> <span class="se">\</span>
  --o-collapsed-table table-l2.qza

qiime composition add-pseudocount <span class="se">\</span>
  --i-table table-l2.qza <span class="se">\</span>
  --o-composition-table comp-table-l2.qza

qiime composition ancom <span class="se">\</span>
  --i-table comp-table-l2.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category BodySite <span class="se">\</span>
  --o-visualization l2-ancom-BodySite.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">comp-table-l2.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcomp-table-l2.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/comp-table-l2.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table-l2.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable-l2.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/table-l2.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">l2-ancom-BodySite.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2.0.6%2Fdata%2Ftutorials%2Fmoving-pictures%2Fl2-ancom-BodySite.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/l2-ancom-BodySite.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-12">
<p class="first admonition-title">Question</p>
<p class="last">What phyla differ in abundance across BodySite? How does this align with what you observed in the <code class="docutils literal"><span class="pre">taxa-bar-plots.qzv</span></code> visualization that was generated above?</p>
</div>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu dropdown-menu-right"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../88soils/">&#8220;88 soils&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../import/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../import-sequence-data/">Importing sequence data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../table-filtering/">Filtering feature tables</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016, QIIME 2 Development Team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>