<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>iqtree-ultrafast-bootstrap: Construct a phylogenetic tree with IQ-TREE with bootstrap supports. &#8212; QIIME 2 2018.11.0 documentation</title>
    
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../../',
        VERSION:     '2018.11.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="top" title="QIIME 2 2018.11.0 documentation" href="../../../../" />
    <link rel="up" title="phylogeny" href="../" />
    <link rel="next" title="midpoint-root: Midpoint root an unrooted phylogenetic tree." href="../midpoint-root/" />
    <link rel="prev" title="iqtree: Construct a phylogenetic tree with IQ-TREE." href="../iqtree/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="iqtree-ultrafast-bootstrap-construct-a-phylogenetic-tree-with-iq-tree-with-bootstrap-supports">
<h1>iqtree-ultrafast-bootstrap: Construct a phylogenetic tree with IQ-TREE with bootstrap supports.<a class="headerlink" href="#iqtree-ultrafast-bootstrap-construct-a-phylogenetic-tree-with-iq-tree-with-bootstrap-supports" title="Permalink to this headline">¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><p id="bibtex-bibliography-plugins/available/phylogeny/iqtree-ultrafast-bootstrap-0"><table class="docutils citation" frame="void" id="phylogeny-iqtree-ultrafast-bootstrap-key3" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:iqtree-ultrafast-bootstrap:HCvH+17]</td><td>Diep&nbsp;Thi Hoang, Olga Chernomor, Arndt von Haeseler, Bui&nbsp;Quang Minh, and Le&nbsp;Sy Vinh. Ufboot2: improving the ultrafast bootstrap approximation. <em>Molecular Biology and Evolution</em>, 35(2):518–522, 2017. <a class="reference external" href="https://doi.org/10.1093/molbev/msx281">doi:10.1093/molbev/msx281</a>.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="phylogeny-iqtree-ultrafast-bootstrap-key1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:iqtree-ultrafast-bootstrap:KMW+17]</td><td>Subha Kalyaanamoorthy, Bui&nbsp;Quang Minh, Thomas K&nbsp;F Wong, Arndt von Haeseler, and Lars&nbsp;S Jermiin. Modelfinder: fast model selection for accurate phylogenetic estimates. <em>Nature Methods</em>, 14(6):587–589, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.4285">http://dx.doi.org/10.1038/nmeth.4285</a>, <a class="reference external" href="https://doi.org/10.1038/nmeth.4285">doi:10.1038/nmeth.4285</a>.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="phylogeny-iqtree-ultrafast-bootstrap-key2" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:iqtree-ultrafast-bootstrap:MNvH13]</td><td>Bui&nbsp;Quang Minh, Minh Anh&nbsp;Thi Nguyen, and Arndt von Haeseler. Ultrafast approximation for phylogenetic bootstrap. <em>Molecular Biology and Evolution</em>, 30(5):1188–1195, 2013. URL: <a class="reference external" href="http://doi.org/10.1093/molbev/mst024">http://doi.org/10.1093/molbev/mst024</a>, <a class="reference external" href="https://doi.org/10.1093/molbev/mst024">doi:10.1093/molbev/mst024</a>.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="phylogeny-iqtree-ultrafast-bootstrap-key0" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:iqtree-ultrafast-bootstrap:NSvHM15]</td><td>Lam-Tung Nguyen, Heiko&nbsp;A. Schmidt, Arndt von Haeseler, and Bui&nbsp;Quang Minh. Iq-tree: a fast and effective stochastic algorithm for estimating maximum-likelihood phylogenies. <em>Molecular Biology and Evolution</em>, 32(1):268–274, 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/molbev/msu300">http://dx.doi.org/10.1093/molbev/msu300</a>, <a class="reference external" href="https://doi.org/10.1093/molbev/msu300">doi:10.1093/molbev/msu300</a>.</td></tr>
</tbody>
</table>
</p>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime phylogeny iqtree-ultrafast-bootstrap [OPTIONS]

  Construct a phylogenetic tree using IQ-TREE (http://www.iqtree.org/) with
  automatic model selection and bootstrap supports.

Options:
  --i-alignment ARTIFACT PATH FeatureData[AlignedSequence]
                                  Aligned sequences to be used for
                                  phylogenetic reconstruction.  [required]
  --p-seed INTEGER                Random number seed. If not set, program
                                  defaults will be used. See IQ-TREE manual
                                  for details.  [optional]
  --p-n-cores INTEGER RANGE       The number of cores to use for parallel
                                  processing. Use '0' to let IQ-TREE
                                  automatically determine the optimal number
                                  of cores to use.  [default: 1]
  --p-n-runs INTEGER RANGE        Number of indepedent runs. Multiple
                                  independent runs (e.g. 10) can outperform a
                                  single run in terms of likelihood
                                  maximisation.  [default: 1]
  --p-substitution-model [TVMe+R4|K80+R6|TNe+R6|TPM3+I|F81+R3|HKY+G|TPM3|TIMe|TIM+R4|F81+R9|GTR|TVMe+R5|K80|TPM3u+R7|TVM+R6|K80+R2|TNe+R10|TIM2+R4|HKY+I+G|TPM2u+G|TNe+R8|TNe+I|TIM2e+R6|K80+R4|TIM3e+I+G|TNe+R5|JC+R6|JC|K81|HKY+R10|GTR+R6|TVMe+R6|F81+I+G|TIMe+I+G|SYM+R6|F81+R5|TPM2+R3|TVM+G|K80+R5|TIM2+R10|TIM2e+R4|SYM+I|TIM+R3|TIM3e+R2|K81+R5|TVM+R2|TPM2u+R8|TPM2+R10|JC+R4|K81u+I|GTR+I+G|TPM3u+R4|K81u+I+G|TPM2u+I+G|TPM2u+R7|TPM2+R9|K81+G|K81u+G|TN+R2|TIM+R6|K81+I+G|TPM2+I+G|TPM3u+I+G|TIMe+R3|TIM2e+R7|HKY+R3|TPM3+R7|TIM2e+I+G|TIM+G|TIM3e+R7|TIM2+R2|TVMe+G|JC+I+G|F81+R8|K80+G|JC+R10|TPM3+R5|TPM3u+R9|TIMe+R8|TIM2e+R8|TVMe|TIM3+R4|TVM|TIM3+G|TN+R7|TPM2+R5|TNe|TIM+R9|TPM3+R8|TIM3+R8|TVMe+R10|K81+R6|GTR+R9|TIM2e+I|TVM+R8|SYM+I+G|TIM3e+R5|TPM2u+R4|K81+R7|TIM+I|TN+R5|TIM3e+R9|TIM2+R8|GTR+R2|TNe+R9|TPM3+R4|K80+I+G|TPM2+R8|TIM2+G|TN+I|K81u|TIMe+R7|TVM+R5|HKY|TIM3+R2|GTR+R5|MFP|TIM+R7|TN+R3|TPM3u+G|JC+R7|TNe+R4|TNe+G|TIM3|HKY+R4|K80+R10|SYM|TVM+R3|TIM2e+R3|TVMe+I|HKY+I|JC+G|TIM2|TIM2e+R2|TPM2u+R2|K81+R9|K81u+R8|K81+I|TPM2+R7|K81+R2|TPM2u+I|TIM3+R6|F81+R2|TN+G|TNe+I+G|K81u+R9|TIMe+R9|TIM+R5|TIM2+R3|SYM+G|TIM3+I+G|JC+R2|TIM3+R5|HKY+R9|TPM2+R6|TVMe+R7|TN+R8|K81+R4|JC+R5|TIM2+I+G|K80+R7|HKY+R6|K81u+R7|TIMe+R2|TPM3+R9|JC+R8|K81u+R4|SYM+R5|GTR+R4|F81+G|HKY+R2|K81u+R10|SYM+R2|TPM2+G|TIM2+R9|K81u+R5|TNe+R2|TIM3e+R3|SYM+R3|TN+I+G|TPM3+G|TIM2+I|TPM3u+R2|TIM2+R5|SYM+R8|TIM+R10|TIM2e+R9|TIMe+G|TVM+R7|TVM+I+G|TIM3e+R4|GTR+R7|TPM2u+R9|TNe+R7|TIMe+R4|TPM2u+R6|TVMe+R3|TVM+R10|TN|SYM+R9|TVM+R4|SYM+R10|K81+R10|TN+R9|TIM3e+R6|F81+R10|TPM2u+R5|F81+R7|K81u+R2|TIM3e+R10|TVM+I|TPM3u+I|K80+R3|TPM2+R4|JC+R9|F81+R6|F81|K81+R8|TIMe+R6|TVMe+R8|TIM3+I|GTR+R3|TIM+R2|K80+R8|HKY+R5|TIMe+I|TIM2e+R5|TPM2u+R10|TIMe+R5|K81+R3|TPM3u+R10|TPM3+R10|HKY+R7|TNe+R3|TPM2u+R3|TIM3+R9|TPM2+I|TIM2e|GTR+R8|TIM|TPM3+R6|TIM2e+G|TIM2+R6|TN+R6|TIM2+R7|F81+I|TPM3+I+G|TPM2|TPM3u+R3|TIM3e+G|TIM3e+R8|SYM+R7|TPM2+R2|TIM3+R7|TPM3u|GTR+G|F81+R4|GTR+R10|TPM3u+R5|TPM3u+R8|TIM+R8|TIM+I+G|HKY+R8|TN+R10|TVMe+R2|TN+R4|K80+R9|TVM+R9|TIM3+R10|JC+R3|TVMe+I+G|TIM3e|TPM3u+R6|TIMe+R10|GTR+I|TIM2e+R10|TEST|JC+I|K80+I|K81u+R6|TPM3+R2|K81u+R3|TPM3+R3|TIM3e+I|TIM3+R3|TVMe+R9|SYM+R4|TPM2u]
                                  Model of Nucleotide Substitution.If not
                                  provided, IQ-TREE will determine the best
                                  fit substitution model automatically.
                                  [default: MFP]
  --p-bootstrap-replicates INTEGER RANGE
                                  The number of bootstrap searches to perform.
                                  Minimum of 1000 recomended.   [default:
                                  1000]
  --p-n-init-pars-trees INTEGER RANGE
                                  Number of initial parsimony trees. If not
                                  set, program defaults will be used. See IQ-
                                  TREE manual for details.  [optional]
  --p-n-top-init-trees INTEGER RANGE
                                  Number of top initial trees. If not set,
                                  program defaults will be used. See IQ-TREE
                                  manual for details.  [optional]
  --p-n-best-retain-trees INTEGER RANGE
                                  Number of best trees retained during search.
                                  If not set, program defaults will be used.
                                  See IQ-TREE manual for details.  [optional]
  --p-stop-iter INTEGER RANGE     Number of unsuccessful iterations to stop.
                                  If not set, program defaults will be used.
                                  See IQ-TREE manual for details.  [optional]
  --p-perturb-nni-strength FLOAT  Perturbation strength for randomized NNI. If
                                  not set, program defaults will be used. See
                                  IQ-TREE manual for details.  [optional]
  --p-spr-radius INTEGER RANGE    Radius for parsimony SPR search. If not set,
                                  program defaults will be used. See IQ-TREE
                                  manual for details.  [optional]
  --p-n-max-ufboot-iter INTEGER RANGE
                                  Maximum number of iterations. If not set,
                                  program defaults will be used. See IQ-TREE
                                  manual for details.  [optional]
  --p-n-ufboot-steps INTEGER RANGE
                                  Number of iterations for UFBoot stopping
                                  rule. If not set, program defaults will be
                                  used.See IQ-TREE manual for details.
                                  [optional]
  --p-min-cor-ufboot FLOAT        Minimum correlation coefficient. If not set,
                                  program defaults will be used.See IQ-TREE
                                  manual for details.  [optional]
  --p-ep-break-ufboot FLOAT       Epsilon value to break tie. If not set,
                                  program defaults will be used. See IQ-TREE
                                  manual for details.  [optional]
  --p-allnni / --p-no-allnni      Perform more thorough NNI search.  [default:
                                  False]
  --p-alrt INTEGER RANGE          Single branch test method. Number of
                                  bootstrap replicates to perform an SH-like
                                  approximate likelihood ratio test (SH-aLRT).
                                  Minimum of 1000 replicates is recomended.
                                  Set to '0' for parametric aLRT test. Can be
                                  used with other 'single branch test
                                  methods'. Values reported in the order of:
                                  alrt, lbp, abayes, ufboot.  [optional]
  --p-abayes / --p-no-abayes      Single branch test method. Performs an
                                  approximate Bayes test. Can be used with
                                  other 'single branch test methods' and
                                  ultrafast bootstrap. Values reported in the
                                  order of: alrt, lbp, abayes, ufboot.
                                  [default: False]
  --p-lbp INTEGER RANGE           Single branch test method. Number of
                                  bootstrap replicates to perform a fast local
                                  bootstrap probability method. Minimum of
                                  1000 replicates is recomended. Can be used
                                  with other 'single branch test methods'.
                                  Values reported in the order of: alrt, lbp,
                                  abayes, ufboot.  [optional]
  --p-bnni / --p-no-bnni          Optimize UFBoot trees by NNI on bootstrap
                                  alignment. This option reduces the risk of
                                  overestimating branch supports with UFBoot
                                  due to severe model violations.  [default:
                                  False]
  --p-safe / --p-no-safe          Safe likelihood kernel to avoid numerical
                                  underflow.  [default: False]
  --o-tree ARTIFACT PATH Phylogeny[Unrooted]
                                  The resulting phylogenetic tree.  [required
                                  if not passing --output-dir]
  --output-dir DIRECTORY          Output unspecified results to a directory
  --cmd-config FILE               Use config file for command options
  --verbose                       Display verbose output to stdout and/or
                                  stderr during execution of this action.
                                  [default: False]
  --quiet                         Silence output if execution is successful
                                  (silence is golden).  [default: False]
  --citations                     Show citations and exit.
  --help                          Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell"><div class="highlight"><pre><span></span>from qiime2.plugins.phylogeny.methods import iqtree_ultrafast_bootstrap
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Construct a phylogenetic tree with IQ-TREE with bootstrap supports.

Construct a phylogenetic tree using IQ-TREE (http://www.iqtree.org/) with
automatic model selection and bootstrap supports.

Parameters
----------
alignment : FeatureData[AlignedSequence]
    Aligned sequences to be used for phylogenetic reconstruction.
seed : Int, optional
    Random number seed. If not set, program defaults will be used. See IQ-
    TREE manual for details.
n_cores : Int % Range(0, None), optional
    The number of cores to use for parallel processing. Use '0' to let IQ-
    TREE automatically determine the optimal number of cores to use.
n_runs : Int % Range(1, None), optional
    Number of indepedent runs. Multiple  independent runs (e.g. 10) can
    outperform a single run in terms of likelihood maximisation.
substitution_model : Str % Choices({'F81', 'F81+G', 'F81+I', 'F81+I+G', 'F81+R10', 'F81+R2', 'F81+R3', 'F81+R4', 'F81+R5', 'F81+R6', 'F81+R7', 'F81+R8', 'F81+R9', 'GTR', 'GTR+G', 'GTR+I', 'GTR+I+G', 'GTR+R10', 'GTR+R2', 'GTR+R3', 'GTR+R4', 'GTR+R5', 'GTR+R6', 'GTR+R7', 'GTR+R8', 'GTR+R9', 'HKY', 'HKY+G', 'HKY+I', 'HKY+I+G', 'HKY+R10', 'HKY+R2', 'HKY+R3', 'HKY+R4', 'HKY+R5', 'HKY+R6', 'HKY+R7', 'HKY+R8', 'HKY+R9', 'JC', 'JC+G', 'JC+I', 'JC+I+G', 'JC+R10', 'JC+R2', 'JC+R3', 'JC+R4', 'JC+R5', 'JC+R6', 'JC+R7', 'JC+R8', 'JC+R9', 'K80', 'K80+G', 'K80+I', 'K80+I+G', 'K80+R10', 'K80+R2', 'K80+R3', 'K80+R4', 'K80+R5', 'K80+R6', 'K80+R7', 'K80+R8', 'K80+R9', 'K81', 'K81+G', 'K81+I', 'K81+I+G', 'K81+R10', 'K81+R2', 'K81+R3', 'K81+R4', 'K81+R5', 'K81+R6', 'K81+R7', 'K81+R8', 'K81+R9', 'K81u', 'K81u+G', 'K81u+I', 'K81u+I+G', 'K81u+R10', 'K81u+R2', 'K81u+R3', 'K81u+R4', 'K81u+R5', 'K81u+R6', 'K81u+R7', 'K81u+R8', 'K81u+R9', 'MFP', 'SYM', 'SYM+G', 'SYM+I', 'SYM+I+G', 'SYM+R10', 'SYM+R2', 'SYM+R3', 'SYM+R4', 'SYM+R5', 'SYM+R6', 'SYM+R7', 'SYM+R8', 'SYM+R9', 'TEST', 'TIM', 'TIM+G', 'TIM+I', 'TIM+I+G', 'TIM+R10', 'TIM+R2', 'TIM+R3', 'TIM+R4', 'TIM+R5', 'TIM+R6', 'TIM+R7', 'TIM+R8', 'TIM+R9', 'TIM2', 'TIM2+G', 'TIM2+I', 'TIM2+I+G', 'TIM2+R10', 'TIM2+R2', 'TIM2+R3', 'TIM2+R4', 'TIM2+R5', 'TIM2+R6', 'TIM2+R7', 'TIM2+R8', 'TIM2+R9', 'TIM2e', 'TIM2e+G', 'TIM2e+I', 'TIM2e+I+G', 'TIM2e+R10', 'TIM2e+R2', 'TIM2e+R3', 'TIM2e+R4', 'TIM2e+R5', 'TIM2e+R6', 'TIM2e+R7', 'TIM2e+R8', 'TIM2e+R9', 'TIM3', 'TIM3+G', 'TIM3+I', 'TIM3+I+G', 'TIM3+R10', 'TIM3+R2', 'TIM3+R3', 'TIM3+R4', 'TIM3+R5', 'TIM3+R6', 'TIM3+R7', 'TIM3+R8', 'TIM3+R9', 'TIM3e', 'TIM3e+G', 'TIM3e+I', 'TIM3e+I+G', 'TIM3e+R10', 'TIM3e+R2', 'TIM3e+R3', 'TIM3e+R4', 'TIM3e+R5', 'TIM3e+R6', 'TIM3e+R7', 'TIM3e+R8', 'TIM3e+R9', 'TIMe', 'TIMe+G', 'TIMe+I', 'TIMe+I+G', 'TIMe+R10', 'TIMe+R2', 'TIMe+R3', 'TIMe+R4', 'TIMe+R5', 'TIMe+R6', 'TIMe+R7', 'TIMe+R8', 'TIMe+R9', 'TN', 'TN+G', 'TN+I', 'TN+I+G', 'TN+R10', 'TN+R2', 'TN+R3', 'TN+R4', 'TN+R5', 'TN+R6', 'TN+R7', 'TN+R8', 'TN+R9', 'TNe', 'TNe+G', 'TNe+I', 'TNe+I+G', 'TNe+R10', 'TNe+R2', 'TNe+R3', 'TNe+R4', 'TNe+R5', 'TNe+R6', 'TNe+R7', 'TNe+R8', 'TNe+R9', 'TPM2', 'TPM2+G', 'TPM2+I', 'TPM2+I+G', 'TPM2+R10', 'TPM2+R2', 'TPM2+R3', 'TPM2+R4', 'TPM2+R5', 'TPM2+R6', 'TPM2+R7', 'TPM2+R8', 'TPM2+R9', 'TPM2u', 'TPM2u+G', 'TPM2u+I', 'TPM2u+I+G', 'TPM2u+R10', 'TPM2u+R2', 'TPM2u+R3', 'TPM2u+R4', 'TPM2u+R5', 'TPM2u+R6', 'TPM2u+R7', 'TPM2u+R8', 'TPM2u+R9', 'TPM3', 'TPM3+G', 'TPM3+I', 'TPM3+I+G', 'TPM3+R10', 'TPM3+R2', 'TPM3+R3', 'TPM3+R4', 'TPM3+R5', 'TPM3+R6', 'TPM3+R7', 'TPM3+R8', 'TPM3+R9', 'TPM3u', 'TPM3u+G', 'TPM3u+I', 'TPM3u+I+G', 'TPM3u+R10', 'TPM3u+R2', 'TPM3u+R3', 'TPM3u+R4', 'TPM3u+R5', 'TPM3u+R6', 'TPM3u+R7', 'TPM3u+R8', 'TPM3u+R9', 'TVM', 'TVM+G', 'TVM+I', 'TVM+I+G', 'TVM+R10', 'TVM+R2', 'TVM+R3', 'TVM+R4', 'TVM+R5', 'TVM+R6', 'TVM+R7', 'TVM+R8', 'TVM+R9', 'TVMe', 'TVMe+G', 'TVMe+I', 'TVMe+I+G', 'TVMe+R10', 'TVMe+R2', 'TVMe+R3', 'TVMe+R4', 'TVMe+R5', 'TVMe+R6', 'TVMe+R7', 'TVMe+R8', 'TVMe+R9'}), optional
    Model of Nucleotide Substitution.If not provided, IQ-TREE will
    determine the best fit substitution model automatically.
bootstrap_replicates : Int % Range(1000, None), optional
    The number of bootstrap searches to perform. Minimum of 1000
    recomended.
n_init_pars_trees : Int % Range(1, None), optional
    Number of initial parsimony trees. If not set, program defaults will be
    used. See IQ-TREE manual for details.
n_top_init_trees : Int % Range(1, None), optional
    Number of top initial trees. If not set, program defaults will be used.
    See IQ-TREE manual for details.
n_best_retain_trees : Int % Range(1, None), optional
    Number of best trees retained during search. If not set, program
    defaults will be used. See IQ-TREE manual for details.
stop_iter : Int % Range(1, None), optional
    Number of unsuccessful iterations to stop. If not set, program defaults
    will be used. See IQ-TREE manual for details.
perturb_nni_strength : Float % Range(0.01, 99), optional
    Perturbation strength for randomized NNI. If not set, program defaults
    will be used. See IQ-TREE manual for details.
spr_radius : Int % Range(1, None), optional
    Radius for parsimony SPR search. If not set, program defaults will be
    used. See IQ-TREE manual for details.
n_max_ufboot_iter : Int % Range(1, None), optional
    Maximum number of iterations. If not set, program defaults will be
    used. See IQ-TREE manual for details.
n_ufboot_steps : Int % Range(1, None), optional
    Number of iterations for UFBoot stopping rule. If not set, program
    defaults will be used.See IQ-TREE manual for details.
min_cor_ufboot : Float % Range(0.51, 0.99), optional
    Minimum correlation coefficient. If not set, program defaults will be
    used.See IQ-TREE manual for details.
ep_break_ufboot : Float % Range(0.01, 0.99), optional
    Epsilon value to break tie. If not set, program defaults will be used.
    See IQ-TREE manual for details.
allnni : Bool, optional
    Perform more thorough NNI search.
alrt : Int % Range(1000, None), optional
    Single branch test method. Number of bootstrap replicates to perform an
    SH-like approximate likelihood ratio test (SH-aLRT). Minimum of 1000
    replicates is recomended. Set to '0' for parametric aLRT test. Can be
    used with other 'single branch test methods'. Values reported in the
    order of: alrt, lbp, abayes, ufboot.
abayes : Bool, optional
    Single branch test method. Performs an approximate Bayes test. Can be
    used with other 'single branch test methods' and ultrafast bootstrap.
    Values reported in the order of: alrt, lbp, abayes, ufboot.
lbp : Int % Range(1000, None), optional
    Single branch test method. Number of bootstrap replicates to perform a
    fast local bootstrap probability method. Minimum of 1000 replicates is
    recomended. Can be used with other 'single branch test methods'. Values
    reported in the order of: alrt, lbp, abayes, ufboot.
bnni : Bool, optional
    Optimize UFBoot trees by NNI on bootstrap alignment. This option
    reduces the risk of overestimating branch supports with UFBoot due to
    severe model violations.
safe : Bool, optional
    Safe likelihood kernel to avoid numerical underflow.

Returns
-------
tree : Phylogeny[Unrooted]
    The resulting phylogenetic tree.</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2018, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>