<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>QIIME 2 for Experienced Microbiome Researchers &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/language_data.js"></script>
    <script src="../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="index" title="Index" href="../../genindex/" />
    <link rel="search" title="Search" href="../../search/" />
    <link rel="next" title="“Moving Pictures” tutorial" href="../moving-pictures/" />
    <link rel="prev" title="Overview of QIIME 2 Plugin Workflows" href="../overview/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="qiime-2-for-experienced-microbiome-researchers">
<h1><a class="toc-backref" href="#id2">QIIME 2 for Experienced Microbiome Researchers</a><a class="headerlink" href="#qiime-2-for-experienced-microbiome-researchers" title="Permalink to this headline">¶</a></h1>
<div class="contents topic" id="id1">
<p class="topic-title">QIIME 2 for Experienced Microbiome Researchers</p>
<ul class="simple">
<li><p><a class="reference internal" href="#qiime-2-for-experienced-microbiome-researchers" id="id2">QIIME 2 for Experienced Microbiome Researchers</a></p>
<ul>
<li><p><a class="reference internal" href="#why-switch-to-qiime-2" id="id3">Why switch to QIIME 2?</a></p>
<ul>
<li><p><a class="reference internal" href="#pro-tips-for-power-users" id="id4">Pro-tips for power users</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#data-processing-steps" id="id5">Data processing steps</a></p>
<ul>
<li><p><a class="reference internal" href="#importing-data-into-qiime-2" id="id6">Importing data into QIIME 2</a></p></li>
<li><p><a class="reference internal" href="#demultiplexing-sequences" id="id7">Demultiplexing sequences</a></p></li>
<li><p><a class="reference internal" href="#merging-reads" id="id8">Merging reads</a></p></li>
<li><p><a class="reference internal" href="#removing-non-biological-sequences" id="id9">Removing non-biological sequences</a></p></li>
<li><p><a class="reference internal" href="#grouping-similar-sequences" id="id10">Grouping similar sequences</a></p></li>
<li><p><a class="reference internal" href="#denoising" id="id11">Denoising</a></p></li>
<li><p><a class="reference internal" href="#otu-clustering" id="id12">OTU Clustering</a></p></li>
<li><p><a class="reference internal" href="#assigning-taxonomy" id="id13">Assigning taxonomy</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#analyze-feature-table-and-gain-insight" id="id14">Analyze feature table and gain insight</a></p>
<ul>
<li><p><a class="reference internal" href="#export-the-data" id="id15">Export the data</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#new-plugins" id="id16">New plugins</a></p></li>
</ul>
</li>
</ul>
</div>
<p>In this document, we’ll go over how to use QIIME 2 to process microbiome data.
This tutorial is intended for experienced microbiome researchers who already know how to process data and need to know the <strong>QIIME 2 commands pertaining to specific steps in 16S processing</strong>.</p>
<p>The <a class="reference internal" href="../overview/"><span class="doc">QIIME 2 overview tutorial</span></a> contains a more theoretical overview of microbiome data processing.</p>
<div class="section" id="why-switch-to-qiime-2">
<h2><a class="toc-backref" href="#id3">Why switch to QIIME 2?</a><a class="headerlink" href="#why-switch-to-qiime-2" title="Permalink to this headline">¶</a></h2>
<p>Transitioning to QIIME 2 can be difficult for users who are used to processing data with their own tools and scripts, and who want fine control over every step in the process.
We understand the frustrating learning curve for experienced microbiome researchers, but believe that
the community, open-sourced nature, and commitment to reproducible science make switching to QIIME 2 worth any initial frustrations.</p>
<p>By providing a common framework for microbiome data analysis, QIIME 2 brings together a <strong>vibrant and inclusive community</strong>.
By joining the QIIME 2 community as an established microbiome researcher, you will automatically be connected to other leaders in this field and can more easily work together to propel the development and implementation of best practices in microbiome research for broad use.
The QIIME 2 community includes both established leaders in microbiome research as well as newcomers: all are encouraged to participate and learn from each other.
The <a class="reference external" href="https://forum.qiime2.org/">QIIME 2 forum</a> contains a wealth of information on how to perform microbiome data processing and analysis, as well as fruitful discussion on best practices in this space.</p>
<p>QIIME 2 also encourages making <strong>microbiome research more reproducible</strong>.
QIIME 2 reduces inappropriate analyses by defining specific data types and restricting methods only to their appropriate data type inputs.
It also keeps track of everything that’s been done to a given data file in the <a class="reference internal" href="../../concepts/"><span class="doc">provenance</span></a> metadata associated with each QIIME 2 artifact.</p>
<p>Furthermore, by wrapping tools into a common framework, <strong>data processing pipelines are streamlined</strong>.
With QIIME 2, most data processing workflows can be consolidated into one (or a few) bash scripts, reducing the number of different programs or executables you need to call and the number of re-formatting data steps that are necessary.</p>
<p>Finally, QIIME 2 is <strong>open-sourced and specifically designed for experienced researchers to contribute and expand the reach of their work.</strong>
Any tool can be added to QIIME 2 as a <a class="reference internal" href="../../plugins/"><span class="doc">plugin</span></a>, which can be written for any software, package, or other installable executables.
Writing a QIIME 2 plugin for a method that you develop instantly makes it accessible and usable by thousands of users.</p>
<div class="section" id="pro-tips-for-power-users">
<h3><a class="toc-backref" href="#id4">Pro-tips for power users</a><a class="headerlink" href="#pro-tips-for-power-users" title="Permalink to this headline">¶</a></h3>
<p>That said, here are a few tips we’ve learned that should substantially improve your experience in transitioning your workflows to QIIME 2:</p>
<p><strong>Pro-tip #1: QIIME 2 artifacts are just zip files</strong>.
If at any point you want to look at what actual files are in the <code class="docutils literal notranslate"><span class="pre">.qza</span></code> artifact, you can use <a class="reference internal" href="../exporting/"><span class="doc">qiime tools export</span></a> to extract the data file directly (which is basically just a wrapper for <code class="docutils literal notranslate"><span class="pre">unzip</span></code>).
Alternatively, you can also unzip your artifact directly (<code class="docutils literal notranslate"><span class="pre">unzip</span> <span class="pre">-k</span> <span class="pre">file.qza</span></code>) and look through the files in the <code class="docutils literal notranslate"><span class="pre">data/</span></code> folder.</p>
<p><strong>Pro-tip #2: the QIIME 2 command line interface tools are slow because they have to unzip and re-zip the data contained in the artifacts each time you call them.</strong>
If you need to process your data more interactively, you might want to use the Python API - it is much faster since objects can be simply stored in memory.
You can learn more about the different <a class="reference external" href="https://docs.qiime2.org/2018.6/interfaces/">QIIME 2 interfaces</a>.</p>
</div>
</div>
<div class="section" id="data-processing-steps">
<h2><a class="toc-backref" href="#id5">Data processing steps</a><a class="headerlink" href="#data-processing-steps" title="Permalink to this headline">¶</a></h2>
<p>The processing steps we’ll cover in this tutorial include:</p>
<ol class="arabic simple">
<li><p>Importing raw sequence (FASTQ) data into QIIME 2</p></li>
<li><p>Demultiplexing data (i.e. mapping each sequence to the sample it came from)</p></li>
<li><p>Removing non-biological parts of the sequences (i.e. primers)</p></li>
<li><p>Performing quality control and:</p>
<ul class="simple">
<li><p>denoising sequences with DADA2 or deblur, and/or</p></li>
<li><p>quality filtering, length trimming, and clustering with VSEARCH or dbOTU</p></li>
</ul>
</li>
<li><p>Assigning taxonomy</p></li>
<li><p>Analyze data and gain insight!</p></li>
</ol>
<p>The <a class="reference internal" href="../overview/"><span class="doc">overview tutorial</span></a> and <a class="reference internal" href="../../plugins/available/"><span class="doc">list of available plugins</span></a> can give you ideas for additional possible processing and analysis steps.</p>
<div class="section" id="importing-data-into-qiime-2">
<h3><a class="toc-backref" href="#id6">Importing data into QIIME 2</a><a class="headerlink" href="#importing-data-into-qiime-2" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugin</strong>: <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">import</span></code></p>
<p>If you’re using QIIME 2 to process your data, the first thing you need to do is get that data into a format that QIIME 2 can understand.
Various importing methods currently available in QIIME 2 are highlighted in the <a class="reference internal" href="../importing/"><span class="doc">QIIME 2 importing tutorial</span></a>.</p>
<p>This step has the potential to be the most confusing part of the QIIME 2 pipeline as there are dozens of import and format types to choose from.
To see a full list of available import/format types use: <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">import</span> <span class="pre">--show-importable-formats</span></code> and <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">import</span> <span class="pre">--show-importable-types</span></code></p>
<p>If you’re importing FASTQ data that you’ve generated, you’ll likely need to generate a <a class="reference internal" href="../importing/#manifest-file"><span class="std std-ref">manifest file</span></a>, which is just a text file that  maps each FASTQ file to its sample ID and direction (if applicable).</p>
<p>If you have sequencing data with one of two very specific formats (<a class="reference internal" href="../importing/#emp-import"><span class="std std-ref">EMP</span></a> or <a class="reference internal" href="../importing/#casava-import"><span class="std std-ref">Casava</span></a>), you can directly import the folder containing your sequencing files with
the <code class="docutils literal notranslate"><span class="pre">--type</span> <span class="pre">EMPSingleEndSequences</span></code> or <code class="docutils literal notranslate"><span class="pre">--type</span> <span class="pre">'SampleData[PairedEndSequencesWithQuality]'</span></code> flags (or their respective paired-end types).
Otherwise if you don’t have one of these two very specific formats, you’ll need to make the manifest file to give <code class="docutils literal notranslate"><span class="pre">import</span></code> instructions on what and how to import your files.</p>
<p>If you want to import FASTA files or a feature table directly, you can also do that by using a different <code class="docutils literal notranslate"><span class="pre">--type</span></code> flag for <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">import</span></code>.
The <a class="reference internal" href="../importing/"><span class="doc">importing tutorial</span></a> goes through all these options in detail.</p>
</div>
<div class="section" id="demultiplexing-sequences">
<h3><a class="toc-backref" href="#id7">Demultiplexing sequences</a><a class="headerlink" href="#demultiplexing-sequences" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugins</strong></p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/demux/"><span class="doc">q2-demux</span></a></p></li>
<li><p><a class="reference internal" href="../../plugins/available/cutadapt/"><span class="doc">cutadapt</span></a></p></li>
</ul>
<p>If you have reads from multiple samples in the same file, you’ll need to demultiplex your sequences.</p>
<p>If your barcodes have already been removed from the reads and are in a separate file, you can use <a class="reference internal" href="../../plugins/available/demux/emp-paired/"><span class="doc">q2-demux</span></a> to demultiplex these.</p>
<p>If your barcodes are still in your sequences, you can use functions from the <a class="reference internal" href="../../plugins/available/cutadapt/"><span class="doc">cutadapt
plugin</span></a>.
The <code class="docutils literal notranslate"><span class="pre">cutadapt</span> <span class="pre">demux-single</span></code> method looks for barcode sequences at the beginning of your reads (5’ end) with a certain error tolerance, removes them, and returns sequence data separated by each sample.
The QIIME 2 forum has a <a class="reference external" href="https://forum.qiime2.org/t/demultiplexing-and-trimming-adapters-from-reads-with-q2-cutadapt/2313">tutorial on various functions available in cutadapt</a>, including demultiplexing.
You can learn more about how <code class="docutils literal notranslate"><span class="pre">cutadapt</span></code> works under the hood by reading their <a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/index.html">documentation</a>.</p>
<p>Note: Currently <code class="docutils literal notranslate"><span class="pre">q2-demux</span></code> and <code class="docutils literal notranslate"><span class="pre">q2-cutadapt</span></code> do not support demultiplexing dual-barcoded paired-end sequences, but only can demultiplex with barcodes in the forward reads.
So for the time being, this type of demultiplexing needs to be done outside of QIIME 2 using other tools, for example
<a class="reference external" href="https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html">bcl2fastq</a>.</p>
</div>
<div class="section" id="merging-reads">
<span id="merge-reads"></span><h3><a class="toc-backref" href="#id8">Merging reads</a><a class="headerlink" href="#merging-reads" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugin</strong>: <a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch</span></a></p>
<p>Whether or not you need to merge reads depends on how you plan to cluster or denoise your sequences into amplicon sequence variants (ASVs) or operational taxonomic units (OTUs).
If you plan to use deblur or OTU clustering methods next, join your sequences now.
If you plan to use dada2 to denoise your sequences, do not merge — dada2 performs read merging automatically after denoising each sequence.</p>
<p>If you need to merge your reads, you can use the QIIME 2 <a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch plugin</span></a>  with the <a class="reference internal" href="../../plugins/available/vsearch/join-pairs/"><span class="doc">join-pairs</span></a> method.</p>
</div>
<div class="section" id="removing-non-biological-sequences">
<span id="remove-non-biological-sequences"></span><h3><a class="toc-backref" href="#id9">Removing non-biological sequences</a><a class="headerlink" href="#removing-non-biological-sequences" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugins</strong></p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/cutadapt/"><span class="doc">q2-cutadapt</span></a></p></li>
<li><p><a class="reference internal" href="../../plugins/available/dada2/"><span class="doc">dada2</span></a></p></li>
</ul>
<p>If your data contains any non-biological sequences (e.g. primers, sequencing adapters, PCR spacers, etc), you should remove these.</p>
<p>The <a class="reference internal" href="../../plugins/available/cutadapt/"><span class="doc">q2-cutadapt</span></a> plugin has comprehensive methods for removing non-biological sequences from <a class="reference internal" href="../../plugins/available/cutadapt/trim-paired/"><span class="doc">paired-end</span></a> or <a class="reference internal" href="../../plugins/available/cutadapt/trim-single/"><span class="doc">single-end</span></a> data.</p>
<p>If you’re going to use DADA2 to denoise your sequences, you can remove biological sequences at the same time as you call the denoising function.
All of DADA2’s <code class="docutils literal notranslate"><span class="pre">denoise</span></code> fuctions have some sort of <code class="docutils literal notranslate"><span class="pre">--p-trim</span></code> parameter you can specify to remove base pairs from the 5’ end of your reads.
(Deblur does not have this functionality yet.)</p>
</div>
<div class="section" id="grouping-similar-sequences">
<h3><a class="toc-backref" href="#id10">Grouping similar sequences</a><a class="headerlink" href="#grouping-similar-sequences" title="Permalink to this headline">¶</a></h3>
<p>There are two main approaches for grouping similar sequences together: denoising and clustering.
The <a class="reference internal" href="../overview/#denoising"><span class="std std-ref">overview tutorial</span></a> provides more in-depth discussion of these approaches.</p>
<p>Regardless of how you group your sequences, the grouping methods will output:</p>
<ol class="arabic simple">
<li><p>A list of representative sequences for each of your OTUs and/or ASVs (QIIME 2 data format <code class="docutils literal notranslate"><span class="pre">FeatureData[Sequence]</span></code>), and</p></li>
<li><p>A feature table which indicates how many reads of each OTU/sequence variants were observed in each sample. (QIIME 2 data format <code class="docutils literal notranslate"><span class="pre">FeatureTable[Frequency]</span></code>)</p></li>
</ol>
<p>DADA2 and deblur will also produce a stats summary file with useful information regarding the filtering and denoising.</p>
</div>
<div class="section" id="denoising">
<h3><a class="toc-backref" href="#id11">Denoising</a><a class="headerlink" href="#denoising" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugins</strong>:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/dada2/"><span class="doc">dada2</span></a></p></li>
<li><p><a class="reference internal" href="../../plugins/available/deblur/"><span class="doc">deblur</span></a></p></li>
</ul>
<p>DADA2 and deblur are currently the two denoising methods available in QIIME 2.
You can read more about the methods in the <a class="reference internal" href="../overview/#denoising"><span class="std std-ref">overview tutorial</span></a>.</p>
<p>DADA2 and deblur both output exact sequence variants, which supposedly represent the true biological sequences present in your data.
Their creators have different terminology for these sequences (DADA2 calls them “amplicon sequence variants” (ASVs) and deblur calls them “sub-OTUs”).
We’ll be using the ASV terminology throughout this tutorial to refer to both outputs.</p>
<div class="section" id="preparing-data-for-denoising">
<h4>Preparing data for denoising<a class="headerlink" href="#preparing-data-for-denoising" title="Permalink to this headline">¶</a></h4>
<p>Denoising requires little data preparation.
Both DADA2 and deblur perform quality filtering, denoising, and chimera removal, so you shouldn’t need to perform any quality screening prior to running them.
That said, the deblur developers recommend doing an initial <a class="reference internal" href="../moving-pictures/#moving-pictures-deblur"><span class="std std-ref">quality-filter</span></a> with default settings prior to using deblur (as illustrated in the <a class="reference internal" href="../moving-pictures/#moving-pictures-deblur"><span class="std std-ref">“Moving Pictures” tutorial</span></a>).
Q-score based filtering is built in to DADA2, so doing this <cite>quality-filter</cite> step prior to denoising with DADA2 is unnecessary.</p>
<p>Both methods have an option to truncate your reads to a constant length (which occurs prior to denoising).
In DADA2, this is the <cite>–p-trunc-len</cite> parameter; in deblur it’s <cite>–p-trim-length</cite>.
The truncating parameter is optional for both DADA2 and deblur (though if you’re using deblur you’ll need to specify <cite>–p-trim-length -1</cite> to disable truncation).
Reads shorter than the truncation length are discarded and reads longer are truncated at that position.
The overview tutorial has more discussion on deciding what length you should truncate to.</p>
</div>
<div class="section" id="denoising-with-dada2">
<h4>Denoising with DADA2<a class="headerlink" href="#denoising-with-dada2" title="Permalink to this headline">¶</a></h4>
<p>The <a class="reference internal" href="../../plugins/available/dada2/"><span class="doc">DADA2 plugin</span></a> has multiple methods to denoise reads:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/dada2/denoise-paired/"><span class="doc">denoise paired-end</span></a> requires unmerged, paired-end reads (i.e. both forward and reverse).</p></li>
<li><p><a class="reference internal" href="../../plugins/available/dada2/denoise-single/"><span class="doc">denoise single-end</span></a> accepts either single-end or unmerged paired-end data. If you give it unmerged paired-end data, it will only use the forward reads (and do nothing with the reverse reads).</p></li>
<li><p><a class="reference internal" href="../../plugins/available/dada2/denoise-pyro/"><span class="doc">denoise-pyro</span></a> accepts ion torrent data.</p></li>
</ul>
<p>Note that DADA2 may be slow on very large datasets. You can increase the number of threads to use with the <code class="docutils literal notranslate"><span class="pre">--p-n-threads</span></code> parameter.</p>
</div>
<div class="section" id="denoising-with-deblur">
<h4>Denoising with deblur<a class="headerlink" href="#denoising-with-deblur" title="Permalink to this headline">¶</a></h4>
<p>The <a class="reference internal" href="../../plugins/available/deblur/"><span class="doc">deblur</span></a> plugin has two methods to denoise sequences:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/deblur/denoise-16S/"><span class="doc">denoise-16S</span></a> denoises 16S sequences.</p></li>
<li><p><a class="reference internal" href="../../plugins/available/deblur/denoise-other/"><span class="doc">denoise-other</span></a> denoises other types of sequences.</p></li>
</ul>
<p>If you use <code class="docutils literal notranslate"><span class="pre">denoise-16S</span></code>, deblur performs an initial positive filtering step where it discards any reads which do not have a minimum 60% identity similarity to sequences from the 85% OTU GreenGenes database.
If you don’t want to do this step, use the <code class="docutils literal notranslate"><span class="pre">denoise-other</span></code> method.</p>
<p>deblur can currently only denoise single-end reads.
It will accept unmerged paired-end reads as input, it just won’t do anything with the reverse reads.
Note that deblur <em>can</em> take in <em>merged</em> reads and treat them as single-end reads, so you might want to merge your reads first if you’re denoising with deblur.</p>
</div>
</div>
<div class="section" id="otu-clustering">
<h3><a class="toc-backref" href="#id12">OTU Clustering</a><a class="headerlink" href="#otu-clustering" title="Permalink to this headline">¶</a></h3>
<p>In this tutorial, we’ll cover QIIME 2 methods that perform <a class="reference internal" href="../../plugins/available/vsearch/cluster-features-de-novo/"><span class="doc">de novo</span></a> and <a class="reference internal" href="../../plugins/available/vsearch/cluster-features-closed-reference/"><span class="doc">closed reference</span></a> clustering.
The QIIME <a class="reference internal" href="../otu-clustering/"><span class="doc">OTU clustering tutorial</span></a> also covers these in more detail.</p>
<p>To cluster your sequences, you need to make sure that:</p>
<ul class="simple">
<li><p>paired-end reads are merged</p></li>
<li><p>non-biological sequences are removed</p></li>
<li><p>reads are all trimmed to the same length</p></li>
<li><p>low-quality reads are discarded</p></li>
</ul>
<p>We discussed merging paired-end reads and removing non-biological sequences above (Sections <a class="reference internal" href="#merge-reads">Merge reads</a> and <a class="reference internal" href="#remove-non-biological-sequences">Remove non-biological sequences</a>).</p>
<p>Once your data are ready, you’ll also need to dereplicate your reads before clustering.</p>
<div class="section" id="length-trimming">
<h4>Length trimming<a class="headerlink" href="#length-trimming" title="Permalink to this headline">¶</a></h4>
<p>If for some reason your raw reads are not already all the same length, you’ll need to trim them to the same length before doing OTU clustering.
There isn’t currently a QIIME 2 function to trim reads to the same length without doing anything else, though you may be able to use functions from the <code class="docutils literal notranslate"><span class="pre">cutadapt</span></code> plugin to do something like that.
(The reason for this is that the <a class="reference internal" href="../overview/#denoising"><span class="std std-ref">QIIME 2 workflow</span></a> recommends first denoising reads - which involves a length trimming step - and then optionally passing the ASVs through a clustering algorithm.)</p>
</div>
<div class="section" id="quality-filtering">
<h4>Quality filtering<a class="headerlink" href="#quality-filtering" title="Permalink to this headline">¶</a></h4>
<p><strong>Relevant plugin</strong>: <a class="reference internal" href="../../plugins/available/quality-filter/"><span class="doc">quality-filter</span></a></p>
<p>You can perform different types of quality filtering with the <a class="reference internal" href="../../plugins/available/quality-filter/"><span class="doc">quality filter</span></a> plugin.
The <cite>q-score</cite> method is for single- or paired-end sequences (i.e. <cite>SampleData[PairedEndSequencesWithQuality | SequencesWithQuality]</cite>) while <cite>q-score-joined</cite> is for joined reads (i.e. <cite>SampleData[JoinedSequencesWithQuality]</cite> after merging).
The option descriptions for each method cover the different types of available quality filtering.</p>
</div>
<div class="section" id="dereplicating-sequences">
<h4>Dereplicating sequences<a class="headerlink" href="#dereplicating-sequences" title="Permalink to this headline">¶</a></h4>
<p><strong>Relevant plugin</strong>: <a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch</span></a></p>
<p>No matter which type of clustering you do, you first need to dereplicate your sequences. The <a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch</span></a> plugin’s method <a class="reference internal" href="../../plugins/available/vsearch/dereplicate-sequences/"><span class="doc">dereplicate-sequences</span></a> performs this step.</p>
</div>
<div class="section" id="de-novo-clustering">
<h4>de novo clustering<a class="headerlink" href="#de-novo-clustering" title="Permalink to this headline">¶</a></h4>
<p><strong>Relevant plugins</strong>:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch</span></a></p></li>
<li><p><a class="reference external" href="https://library.qiime2.org/plugins/q2-dbotu/">q2-dbotu</a></p></li>
</ul>
<p>Sequences can be clustered <em>de novo</em> based on their genetic similarity alone (i.e. with VSEARCH) or based on a combination of their genetic similarity and abundance distributions (i.e. with distribution-based clustering).</p>
<ul class="simple">
<li><p><strong>Similarity-based clustering.</strong> The <cite>q2-vsearch</cite> plugin method <a class="reference internal" href="../../plugins/available/vsearch/cluster-features-de-novo/"><span class="doc">cluster-features-de-novo</span></a> clusters OTUs. You can change the genetic similarity threshold with the <code class="docutils literal notranslate"><span class="pre">--p-perc-identity</span></code> parameter. The plugin wraps the VSEARCH <code class="docutils literal notranslate"><span class="pre">--cluster_size</span></code> function.</p></li>
<li><p><strong>Distribution-based clustering</strong> incorporates the similarity between sequences and their abundance distribution to identify ecologically meaningful populations. You can learn more about this method in the <a class="reference external" href="https://github.com/cduvallet/q2-dbotu">plugin documentation</a>, <a class="reference external" href="http://dx.doi.org/10.1128/AEM.00342-13">original paper</a>, and the <a class="reference external" href="https://doi.org/10.1371/journal.pone.0176335">re-implementation update paper</a>. The <code class="docutils literal notranslate"><span class="pre">call-otus</span></code> function in the <a class="reference external" href="https://github.com/cduvallet/q2-dbotu">q2-dbotu</a> plugin performs distribution-based clustering on input data.</p></li>
</ul>
<p>Both of these functions take as input the output of <code class="docutils literal notranslate"><span class="pre">q2-vsearch</span> <span class="pre">dereplicate-sequences</span></code>, which are dereplicated sequences with QIIME 2 data type <code class="docutils literal notranslate"><span class="pre">'FeatureData[Sequence]'</span></code>, and a table of counts with QIIME 2 data type <code class="docutils literal notranslate"><span class="pre">'FeatureTable[Frequency]'</span></code>.</p>
</div>
<div class="section" id="closed-reference-clustering">
<h4>closed reference clustering<a class="headerlink" href="#closed-reference-clustering" title="Permalink to this headline">¶</a></h4>
<p><strong>Relevant plugin</strong>: <a class="reference internal" href="../../plugins/available/vsearch/"><span class="doc">q2-vsearch</span></a></p>
<p>Closed reference clustering groups sequences together which match the same reference sequence in a database with a certain similarity.</p>
<p>VSEARCH can do closed reference clustering with the <a class="reference internal" href="../../plugins/available/vsearch/cluster-features-closed-reference/"><span class="doc">cluster-features-closed-reference</span></a> method.
This method wraps the <code class="docutils literal notranslate"><span class="pre">--usearch_global</span></code> VSEARCH function.
You can decide which reference database to cluster against with the <code class="docutils literal notranslate"><span class="pre">--i-reference-sequences</span></code> flag.
The input file to this flag should be a <code class="docutils literal notranslate"><span class="pre">.qza</span></code> file containing a fasta file with the sequences to use as references, with QIIME 2 data type <code class="docutils literal notranslate"><span class="pre">FeatureData[Sequence]</span></code>.
Most people use GreenGenes or SILVA for 16S rRNA gene sequences, but others curate their own databases or use other standard references (e.g. UNITE for ITS data).
You can download the references from the links on the <a class="reference internal" href="../../data-resources/#marker-gene-db"><span class="std std-ref">QIIME 2 data resources page</span></a>.
You’ll need to unzip/untar and import them as <code class="docutils literal notranslate"><span class="pre">FeatureData[Sequence]</span></code> artifacts, since they’re provided as raw data files.</p>
</div>
</div>
<div class="section" id="assigning-taxonomy">
<h3><a class="toc-backref" href="#id13">Assigning taxonomy</a><a class="headerlink" href="#assigning-taxonomy" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugin</strong>: <a class="reference internal" href="../../plugins/available/feature-classifier/"><span class="doc">feature-classifier</span></a></p>
<p>Assigning taxonomy to ASV or OTU representative sequences is covered in the <a class="reference internal" href="../overview/#taxonomy"><span class="std std-ref">taxonomy classification
tutorial</span></a>.
All taxonomy assignment methods are in the <a class="reference internal" href="../../plugins/available/feature-classifier/"><span class="doc">feature-classifier plugin</span></a>.</p>
<p>There are two main approaches for assigning taxonomy, each with multiple methods available.</p>
<p>The first involves aligning reads to reference databases directly:</p>
<ul class="simple">
<li><p><a class="reference internal" href="../../plugins/available/feature-classifier/classify-consensus-blast/"><span class="doc">classify-consensus-blast</span></a>: BLAST+ local alignment</p></li>
<li><p><a class="reference internal" href="../../plugins/available/feature-classifier/classify-consensus-vsearch/"><span class="doc">classify-consensus-vsearch</span></a>:  VSEARCH global alignment</p></li>
</ul>
<p>Both use the <em>consensus</em> approach of taxonomy assignment, which you can learn more about in the <a class="reference internal" href="../overview/#taxonomy"><span class="std std-ref">overview</span></a> and tweak with the <code class="docutils literal notranslate"><span class="pre">maxaccepts</span></code>, <code class="docutils literal notranslate"><span class="pre">perc-identity</span></code>, and <code class="docutils literal notranslate"><span class="pre">min-consensus</span></code> parameters.</p>
<p>The second approach uses a machine learning classifier to assign likely taxonomies to reads, and can be used through the <a class="reference internal" href="../../plugins/available/feature-classifier/classify-sklearn/"><span class="doc">classify-sklearn</span></a> method.</p>
<p>This method needs a pre-trained model to classify the sequences: you can either download one of the pre-trained taxonomy classifiers from the <a class="reference internal" href="../../data-resources/"><span class="doc">data resources page</span></a>, or train one yourself (following the steps outlined in the <a class="reference internal" href="../feature-classifier/"><span class="doc">feature classifier tutorial</span></a>).
(You can also learn a lot more about the specific models implemented in <a class="reference external" href="https://doi.org/10.1186/s40168-018-0470-z">the plugin’s associated paper</a>.)</p>
</div>
</div>
<div class="section" id="analyze-feature-table-and-gain-insight">
<h2><a class="toc-backref" href="#id14">Analyze feature table and gain insight</a><a class="headerlink" href="#analyze-feature-table-and-gain-insight" title="Permalink to this headline">¶</a></h2>
<p><strong>Relevant plugins</strong>: <a class="reference internal" href="../../plugins/available/"><span class="doc">Many!</span></a></p>
<p>At this point, you should be ready to analyze your feature table to answer your scientific questions!
QIIME 2 offers multiple built-in functions to analyze your data, and you can also <a class="reference external" href="Exportthedata">export</a> it to do downstream analyses in your preferred coding language or software package.</p>
<p>Some general things you can do with QIIME 2 are:</p>
<ul class="simple">
<li><p><strong>Look at the data:</strong> QIIME 2 has a nice <a class="reference internal" href="../../plugins/available/taxa/barplot/"><span class="doc">taxa barplot visualizer</span></a> to make visually exploring your data easy. You can also visualize your data on a PCoA plot with the <a class="reference internal" href="../../plugins/available/emperor/plot/"><span class="doc">emperor</span></a> plugin (after calculating beta diversity between samples).</p></li>
<li><p><strong>Build a phylogenetic tree:</strong> QIIME 2 has a <a class="reference internal" href="../../plugins/available/phylogeny/"><span class="doc">phylogeny</span></a> plugin with different tree-building methods.</p></li>
<li><p><strong>Calculate alpha diversity of your samples:</strong> the <a class="reference internal" href="../../plugins/available/diversity/"><span class="doc">diversity plugin</span></a> has many <a class="reference external" href="https://forum.qiime2.org/t/alpha-and-beta-diversity-explanations-and-commands/2282">alpha diversity metrics</a> available through the <code class="docutils literal notranslate"><span class="pre">alpha</span></code> and <code class="docutils literal notranslate"><span class="pre">alpha-phylogenetic</span></code> methods.</p></li>
<li><p><strong>Calculate beta diversity between samples:</strong> the <a class="reference internal" href="../../plugins/available/diversity/"><span class="doc">diversity plugin</span></a> also has these metrics available in the <code class="docutils literal notranslate"><span class="pre">beta</span></code>, <code class="docutils literal notranslate"><span class="pre">beta-phylogenetic</span></code>, and <code class="docutils literal notranslate"><span class="pre">beta-phylogenetic-alt</span></code> methods. Note that the <code class="docutils literal notranslate"><span class="pre">diversity</span> <span class="pre">core-metrics</span></code> and <code class="docutils literal notranslate"><span class="pre">diversity</span> <span class="pre">core-metrics-phylogenetic</span></code> pipelines are a handy wrapper for alpha and beta diversity analyses. These are described in the <a class="reference internal" href="../overview/#diversity"><span class="std std-ref">overview tutorial</span></a>.</p></li>
<li><p><strong>Test for differences between samples</strong>, through differential abundance or distribution testing: PERMANOVA, ANOSIM, ANCOM, and Gneiss are some of the relevant methods which are available in QIIME 2. PERMANOVA and ANOSIM can be done with the <a class="reference internal" href="../../plugins/available/diversity/beta-group-significance/"><span class="doc">beta-group-significance</span></a> method in the <code class="docutils literal notranslate"><span class="pre">diversity</span></code> plugin. ANCOM is available in the <a class="reference internal" href="../../plugins/available/composition/"><span class="doc">composition</span></a> plugin. Gneiss is available in the <a class="reference internal" href="../../plugins/available/gneiss/"><span class="doc">gneiss</span></a> plugin, and has an associated tutorial, <a class="reference internal" href="../gneiss/"><span class="doc">“Differential abundance  analysis with gneiss”</span></a>.</p></li>
<li><p><strong>Build machine learning classifiers and regressors to make predictions:</strong> the <a class="reference internal" href="../../plugins/available/sample-classifier/"><span class="doc">q2-sample-classifier</span></a> plugin has several actions for building classifiers and regressors, and the associated <a class="reference internal" href="../sample-classifier/"><span class="doc">“Predicting sample metadata values with q2-sample-classifier” tutorial</span></a> provides more details.</p></li>
</ul>
<div class="section" id="export-the-data">
<h3><a class="toc-backref" href="#id15">Export the data</a><a class="headerlink" href="#export-the-data" title="Permalink to this headline">¶</a></h3>
<p><strong>Relevant plugin</strong>: <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">export</span></code></p>
<p>If you’re a veteran microbiome scientist and don’t want to use QIIME 2 for your analyses, you can extract your feature table and sequences from the artifact using the <a class="reference internal" href="../exporting/"><span class="doc">export</span></a> tool.
While <code class="docutils literal notranslate"><span class="pre">export</span></code> only outputs the data, the <a class="reference internal" href="../exporting/#export-vs-extract"><span class="std std-ref">extract</span></a> tool allows you to also extract other metadata such as the citations, provenance etc.</p>
<p>Note that this places generically named files (e.g. <code class="docutils literal notranslate"><span class="pre">feature-table.txt</span></code>) into the output directory, so you may want to immediately rename the files to something more information (or somehow ensure that they stay in their original directory)!</p>
<p>You can also use the handy <a class="reference external" href="https://github.com/jbisanz/qiime2R">qiime2R</a> package to import QIIME 2 artifacts directly into R.</p>
</div>
</div>
<div class="section" id="new-plugins">
<h2><a class="toc-backref" href="#id16">New plugins</a><a class="headerlink" href="#new-plugins" title="Permalink to this headline">¶</a></h2>
<p>You can explore QIIME 2’s ever-growing list of
<a class="reference internal" href="../../plugins/available/"><span class="doc">plugins</span></a> to find other methods to apply to your data.</p>
<p>And remember that you can also <a class="reference internal" href="../../plugins/developing/"><span class="doc">make your own QIIME 2 plugins</span></a> to add functionality to QIIME 2 and share it with the community!</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">QIIME 2 for Experienced Microbiome Researchers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">“Moving Pictures” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">“Atacama soil microbiome” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pd-mice/">Parkinson’s Mouse Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../otu-clustering/">Clustering sequences into OTUs using q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../utilities/">Utilities in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../phylogeny/">Phylogenetic inference with q2-phylogeny</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>