<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Phylogenetic inference with q2-phylogeny &#8212; QIIME 2 2019.10.0 documentation</title>
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/language_data.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="index" title="Index" href="../../genindex/" />
    <link rel="search" title="Search" href="../../search/" />
    <link rel="next" title="Interfaces" href="../../interfaces/" />
    <link rel="prev" title="Clustering sequences into OTUs using q2-vsearch" href="../otu-clustering/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="phylogenetic-inference-with-q2-phylogeny">
<h1><a class="toc-backref" href="#id4">Phylogenetic inference with q2-phylogeny</a><a class="headerlink" href="#phylogenetic-inference-with-q2-phylogeny" title="Permalink to this headline">¶</a></h1>
<div class="contents topic" id="id1">
<p class="topic-title first">Phylogenetic inference with q2-phylogeny</p>
<ul class="simple">
<li><p><a class="reference internal" href="#phylogenetic-inference-with-q2-phylogeny" id="id4">Phylogenetic inference with q2-phylogeny</a></p>
<ul>
<li><p><a class="reference internal" href="#inferring-phylogenies" id="id5">Inferring phylogenies</a></p>
<ul>
<li><p><a class="reference internal" href="#sequence-alignment" id="id6">Sequence Alignment</a></p></li>
<li><p><a class="reference internal" href="#reducing-alignment-ambiguity-masking-and-reference-alignments" id="id7">Reducing alignment ambiguity: masking and reference alignments</a></p></li>
<li><p><a class="reference internal" href="#construct-a-phylogeny" id="id8">Construct a phylogeny</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#methods" id="id9">Methods</a></p>
<ul>
<li><p><a class="reference internal" href="#fasttree" id="id10">fasttree</a></p></li>
<li><p><a class="reference internal" href="#raxml" id="id11">raxml</a></p>
<ul>
<li><p><a class="reference internal" href="#perform-multiple-searches-using-raxml" id="id12">Perform multiple searches using raxml</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#raxml-rapid-bootstrap" id="id13">raxml-rapid-bootstrap</a></p></li>
<li><p><a class="reference internal" href="#iqtree" id="id14">iqtree</a></p>
<ul>
<li><p><a class="reference internal" href="#specifying-a-substitution-model" id="id15">Specifying a substitution model</a></p></li>
<li><p><a class="reference internal" href="#single-branch-tests" id="id16">Single branch tests</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#iqtree-ultrafast-bootstrap" id="id17">iqtree-ultrafast-bootstrap</a></p>
<ul>
<li><p><a class="reference internal" href="#perform-single-branch-tests-alongside-ufboot" id="id18">Perform single branch tests alongside ufboot</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#root-the-phylogeny" id="id19">Root the phylogeny</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#pipelines" id="id20">Pipelines</a></p></li>
</ul>
</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tutorial assumes, you’ve read through the <a class="reference internal" href="../overview/"><span class="doc">QIIME 2 Overview</span></a> documentation and have at least worked through some of the other
<a class="reference internal" href="../"><span class="doc">Tutorials</span></a>.</p>
</div>
<div class="section" id="inferring-phylogenies">
<h2><a class="toc-backref" href="#id5">Inferring phylogenies</a><a class="headerlink" href="#inferring-phylogenies" title="Permalink to this headline">¶</a></h2>
<p>Several downstream diversity metrics, available within QIIME 2, require that a
phylogenetic tree be constructed using the Operational Taxonomic Units
(<a class="reference external" href="https://en.wikipedia.org/wiki/Operational_taxonomic_unit">OTUs</a>) or Exact Sequence Variants (<a class="reference external" href="https://doi.org/10.1038/ismej.2019.119">ESVs</a>) being investigated.</p>
<p><em>But how do we proceed to construct a phylogeny from our sequence data?</em></p>
<p>Well, there are two phylogeny-based approaches we can use. Deciding upon which
to use is largely dependent on your study questions:</p>
<p>1. A reference-based <a class="reference external" href="https://doi.org/10.1128/mSystems.00021-18">fragment insertion</a> approach. Which, is likely the
ideal choice. Especially, if your reference phylogeny (and associated
representative sequences) encompass neighboring relatives of which your
sequences can be reliably inserted. Any sequences that do not match well enough
to the reference are not inserted. For example, this approach may not work well
if your data contain sequences that are not well represented within your
reference phylogeny (e.g. missing clades, etc.). For more information, check
out these great <a class="reference external" href="https://library.qiime2.org/plugins/q2-fragment-insertion/16/">fragment insertion examples</a>.</p>
<p>2. A <em>de novo</em> approach. Marker genes that can be globally aligned across
divergent taxa, are usually amenable to sequence alignment and phylogenetic
investigation through this approach. Be mindful of the length of your
sequences when constructing a <em>de novo</em> phylogeny, short reads many not have
enough phylogenetic information to capture a meaningful phylogeny. This
community tutorial will focus on the <em>de novo</em> approaches.</p>
<p>Here, you will learn how to make use of <em>de novo</em> phylogenetic approaches
to:</p>
<ol class="arabic simple">
<li><p>generate a sequence alignment within QIIME 2</p></li>
<li><p>mask the alignment if needed</p></li>
<li><p>construct a phylogenetic tree</p></li>
<li><p>root the phylogenetic tree</p></li>
</ol>
<p>If you would like to substitute any of the steps outlined here by making use of
tools external to QIIME 2, please see the <a class="reference internal" href="../importing/"><span class="doc">import</span></a>,
<a class="reference internal" href="../exporting/"><span class="doc">export</span></a>, and <a class="reference internal" href="../filtering/"><span class="doc">filtering</span></a> documentation
where appropriate.</p>
<div class="section" id="sequence-alignment">
<h3><a class="toc-backref" href="#id6">Sequence Alignment</a><a class="headerlink" href="#sequence-alignment" title="Permalink to this headline">¶</a></h3>
<p>Prior to constructing a <a class="reference external" href="https://simple.wikipedia.org/wiki/Phylogeny">phylogeny</a> we must generate a multiple sequence
alignment (<a class="reference external" href="https://en.wikipedia.org/wiki/Multiple_sequence_alignment">MSA</a>). When constructing a MSA we are making a statement about the
<a class="reference external" href="http://doi.org/10.1006/mpev.2000.0785">putative homology</a> of the aligned residues (columns of the MSA) by virtue of
their <a class="reference external" href="http://doi.org/10.1002/0471250953.bi0301s42">sequence similarity</a>.</p>
<p>The number of algorithms to construct a MSA are legion. We will make use of
<a class="reference external" href="https://en.wikipedia.org/wiki/MAFFT">MAFFT</a> (Multiple Alignment using Fast Fourier Transform)) via the
<a class="reference internal" href="../../plugins/available/alignment/"><span class="doc">q2-alignment</span></a> plugin. For more
information checkout the <a class="reference external" href="http://doi.org/10.1093/molbev/mst010">MAFFT paper</a>.</p>
<p>Let’s start by creating a directory to work in:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir qiime2-phylogeny-tutorial
<span class="nb">cd</span> qiime2-phylogeny-tutorial
</pre></div>
</div>
<p>Next, download the data:</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2019.10/tutorials/phylogeny/rep-seqs.qza">
            https://data.qiime2.org/2019.10/tutorials/phylogeny/rep-seqs.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> rep-seqs.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "rep-seqs.qza" \
  "https://data.qiime2.org/2019.10/tutorials/phylogeny/rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2019.10/tutorials/phylogeny/rep-seqs.qza" > \
  "rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p><strong>Run MAFFT</strong></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime alignment mafft <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --o-alignment aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/rep-seqs.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Faligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/aligned-rep-seqs.qza">download</a></p></li>
</ul>
</div>
</div>
<div class="section" id="reducing-alignment-ambiguity-masking-and-reference-alignments">
<h3><a class="toc-backref" href="#id7">Reducing alignment ambiguity: masking and reference alignments</a><a class="headerlink" href="#reducing-alignment-ambiguity-masking-and-reference-alignments" title="Permalink to this headline">¶</a></h3>
<p><em>Why mask an alignment?</em></p>
<p>Masking helps to eliminate alignment columns that are phylogenetically
uninformative or misleading before phylogenetic analysis. Much of the time
alignment errors can introduce noise and confound phylogenetic inference. It is
common practice to mask (remove) these ambiguously aligned regions prior to
performing phylogenetic inference. In particular, <a class="reference external" href="http://www.worldcat.org/title/nucleic-acid-techniques-in-bacterial-systematics/oclc/22310197">David Lane’s (1991)</a>
chapter <a class="reference external" href="http://catdir.loc.gov/catdir/toc/onix05/90012998.html">16S/23S rRNA sequencing</a> proposed masking SSU data prior to
phylogenetic analysis. However, knowing how to deal with ambiguously aligned
regions and when to apply masks largely depends on the marker genes being
analyzed and the question being asked of the data.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Keep in mind that this is still an active area of discussion, as
highlighted by the following non-exhaustive list of articles: <a class="reference external" href="https://doi.org/10.1371/journal.pone.0030288">Wu <em>et al</em>. 2012</a>, <a class="reference external" href="https://doi.org/10.1093/sysbio/syy036">Ashkenazy <em>et al</em>. 2018</a>, <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1000844">Schloss 2010</a>, <a class="reference external" href="https://doi.org/10.1093/sysbio/syv033">Tan <em>et al</em>. 2015</a>,
<a class="reference external" href="https://doi.org/10.1093/molbev/mss264">Rajan 2015</a>.</p>
</div>
<p><em>How to mask alignment.</em></p>
<p>For our purposes, we’ll assume that we have ambiguously aligned columns in the
MAFFT alignment we produced above. The default settings for the
<code class="docutils literal notranslate"><span class="pre">--p-min-conservation</span></code> of the
<a class="reference internal" href="../../plugins/available/alignment/mask/"><span class="doc">alignment mask</span></a> approximates the
Lane mask filtering of QIIME 1. Keep an eye out for updates to the alignment
plugin.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime alignment mask <span class="se">\</span>
  --i-alignment aligned-rep-seqs.qza <span class="se">\</span>
  --o-masked-alignment masked-aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">masked-aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fmasked-aligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/masked-aligned-rep-seqs.qza">download</a></p></li>
</ul>
</div>
<p><em>Reference based alignments</em></p>
<p>There are a variety of tools such as <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btp636">PyNAST</a>) (using <a class="reference external" href="https://doi.org/10.1093/nar/gkl244">NAST</a>), <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btt509">Infernal</a>,
and <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts252">SINA</a>, etc., that attempt to reduce the amount of ambiguously aligned
regions by using curated reference alignments (e.g. <a class="reference external" href="https://www.arb-silva.de/">SILVA</a>. Reference
alignments are particularly powerful for rRNA gene sequence data, as knowledge
of secondary structure is incorporated into the curation process, thus
increasing alignment quality. For a more in-depth and eloquent overview of
reference-based alignment approaches, check out the great <a class="reference external" href="https://forum.qiime2.org/t/q2-alignment-reference-based-alignment-using-sina/6220">SINA community
tutorial</a>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Alignments constructed using reference based alignment approaches can
be masked too, just like the above MAFFT example. Also, the reference
alignment approach we are discussing here is distinct from the reference
phylogeny approach (i.e.
<a class="reference internal" href="../../plugins/available/fragment-insertion/"><span class="doc">q2-fragment-insertion</span></a>)
we mentioned earlier. That is, we are not inserting our data into an
existing tree, but simply trying to create a more robust alignment for
making a better <em>de novo</em> phylogeny.</p>
</div>
</div>
<div class="section" id="construct-a-phylogeny">
<h3><a class="toc-backref" href="#id8">Construct a phylogeny</a><a class="headerlink" href="#construct-a-phylogeny" title="Permalink to this headline">¶</a></h3>
<p>As with MSA algorithms, phylogenetic inference tools are also legion.
Fortunately, there are many great resources to learn about phylogentics. Below
are just a few introductory resources to get you started:</p>
<ol class="arabic simple">
<li><p><a class="reference external" href="http://doi.org/10.1016/S0168-9525(03)00112-4">Phylogeny for the faint of heart - a tutorial</a></p></li>
<li><p><a class="reference external" href="http://dx.doi.org/10.1038/nrg3186">Molecular phylogenetics - principles and practice</a></p></li>
<li><p><a class="reference external" href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics">Phylogenetics - An Introduction</a></p></li>
</ol>
<p>There are several methods / pipelines available through the
<a class="reference internal" href="../../plugins/available/phylogeny/"><span class="doc">q2-phylogeny</span></a> plugin of :qiime2:.
These are based on the following tools:</p>
<ol class="arabic simple">
<li><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0009490">FastTree</a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu033">RAxML</a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/molbev/msu300">IQ-TREE</a></p></li>
</ol>
</div>
</div>
<div class="section" id="methods">
<h2><a class="toc-backref" href="#id9">Methods</a><a class="headerlink" href="#methods" title="Permalink to this headline">¶</a></h2>
<div class="section" id="fasttree">
<h3><a class="toc-backref" href="#id10">fasttree</a><a class="headerlink" href="#fasttree" title="Permalink to this headline">¶</a></h3>
<p>FastTree is able to construct phylogenies from large sequence alignments quite
rapidly. It does this by using the using a <a class="reference external" href="https://doi.org/10.1109/IPDPS.2006.1639535">CAT-like</a> rate category
approximation, which is also available through RAxML (discussed below). Check
out the <a class="reference external" href="http://www.microbesonline.org/fasttree/">FastTree online manual</a> for more information.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny fasttree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree fasttree-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">fasttree-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Ffasttree-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/fasttree-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For an easy and direct way to view your <code class="docutils literal notranslate"><span class="pre">tree.qza</span></code> files, upload
them to <a class="reference external" href="https://itol.embl.de/">iTOL</a>. Here, you can interactively view and manipulate your
phylogeny. Even better, while viewing the tree topology in “Normal mode”,
you can drag and drop your associated <code class="docutils literal notranslate"><span class="pre">alignment.qza</span></code> (the one you used to
build the phylogeny) or a relevent <code class="docutils literal notranslate"><span class="pre">taxonomy.qza</span></code> file onto the iTOL tree
visualization. This will allow you to directly view the sequence alignment
or taxonomy alongside the phylogeny. 🕶️</p>
</div>
</div>
<div class="section" id="raxml">
<h3><a class="toc-backref" href="#id11">raxml</a><a class="headerlink" href="#raxml" title="Permalink to this headline">¶</a></h3>
<p>Like <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>, <code class="docutils literal notranslate"><span class="pre">raxml</span></code> will perform a single phylogentic inference and
return a tree. Note, the default model for <code class="docutils literal notranslate"><span class="pre">raxml</span></code> is
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span> <span class="pre">GTRGAMMA</span></code>. If you’d like to construct a tree using
the CAT model like <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>, simply replace <code class="docutils literal notranslate"><span class="pre">GTRGAMMA</span></code> with <code class="docutils literal notranslate"><span class="pre">GTRCAT</span></code> as
shown below:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --o-tree raxml-cat-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid hill-climbing mode

Using 1 distinct models/data partitions with joint branch length optimization


Executing 1 inferences on the original alignment using 1 distinct randomized MP trees

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -m GTRCAT -p 6817 -N 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-30edg2qt/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpmcuswmfx -n q2 


Partition: 0 with name: No Name Provided
Base frequencies: 0.243 0.182 0.319 0.256 

Inference[0]: Time 0.485824 CAT-based likelihood -1258.134859, best rearrangement setting 5


Conducting final model optimizations on all 1 trees under GAMMA-based models ....

Inference[0] final GAMMA-based Likelihood: -1387.511656 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpmcuswmfx/RAxML_result.q2


Starting final GAMMA-based thorough Optimization on tree 0 likelihood -1387.511656 .... 

Final GAMMA-based Score of best tree -1387.321440

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpmcuswmfx/RAxML_info.q2
Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpmcuswmfx/RAxML_bestTree.q2

Overall execution time: 1.051118 secs or 0.000292 hours or 0.000012 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -m GTRCAT -p 6817 -N 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-30edg2qt/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpmcuswmfx -n q2

Saved Phylogeny[Unrooted] to: raxml-cat-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/raxml-cat-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="perform-multiple-searches-using-raxml">
<h4><a class="toc-backref" href="#id12">Perform multiple searches using raxml</a><a class="headerlink" href="#perform-multiple-searches-using-raxml" title="Permalink to this headline">¶</a></h4>
<p>If you’d like to perform a more thorough search of “tree space” you can
instruct <code class="docutils literal notranslate"><span class="pre">raxml</span></code> to perform multiple independent searches on the full
alignment by using <code class="docutils literal notranslate"><span class="pre">--p-n-searches</span> <span class="pre">5</span></code>. Once these 5 independent searches are
completed, only the single best scoring tree will be returned. <em>Note, we are
not bootstrapping here, we’ll do that in a later example.</em> Let’s set
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span> <span class="pre">GTRCAT</span></code>. Finally, let’s also manually set a seed via
<code class="docutils literal notranslate"><span class="pre">--p-seed</span></code>. By setting our seed, we allow other users the ability to
reproduce our phylogeny. That is, anyone using the same sequence alignment and
substitution model, will generate the same tree as long as they set the same
seed value. Although, <code class="docutils literal notranslate"><span class="pre">--p-seed</span></code> is not a required argument, it is generally
a good idea to set this value.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --p-seed <span class="m">1723</span> <span class="se">\</span>
  --p-n-searches <span class="m">5</span> <span class="se">\</span>
  --o-tree raxml-cat-searches-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid hill-climbing mode

Using 1 distinct models/data partitions with joint branch length optimization


Executing 5 inferences on the original alignment using 5 distinct randomized MP trees

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -m GTRCAT -p 1723 -N 5 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-nopa3msr/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci -n q2 


Partition: 0 with name: No Name Provided
Base frequencies: 0.243 0.182 0.319 0.256 

Inference[0]: Time 0.616283 CAT-based likelihood -1238.242991, best rearrangement setting 5
Inference[1]: Time 0.506815 CAT-based likelihood -1249.502284, best rearrangement setting 5
Inference[2]: Time 0.547314 CAT-based likelihood -1242.978035, best rearrangement setting 5
Inference[3]: Time 0.657815 CAT-based likelihood -1243.159855, best rearrangement setting 5
Inference[4]: Time 0.500320 CAT-based likelihood -1261.321621, best rearrangement setting 5


Conducting final model optimizations on all 5 trees under GAMMA-based models ....

Inference[0] final GAMMA-based Likelihood: -1388.324037 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_result.q2.RUN.0
Inference[1] final GAMMA-based Likelihood: -1392.813982 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_result.q2.RUN.1
Inference[2] final GAMMA-based Likelihood: -1388.073642 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_result.q2.RUN.2
Inference[3] final GAMMA-based Likelihood: -1387.945266 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_result.q2.RUN.3
Inference[4] final GAMMA-based Likelihood: -1387.557031 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_result.q2.RUN.4


Starting final GAMMA-based thorough Optimization on tree 4 likelihood -1387.557031 .... 

Final GAMMA-based Score of best tree -1387.385075

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_info.q2
Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci/RAxML_bestTree.q2

Overall execution time: 3.586809 secs or 0.000996 hours or 0.000042 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -m GTRCAT -p 1723 -N 5 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-nopa3msr/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwrbcnmci -n q2

Saved Phylogeny[Unrooted] to: raxml-cat-searches-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-searches-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-searches-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/raxml-cat-searches-tree.qza">download</a></p></li>
</ul>
</div>
</div>
</div>
<div class="section" id="raxml-rapid-bootstrap">
<h3><a class="toc-backref" href="#id13">raxml-rapid-bootstrap</a><a class="headerlink" href="#raxml-rapid-bootstrap" title="Permalink to this headline">¶</a></h3>
<p>In phylogenetics, it is good practice to check how well the <a class="reference external" href="https://en.wikipedia.org/wiki/Split_(phylogenetics)">splits /
bipartitions</a> in your phylogeny are supported. Often one is interested in
which clades are robustly separated from other clades in the phylogeny. One
way, of doing this is via bootstrapping (See the <em>Bootstrapping</em> section of the
first introductory link above). In QIIME 2, we’ve provided access to the RAxML
<a class="reference external" href="http://dx.doi.org/10.1080/10635150802429642">rapid bootstrap</a> feature. The only difference between this command and the
previous are the additional flags <code class="docutils literal notranslate"><span class="pre">--p-bootstrap-replicates</span></code> and
<code class="docutils literal notranslate"><span class="pre">--p-rapid-bootstrap-seed</span></code>. It is quite common to perform anywhere from 100 -
1000 bootstrap replicates. The <code class="docutils literal notranslate"><span class="pre">--p-rapid-bootstrap-seed</span></code> works very much
like the <code class="docutils literal notranslate"><span class="pre">--p-seed</span></code> argument from above except that it allows anyone to
reproduce the bootstrapping process and the associated supports for your
splits.</p>
<p>As per the <a class="reference external" href="https://sco.h-its.org/exelixis/web/software/raxml/hands_on.html">RAxML online documentation</a> and the <a class="reference external" href="https://sco.h-its.org/exelixis/resource/download/NewManual.pdf">RAxML manual</a>, the rapid
bootstrapping command that we will execute below will do the following:</p>
<ol class="arabic simple">
<li><p>Bootstrap the input alignment 100 times and perform a Maximum Likelihood
(ML) search on each.</p></li>
<li><p>Find best scoring ML tree through multiple independent searches using the
original input alignment. The number of independent searches is determined
by the number of bootstrap replicates set in the 1st step. That is, your
search becomes more thorough with increasing bootstrap replicates. The ML
optimization of RAxML uses every 5th bootstrap tree as the starting tree for
an ML search on the original alignment.</p></li>
<li><p>Map the bipartitions (bootstrap supports, 1st step) onto the best scoring ML
tree (2nd step).</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml-rapid-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-seed <span class="m">1723</span> <span class="se">\</span>
  --p-rapid-bootstrap-seed <span class="m">9384</span> <span class="se">\</span>
  --p-bootstrap-replicates <span class="m">100</span> <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --o-tree raxml-cat-bootstrap-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid bootstrapping and subsequent ML search

Using 1 distinct models/data partitions with joint branch length optimization



Executing 100 rapid bootstrap inferences and thereafter a thorough ML search 

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -f a -m GTRCAT -p 1723 -x 9384 -N 100 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-rqm281x4/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr -n q2bootstrap 



Time for BS model parameter optimization 0.051422
Bootstrap[0]: Time 0.180020 seconds, bootstrap likelihood -1199.758796, best rearrangement setting 12
Bootstrap[1]: Time 0.123928 seconds, bootstrap likelihood -1344.229251, best rearrangement setting 6
Bootstrap[2]: Time 0.108201 seconds, bootstrap likelihood -1295.343000, best rearrangement setting 8
Bootstrap[3]: Time 0.094177 seconds, bootstrap likelihood -1273.768320, best rearrangement setting 8
Bootstrap[4]: Time 0.105058 seconds, bootstrap likelihood -1253.402952, best rearrangement setting 6
Bootstrap[5]: Time 0.118345 seconds, bootstrap likelihood -1260.866113, best rearrangement setting 10
Bootstrap[6]: Time 0.112941 seconds, bootstrap likelihood -1293.636299, best rearrangement setting 14
Bootstrap[7]: Time 0.101680 seconds, bootstrap likelihood -1227.178693, best rearrangement setting 6
Bootstrap[8]: Time 0.110852 seconds, bootstrap likelihood -1321.820787, best rearrangement setting 13
Bootstrap[9]: Time 0.117767 seconds, bootstrap likelihood -1147.233446, best rearrangement setting 6
Bootstrap[10]: Time 0.085118 seconds, bootstrap likelihood -1220.766493, best rearrangement setting 13
Bootstrap[11]: Time 0.124581 seconds, bootstrap likelihood -1200.006355, best rearrangement setting 8
Bootstrap[12]: Time 0.132835 seconds, bootstrap likelihood -1346.392834, best rearrangement setting 14
Bootstrap[13]: Time 0.107955 seconds, bootstrap likelihood -1301.111096, best rearrangement setting 14
Bootstrap[14]: Time 0.114613 seconds, bootstrap likelihood -1262.253559, best rearrangement setting 11
Bootstrap[15]: Time 0.114745 seconds, bootstrap likelihood -1215.017551, best rearrangement setting 14
Bootstrap[16]: Time 0.108195 seconds, bootstrap likelihood -1238.832009, best rearrangement setting 7
Bootstrap[17]: Time 0.100415 seconds, bootstrap likelihood -1393.989732, best rearrangement setting 12
Bootstrap[18]: Time 0.104471 seconds, bootstrap likelihood -1173.921002, best rearrangement setting 15
Bootstrap[19]: Time 0.108229 seconds, bootstrap likelihood -1185.726976, best rearrangement setting 11
Bootstrap[20]: Time 0.096617 seconds, bootstrap likelihood -1158.491940, best rearrangement setting 6
Bootstrap[21]: Time 0.093788 seconds, bootstrap likelihood -1154.664272, best rearrangement setting 11
Bootstrap[22]: Time 0.104610 seconds, bootstrap likelihood -1244.159837, best rearrangement setting 10
Bootstrap[23]: Time 0.124368 seconds, bootstrap likelihood -1211.171036, best rearrangement setting 15
Bootstrap[24]: Time 0.105563 seconds, bootstrap likelihood -1261.440677, best rearrangement setting 12
Bootstrap[25]: Time 0.106259 seconds, bootstrap likelihood -1331.836715, best rearrangement setting 15
Bootstrap[26]: Time 0.109827 seconds, bootstrap likelihood -1129.144509, best rearrangement setting 5
Bootstrap[27]: Time 0.137045 seconds, bootstrap likelihood -1226.624056, best rearrangement setting 7
Bootstrap[28]: Time 0.135334 seconds, bootstrap likelihood -1221.046176, best rearrangement setting 12
Bootstrap[29]: Time 0.088339 seconds, bootstrap likelihood -1211.791204, best rearrangement setting 14
Bootstrap[30]: Time 0.106728 seconds, bootstrap likelihood -1389.442380, best rearrangement setting 5
Bootstrap[31]: Time 0.112468 seconds, bootstrap likelihood -1303.638592, best rearrangement setting 12
Bootstrap[32]: Time 0.119762 seconds, bootstrap likelihood -1172.859456, best rearrangement setting 12
Bootstrap[33]: Time 0.105989 seconds, bootstrap likelihood -1244.617135, best rearrangement setting 9
Bootstrap[34]: Time 0.101197 seconds, bootstrap likelihood -1211.871717, best rearrangement setting 15
Bootstrap[35]: Time 0.121801 seconds, bootstrap likelihood -1299.862912, best rearrangement setting 5
Bootstrap[36]: Time 0.097434 seconds, bootstrap likelihood -1141.967505, best rearrangement setting 5
Bootstrap[37]: Time 0.121257 seconds, bootstrap likelihood -1283.923198, best rearrangement setting 12
Bootstrap[38]: Time 0.097948 seconds, bootstrap likelihood -1304.250946, best rearrangement setting 5
Bootstrap[39]: Time 0.089179 seconds, bootstrap likelihood -1407.084376, best rearrangement setting 15
Bootstrap[40]: Time 0.110386 seconds, bootstrap likelihood -1277.946299, best rearrangement setting 13
Bootstrap[41]: Time 0.105007 seconds, bootstrap likelihood -1279.006200, best rearrangement setting 7
Bootstrap[42]: Time 0.104400 seconds, bootstrap likelihood -1160.274606, best rearrangement setting 6
Bootstrap[43]: Time 0.126724 seconds, bootstrap likelihood -1216.079259, best rearrangement setting 14
Bootstrap[44]: Time 0.098282 seconds, bootstrap likelihood -1382.278311, best rearrangement setting 8
Bootstrap[45]: Time 0.110563 seconds, bootstrap likelihood -1099.004439, best rearrangement setting 11
Bootstrap[46]: Time 0.089856 seconds, bootstrap likelihood -1296.527478, best rearrangement setting 8
Bootstrap[47]: Time 0.138094 seconds, bootstrap likelihood -1291.322658, best rearrangement setting 9
Bootstrap[48]: Time 0.087200 seconds, bootstrap likelihood -1161.908080, best rearrangement setting 6
Bootstrap[49]: Time 0.117201 seconds, bootstrap likelihood -1257.348428, best rearrangement setting 13
Bootstrap[50]: Time 0.135139 seconds, bootstrap likelihood -1309.422533, best rearrangement setting 13
Bootstrap[51]: Time 0.098555 seconds, bootstrap likelihood -1197.633097, best rearrangement setting 11
Bootstrap[52]: Time 0.111778 seconds, bootstrap likelihood -1347.123005, best rearrangement setting 8
Bootstrap[53]: Time 0.099514 seconds, bootstrap likelihood -1234.934890, best rearrangement setting 14
Bootstrap[54]: Time 0.118527 seconds, bootstrap likelihood -1227.092434, best rearrangement setting 6
Bootstrap[55]: Time 0.123914 seconds, bootstrap likelihood -1280.635747, best rearrangement setting 7
Bootstrap[56]: Time 0.099109 seconds, bootstrap likelihood -1225.911449, best rearrangement setting 6
Bootstrap[57]: Time 0.089101 seconds, bootstrap likelihood -1236.213347, best rearrangement setting 11
Bootstrap[58]: Time 0.134924 seconds, bootstrap likelihood -1393.245723, best rearrangement setting 14
Bootstrap[59]: Time 0.106722 seconds, bootstrap likelihood -1212.039371, best rearrangement setting 6
Bootstrap[60]: Time 0.093722 seconds, bootstrap likelihood -1248.692011, best rearrangement setting 10
Bootstrap[61]: Time 0.109004 seconds, bootstrap likelihood -1172.820979, best rearrangement setting 13
Bootstrap[62]: Time 0.129389 seconds, bootstrap likelihood -1126.745788, best rearrangement setting 14
Bootstrap[63]: Time 0.099084 seconds, bootstrap likelihood -1267.434444, best rearrangement setting 12
Bootstrap[64]: Time 0.095805 seconds, bootstrap likelihood -1340.680748, best rearrangement setting 5
Bootstrap[65]: Time 0.097272 seconds, bootstrap likelihood -1072.671059, best rearrangement setting 5
Bootstrap[66]: Time 0.123355 seconds, bootstrap likelihood -1234.294838, best rearrangement setting 8
Bootstrap[67]: Time 0.119166 seconds, bootstrap likelihood -1109.249439, best rearrangement setting 15
Bootstrap[68]: Time 0.093235 seconds, bootstrap likelihood -1314.493588, best rearrangement setting 8
Bootstrap[69]: Time 0.095402 seconds, bootstrap likelihood -1173.850035, best rearrangement setting 13
Bootstrap[70]: Time 0.107129 seconds, bootstrap likelihood -1231.066465, best rearrangement setting 10
Bootstrap[71]: Time 0.101057 seconds, bootstrap likelihood -1146.861379, best rearrangement setting 9
Bootstrap[72]: Time 0.087689 seconds, bootstrap likelihood -1148.753369, best rearrangement setting 8
Bootstrap[73]: Time 0.100889 seconds, bootstrap likelihood -1333.374056, best rearrangement setting 9
Bootstrap[74]: Time 0.090034 seconds, bootstrap likelihood -1259.382378, best rearrangement setting 5
Bootstrap[75]: Time 0.097727 seconds, bootstrap likelihood -1319.944496, best rearrangement setting 6
Bootstrap[76]: Time 0.110285 seconds, bootstrap likelihood -1309.042165, best rearrangement setting 14
Bootstrap[77]: Time 0.134228 seconds, bootstrap likelihood -1232.061289, best rearrangement setting 8
Bootstrap[78]: Time 0.114384 seconds, bootstrap likelihood -1261.333984, best rearrangement setting 9
Bootstrap[79]: Time 0.113600 seconds, bootstrap likelihood -1194.644341, best rearrangement setting 13
Bootstrap[80]: Time 0.102618 seconds, bootstrap likelihood -1214.037389, best rearrangement setting 9
Bootstrap[81]: Time 0.110836 seconds, bootstrap likelihood -1224.527657, best rearrangement setting 8
Bootstrap[82]: Time 0.127031 seconds, bootstrap likelihood -1241.464826, best rearrangement setting 11
Bootstrap[83]: Time 0.097515 seconds, bootstrap likelihood -1230.730558, best rearrangement setting 6
Bootstrap[84]: Time 0.100665 seconds, bootstrap likelihood -1219.034592, best rearrangement setting 10
Bootstrap[85]: Time 0.111387 seconds, bootstrap likelihood -1280.071994, best rearrangement setting 8
Bootstrap[86]: Time 0.092283 seconds, bootstrap likelihood -1444.747777, best rearrangement setting 9
Bootstrap[87]: Time 0.094826 seconds, bootstrap likelihood -1245.890035, best rearrangement setting 14
Bootstrap[88]: Time 0.110137 seconds, bootstrap likelihood -1287.832766, best rearrangement setting 7
Bootstrap[89]: Time 0.130631 seconds, bootstrap likelihood -1325.245976, best rearrangement setting 5
Bootstrap[90]: Time 0.116055 seconds, bootstrap likelihood -1227.883697, best rearrangement setting 5
Bootstrap[91]: Time 0.116502 seconds, bootstrap likelihood -1273.489392, best rearrangement setting 8
Bootstrap[92]: Time 0.048078 seconds, bootstrap likelihood -1234.725870, best rearrangement setting 7
Bootstrap[93]: Time 0.119031 seconds, bootstrap likelihood -1235.733064, best rearrangement setting 11
Bootstrap[94]: Time 0.097571 seconds, bootstrap likelihood -1204.319488, best rearrangement setting 15
Bootstrap[95]: Time 0.095067 seconds, bootstrap likelihood -1183.328582, best rearrangement setting 11
Bootstrap[96]: Time 0.111702 seconds, bootstrap likelihood -1196.298898, best rearrangement setting 13
Bootstrap[97]: Time 0.118947 seconds, bootstrap likelihood -1339.251746, best rearrangement setting 12
Bootstrap[98]: Time 0.045710 seconds, bootstrap likelihood -1404.363552, best rearrangement setting 7
Bootstrap[99]: Time 0.058280 seconds, bootstrap likelihood -1270.157811, best rearrangement setting 7


Overall Time for 100 Rapid Bootstraps 10.784172 seconds
Average Time per Rapid Bootstrap 0.107842 seconds

Starting ML Search ...

Fast ML optimization finished

Fast ML search Time: 4.332645 seconds

Slow ML Search 0 Likelihood: -1387.994678
Slow ML Search 1 Likelihood: -1387.994678
Slow ML Search 2 Likelihood: -1387.994676
Slow ML Search 3 Likelihood: -1387.994650
Slow ML Search 4 Likelihood: -1387.994685
Slow ML Search 5 Likelihood: -1388.092954
Slow ML Search 6 Likelihood: -1388.182551
Slow ML Search 7 Likelihood: -1388.182563
Slow ML Search 8 Likelihood: -1388.182547
Slow ML Search 9 Likelihood: -1387.994723
Slow ML optimization finished

Slow ML search Time: 2.209532 seconds
Thorough ML search Time: 0.582538 seconds

Final ML Optimization Likelihood: -1387.204993

Model Information:

Model Parameters of Partition 0, Name: No Name Provided, Type of Data: DNA
alpha: 1.227800
Tree-Length: 7.823400
rate A &lt;-&gt; C: 0.332564
rate A &lt;-&gt; G: 2.312784
rate A &lt;-&gt; T: 2.215466
rate C &lt;-&gt; G: 1.243321
rate C &lt;-&gt; T: 3.278770
rate G &lt;-&gt; T: 1.000000

freq pi(A): 0.243216
freq pi(C): 0.181967
freq pi(G): 0.319196
freq pi(T): 0.255621


ML search took 7.131437 secs or 0.001981 hours

Combined Bootstrap and ML search took 17.915850 secs or 0.004977 hours

Drawing Bootstrap Support Values on best-scoring ML tree ...



Found 1 tree in File /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bestTree.q2bootstrap



Found 1 tree in File /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bestTree.q2bootstrap

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_info.q2bootstrap
All 100 bootstrapped trees written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bootstrap.q2bootstrap

Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bestTree.q2bootstrap

Best-scoring ML tree with support values written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bipartitions.q2bootstrap

Best-scoring ML tree with support values as branch labels written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr/RAxML_bipartitionsBranchLabels.q2bootstrap

Overall execution time for full ML analysis: 17.928981 secs or 0.004980 hours or 0.000208 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -f a -m GTRCAT -p 1723 -x 9384 -N 100 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-rqm281x4/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpp9_g7icr -n q2bootstrap

Saved Phylogeny[Unrooted] to: raxml-cat-bootstrap-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-bootstrap-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-bootstrap-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/raxml-cat-bootstrap-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>Optimizing RAxML Run Time.</strong>
You may gave noticed that we haven’t added the flag <code class="docutils literal notranslate"><span class="pre">--p-raxml-version</span></code> to
the RAxML methods. This parameter provides a means to access versions of
RAxML that have optimized vector instructions for various modern x86
processor architectures. Paraphrased from the RAxML manual and help
documentation: Firstly, most recent processors will support SSE3 vector
instructions (i.e. will likely support the faster AVX2 vector instructions).
Secondly, these instructions will substantially accelerate the likelihood
and parsimony computations. In general, SSE3 versions will run approximately
40% faster than the standard version. The AVX2 version will run 10-30%
faster than the SSE3 version. Additionally, keep in mind that using more
cores / threads will not necessarily decrease run time. The RAxML manual
suggests using 1 core per ~500 DNA alignment patterns. Alignment pattern
information is usually visible on screen, when the <code class="docutils literal notranslate"><span class="pre">--verbose</span></code> option is
used. Additionally, try using a rate category (CAT model; via
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span></code>), which results in equally good trees as the
GAMMA models and is approximately 4 times faster. See the <a class="reference external" href="https://doi.org/10.1109/IPDPS.2006.1639535">CAT paper</a>. The
CAT approximation is also Ideal for alignments containing <a class="reference external" href="https://doi.org/10.1186/1471-2105-12-470">10,000 or more
taxa</a>, and is very much similar the <a class="reference external" href="https://doi.org/10.1371/journal.pone.0009490">CAT-like model of FastTree2</a>.</p>
</div>
</div>
<div class="section" id="iqtree">
<h3><a class="toc-backref" href="#id14">iqtree</a><a class="headerlink" href="#iqtree" title="Permalink to this headline">¶</a></h3>
<p>Similar to the <code class="docutils literal notranslate"><span class="pre">raxml</span></code> and <code class="docutils literal notranslate"><span class="pre">raxml-rapid-bootstrap</span></code> methods above, we
provide similar functionality for <a class="reference external" href="https://doi.org/10.1093/molbev/msu300">IQ-TREE</a>: <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> and
<code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code>. IQ-TREE is unique compared to the <code class="docutils literal notranslate"><span class="pre">fastree</span></code>
and <code class="docutils literal notranslate"><span class="pre">raxml</span></code> options, in that it provides access to 286 <a class="reference external" href="https://doi.org/10.1016/j.dci.2004.07.007">models of nucleotide
substitution</a>! IQ-TREE can also determine which of these models best fits your
dataset prior to constructing your tree via its built-in <a class="reference external" href="https://doi.org/10.1038/nmeth.4285">ModelFinder</a>
algorithm. This is the default in QIIME 2, but do not worry, you can set any
one of the 286 models of nucleotide substitution via the
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span></code> flag, e.g. you can set the model as <code class="docutils literal notranslate"><span class="pre">HKY+I+G</span></code>
instead of the default <code class="docutils literal notranslate"><span class="pre">MFP</span></code> (a basic short-hand for: “build a phylogeny
after determining the best fit model as determined by ModelFinder”). Keep in
mind the additional computational time required for model testing via
ModelFinder.</p>
<p>The simplest way to run the
<a class="reference internal" href="../../plugins/available/phylogeny/iqtree/"><span class="doc">iqtree command</span></a> with default
settings and automatic model selection (<code class="docutils literal notranslate"><span class="pre">MFP</span></code>) is like so:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree iqt-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-lc6ec9s2/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree -nt 1
Seed:    914763 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:40:44 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-lc6ec9s2/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1425.579     37  2925.158     2941.136     3049.700
  2  JC+I          1424.363     38  2924.727     2941.664     3052.634
  3  JC+G4         1417.262     38  2910.525     2927.462     3038.432
  4  JC+I+G4       1417.678     39  2913.356     2931.287     3044.629
  5  JC+R2         1413.758     39  2905.516     2923.447     3036.789
  6  JC+R3         1413.670     41  2909.339     2929.363     3047.344
 14  F81+F         1422.791     40  2925.582     2944.542     3060.221
 15  F81+F+I       1421.596     41  2925.192     2945.215     3063.197
 16  F81+F+G4      1413.765     41  2909.531     2929.554     3047.536
 17  F81+F+I+G4    1414.147     42  2912.295     2933.418     3053.666
 18  F81+F+R2      1409.425     42  2902.850     2923.973     3044.221
 19  F81+F+R3      1409.417     44  2906.834     2930.266     3054.937
 27  K2P           1416.855     38  2909.709     2926.646     3037.616
 28  K2P+I         1415.842     39  2909.684     2927.615     3040.957
 29  K2P+G4        1407.695     39  2893.391     2911.322     3024.664
 30  K2P+I+G4      1408.037     40  2896.075     2915.034     3030.714
 31  K2P+R2        1402.994     40  2885.987     2904.947     3020.626
 32  K2P+R3        1402.982     42  2889.963     2911.086     3031.334
 40  HKY+F         1414.238     41  2910.476     2930.499     3048.481
 41  HKY+F+I       1413.003     42  2910.006     2931.129     3051.377
 42  HKY+F+G4      1404.192     42  2892.385     2913.508     3033.756
 43  HKY+F+I+G4    1404.423     43  2894.845     2917.104     3039.582
 44  HKY+F+R2      1398.834     43  2883.668     2905.927     3028.405
 45  HKY+F+R3      1398.831     45  2887.663     2912.306     3039.132
 53  TNe           1416.748     39  2911.496     2929.427     3042.769
 54  TNe+I         1415.603     40  2911.205     2930.165     3045.844
 55  TNe+G4        1407.646     40  2895.291     2914.251     3029.930
 56  TNe+I+G4      1407.833     41  2897.666     2917.689     3035.671
 57  TNe+R2        1402.979     41  2887.958     2907.981     3025.963
 58  TNe+R3        1402.976     43  2891.952     2914.211     3036.689
 66  TN+F          1413.880     42  2911.760     2932.883     3053.131
 67  TN+F+I        1412.767     43  2911.534     2933.793     3056.271
 68  TN+F+G4       1403.365     43  2892.729     2914.988     3037.466
 69  TN+F+I+G4     1403.538     44  2895.076     2918.508     3043.179
 70  TN+F+R2       1397.639     44  2883.279     2906.711     3031.382
 71  TN+F+R3       1397.575     46  2887.149     2913.041     3041.984
 79  K3P           1415.370     39  2908.741     2926.672     3040.014
 80  K3P+I         1414.342     40  2908.685     2927.644     3043.324
 81  K3P+G4        1405.099     40  2890.197     2909.157     3024.836
 82  K3P+I+G4      1405.242     41  2892.485     2912.508     3030.490
 83  K3P+R2        1400.348     41  2882.696     2902.719     3020.701
 84  K3P+R3        1400.308     43  2886.616     2908.875     3031.353
 92  K3Pu+F        1412.388     42  2908.776     2929.899     3050.147
 93  K3Pu+F+I      1411.038     43  2908.077     2930.336     3052.814
 94  K3Pu+F+G4     1400.765     43  2887.530     2909.789     3032.267
 95  K3Pu+F+I+G4   1400.883     44  2889.765     2913.197     3037.868
 96  K3Pu+F+R2     1395.248     44  2878.496     2901.928     3026.599
 97  K3Pu+F+R3     1395.237     46  2882.473     2908.365     3037.308
105  TPM2+F        1414.141     42  2912.282     2933.405     3053.653
106  TPM2+F+I      1412.886     43  2911.772     2934.031     3056.509
107  TPM2+F+G4     1403.908     43  2893.816     2916.075     3038.553
108  TPM2+F+I+G4   1404.004     44  2896.008     2919.440     3044.111
109  TPM2+F+R2     1398.374     44  2884.748     2908.180     3032.851
110  TPM2+F+R3     1398.361     46  2888.722     2914.614     3043.557
118  TPM2u+F       1414.141     42  2912.283     2933.406     3053.654
119  TPM2u+F+I     1412.886     43  2911.772     2934.031     3056.509
120  TPM2u+F+G4    1403.908     43  2893.817     2916.076     3038.554
121  TPM2u+F+I+G4  1403.988     44  2895.976     2919.408     3044.079
122  TPM2u+F+R2    1398.373     44  2884.746     2908.178     3032.849
123  TPM2u+F+R3    1398.360     46  2888.721     2914.613     3043.555
131  TPM3+F        1410.239     42  2904.478     2925.601     3045.849
132  TPM3+F+I      1409.299     43  2904.597     2926.856     3049.334
133  TPM3+F+G4     1400.570     43  2887.140     2909.398     3031.877
134  TPM3+F+I+G4   1400.639     44  2889.278     2912.709     3037.380
135  TPM3+F+R2     1395.750     44  2879.501     2902.933     3027.604
136  TPM3+F+R3     1395.743     46  2883.485     2909.377     3038.320
144  TPM3u+F       1410.247     42  2904.495     2925.618     3045.866
145  TPM3u+F+I     1409.296     43  2904.592     2926.851     3049.329
146  TPM3u+F+G4    1400.573     43  2887.145     2909.404     3031.882
147  TPM3u+F+I+G4  1400.626     44  2889.252     2912.684     3037.355
148  TPM3u+F+R2    1395.750     44  2879.501     2902.933     3027.604
149  TPM3u+F+R3    1395.741     46  2883.482     2909.374     3038.317
157  TIMe          1415.276     40  2910.552     2929.511     3045.191
158  TIMe+I        1414.108     41  2910.215     2930.239     3048.220
159  TIMe+G4       1405.067     41  2892.133     2912.156     3030.138
160  TIMe+I+G4     1405.110     42  2894.220     2915.343     3035.591
161  TIMe+R2       1400.240     42  2884.480     2905.603     3025.851
162  TIMe+R3       1400.231     44  2888.462     2911.894     3036.565
170  TIM+F         1412.007     43  2910.014     2932.273     3054.751
171  TIM+F+I       1410.783     44  2909.565     2932.997     3057.668
172  TIM+F+G4      1399.776     44  2887.553     2910.985     3035.656
173  TIM+F+I+G4    1399.816     45  2889.632     2914.275     3041.101
174  TIM+F+R2      1393.932     45  2877.864     2902.507     3029.333
175  TIM+F+R3      1393.906     47  2881.811     2908.992     3040.012
183  TIM2e         1416.718     40  2913.436     2932.395     3048.075
184  TIM2e+I       1415.562     41  2913.124     2933.147     3051.129
185  TIM2e+G4      1407.650     41  2897.300     2917.323     3035.305
186  TIM2e+I+G4    1407.668     42  2899.335     2920.458     3040.706
187  TIM2e+R2      1402.916     42  2889.832     2910.955     3031.203
188  TIM2e+R3      1402.909     44  2893.819     2917.251     3041.922
196  TIM2+F        1413.778     43  2913.555     2935.814     3058.292
197  TIM2+F+I      1412.645     44  2913.291     2936.723     3061.394
198  TIM2+F+G4     1403.094     44  2894.189     2917.621     3042.291
199  TIM2+F+I+G4   1403.108     45  2896.215     2920.858     3047.684
200  TIM2+F+R2     1397.116     45  2884.232     2908.875     3035.701
201  TIM2+F+R3     1397.086     47  2888.172     2915.353     3046.373
209  TIM3e         1408.974     40  2897.948     2916.908     3032.587
210  TIM3e+I       1408.095     41  2898.190     2918.213     3036.195
211  TIM3e+G4      1399.835     41  2881.671     2901.694     3019.676
212  TIM3e+I+G4    1399.852     42  2883.703     2904.826     3025.074
213  TIM3e+R2      1396.088     42  2876.176     2897.299     3017.547
214  TIM3e+R3      1396.084     44  2880.167     2903.599     3028.270
222  TIM3+F        1409.829     43  2905.657     2927.916     3050.394
223  TIM3+F+I      1408.991     44  2905.982     2929.414     3054.085
224  TIM3+F+G4     1399.780     44  2887.560     2910.992     3035.663
225  TIM3+F+I+G4   1399.774     45  2889.547     2914.190     3041.016
226  TIM3+F+R2     1394.419     45  2878.838     2903.481     3030.307
227  TIM3+F+R3     1394.413     47  2882.826     2910.006     3041.027
235  TVMe          1406.272     41  2894.543     2914.567     3032.548
236  TVMe+I        1405.467     42  2894.933     2916.056     3036.304
237  TVMe+G4       1395.955     42  2875.910     2897.033     3017.281
238  TVMe+I+G4     1395.951     43  2877.902     2900.161     3022.639
239  TVMe+R2       1392.313     43  2870.626     2892.885     3015.363
240  TVMe+R3       1392.312     45  2874.625     2899.267     3026.093
248  TVM+F         1407.307     44  2902.614     2926.046     3050.717
249  TVM+F+I       1406.269     45  2902.538     2927.181     3054.007
250  TVM+F+G4      1396.061     45  2882.123     2906.766     3033.592
251  TVM+F+I+G4    1396.053     46  2884.107     2909.999     3038.942
252  TVM+F+R2      1391.400     46  2874.799     2900.692     3029.634
253  TVM+F+R3      1391.390     48  2878.780     2907.289     3040.347
261  SYM           1406.210     42  2896.421     2917.544     3037.792
262  SYM+I         1405.321     43  2896.642     2918.901     3041.379
263  SYM+G4        1395.918     43  2877.836     2900.095     3022.573
264  SYM+I+G4      1395.916     44  2879.832     2903.264     3027.935
265  SYM+R2        1392.308     44  2872.615     2896.047     3020.718
266  SYM+R3        1392.302     46  2876.603     2902.495     3031.438
274  GTR+F         1406.859     45  2903.717     2928.360     3055.186
275  GTR+F+I       1405.970     46  2903.939     2929.831     3058.774
276  GTR+F+G4      1395.211     46  2882.423     2908.315     3037.258
277  GTR+F+I+G4    1395.206     47  2884.413     2911.594     3042.614
278  GTR+F+R2      1390.151     47  2874.302     2901.483     3032.503
279  GTR+F+R3      1390.132     49  2878.263     2908.141     3043.196
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree.model.gz
CPU time for ModelFinder: 0.956 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.969 seconds (0h:0m:0s)

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1418.951
2. Current log-likelihood: -1397.991
3. Current log-likelihood: -1393.472
4. Current log-likelihood: -1392.886
5. Current log-likelihood: -1392.619
6. Current log-likelihood: -1392.468
Optimal log-likelihood: -1392.377
Rate parameters:  A-C: 0.14236  A-G: 1.85682  A-T: 1.51945  C-G: 0.73999  C-T: 1.85682  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.672,0.382) (0.328,2.264)
Parameters optimization took 6 rounds (0.035 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.227
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.091 second
Computing log-likelihood of 98 initial trees ... 0.144 seconds
Current best score: -1389.227

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.389
UPDATE BEST LOG-LIKELIHOOD: -1382.389
Iteration 10 / LogL: -1382.398 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.391 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.389 / CPU time: 0.485
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1391.358 / Time: 0h:0m:0s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.388
Iteration 40 / LogL: -1403.595 / Time: 0h:0m:0s (0h:0m:1s left)
Iteration 50 / LogL: -1382.521 / Time: 0h:0m:1s (0h:0m:1s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 53: -1382.119
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 55: -1382.044
Iteration 60 / LogL: -1382.069 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 70 / LogL: -1391.264 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 80 / LogL: -1382.445 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 90 / LogL: -1382.096 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 100 / LogL: -1382.402 / Time: 0h:0m:1s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.043
Iteration 110 / LogL: -1382.518 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 120 / LogL: -1382.607 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 130 / LogL: -1382.445 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 140 / LogL: -1382.222 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 150 / LogL: -1382.443 / Time: 0h:0m:2s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 156 ITERATIONS / Time: 0h:0m:2s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.043
Optimal log-likelihood: -1382.037
Rate parameters:  A-C: 0.18887  A-G: 1.83050  A-T: 1.53116  C-G: 0.77209  C-T: 1.83050  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.719,0.420) (0.281,2.487)
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1382.037
Total tree length: 6.919

Total number of iterations: 156
CPU time used for tree search: 2.585 sec (0h:0m:2s)
Wall-clock time used for tree search: 2.588 sec (0h:0m:2s)
Total CPU time used: 2.640 sec (0h:0m:2s)
Total wall-clock time used: 2.646 sec (0h:0m:2s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree.log

Date and Time: Fri Nov  1 07:40:48 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-lc6ec9s2/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwa02_fgr/q2iqtree -nt 1

Saved Phylogeny[Unrooted] to: iqt-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="specifying-a-substitution-model">
<h4><a class="toc-backref" href="#id15">Specifying a substitution model</a><a class="headerlink" href="#specifying-a-substitution-model" title="Permalink to this headline">¶</a></h4>
<p>We can also set a substitution model of our choosing. You may have noticed
while watching the onscreen output of the previous command that the best
fitting model selected by ModelFinder is noted. For the sake of argument, let’s
say the best selected model was shown as <code class="docutils literal notranslate"><span class="pre">GTR+F+I+G4</span></code>. The <code class="docutils literal notranslate"><span class="pre">F</span></code> is only a
notation to let us know that <em>if</em> a given model supports <em>unequal base
frequencies</em>, then the <em>empirical base frequencies</em> will be used by default.
Using empirical base frequencies (<code class="docutils literal notranslate"><span class="pre">F</span></code>), rather than estimating them, greatly
reduces computational time. The <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> plugin will not accept <code class="docutils literal notranslate"><span class="pre">F</span></code> within
the model notation supplied at the command line, as this will always be implied
automatically for the appropriate model. Also, the <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> plugin only
accepts <code class="docutils literal notranslate"><span class="pre">G</span></code> <em>not</em> <code class="docutils literal notranslate"><span class="pre">G4</span></code> to be specified within the model notation. The <code class="docutils literal notranslate"><span class="pre">4</span></code>
is simply another explicit notation to remind us that four rate categories are
being assumed by default. The notation approach used by the plugin simply helps
to retain simplicity and familiarity when supplying model notations on the
command line. So, in brief, we only have to type <code class="docutils literal notranslate"><span class="pre">GTR+I+G</span></code> as our input
model:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-gtrig-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-67hgp5u_/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree -nt 1
Seed:    791263 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:40:52 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-67hgp5u_/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.281 / LogL: -1392.558
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.008, 1.381 / LogL: -1392.827
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.009, 1.392 / LogL: -1392.900
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.009, 1.387 / LogL: -1392.892
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.858
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.387 / LogL: -1392.886
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.008, 1.378 / LogL: -1392.831
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.382 / LogL: -1392.845
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.008, 1.382 / LogL: -1392.850
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.860
Optimal pinv,alpha: 0.000, 1.281 / LogL: -1392.558

Parameters optimization took 0.480 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.710
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.089 second
Computing log-likelihood of 98 initial trees ... 0.199 seconds
Current best score: -1392.558

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.259
Iteration 10 / LogL: -1387.287 / Time: 0h:0m:0s
Iteration 20 / LogL: -1387.269 / Time: 0h:0m:1s
Finish initializing candidate tree set (2)
Current best tree score: -1387.259 / CPU time: 0.599
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1396.818 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 40 / LogL: -1387.262 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 50 / LogL: -1387.350 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 60 / LogL: -1387.451 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 70 / LogL: -1406.372 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 80 / LogL: -1392.423 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 90 / LogL: -1387.352 / Time: 0h:0m:3s (0h:0m:0s left)
Iteration 100 / LogL: -1387.350 / Time: 0h:0m:3s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 102 ITERATIONS / Time: 0h:0m:3s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.259
Optimal log-likelihood: -1387.253
Rate parameters:  A-C: 0.33196  A-G: 2.27261  A-T: 2.15113  C-G: 1.18143  C-T: 3.30120  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.324
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1387.253
Total tree length: 6.714

Total number of iterations: 102
CPU time used for tree search: 3.027 sec (0h:0m:3s)
Wall-clock time used for tree search: 3.030 sec (0h:0m:3s)
Total CPU time used: 3.524 sec (0h:0m:3s)
Total wall-clock time used: 3.538 sec (0h:0m:3s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree.log

Date and Time: Fri Nov  1 07:40:56 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-67hgp5u_/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpbdvurvku/q2iqtree -nt 1

Saved Phylogeny[Unrooted] to: iqt-gtrig-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-gtrig-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-gtrig-tree.qza">download</a></p></li>
</ul>
</div>
<p>Let’s rerun the command above and add the <code class="docutils literal notranslate"><span class="pre">--p-fast</span></code> option. This option,
only compatible with the <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> method, resembles the fast search performed
by <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>. 🏎️ Secondly, let’s also perform multiple tree searches and
keep the best of those trees (as we did earlier with the
<code class="docutils literal notranslate"><span class="pre">raxml</span> <span class="pre">--p-n-searches</span> <span class="pre">...</span></code> command):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --p-fast <span class="se">\</span>
  --p-n-runs <span class="m">10</span> <span class="se">\</span>
  --o-tree iqt-gtrig-fast-ms-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 10 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-7bbn7mkx/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree -nt 1 -fast
Seed:    589875 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:41:00 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-7bbn7mkx/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

---&gt; START RUN NUMBER 1 (seed: 589875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.243 / LogL: -1394.457
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.018, 1.300 / LogL: -1395.193
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.023, 1.343 / LogL: -1395.423
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.022, 1.352 / LogL: -1395.409
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.025, 1.352 / LogL: -1395.581
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.027, 1.353 / LogL: -1395.694
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.023, 1.350 / LogL: -1395.451
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.024, 1.352 / LogL: -1395.506
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.025, 1.352 / LogL: -1395.553
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.026, 1.353 / LogL: -1395.597
Optimal pinv,alpha: 0.000, 1.243 / LogL: -1394.457

Parameters optimization took 0.305 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.886
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.284
Finish initializing candidate tree set (3)
Current best tree score: -1387.284 / CPU time: 0.059
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.284
Optimal log-likelihood: -1387.262
Rate parameters:  A-C: 0.33572  A-G: 2.25727  A-T: 2.13258  C-G: 1.17706  C-T: 3.27735  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.306
Parameters optimization took 1 rounds (0.005 sec)
BEST SCORE FOUND : -1387.262
Total tree length: 6.787

Total number of iterations: 2
CPU time used for tree search: 0.059 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.060 sec (0h:0m:0s)
Total CPU time used: 0.388 sec (0h:0m:0s)
Total wall-clock time used: 0.392 sec (0h:0m:0s)

---&gt; START RUN NUMBER 2 (seed: 590875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1481.570
2. Current log-likelihood: -1401.790
3. Current log-likelihood: -1396.547
4. Current log-likelihood: -1395.118
5. Current log-likelihood: -1394.384
Optimal log-likelihood: -1393.813
Rate parameters:  A-C: 0.27488  A-G: 2.41340  A-T: 2.17693  C-G: 1.25270  C-T: 3.29513  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.421
Parameters optimization took 5 rounds (0.037 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.791
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.211
Finish initializing candidate tree set (3)
Current best tree score: -1388.211 / CPU time: 0.046
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.211
2. Current log-likelihood: -1388.012
3. Current log-likelihood: -1387.866
4. Current log-likelihood: -1387.758
5. Current log-likelihood: -1387.675
6. Current log-likelihood: -1387.610
7. Current log-likelihood: -1387.559
Optimal log-likelihood: -1387.518
Rate parameters:  A-C: 0.35582  A-G: 2.35557  A-T: 2.14271  C-G: 1.20481  C-T: 3.37552  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.361
Parameters optimization took 7 rounds (0.027 sec)
BEST SCORE FOUND : -1387.518
Total tree length: 6.815

Total number of iterations: 2
CPU time used for tree search: 0.045 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.046 sec (0h:0m:0s)
Total CPU time used: 0.129 sec (0h:0m:0s)
Total wall-clock time used: 0.132 sec (0h:0m:0s)

---&gt; START RUN NUMBER 3 (seed: 591875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.521
2. Current log-likelihood: -1403.150
3. Current log-likelihood: -1398.384
4. Current log-likelihood: -1396.968
5. Current log-likelihood: -1396.253
Optimal log-likelihood: -1395.745
Rate parameters:  A-C: 0.23809  A-G: 2.08501  A-T: 1.96789  C-G: 1.08192  C-T: 2.78800  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.364
Parameters optimization took 5 rounds (0.040 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.993
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.951
Finish initializing candidate tree set (4)
Current best tree score: -1387.951 / CPU time: 0.078
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.951
2. Current log-likelihood: -1387.792
3. Current log-likelihood: -1387.673
4. Current log-likelihood: -1387.583
5. Current log-likelihood: -1387.515
6. Current log-likelihood: -1387.462
Optimal log-likelihood: -1387.420
Rate parameters:  A-C: 0.33363  A-G: 2.24758  A-T: 2.12142  C-G: 1.16644  C-T: 3.24795  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.358
Parameters optimization took 6 rounds (0.024 sec)
BEST SCORE FOUND : -1387.420
Total tree length: 6.700

Total number of iterations: 2
CPU time used for tree search: 0.078 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.078 sec (0h:0m:0s)
Total CPU time used: 0.161 sec (0h:0m:0s)
Total wall-clock time used: 0.164 sec (0h:0m:0s)

---&gt; START RUN NUMBER 4 (seed: 592875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.586
2. Current log-likelihood: -1403.151
3. Current log-likelihood: -1398.384
4. Current log-likelihood: -1396.968
5. Current log-likelihood: -1396.253
Optimal log-likelihood: -1395.745
Rate parameters:  A-C: 0.23809  A-G: 2.08501  A-T: 1.96788  C-G: 1.08193  C-T: 2.78798  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.364
Parameters optimization took 5 rounds (0.044 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.993
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.951
Finish initializing candidate tree set (4)
Current best tree score: -1387.951 / CPU time: 0.080
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.951
2. Current log-likelihood: -1387.792
3. Current log-likelihood: -1387.673
4. Current log-likelihood: -1387.583
5. Current log-likelihood: -1387.515
6. Current log-likelihood: -1387.462
Optimal log-likelihood: -1387.420
Rate parameters:  A-C: 0.33363  A-G: 2.24758  A-T: 2.12142  C-G: 1.16644  C-T: 3.24795  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.358
Parameters optimization took 6 rounds (0.024 sec)
BEST SCORE FOUND : -1387.420
Total tree length: 6.700

Total number of iterations: 2
CPU time used for tree search: 0.079 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.080 sec (0h:0m:0s)
Total CPU time used: 0.163 sec (0h:0m:0s)
Total wall-clock time used: 0.166 sec (0h:0m:0s)

---&gt; START RUN NUMBER 5 (seed: 593875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.521
2. Current log-likelihood: -1403.150
3. Current log-likelihood: -1398.384
4. Current log-likelihood: -1396.968
5. Current log-likelihood: -1396.253
Optimal log-likelihood: -1395.745
Rate parameters:  A-C: 0.23809  A-G: 2.08501  A-T: 1.96789  C-G: 1.08192  C-T: 2.78800  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.364
Parameters optimization took 5 rounds (0.036 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.993
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.951
Finish initializing candidate tree set (4)
Current best tree score: -1387.951 / CPU time: 0.082
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.951
2. Current log-likelihood: -1387.792
3. Current log-likelihood: -1387.673
4. Current log-likelihood: -1387.583
5. Current log-likelihood: -1387.515
6. Current log-likelihood: -1387.462
Optimal log-likelihood: -1387.420
Rate parameters:  A-C: 0.33363  A-G: 2.24758  A-T: 2.12142  C-G: 1.16644  C-T: 3.24795  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.358
Parameters optimization took 6 rounds (0.021 sec)
BEST SCORE FOUND : -1387.420
Total tree length: 6.700

Total number of iterations: 2
CPU time used for tree search: 0.081 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.082 sec (0h:0m:0s)
Total CPU time used: 0.158 sec (0h:0m:0s)
Total wall-clock time used: 0.161 sec (0h:0m:0s)

---&gt; START RUN NUMBER 6 (seed: 594875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1481.723
2. Current log-likelihood: -1403.056
3. Current log-likelihood: -1398.274
4. Current log-likelihood: -1396.896
5. Current log-likelihood: -1396.165
Optimal log-likelihood: -1395.664
Rate parameters:  A-C: 0.23525  A-G: 2.07316  A-T: 1.96748  C-G: 1.07231  C-T: 2.84580  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.026
Gamma shape alpha: 1.337
Parameters optimization took 5 rounds (0.034 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.979
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.172
Finish initializing candidate tree set (3)
Current best tree score: -1388.172 / CPU time: 0.038
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.172
2. Current log-likelihood: -1387.951
3. Current log-likelihood: -1387.788
4. Current log-likelihood: -1387.670
5. Current log-likelihood: -1387.581
6. Current log-likelihood: -1387.513
7. Current log-likelihood: -1387.460
Optimal log-likelihood: -1387.418
Rate parameters:  A-C: 0.33576  A-G: 2.25513  A-T: 2.12939  C-G: 1.17073  C-T: 3.26109  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.356
Parameters optimization took 7 rounds (0.028 sec)
BEST SCORE FOUND : -1387.418
Total tree length: 6.706

Total number of iterations: 2
CPU time used for tree search: 0.038 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.039 sec (0h:0m:0s)
Total CPU time used: 0.122 sec (0h:0m:0s)
Total wall-clock time used: 0.125 sec (0h:0m:0s)

---&gt; START RUN NUMBER 7 (seed: 595875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.860
2. Current log-likelihood: -1401.866
3. Current log-likelihood: -1396.558
4. Current log-likelihood: -1395.110
5. Current log-likelihood: -1394.382
Optimal log-likelihood: -1393.810
Rate parameters:  A-C: 0.27492  A-G: 2.41474  A-T: 2.17844  C-G: 1.25256  C-T: 3.29094  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.026
Gamma shape alpha: 1.421
Parameters optimization took 5 rounds (0.036 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.790
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.211
Finish initializing candidate tree set (3)
Current best tree score: -1388.211 / CPU time: 0.048
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.211
2. Current log-likelihood: -1388.012
3. Current log-likelihood: -1387.866
4. Current log-likelihood: -1387.758
5. Current log-likelihood: -1387.675
6. Current log-likelihood: -1387.610
7. Current log-likelihood: -1387.559
Optimal log-likelihood: -1387.518
Rate parameters:  A-C: 0.35576  A-G: 2.35519  A-T: 2.14234  C-G: 1.20461  C-T: 3.37491  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.361
Parameters optimization took 7 rounds (0.027 sec)
BEST SCORE FOUND : -1387.518
Total tree length: 6.815

Total number of iterations: 2
CPU time used for tree search: 0.048 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.049 sec (0h:0m:0s)
Total CPU time used: 0.127 sec (0h:0m:0s)
Total wall-clock time used: 0.130 sec (0h:0m:0s)

---&gt; START RUN NUMBER 8 (seed: 596875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.403
2. Current log-likelihood: -1404.572
3. Current log-likelihood: -1399.236
4. Current log-likelihood: -1397.831
5. Current log-likelihood: -1397.061
Optimal log-likelihood: -1396.470
Rate parameters:  A-C: 0.24622  A-G: 2.08130  A-T: 1.99445  C-G: 1.06444  C-T: 2.85652  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.437
Parameters optimization took 5 rounds (0.041 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.961
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.181
Finish initializing candidate tree set (3)
Current best tree score: -1388.181 / CPU time: 0.055
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.955
3. Current log-likelihood: -1387.793
4. Current log-likelihood: -1387.674
5. Current log-likelihood: -1387.584
6. Current log-likelihood: -1387.516
7. Current log-likelihood: -1387.463
Optimal log-likelihood: -1387.421
Rate parameters:  A-C: 0.33228  A-G: 2.23486  A-T: 2.10851  C-G: 1.15950  C-T: 3.22710  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.360
Parameters optimization took 7 rounds (0.028 sec)
BEST SCORE FOUND : -1387.421
Total tree length: 6.696

Total number of iterations: 2
CPU time used for tree search: 0.055 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.055 sec (0h:0m:0s)
Total CPU time used: 0.143 sec (0h:0m:0s)
Total wall-clock time used: 0.146 sec (0h:0m:0s)

---&gt; START RUN NUMBER 9 (seed: 597875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.977
2. Current log-likelihood: -1402.079
3. Current log-likelihood: -1396.806
4. Current log-likelihood: -1395.378
5. Current log-likelihood: -1394.645
Optimal log-likelihood: -1394.074
Rate parameters:  A-C: 0.27526  A-G: 2.39376  A-T: 2.12146  C-G: 1.20995  C-T: 3.30275  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.394
Parameters optimization took 5 rounds (0.039 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.804
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.216
BETTER TREE FOUND at iteration 2: -1388.181
Finish initializing candidate tree set (4)
Current best tree score: -1388.181 / CPU time: 0.063
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.974
3. Current log-likelihood: -1387.831
4. Current log-likelihood: -1387.726
5. Current log-likelihood: -1387.646
6. Current log-likelihood: -1387.585
Optimal log-likelihood: -1387.535
Rate parameters:  A-C: 0.36870  A-G: 2.32160  A-T: 2.12854  C-G: 1.22861  C-T: 3.29569  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.006
Gamma shape alpha: 1.331
Parameters optimization took 6 rounds (0.022 sec)
BEST SCORE FOUND : -1387.535
Total tree length: 7.507

Total number of iterations: 2
CPU time used for tree search: 0.062 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.063 sec (0h:0m:0s)
Total CPU time used: 0.143 sec (0h:0m:0s)
Total wall-clock time used: 0.146 sec (0h:0m:0s)

---&gt; START RUN NUMBER 10 (seed: 598875)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.110
2. Current log-likelihood: -1403.534
3. Current log-likelihood: -1398.492
4. Current log-likelihood: -1397.024
5. Current log-likelihood: -1396.211
Optimal log-likelihood: -1395.651
Rate parameters:  A-C: 0.23242  A-G: 2.05799  A-T: 1.95257  C-G: 1.07461  C-T: 2.81853  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.346
Parameters optimization took 5 rounds (0.040 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.989
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.946
Finish initializing candidate tree set (4)
Current best tree score: -1387.946 / CPU time: 0.067
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.946
2. Current log-likelihood: -1387.788
3. Current log-likelihood: -1387.670
4. Current log-likelihood: -1387.581
5. Current log-likelihood: -1387.513
6. Current log-likelihood: -1387.460
Optimal log-likelihood: -1387.418
Rate parameters:  A-C: 0.33477  A-G: 2.25224  A-T: 2.12637  C-G: 1.16912  C-T: 3.25669  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.357
Parameters optimization took 6 rounds (0.020 sec)
BEST SCORE FOUND : -1387.418
Total tree length: 6.704

Total number of iterations: 2
CPU time used for tree search: 0.066 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.067 sec (0h:0m:0s)
Total CPU time used: 0.146 sec (0h:0m:0s)
Total wall-clock time used: 0.149 sec (0h:0m:0s)

---&gt; SUMMARIZE RESULTS FROM 10 RUNS

Run 1 gave best log-likelihood: -1387.262
Total CPU time for 10 runs: 1.696 seconds.
Total wall-clock time for 10 runs: 1.734 seconds.


Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree.treefile
  Trees from independent runs:   /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree.runtrees
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree.log

Date and Time: Fri Nov  1 07:41:02 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 10 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-7bbn7mkx/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp1gz3ic6z/q2iqtree -nt 1 -fast

Saved Phylogeny[Unrooted] to: iqt-gtrig-fast-ms-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-gtrig-fast-ms-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-fast-ms-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-gtrig-fast-ms-tree.qza">download</a></p></li>
</ul>
</div>
</div>
<div class="section" id="single-branch-tests">
<h4><a class="toc-backref" href="#id16">Single branch tests</a><a class="headerlink" href="#single-branch-tests" title="Permalink to this headline">¶</a></h4>
<p>IQ-TREE provides access to a few <a class="reference external" href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-single-branch-tests">single branch testing methods</a></p>
<ol class="arabic simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/sysbio/syq010">SH-aLRT</a> via <code class="docutils literal notranslate"><span class="pre">--p-alrt</span> <span class="pre">[INT</span> <span class="pre">&gt;=</span> <span class="pre">1000]</span></code></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/sysbio/syr041">aBayes</a> via <code class="docutils literal notranslate"><span class="pre">--p-abayes</span> <span class="pre">[TRUE</span> <span class="pre">|</span> <span class="pre">FALSE]</span></code></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1007/BF0249864">local bootstrap test</a> via <code class="docutils literal notranslate"><span class="pre">--p-lbp</span> <span class="pre">[INT</span> <span class="pre">&gt;=</span> <span class="pre">1000]</span></code></p></li>
</ol>
<p>Single branch tests are commonly used as an alternative to the bootstrapping
approach we’ve discussed above, as they are substantially faster and <a class="reference external" href="http://www.iqtree.org/doc/Command-Reference#single-branch-tests">often
recommended</a> when constructing large phylogenies (e.g. &gt;10,000 taxa). All
three of these methods can be applied simultaneously and viewed within <a class="reference external" href="https://itol.embl.de/">iTOL</a>
as separate bootstrap support values. These values are always in listed in the
following order of <em>alrt / lbp / abayes</em>. We’ll go ahead and apply all of the
branch tests in our next command, while specifying the same substitution model
as above. Feel free to combine this with the <code class="docutils literal notranslate"><span class="pre">--p-fast</span></code> option. 😉</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-alrt <span class="m">1000</span> <span class="se">\</span>
  --p-abayes <span class="se">\</span>
  --p-lbp <span class="m">1000</span> <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-sbt-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-2_nv7xic/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree -nt 1 -alrt 1000 -abayes -lbp 1000
Seed:    584004 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:41:06 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-2_nv7xic/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.237 / LogL: -1394.542
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.010, 1.344 / LogL: -1394.883
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.010, 1.353 / LogL: -1394.887
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.010, 1.352 / LogL: -1394.874
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.009, 1.349 / LogL: -1394.839
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.351 / LogL: -1394.865
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.010, 1.352 / LogL: -1394.886
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.346 / LogL: -1394.827
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.009, 1.347 / LogL: -1394.838
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.009, 1.347 / LogL: -1394.849
Optimal pinv,alpha: 0.000, 1.237 / LogL: -1394.542

Parameters optimization took 0.462 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.865
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.089 second
Computing log-likelihood of 98 initial trees ... 0.213 seconds
Current best score: -1392.865

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.265
Iteration 10 / LogL: -1387.304 / Time: 0h:0m:0s
Iteration 20 / LogL: -1387.279 / Time: 0h:0m:1s
Finish initializing candidate tree set (1)
Current best tree score: -1387.265 / CPU time: 0.655
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1389.882 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 40 / LogL: -1387.757 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 50 / LogL: -1388.533 / Time: 0h:0m:2s (0h:0m:2s left)
Iteration 60 / LogL: -1387.407 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 70 / LogL: -1387.616 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 80 / LogL: -1387.419 / Time: 0h:0m:3s (0h:0m:0s left)
Iteration 90 / LogL: -1387.407 / Time: 0h:0m:3s (0h:0m:0s left)
Iteration 100 / LogL: -1387.288 / Time: 0h:0m:3s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 102 ITERATIONS / Time: 0h:0m:3s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.265
Optimal log-likelihood: -1387.256
Rate parameters:  A-C: 0.32900  A-G: 2.26254  A-T: 2.14102  C-G: 1.17611  C-T: 3.29231  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.318
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1387.256

Testing tree branches by SH-like aLRT with 1000 replicates...
Testing tree branches by local-BP test with 1000 replicates...
Testing tree branches by aBayes parametric test...
0.058 sec.
Total tree length: 6.743

Total number of iterations: 102
CPU time used for tree search: 3.193 sec (0h:0m:3s)
Wall-clock time used for tree search: 3.196 sec (0h:0m:3s)
Total CPU time used: 3.737 sec (0h:0m:3s)
Total wall-clock time used: 3.743 sec (0h:0m:3s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree.log

Date and Time: Fri Nov  1 07:41:10 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-2_nv7xic/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpxom3yfex/q2iqtree -nt 1 -alrt 1000 -abayes -lbp 1000

Saved Phylogeny[Unrooted] to: iqt-sbt-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-sbt-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-sbt-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-sbt-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>IQ-TREE search settings</strong>.
There are quite a few adjustable parameters available for <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> that
can be modified improve searches through “tree space” and prevent the search
algorithms from getting stuck in local optima. One particular <a class="reference external" href="https://groups.google.com/forum/#!searchin/iqtree/iterations|sort:date/iqtree/0mwGhDokNns/vlBryIwXHAAJ">best
practice</a> to aid in this regard, is to adjust the following parameters:
<code class="docutils literal notranslate"><span class="pre">--p-perturb-nni-strength</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-stop-iter</span></code> (each respectively maps
to the <code class="docutils literal notranslate"><span class="pre">-pers</span></code> and <code class="docutils literal notranslate"><span class="pre">-nstop</span></code> flags of <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> ). In brief, the larger
the value for NNI (nearest-neighbor interchange) perturbation, the larger
the jumps in “tree space”. This value should be set high enough to allow the
search algorithm to avoid being trapped in local optima, but not to high
that the search is haphazardly jumping around “tree space”. That is, like
Goldilocks and the three 🐻s you need to find a setting that is “just
right”, or at least within a set of reasonable bounds. One way of assessing
this, is to do a few short trial runs using the <code class="docutils literal notranslate"><span class="pre">--verbose</span></code> flag. If you
see that the likelihood values are jumping around to much, then lowering the
value for <code class="docutils literal notranslate"><span class="pre">--p-perturb-nni-strength</span></code> may be warranted. As for the stopping
criteria, i.e. <code class="docutils literal notranslate"><span class="pre">--p-stop-iter</span></code>, the higher this value, the more thorough
your search in “tree space”. Be aware, increasing this value may also
increase the run time. That is, the search will continue until it has
sampled a number of trees, say 100 (default), without finding a better
scoring tree. If a better tree is found, then the counter resets, and the
search continues. These two parameters deserve special consideration when a
given data set contains many short sequences, quite common for microbiome
survey data. We can modify our original command to include these extra
parameters with the recommended modifications for short sequences, i.e. a
lower value for perturbation strength (shorter reads do not contain as much
phylogenetic information, thus we should limit how far we jump around in
“tree space”) and a larger number of stop iterations. See the <a class="reference external" href="http://www.iqtree.org/doc/Command-Reference">IQ-TREE
command reference</a> for more details about default parameter settings.
Finally, we’ll let <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> perform the model testing, and automatically
determine the optimal number of CPU cores to use.</p>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --o-tree iqt-nnisi-fast-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g464_3bj/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree -nt 1 -nstop 200 -pers 0.200000
Seed:    582488 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:41:14 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g464_3bj/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1425.486     37  2924.972     2940.949     3049.513
  2  JC+I          1424.270     38  2924.540     2941.477     3052.447
  3  JC+G4         1417.170     38  2910.340     2927.277     3038.247
  4  JC+I+G4       1417.586     39  2913.172     2931.103     3044.445
  5  JC+R2         1413.666     39  2905.331     2923.262     3036.604
  6  JC+R3         1413.577     41  2909.155     2929.178     3047.160
 14  F81+F         1422.698     40  2925.395     2944.355     3060.034
 15  F81+F+I       1421.502     41  2925.005     2945.028     3063.010
 16  F81+F+G4      1413.672     41  2909.345     2929.368     3047.350
 17  F81+F+I+G4    1414.055     42  2912.110     2933.232     3053.481
 18  F81+F+R2      1409.332     42  2902.664     2923.787     3044.035
 19  F81+F+R3      1409.325     44  2906.649     2930.081     3054.752
 27  K2P           1416.820     38  2909.640     2926.578     3037.548
 28  K2P+I         1415.808     39  2909.616     2927.547     3040.889
 29  K2P+G4        1407.667     39  2893.335     2911.266     3024.608
 30  K2P+I+G4      1408.009     40  2896.018     2914.978     3030.657
 31  K2P+R2        1402.967     40  2885.935     2904.894     3020.574
 32  K2P+R3        1402.955     42  2889.911     2911.033     3031.282
 40  HKY+F         1414.203     41  2910.405     2930.428     3048.410
 41  HKY+F+I       1412.969     42  2909.937     2931.060     3051.308
 42  HKY+F+G4      1404.164     42  2892.328     2913.450     3033.699
 43  HKY+F+I+G4    1404.394     43  2894.788     2917.047     3039.525
 44  HKY+F+R2      1398.807     43  2883.614     2905.873     3028.351
 45  HKY+F+R3      1398.805     45  2887.609     2912.252     3039.078
 53  TNe           1416.710     39  2911.420     2929.351     3042.693
 54  TNe+I         1415.563     40  2911.125     2930.085     3045.764
 55  TNe+G4        1407.614     40  2895.229     2914.189     3029.868
 56  TNe+I+G4      1407.802     41  2897.604     2917.627     3035.609
 57  TNe+R2        1402.953     41  2887.906     2907.929     3025.911
 58  TNe+R3        1402.950     43  2891.900     2914.159     3036.637
 66  TN+F          1413.851     42  2911.702     2932.825     3053.073
 67  TN+F+I        1412.738     43  2911.476     2933.735     3056.213
 68  TN+F+G4       1403.346     43  2892.693     2914.952     3037.430
 69  TN+F+I+G4     1403.520     44  2895.040     2918.472     3043.143
 70  TN+F+R2       1397.623     44  2883.247     2906.678     3031.349
 71  TN+F+R3       1397.558     46  2887.116     2913.008     3041.950
 79  K3P           1415.371     39  2908.742     2926.673     3040.015
 80  K3P+I         1414.342     40  2908.684     2927.644     3043.323
 81  K3P+G4        1405.098     40  2890.196     2909.156     3024.836
 82  K3P+I+G4      1405.242     41  2892.485     2912.508     3030.490
 83  K3P+R2        1400.348     41  2882.697     2902.720     3020.702
 84  K3P+R3        1400.308     43  2886.616     2908.875     3031.353
 92  K3Pu+F        1412.388     42  2908.776     2929.899     3050.147
 93  K3Pu+F+I      1411.038     43  2908.077     2930.335     3052.814
 94  K3Pu+F+G4     1400.765     43  2887.530     2909.789     3032.267
 95  K3Pu+F+I+G4   1400.883     44  2889.765     2913.197     3037.868
 96  K3Pu+F+R2     1395.248     44  2878.495     2901.927     3026.598
 97  K3Pu+F+R3     1395.237     46  2882.474     2908.366     3037.309
105  TPM2+F        1414.108     42  2912.216     2933.339     3053.587
106  TPM2+F+I      1412.854     43  2911.709     2933.968     3056.446
107  TPM2+F+G4     1403.884     43  2893.768     2916.027     3038.505
108  TPM2+F+I+G4   1403.980     44  2895.960     2919.392     3044.063
109  TPM2+F+R2     1398.351     44  2884.702     2908.134     3032.805
110  TPM2+F+R3     1398.338     46  2888.677     2914.569     3043.512
118  TPM2u+F       1414.108     42  2912.217     2933.339     3053.588
119  TPM2u+F+I     1412.854     43  2911.709     2933.968     3056.446
120  TPM2u+F+G4    1403.884     43  2893.769     2916.027     3038.506
121  TPM2u+F+I+G4  1403.964     44  2895.928     2919.360     3044.031
122  TPM2u+F+R2    1398.350     44  2884.701     2908.133     3032.804
123  TPM2u+F+R3    1398.338     46  2888.676     2914.568     3043.510
131  TPM3+F        1409.702     42  2903.405     2924.528     3044.776
132  TPM3+F+I      1408.762     43  2903.525     2925.784     3048.262
133  TPM3+F+G4     1399.930     43  2885.861     2908.120     3030.598
134  TPM3+F+I+G4   1399.997     44  2887.994     2911.426     3036.097
135  TPM3+F+R2     1395.174     44  2878.348     2901.780     3026.451
136  TPM3+F+R3     1395.168     46  2882.335     2908.227     3037.170
144  TPM3u+F       1409.713     42  2903.425     2924.548     3044.796
145  TPM3u+F+I     1408.758     43  2903.515     2925.774     3048.252
146  TPM3u+F+G4    1399.933     43  2885.866     2908.125     3030.603
147  TPM3u+F+I+G4  1399.985     44  2887.970     2911.402     3036.073
148  TPM3u+F+R2    1395.174     44  2878.347     2901.779     3026.450
149  TPM3u+F+R3    1395.165     46  2882.329     2908.222     3037.164
157  TIMe          1415.278     40  2910.556     2929.515     3045.195
158  TIMe+I        1414.107     41  2910.215     2930.238     3048.220
159  TIMe+G4       1405.066     41  2892.132     2912.155     3030.137
160  TIMe+I+G4     1405.109     42  2894.218     2915.341     3035.589
161  TIMe+R2       1400.241     42  2884.482     2905.605     3025.853
162  TIMe+R3       1400.232     44  2888.464     2911.896     3036.567
170  TIM+F         1412.007     43  2910.014     2932.272     3054.751
171  TIM+F+I       1410.782     44  2909.565     2932.997     3057.668
172  TIM+F+G4      1399.776     44  2887.552     2910.984     3035.655
173  TIM+F+I+G4    1399.814     45  2889.628     2914.271     3041.097
174  TIM+F+R2      1393.933     45  2877.866     2902.509     3029.335
175  TIM+F+R3      1393.893     47  2881.786     2908.967     3039.987
183  TIM2e         1416.678     40  2913.356     2932.316     3047.995
184  TIM2e+I       1415.520     41  2913.040     2933.063     3051.045
185  TIM2e+G4      1407.618     41  2897.236     2917.259     3035.241
186  TIM2e+I+G4    1407.636     42  2899.272     2920.395     3040.643
187  TIM2e+R2      1402.891     42  2889.781     2910.904     3031.152
188  TIM2e+R3      1402.882     44  2893.764     2917.196     3041.867
196  TIM2+F        1413.751     43  2913.502     2935.761     3058.239
197  TIM2+F+I      1412.619     44  2913.238     2936.670     3061.341
198  TIM2+F+G4     1403.079     44  2894.158     2917.590     3042.261
199  TIM2+F+I+G4   1403.093     45  2896.185     2920.828     3047.654
200  TIM2+F+R2     1397.104     45  2884.207     2908.850     3035.676
201  TIM2+F+R3     1397.063     47  2888.125     2915.306     3046.326
209  TIM3e         1408.186     40  2896.371     2915.331     3031.010
210  TIM3e+I       1407.301     41  2896.602     2916.625     3034.607
211  TIM3e+G4      1398.875     41  2879.751     2899.774     3017.756
212  TIM3e+I+G4    1398.888     42  2881.776     2902.898     3023.147
213  TIM3e+R2      1395.154     42  2874.308     2895.431     3015.679
214  TIM3e+R3      1395.147     44  2878.295     2901.727     3026.398
222  TIM3+F        1409.260     43  2904.521     2926.779     3049.258
223  TIM3+F+I      1408.433     44  2904.867     2928.299     3052.970
224  TIM3+F+G4     1399.112     44  2886.223     2909.655     3034.326
225  TIM3+F+I+G4   1399.103     45  2888.206     2912.849     3039.675
226  TIM3+F+R2     1393.777     45  2877.554     2902.196     3029.023
227  TIM3+F+R3     1393.769     47  2881.539     2908.720     3039.740
235  TVMe          1405.952     41  2893.904     2913.927     3031.909
236  TVMe+I        1405.133     42  2894.265     2915.388     3035.636
237  TVMe+G4       1395.646     42  2875.291     2896.414     3016.662
238  TVMe+I+G4     1395.641     43  2877.282     2899.541     3022.019
239  TVMe+R2       1392.048     43  2870.095     2892.354     3014.832
240  TVMe+R3       1392.046     45  2874.092     2898.735     3025.561
248  TVM+F         1407.238     44  2902.475     2925.907     3050.578
249  TVM+F+I       1406.201     45  2902.402     2927.045     3053.871
250  TVM+F+G4      1396.059     45  2882.118     2906.761     3033.587
251  TVM+F+I+G4    1396.045     46  2884.090     2909.982     3038.925
252  TVM+F+R2      1391.400     46  2874.800     2900.692     3029.635
253  TVM+F+R3      1391.381     48  2878.762     2907.272     3040.329
261  SYM           1405.900     42  2895.800     2916.923     3037.171
262  SYM+I         1405.014     43  2896.028     2918.286     3040.765
263  SYM+G4        1395.617     43  2877.234     2899.493     3021.971
264  SYM+I+G4      1395.614     44  2879.228     2902.660     3027.331
265  SYM+R2        1392.038     44  2872.076     2895.508     3020.179
266  SYM+R3        1392.030     46  2876.060     2901.953     3030.895
274  GTR+F         1406.799     45  2903.598     2928.241     3055.067
275  GTR+F+I       1405.902     46  2903.804     2929.696     3058.639
276  GTR+F+G4      1395.211     46  2882.423     2908.315     3037.258
277  GTR+F+I+G4    1395.203     47  2884.406     2911.587     3042.607
278  GTR+F+R2      1390.154     47  2874.308     2901.488     3032.509
279  GTR+F+R3      1390.122     49  2878.245     2908.123     3043.177
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree.model.gz
CPU time for ModelFinder: 0.952 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.957 seconds (0h:0m:0s)

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1419.055
2. Current log-likelihood: -1399.481
3. Current log-likelihood: -1393.449
4. Current log-likelihood: -1392.617
5. Current log-likelihood: -1392.346
6. Current log-likelihood: -1392.194
Optimal log-likelihood: -1392.107
Rate parameters:  A-C: 0.13866  A-G: 1.75800  A-T: 1.42775  C-G: 0.65275  C-T: 1.75800  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.671,0.384) (0.329,2.259)
Parameters optimization took 6 rounds (0.036 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.426
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.087 second
Computing log-likelihood of 98 initial trees ... 0.142 seconds
Current best score: -1389.426

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.399
UPDATE BEST LOG-LIKELIHOOD: -1382.399
Iteration 10 / LogL: -1382.407 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.409 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.399 / CPU time: 0.461
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
UPDATE BEST LOG-LIKELIHOOD: -1382.398
Iteration 30 / LogL: -1382.910 / Time: 0h:0m:0s (0h:0m:3s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 32: -1382.125
Iteration 40 / LogL: -1382.899 / Time: 0h:0m:0s (0h:0m:3s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 48: -1382.051
Iteration 50 / LogL: -1382.127 / Time: 0h:0m:0s (0h:0m:3s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.049
Iteration 60 / LogL: -1382.081 / Time: 0h:0m:0s (0h:0m:3s left)
Iteration 70 / LogL: -1382.125 / Time: 0h:0m:1s (0h:0m:2s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.049
Iteration 80 / LogL: -1382.066 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 90 / LogL: -1382.460 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 100 / LogL: -1382.052 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 110 / LogL: -1382.151 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 120 / LogL: -1382.525 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 130 / LogL: -1382.063 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 140 / LogL: -1382.170 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 150 / LogL: -1382.465 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 160 / LogL: -1382.890 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 170 / LogL: -1384.907 / Time: 0h:0m:2s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.049
Iteration 180 / LogL: -1382.294 / Time: 0h:0m:2s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.049
Iteration 190 / LogL: -1382.049 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 200 / LogL: -1382.083 / Time: 0h:0m:2s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.049
Iteration 210 / LogL: -1382.386 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 220 / LogL: -1382.085 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 230 / LogL: -1382.049 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 240 / LogL: -1382.955 / Time: 0h:0m:2s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 249 ITERATIONS / Time: 0h:0m:2s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.049
Optimal log-likelihood: -1382.043
Rate parameters:  A-C: 0.18904  A-G: 1.81946  A-T: 1.51719  C-G: 0.76290  C-T: 1.81946  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.719,0.422) (0.281,2.481)
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1382.043
Total tree length: 6.894

Total number of iterations: 249
CPU time used for tree search: 2.851 sec (0h:0m:2s)
Wall-clock time used for tree search: 2.856 sec (0h:0m:2s)
Total CPU time used: 2.905 sec (0h:0m:2s)
Total wall-clock time used: 2.912 sec (0h:0m:2s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree.log

Date and Time: Fri Nov  1 07:41:18 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g464_3bj/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp9crp_hew/q2iqtree -nt 1 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-fast-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-fast-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-fast-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-nnisi-fast-tree.qza">download</a></p></li>
</ul>
</div>
</div>
</div>
<div class="section" id="iqtree-ultrafast-bootstrap">
<h3><a class="toc-backref" href="#id17">iqtree-ultrafast-bootstrap</a><a class="headerlink" href="#iqtree-ultrafast-bootstrap" title="Permalink to this headline">¶</a></h3>
<p>As per our discussion in the <code class="docutils literal notranslate"><span class="pre">raxml-rapid-bootstrap</span></code> section above, we can
also use IQ-TREE to evaluate how well our splits / bipartitions are supported
within our phylogeny via the <a class="reference external" href="https://doi.org/10.1093/molbev/msx281">ultrafast bootstrap algorithm</a>. Below, we’ll
apply the plugin’s
<a class="reference internal" href="../../plugins/available/phylogeny/iqtree-ultrafast-bootstrap/"><span class="doc">ultrafast bootstrap command</span></a>:
automatic model selection (<code class="docutils literal notranslate"><span class="pre">MFP</span></code>), perform <code class="docutils literal notranslate"><span class="pre">1000</span></code> bootstrap replicates
(minimum required), set the same generally suggested parameters for
constructing a phylogeny from short sequences, and automatically determine the
optimal number of CPU cores to use:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree-ultrafast-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --o-tree iqt-nnisi-bootstrap-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-9tle8d2n/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot -nt 1 -nstop 200 -pers 0.200000
Seed:    709837 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:41:22 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-9tle8d2n/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1425.951     37  2925.902     2941.879     3050.443
  2  JC+I          1424.734     38  2925.467     2942.404     3053.374
  3  JC+G4         1416.622     38  2909.244     2926.181     3037.151
  4  JC+I+G4       1416.980     39  2911.960     2929.891     3043.233
  5  JC+R2         1412.279     39  2902.559     2920.490     3033.832
  6  JC+R3         1412.210     41  2906.420     2926.444     3044.425
 14  F81+F         1423.050     40  2926.099     2945.059     3060.738
 15  F81+F+I       1421.729     41  2925.458     2945.481     3063.463
 16  F81+F+G4      1412.578     41  2907.156     2927.179     3045.161
 17  F81+F+I+G4    1412.897     42  2909.793     2930.916     3051.164
 18  F81+F+R2      1407.514     42  2899.028     2920.151     3040.399
 19  F81+F+R3      1407.499     44  2902.998     2926.430     3051.101
 27  K2P           1417.585     38  2911.171     2928.108     3039.078
 28  K2P+I         1416.545     39  2911.090     2929.021     3042.363
 29  K2P+G4        1407.275     39  2892.551     2910.482     3023.824
 30  K2P+I+G4      1407.564     40  2895.128     2914.088     3029.767
 31  K2P+R2        1401.745     40  2883.489     2902.449     3018.128
 32  K2P+R3        1401.738     42  2887.476     2908.599     3028.847
 40  HKY+F         1414.905     41  2911.810     2931.833     3049.815
 41  HKY+F+I       1413.632     42  2911.264     2932.387     3052.635
 42  HKY+F+G4      1403.307     42  2890.615     2911.738     3031.986
 43  HKY+F+I+G4    1403.502     43  2893.005     2915.264     3037.742
 44  HKY+F+R2      1397.137     43  2880.274     2902.533     3025.011
 45  HKY+F+R3      1397.135     45  2884.270     2908.913     3035.739
 53  TNe           1417.468     39  2912.937     2930.868     3044.210
 54  TNe+I         1416.284     40  2912.569     2931.528     3047.208
 55  TNe+G4        1407.222     40  2894.443     2913.403     3029.082
 56  TNe+I+G4      1407.382     41  2896.763     2916.787     3034.768
 57  TNe+R2        1401.732     41  2885.465     2905.488     3023.470
 58  TNe+R3        1401.730     43  2889.460     2911.719     3034.197
 66  TN+F          1414.573     42  2913.146     2934.269     3054.517
 67  TN+F+I        1413.423     43  2912.846     2935.105     3057.583
 68  TN+F+G4       1402.504     43  2891.008     2913.267     3035.745
 69  TN+F+I+G4     1402.653     44  2893.306     2916.738     3041.409
 70  TN+F+R2       1395.971     44  2879.941     2903.373     3028.044
 71  TN+F+R3       1395.969     46  2883.937     2909.829     3038.772
 79  K3P           1415.881     39  2909.762     2927.693     3041.035
 80  K3P+I         1414.835     40  2909.669     2928.629     3044.308
 81  K3P+G4        1404.651     40  2889.301     2908.261     3023.940
 82  K3P+I+G4      1404.768     41  2891.536     2911.559     3029.541
 83  K3P+R2        1399.218     41  2880.436     2900.460     3018.441
 84  K3P+R3        1399.217     43  2884.434     2906.693     3029.171
 92  K3Pu+F        1412.817     42  2909.634     2930.756     3051.005
 93  K3Pu+F+I      1411.432     43  2908.865     2931.123     3053.601
 94  K3Pu+F+G4     1399.864     43  2885.728     2907.987     3030.465
 95  K3Pu+F+I+G4   1399.963     44  2887.926     2911.358     3036.029
 96  K3Pu+F+R2     1393.894     44  2875.787     2899.219     3023.890
 97  K3Pu+F+R3     1393.767     46  2879.534     2905.426     3034.369
105  TPM2+F        1414.863     42  2913.726     2934.848     3055.097
106  TPM2+F+I      1413.579     43  2913.157     2935.416     3057.894
107  TPM2+F+G4     1403.089     43  2892.177     2914.436     3036.914
108  TPM2+F+I+G4   1403.170     44  2894.341     2917.773     3042.444
109  TPM2+F+R2     1396.927     44  2881.855     2905.287     3029.958
110  TPM2+F+R3     1396.819     46  2885.638     2911.530     3040.473
118  TPM2u+F       1414.863     42  2913.726     2934.849     3055.097
119  TPM2u+F+I     1413.578     43  2913.157     2935.416     3057.894
120  TPM2u+F+G4    1403.090     43  2892.180     2914.439     3036.917
121  TPM2u+F+I+G4  1403.158     44  2894.315     2917.747     3042.418
122  TPM2u+F+R2    1396.926     44  2881.851     2905.283     3029.954
123  TPM2u+F+R3    1396.816     46  2885.632     2911.525     3040.467
131  TPM3+F        1411.656     42  2907.312     2928.435     3048.683
132  TPM3+F+I      1410.677     43  2907.354     2929.613     3052.091
133  TPM3+F+G4     1399.788     43  2885.576     2907.835     3030.313
134  TPM3+F+I+G4   1399.831     44  2887.663     2911.094     3035.765
135  TPM3+F+R2     1394.166     44  2876.333     2899.765     3024.436
136  TPM3+F+R3     1394.077     46  2880.155     2906.047     3034.990
144  TPM3u+F       1411.674     42  2907.348     2928.470     3048.719
145  TPM3u+F+I     1410.672     43  2907.344     2929.603     3052.081
146  TPM3u+F+G4    1399.789     43  2885.579     2907.838     3030.316
147  TPM3u+F+I+G4  1399.821     44  2887.643     2911.075     3035.746
148  TPM3u+F+R2    1394.166     44  2876.333     2899.764     3024.435
149  TPM3u+F+R3    1394.077     46  2880.155     2906.047     3034.990
157  TIMe          1415.785     40  2911.571     2930.530     3046.210
158  TIMe+I        1414.585     41  2911.170     2931.193     3049.175
159  TIMe+G4       1404.613     41  2891.226     2911.249     3029.231
160  TIMe+I+G4     1404.650     42  2893.299     2914.422     3034.670
161  TIMe+R2       1399.218     42  2882.436     2903.559     3023.807
162  TIMe+R3       1399.148     44  2886.297     2909.729     3034.400
170  TIM+F         1412.458     43  2910.915     2933.174     3055.652
171  TIM+F+I       1411.201     44  2910.402     2933.834     3058.505
172  TIM+F+G4      1398.908     44  2885.815     2909.247     3033.918
173  TIM+F+I+G4    1398.937     45  2887.873     2912.516     3039.342
174  TIM+F+R2      1392.657     45  2875.314     2899.957     3026.783
175  TIM+F+R3      1392.516     47  2879.031     2906.212     3037.232
183  TIM2e         1417.385     40  2914.771     2933.730     3049.410
184  TIM2e+I       1416.186     41  2914.371     2934.394     3052.376
185  TIM2e+G4      1407.204     41  2896.407     2916.431     3034.412
186  TIM2e+I+G4    1407.224     42  2898.448     2919.571     3039.819
187  TIM2e+R2      1401.729     42  2887.458     2908.581     3028.829
188  TIM2e+R3      1401.687     44  2891.375     2914.807     3039.478
196  TIM2+F        1414.529     43  2915.059     2937.318     3059.796
197  TIM2+F+I      1413.365     44  2914.731     2938.163     3062.834
198  TIM2+F+G4     1402.305     44  2892.611     2916.043     3040.714
199  TIM2+F+I+G4   1402.320     45  2894.640     2919.282     3046.108
200  TIM2+F+R2     1395.758     45  2881.515     2906.158     3032.984
201  TIM2+F+R3     1395.633     47  2885.267     2912.447     3043.468
209  TIM3e         1410.612     40  2901.223     2920.183     3035.862
210  TIM3e+I       1409.721     41  2901.442     2921.466     3039.447
211  TIM3e+G4      1399.508     41  2881.017     2901.040     3019.022
212  TIM3e+I+G4    1399.522     42  2883.044     2904.167     3024.415
213  TIM3e+R2      1394.831     42  2873.661     2894.784     3015.032
214  TIM3e+R3      1394.801     44  2877.602     2901.034     3025.705
222  TIM3+F        1411.254     43  2908.507     2930.766     3053.244
223  TIM3+F+I      1410.409     44  2908.818     2932.250     3056.921
224  TIM3+F+G4     1399.010     44  2886.019     2909.451     3034.122
225  TIM3+F+I+G4   1398.998     45  2887.996     2912.639     3039.465
226  TIM3+F+R2     1392.871     45  2875.742     2900.385     3027.211
227  TIM3+F+R3     1392.767     47  2879.535     2906.715     3037.736
235  TVMe          1407.398     41  2896.797     2916.820     3034.802
236  TVMe+I        1406.564     42  2897.128     2918.251     3038.499
237  TVMe+G4       1395.541     42  2875.082     2896.205     3016.453
238  TVMe+I+G4     1395.536     43  2877.073     2899.331     3021.810
239  TVMe+R2       1391.107     43  2868.214     2890.473     3012.951
240  TVMe+R3       1391.081     45  2872.161     2896.804     3023.630
248  TVM+F         1408.335     44  2904.670     2928.102     3052.773
249  TVM+F+I       1407.262     45  2904.525     2929.168     3055.994
250  TVM+F+G4      1395.255     45  2880.509     2905.152     3031.978
251  TVM+F+I+G4    1395.266     46  2882.533     2908.425     3037.368
252  TVM+F+R2      1389.997     46  2871.994     2897.886     3026.829
253  TVM+F+R3      1389.897     48  2875.795     2904.304     3037.362
261  SYM           1407.311     42  2898.622     2919.745     3039.993
262  SYM+I         1406.388     43  2898.776     2921.034     3043.513
263  SYM+G4        1395.491     43  2876.983     2899.242     3021.720
264  SYM+I+G4      1395.502     44  2879.003     2902.435     3027.106
265  SYM+R2        1391.101     44  2870.201     2893.633     3018.304
266  SYM+R3        1391.069     46  2874.138     2900.030     3028.972
274  GTR+F         1407.885     45  2905.771     2930.414     3057.240
275  GTR+F+I       1406.999     46  2905.999     2931.891     3060.834
276  GTR+F+G4      1394.439     46  2880.878     2906.770     3035.713
277  GTR+F+I+G4    1394.444     47  2882.888     2910.068     3041.088
278  GTR+F+R2      1388.761     47  2871.522     2898.703     3029.723
279  GTR+F+R3      1388.649     49  2875.298     2905.176     3040.231
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.model.gz
CPU time for ModelFinder: 0.966 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.972 seconds (0h:0m:0s)
Generating 1000 samples for ultrafast bootstrap (seed: 709837)...

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1417.970
2. Current log-likelihood: -1393.853
3. Current log-likelihood: -1392.324
4. Current log-likelihood: -1391.652
5. Current log-likelihood: -1391.390
6. Current log-likelihood: -1391.246
Optimal log-likelihood: -1391.145
Rate parameters:  A-C: 0.13929  A-G: 1.83346  A-T: 1.48449  C-G: 0.74342  C-T: 1.83346  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.682,0.380) (0.318,2.330)
Parameters optimization took 6 rounds (0.038 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.211
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.087 second
Computing log-likelihood of 98 initial trees ... 0.137 seconds
Current best score: -1389.211

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.344
Iteration 10 / LogL: -1382.351 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.354 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.344 / CPU time: 0.609
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
UPDATE BEST LOG-LIKELIHOOD: -1382.343
UPDATE BEST LOG-LIKELIHOOD: -1382.343
UPDATE BEST LOG-LIKELIHOOD: -1382.343
Iteration 30 / LogL: -1383.049 / Time: 0h:0m:0s (0h:0m:5s left)
Iteration 40 / LogL: -1382.871 / Time: 0h:0m:1s (0h:0m:4s left)
Iteration 50 / LogL: -1382.907 / Time: 0h:0m:1s (0h:0m:3s left)
Log-likelihood cutoff on original alignment: -1414.398
Iteration 60 / LogL: -1382.877 / Time: 0h:0m:1s (0h:0m:3s left)
Estimate model parameters (epsilon = 0.100)
UPDATE BEST LOG-LIKELIHOOD: -1382.098
Iteration 70 / LogL: -1382.915 / Time: 0h:0m:1s (0h:0m:3s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 71: -1382.026
UPDATE BEST LOG-LIKELIHOOD: -1382.019
UPDATE BEST LOG-LIKELIHOOD: -1382.019
Iteration 80 / LogL: -1382.097 / Time: 0h:0m:1s (0h:0m:4s left)
Iteration 90 / LogL: -1382.424 / Time: 0h:0m:2s (0h:0m:4s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.019
Iteration 100 / LogL: -1382.020 / Time: 0h:0m:2s (0h:0m:3s left)
Log-likelihood cutoff on original alignment: -1415.123
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.976
NOTE: UFBoot does not converge, continue at least 100 more iterations
Iteration 110 / LogL: -1382.875 / Time: 0h:0m:2s (0h:0m:3s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.018
Iteration 120 / LogL: -1382.018 / Time: 0h:0m:2s (0h:0m:3s left)
Iteration 130 / LogL: -1382.440 / Time: 0h:0m:2s (0h:0m:3s left)
Iteration 140 / LogL: -1382.441 / Time: 0h:0m:3s (0h:0m:2s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.018
Iteration 150 / LogL: -1382.423 / Time: 0h:0m:3s (0h:0m:2s left)
Log-likelihood cutoff on original alignment: -1415.123
Iteration 160 / LogL: -1382.065 / Time: 0h:0m:3s (0h:0m:2s left)
Iteration 170 / LogL: -1382.100 / Time: 0h:0m:3s (0h:0m:2s left)
Iteration 180 / LogL: -1382.646 / Time: 0h:0m:3s (0h:0m:1s left)
Iteration 190 / LogL: -1382.103 / Time: 0h:0m:4s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.018
Iteration 200 / LogL: -1382.019 / Time: 0h:0m:4s (0h:0m:1s left)
Log-likelihood cutoff on original alignment: -1415.806
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.997
Iteration 210 / LogL: -1382.030 / Time: 0h:0m:4s (0h:0m:1s left)
Iteration 220 / LogL: -1382.031 / Time: 0h:0m:4s (0h:0m:1s left)
Iteration 230 / LogL: -1382.031 / Time: 0h:0m:4s (0h:0m:1s left)
Iteration 240 / LogL: -1382.096 / Time: 0h:0m:5s (0h:0m:1s left)
Iteration 250 / LogL: -1382.019 / Time: 0h:0m:5s (0h:0m:1s left)
Log-likelihood cutoff on original alignment: -1415.806
Iteration 260 / LogL: -1385.409 / Time: 0h:0m:5s (0h:0m:0s left)
Iteration 270 / LogL: -1382.044 / Time: 0h:0m:5s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 272 ITERATIONS / Time: 0h:0m:5s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.018
Optimal log-likelihood: -1382.012
Rate parameters:  A-C: 0.18912  A-G: 1.82226  A-T: 1.51736  C-G: 0.76287  C-T: 1.82226  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.721,0.413) (0.279,2.515)
Parameters optimization took 1 rounds (0.005 sec)
BEST SCORE FOUND : -1382.012
Creating bootstrap support values...
Split supports printed to NEXUS file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.splits.nex
Total tree length: 7.029

Total number of iterations: 272
CPU time used for tree search: 5.817 sec (0h:0m:5s)
Wall-clock time used for tree search: 5.833 sec (0h:0m:5s)
Total CPU time used: 5.954 sec (0h:0m:5s)
Total wall-clock time used: 5.975 sec (0h:0m:5s)

Computing bootstrap consensus tree...
Reading input file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.splits.nex...
20 taxa and 145 splits.
Consensus tree written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.contree
Reading input trees file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.contree
Log-likelihood of consensus tree: -1382.012

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.splits.nex
  Consensus tree:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.contree
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot.log

Date and Time: Fri Nov  1 07:41:29 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-9tle8d2n/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpnrx1h1ib/q2iqtreeufboot -nt 1 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-bootstrap-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-nnisi-bootstrap-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="perform-single-branch-tests-alongside-ufboot">
<h4><a class="toc-backref" href="#id18">Perform single branch tests alongside ufboot</a><a class="headerlink" href="#perform-single-branch-tests-alongside-ufboot" title="Permalink to this headline">¶</a></h4>
<p>We can also apply single branch test methods concurrently with ultrafast
bootstrapping. The support values will always be represented in the following
order: <em>alrt / lbp / abayes / ufboot</em>. Again, these values can be seen as
separately listed bootstrap values in <a class="reference external" href="https://itol.embl.de/">iTOL</a>. We’ll also specify a model as we
did earlier.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree-ultrafast-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --p-alrt <span class="m">1000</span> <span class="se">\</span>
  --p-abayes <span class="se">\</span>
  --p-lbp <span class="m">1000</span> <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-hekktd9a/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot -nt 1 -alrt 1000 -abayes -lbp 1000 -nstop 200 -pers 0.200000
Seed:    960019 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Nov  1 07:41:33 2019
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-hekktd9a/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
Generating 1000 samples for ultrafast bootstrap (seed: 960019)...

NOTE: 1 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.237 / LogL: -1394.542
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.010, 1.344 / LogL: -1394.883
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.010, 1.353 / LogL: -1394.887
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.010, 1.352 / LogL: -1394.874
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.009, 1.348 / LogL: -1394.840
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.351 / LogL: -1394.865
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.010, 1.352 / LogL: -1394.886
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.346 / LogL: -1394.827
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.009, 1.347 / LogL: -1394.838
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.009, 1.347 / LogL: -1394.849
Optimal pinv,alpha: 0.000, 1.237 / LogL: -1394.542

Parameters optimization took 0.465 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.865
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.084 second
Computing log-likelihood of 98 initial trees ... 0.200 seconds
Current best score: -1392.865

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.265
UPDATE BEST LOG-LIKELIHOOD: -1387.265
Iteration 10 / LogL: -1387.265 / Time: 0h:0m:1s
Iteration 20 / LogL: -1387.279 / Time: 0h:0m:1s
Finish initializing candidate tree set (1)
Current best tree score: -1387.265 / CPU time: 0.799
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
UPDATE BEST LOG-LIKELIHOOD: -1387.259
UPDATE BEST LOG-LIKELIHOOD: -1387.257
Iteration 30 / LogL: -1387.347 / Time: 0h:0m:1s (0h:0m:9s left)
Iteration 40 / LogL: -1387.332 / Time: 0h:0m:1s (0h:0m:7s left)
Iteration 50 / LogL: -1387.414 / Time: 0h:0m:1s (0h:0m:6s left)
Log-likelihood cutoff on original alignment: -1405.359
Iteration 60 / LogL: -1387.347 / Time: 0h:0m:2s (0h:0m:5s left)
Iteration 70 / LogL: -1387.347 / Time: 0h:0m:2s (0h:0m:5s left)
Iteration 80 / LogL: -1387.348 / Time: 0h:0m:3s (0h:0m:4s left)
Iteration 90 / LogL: -1387.347 / Time: 0h:0m:3s (0h:0m:4s left)
Iteration 100 / LogL: -1389.337 / Time: 0h:0m:3s (0h:0m:3s left)
Log-likelihood cutoff on original alignment: -1406.775
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998
Iteration 110 / LogL: -1387.347 / Time: 0h:0m:3s (0h:0m:3s left)
Iteration 120 / LogL: -1387.614 / Time: 0h:0m:4s (0h:0m:2s left)
Iteration 130 / LogL: -1387.347 / Time: 0h:0m:4s (0h:0m:2s left)
Iteration 140 / LogL: -1387.446 / Time: 0h:0m:4s (0h:0m:2s left)
Iteration 150 / LogL: -1387.543 / Time: 0h:0m:5s (0h:0m:1s left)
Log-likelihood cutoff on original alignment: -1406.775
Iteration 160 / LogL: -1387.347 / Time: 0h:0m:5s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.256
Iteration 170 / LogL: -1387.349 / Time: 0h:0m:5s (0h:0m:1s left)
Iteration 180 / LogL: -1387.347 / Time: 0h:0m:6s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.256
Iteration 190 / LogL: -1387.415 / Time: 0h:0m:6s (0h:0m:0s left)
Iteration 200 / LogL: -1387.361 / Time: 0h:0m:6s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -1406.708
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998
TREE SEARCH COMPLETED AFTER 202 ITERATIONS / Time: 0h:0m:6s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.256
Optimal log-likelihood: -1387.253
Rate parameters:  A-C: 0.32957  A-G: 2.26332  A-T: 2.14163  C-G: 1.17419  C-T: 3.29400  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.318
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1387.253

Testing tree branches by SH-like aLRT with 1000 replicates...
Testing tree branches by local-BP test with 1000 replicates...
Testing tree branches by aBayes parametric test...
0.055 sec.
Creating bootstrap support values...
Split supports printed to NEXUS file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.splits.nex
Total tree length: 6.746

Total number of iterations: 202
CPU time used for tree search: 6.472 sec (0h:0m:6s)
Wall-clock time used for tree search: 6.486 sec (0h:0m:6s)
Total CPU time used: 7.094 sec (0h:0m:7s)
Total wall-clock time used: 7.114 sec (0h:0m:7s)

Computing bootstrap consensus tree...
Reading input file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.splits.nex...
20 taxa and 147 splits.
Consensus tree written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.contree
Reading input trees file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.contree
Log-likelihood of consensus tree: -1387.430

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.splits.nex
  Consensus tree:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.contree
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot.log

Date and Time: Fri Nov  1 07:41:41 2019
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-hekktd9a/f3132e20-be7c-4b66-9769-15edeb3b3842/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpwbblgu6y/q2iqtreeufboot -nt 1 -alrt 1000 -abayes -lbp 1000 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-bootstrap-sbt-gtrig-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-sbt-gtrig-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If there is a need to reduce the impact of <a class="reference external" href="http://www.iqtree.org/doc/Tutorial#reducing-impact-of-severe-model-violations-with-ufboot">potential model
violations</a> that occur during a <a class="reference external" href="https://doi.org/10.1093/molbev/msx281">UFBoot search</a>, and / or would simply
like to be more rigorous, we can add the <code class="docutils literal notranslate"><span class="pre">--p-bnni</span></code> option to any of the
<code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code> commands above.</p>
</div>
</div>
</div>
<div class="section" id="root-the-phylogeny">
<h3><a class="toc-backref" href="#id19">Root the phylogeny</a><a class="headerlink" href="#root-the-phylogeny" title="Permalink to this headline">¶</a></h3>
<p>In order to make proper use of diversity metrics such as UniFrac, the phylogeny
must be <a class="reference external" href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics/what-phylogeny/aspects-phylogenies/nodes/root">rooted</a>. Typically an <a class="reference external" href="http://phylobotanist.blogspot.com/2015/01/how-to-root-phylogenetic-tree-outgroup.html">outgroup</a> is chosen when rooting a tree. In
general, phylogenetic inference tools using Maximum Likelihood often return an
unrooted tree by default.</p>
<p>QIIME 2 provides a way to
<a class="reference internal" href="../../plugins/available/phylogeny/midpoint-root/"><span class="doc">mid-point root</span></a> our
phylogeny. Other rooting options may be available in the future. For now, we’ll
root our bootstrap tree from <code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code> like so:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny midpoint-root <span class="se">\</span>
  --i-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza <span class="se">\</span>
  --o-rooted-tree iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>iTOL viewing Reminder</strong>. We can view our tree and its associated
alignment via <a class="reference external" href="https://itol.embl.de/">iTOL</a>. All you need to do is upload the
<cite>iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza</cite> tree file. Display the tree
in <cite>Normal</cite> mode. Then drag and drop the <cite>masked-aligned-rep-seqs.qza</cite> file
onto the visualization. Now you can view the phylogeny alongside the
alignment.</p>
</div>
</div>
</div>
<div class="section" id="pipelines">
<h2><a class="toc-backref" href="#id20">Pipelines</a><a class="headerlink" href="#pipelines" title="Permalink to this headline">¶</a></h2>
<p>Here we will outline the use of the phylogeny pipeline
<a class="reference internal" href="../../plugins/available/phylogeny/align-to-tree-mafft-fasttree/"><span class="doc">align-to-tree-mafft-fasttree</span></a></p>
<p>One advantage of pipelines is that they combine ordered sets of commonly used
commands, into one condensed simple command. To keep these “convenience”
pipelines easy to use, it is quite common to only expose a few options to the
user. That is, most of the commands executed via pipelines are often configured
to use default option settings. However, options that are deemed important
enough for the user to consider setting, are made available. The options
exposed via a given pipeline will largely depend upon what it is doing.
Pipelines are also a great way for new users to get started, as it helps to lay
a foundation of good practices in setting up standard operating procedures.</p>
<p>Rather than run one or more of the following QIIME 2 commands listed below:</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">alignment</span> <span class="pre">mafft</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">alignment</span> <span class="pre">mask</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">phylogeny</span> <span class="pre">fasttree</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">phylogeny</span> <span class="pre">midpoint-root</span> <span class="pre">...</span></code></p></li>
</ol>
<p>We can make use of the pipeline
<a class="reference internal" href="../../plugins/available/phylogeny/align-to-tree-mafft-fasttree/"><span class="doc">align-to-tree-mafft-fasttree</span></a>
to automate the above four steps in one go. Here is the description taken from
the pipeline help doc:</p>
<blockquote>
<div><p>This pipeline will start by creating a sequence alignment using MAFFT,
after which any alignment columns that are phylogenetically uninformative
or ambiguously aligned will be removed (masked). The resulting masked
alignment will be used to infer a phylogenetic tree and then subsequently
rooted at its midpoint. Output files from each step of the pipeline will be
saved. This includes both the unmasked and masked MAFFT alignment from
q2-alignment methods, and both the rooted and unrooted phylogenies from
q2-phylogeny methods.</p>
</div></blockquote>
<p>This can all be accomplished by simply running the following:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny align-to-tree-mafft-fasttree <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --output-dir mafft-fasttree-output
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/masked_alignment.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Fmasked_alignment.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/mafft-fasttree-output/masked_alignment.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/rooted_tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Frooted_tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/mafft-fasttree-output/rooted_tree.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/alignment.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Falignment.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/mafft-fasttree-output/alignment.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2019.10%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Ftree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2019.10/data/tutorials/phylogeny/mafft-fasttree-output/tree.qza">download</a></p></li>
</ul>
</div>
<p><strong>Congratulations! You now know how to construct a phylogeny in QIIME 2!</strong></p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="../qiime2-for-experienced-microbiome-researchers/">QIIME 2 for Experienced Microbiome Researchers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">“Moving Pictures” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">“Atacama soil microbiome” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pd-mice/">Parkinson’s Mouse Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../otu-clustering/">Clustering sequences into OTUs using q2-vsearch</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Phylogenetic inference with q2-phylogeny</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2019, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.1.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>