<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>classify-hybrid-vsearch-sklearn: ALPHA Hybrid classifier: VSEARCH exact match + sklearn classifier &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script src="../../../../_static/jquery.js"></script>
    <script src="../../../../_static/underscore.js"></script>
    <script src="../../../../_static/doctools.js"></script>
    <script src="../../../../_static/language_data.js"></script>
    <script src="../../../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="index" title="Index" href="../../../../genindex/" />
    <link rel="search" title="Search" href="../../../../search/" />
    <link rel="next" title="classify-consensus-blast: BLAST+ consensus taxonomy classifier" href="../classify-consensus-blast/" />
    <link rel="prev" title="feature-classifier" href="../" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="classify-hybrid-vsearch-sklearn-alpha-hybrid-classifier-vsearch-exact-match-sklearn-classifier">
<h1>classify-hybrid-vsearch-sklearn: ALPHA Hybrid classifier: VSEARCH exact match + sklearn classifier<a class="headerlink" href="#classify-hybrid-vsearch-sklearn-alpha-hybrid-classifier-vsearch-exact-match-sklearn-classifier" title="Permalink to this headline">Â¶</a></h1>
<div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime feature-classifier classify-hybrid-vsearch-sklearn
           [OPTIONS]

  NOTE: THIS PIPELINE IS AN ALPHA RELEASE. Please report bugs to
  https://forum.qiime2.org! Assign taxonomy to query sequences using hybrid
  classifier. First performs rough positive filter to remove artifact and
  low-coverage sequences (use "prefilter" parameter to toggle this step on
  or off). Second, performs VSEARCH exact match between query and
  reference_reads to find exact matches, followed by least common ancestor
  consensus taxonomy assignment from among maxaccepts top hits,
  min_consensus of which share that taxonomic assignment. Query sequences
  without an exact match are then classified with a pre-trained sklearn
  taxonomy classifier to predict the most likely taxonomic lineage.

Inputs:
  --i-query ARTIFACT FeatureData[Sequence]
                        Sequences to classify taxonomically.        [required]
  --i-reference-reads ARTIFACT FeatureData[Sequence]
                        reference sequences.                        [required]
  --i-reference-taxonomy ARTIFACT FeatureData[Taxonomy]
                        reference taxonomy labels.                  [required]
  --i-classifier ARTIFACT TaxonomicClassifier
                        Pre-trained sklearn taxonomic classifier for
                        classifying the reads.                      [required]
Parameters:
  --p-maxaccepts VALUE Int % Range(1, None) | Str % Choices('all')
                        Maximum number of hits to keep for each query. Set to
                        "all" to keep all hits > perc-identity similarity.
                        Note that if strand=both, maxaccepts will keep N hits
                        for each direction (if searches in the opposite
                        direction yield results that exceed the minimum
                        perc-identity). In those cases use maxhits to control
                        the total number of hits returned. This option works
                        in pair with maxrejects. The search process sorts
                        target sequences by decreasing number of k-mers they
                        have in common with the query sequence, using that
                        information as a proxy for sequence similarity. After
                        pairwise alignments, if the first target sequence
                        passes the acceptation criteria, it is accepted as
                        best hit and the search process stops for that query.
                        If maxaccepts is set to a higher value, more hits are
                        accepted. If maxaccepts and maxrejects are both set to
                        "all", the complete database is searched.
                                                                 [default: 10]
  --p-perc-identity PROPORTION Range(0.0, 1.0, inclusive_end=True)
                        Percent sequence similarity to use for PREFILTER.
                        Reject match if percent identity to query is lower.
                        Set to a lower value to perform a rough pre-filter.
                        This parameter is ignored if `prefilter` is disabled.
                                                                [default: 0.5]
  --p-query-cov PROPORTION Range(0.0, 1.0, inclusive_end=True)
                        Query coverage threshold to use for PREFILTER. Reject
                        match if query alignment coverage per high-scoring
                        pair is lower. Set to a lower value to perform a rough
                        pre-filter. This parameter is ignored if `prefilter`
                        is disabled.                            [default: 0.8]
  --p-strand TEXT Choices('both', 'plus')
                        Align against reference sequences in forward ("plus")
                        or both directions ("both").         [default: 'both']
  --p-min-consensus NUMBER Range(0.5, 1.0, inclusive_start=False,
    inclusive_end=True) Minimum fraction of assignments must match top hit to
                        be accepted as consensus assignment.   [default: 0.51]
  --p-maxhits VALUE Int % Range(1, None) | Str % Choices('all')
                                                              [default: 'all']
  --p-maxrejects VALUE Int % Range(1, None) | Str % Choices('all')
                                                              [default: 'all']
  --p-reads-per-batch INTEGER
    Range(0, None)      Number of reads to process in each batch for sklearn
                        classification. If "auto", this parameter is
                        autoscaled to min(number of query sequences / threads,
                        20000).                                   [default: 0]
  --p-confidence VALUE Float % Range(0, 1, inclusive_end=True) | Str %
    Choices('disable')  Confidence threshold for limiting taxonomic depth.
                        Set to "disable" to disable confidence calculation, or
                        0 to calculate confidence but not apply it to limit
                        the taxonomic depth of the assignments. [default: 0.7]
  --p-read-orientation TEXT Choices('same', 'reverse-complement', 'auto')
                        Direction of reads with respect to reference
                        sequences in pre-trained sklearn classifier. same will
                        cause reads to be classified unchanged;
                        reverse-complement will cause reads to be reversed and
                        complemented prior to classification. "auto" will
                        autodetect orientation based on the confidence
                        estimates for the first 100 reads.   [default: 'auto']
  --p-threads INTEGER   Number of threads to use for job parallelization.
    Range(1, None)                                                [default: 1]
  --p-prefilter / --p-no-prefilter
                        Toggle positive filter of query sequences on or off.
                                                               [default: True]
  --p-sample-size INTEGER
    Range(1, None)      Randomly extract the given number of sequences from
                        the reference database to use for prefiltering. This
                        parameter is ignored if `prefilter` is disabled.
                                                               [default: 1000]
  --p-randseed INTEGER  Use integer as a seed for the pseudo-random generator
    Range(0, None)      used during prefiltering. A given seed always produces
                        the same output, which is useful for replicability.
                        Set to 0 to use a pseudo-random seed. This parameter
                        is ignored if `prefilter` is disabled.    [default: 0]
Outputs:
  --o-classification ARTIFACT FeatureData[Taxonomy]
                        The resulting taxonomy classifications.     [required]
Miscellaneous:
  --output-dir PATH     Output unspecified results to a directory
  --verbose / --quiet   Display verbose output to stdout and/or stderr during
                        execution of this action. Or silence output if
                        execution is successful (silence is golden).
  --citations           Show citations and exit.
  --help                Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>from qiime2.plugins.feature_classifier.pipelines import classify_hybrid_vsearch_sklearn
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>ALPHA Hybrid classifier: VSEARCH exact match + sklearn classifier

NOTE: THIS PIPELINE IS AN ALPHA RELEASE. Please report bugs to
https://forum.qiime2.org! Assign taxonomy to query sequences using hybrid
classifier. First performs rough positive filter to remove artifact and
low-coverage sequences (use "prefilter" parameter to toggle this step on or
off). Second, performs VSEARCH exact match between query and
reference_reads to find exact matches, followed by least common ancestor
consensus taxonomy assignment from among maxaccepts top hits, min_consensus
of which share that taxonomic assignment. Query sequences without an exact
match are then classified with a pre-trained sklearn taxonomy classifier to
predict the most likely taxonomic lineage.

Parameters
----------
query : FeatureData[Sequence]
    Sequences to classify taxonomically.
reference_reads : FeatureData[Sequence]
    reference sequences.
reference_taxonomy : FeatureData[Taxonomy]
    reference taxonomy labels.
classifier : TaxonomicClassifier
    Pre-trained sklearn taxonomic classifier for classifying the reads.
maxaccepts : Int % Range(1, None) | Str % Choices('all'), optional
    Maximum number of hits to keep for each query. Set to "all" to keep all
    hits > perc_identity similarity. Note that if strand=both, maxaccepts
    will keep N hits for each direction (if searches in the opposite
    direction yield results that exceed the minimum perc_identity). In
    those cases use maxhits to control the total number of hits returned.
    This option works in pair with maxrejects. The search process sorts
    target sequences by decreasing number of k-mers they have in common
    with the query sequence, using that information as a proxy for sequence
    similarity. After pairwise alignments, if the first target sequence
    passes the acceptation criteria, it is accepted as best hit and the
    search process stops for that query. If maxaccepts is set to a higher
    value, more hits are accepted. If maxaccepts and maxrejects are both
    set to "all", the complete database is searched.
perc_identity : Float % Range(0.0, 1.0, inclusive_end=True), optional
    Percent sequence similarity to use for PREFILTER. Reject match if
    percent identity to query is lower. Set to a lower value to perform a
    rough pre-filter. This parameter is ignored if `prefilter` is disabled.
query_cov : Float % Range(0.0, 1.0, inclusive_end=True), optional
    Query coverage threshold to use for PREFILTER. Reject match if query
    alignment coverage per high-scoring pair is lower. Set to a lower value
    to perform a rough pre-filter. This parameter is ignored if `prefilter`
    is disabled.
strand : Str % Choices('both', 'plus'), optional
    Align against reference sequences in forward ("plus") or both
    directions ("both").
min_consensus : Float % Range(0.5, 1.0, inclusive_start=False, inclusive_end=True), optional
    Minimum fraction of assignments must match top hit to be accepted as
    consensus assignment.
maxhits : Int % Range(1, None) | Str % Choices('all'), optional
maxrejects : Int % Range(1, None) | Str % Choices('all'), optional
reads_per_batch : Int % Range(0, None), optional
    Number of reads to process in each batch for sklearn classification. If
    "auto", this parameter is autoscaled to min(number of query sequences /
    threads, 20000).
confidence : Float % Range(0, 1, inclusive_end=True) | Str % Choices('disable'), optional
    Confidence threshold for limiting taxonomic depth. Set to "disable" to
    disable confidence calculation, or 0 to calculate confidence but not
    apply it to limit the taxonomic depth of the assignments.
read_orientation : Str % Choices('same', 'reverse-complement', 'auto'), optional
    Direction of reads with respect to reference sequences in pre-trained
    sklearn classifier. same will cause reads to be classified unchanged;
    reverse-complement will cause reads to be reversed and complemented
    prior to classification. "auto" will autodetect orientation based on
    the confidence estimates for the first 100 reads.
threads : Int % Range(1, None), optional
    Number of threads to use for job parallelization.
prefilter : Bool, optional
    Toggle positive filter of query sequences on or off.
sample_size : Int % Range(1, None), optional
    Randomly extract the given number of sequences from the reference
    database to use for prefiltering. This parameter is ignored if
    `prefilter` is disabled.
randseed : Int % Range(0, None), optional
    Use integer as a seed for the pseudo-random generator used during
    prefiltering. A given seed always produces the same output, which is
    useful for replicability. Set to 0 to use a pseudo-random seed. This
    parameter is ignored if `prefilter` is disabled.

Returns
-------
classification : FeatureData[Taxonomy]
    The resulting taxonomy classifications.</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>