<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>evaluate-taxonomy: Evaluate expected vs. observed taxonomic assignments &#8212; QIIME 2 2018.11.0 documentation</title>
    
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../../',
        VERSION:     '2018.11.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="top" title="QIIME 2 2018.11.0 documentation" href="../../../../" />
    <link rel="up" title="quality-control" href="../" />
    <link rel="next" title="quality-filter" href="../../quality-filter/" />
    <link rel="prev" title="evaluate-seqs: Compare query (observed) vs. reference (expected) sequences." href="../evaluate-seqs/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="evaluate-taxonomy-evaluate-expected-vs-observed-taxonomic-assignments">
<h1>evaluate-taxonomy: Evaluate expected vs. observed taxonomic assignments<a class="headerlink" href="#evaluate-taxonomy-evaluate-expected-vs-observed-taxonomic-assignments" title="Permalink to this headline">Â¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><p id="bibtex-bibliography-plugins/available/quality-control/evaluate-taxonomy-0"><table class="docutils citation" frame="void" id="quality-control-evaluate-taxonomy-key0" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[quality-control:evaluate-taxonomy:BKR+18]</td><td>Nicholas&nbsp;A Bokulich, Benjamin&nbsp;D Kaehler, Jai&nbsp;Ram Rideout, Matthew Dillon, Evan Bolyen, Rob Knight, Gavin&nbsp;A Huttley, and J&nbsp;Gregory Caporaso. Optimizing taxonomic classification of marker-gene amplicon sequences with qiime 2&#8217;s q2-feature-classifier plugin. <em>Microbiome</em>, 2018.</td></tr>
</tbody>
</table>
</p>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime quality-control evaluate-taxonomy [OPTIONS]

  This visualizer compares a pair of observed and expected taxonomic
  assignments to calculate precision, recall, and F-measure at each
  taxonomic level, up to maximum level specified by the depth parameter.
  These metrics are calculated at each semicolon-delimited rank. This action
  is useful for comparing the accuracy of taxonomic assignment, e.g.,
  between different taxonomy classifiers or other bioinformatics methods.
  Expected taxonomies should be derived from simulated or mock community
  sequences that have known taxonomic affiliations.

Options:
  --i-expected-taxa ARTIFACT PATH FeatureData[Taxonomy]
                                  Expected taxonomic assignments  [required]
  --i-observed-taxa ARTIFACT PATH FeatureData[Taxonomy]
                                  Observed taxonomic assignments  [required]
  --p-depth INTEGER               Maximum depth of semicolon-delimited
                                  taxonomic ranks to test (e.g., 1 = root, 7 =
                                  species for the greengenes reference
                                  sequence database).  [required]
  --p-palette [Pastel1|Set3|Set1|tab20|Pastel2|magma|rainbow|Paired|inferno|Accent|tab10|plasma|Dark2|Set2|tab20b|tab20c|terrain|viridis]
                                  Color palette to utilize for plotting.
                                  [default: Set1]
  --p-require-exp-ids / --p-no-require-exp-ids
                                  Require that all features found in observed
                                  taxa must be found in expected taxa or raise
                                  error.  [default: True]
  --p-require-obs-ids / --p-no-require-obs-ids
                                  Require that all features found in expected
                                  taxa must be found in observed taxa or raise
                                  error.  [default: True]
  --i-feature-table ARTIFACT PATH FeatureTable[RelativeFrequency]
                                  Optional feature table containing relative
                                  frequency of each feature, used to weight
                                  accuracy scores by frequency. Must contain
                                  all features found in expected and/or
                                  observed taxa. Features found in the table
                                  but not the expected/observed taxa will be
                                  dropped prior to analysis.  [optional]
  --p-sample-id TEXT              Optional sample ID to use for extracting
                                  frequency data from feature table, and for
                                  labeling accuracy results. If no sample_id
                                  is provided, feature frequencies are derived
                                  from the sum of all samples present in the
                                  feature table.  [optional]
  --o-visualization VISUALIZATION PATH
                                  [required if not passing --output-dir]
  --output-dir DIRECTORY          Output unspecified results to a directory
  --cmd-config FILE               Use config file for command options
  --verbose                       Display verbose output to stdout and/or
                                  stderr during execution of this action.
                                  [default: False]
  --quiet                         Silence output if execution is successful
                                  (silence is golden).  [default: False]
  --citations                     Show citations and exit.
  --help                          Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell"><div class="highlight"><pre><span></span>from qiime2.plugins.quality_control.visualizers import evaluate_taxonomy
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Evaluate expected vs. observed taxonomic assignments

This visualizer compares a pair of observed and expected taxonomic
assignments to calculate precision, recall, and F-measure at each taxonomic
level, up to maximum level specified by the depth parameter. These metrics
are calculated at each semicolon-delimited rank. This action is useful for
comparing the accuracy of taxonomic assignment, e.g., between different
taxonomy classifiers or other bioinformatics methods. Expected taxonomies
should be derived from simulated or mock community sequences that have
known taxonomic affiliations.

Parameters
----------
expected_taxa : FeatureData[Taxonomy]
    Expected taxonomic assignments
observed_taxa : FeatureData[Taxonomy]
    Observed taxonomic assignments
depth : Int
    Maximum depth of semicolon-delimited taxonomic ranks to test (e.g., 1 =
    root, 7 = species for the greengenes reference sequence database).
palette : Str % Choices({'Accent', 'Dark2', 'Paired', 'Pastel1', 'Pastel2', 'Set1', 'Set2', 'Set3', 'inferno', 'magma', 'plasma', 'rainbow', 'tab10', 'tab20', 'tab20b', 'tab20c', 'terrain', 'viridis'}), optional
    Color palette to utilize for plotting.
require_exp_ids : Bool, optional
    Require that all features found in observed taxa must be found in
    expected taxa or raise error.
require_obs_ids : Bool, optional
    Require that all features found in expected taxa must be found in
    observed taxa or raise error.
feature_table : FeatureTable[RelativeFrequency], optional
    Optional feature table containing relative frequency of each feature,
    used to weight accuracy scores by frequency. Must contain all features
    found in expected and/or observed taxa. Features found in the table but
    not the expected/observed taxa will be dropped prior to analysis.
sample_id : Str, optional
    Optional sample ID to use for extracting frequency data from feature
    table, and for labeling accuracy results. If no sample_id is provided,
    feature frequencies are derived from the sum of all samples present in
    the feature table.

Returns
-------
visualization : Visualization</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2018, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>