<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Differential abundance analysis with gneiss &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/language_data.js"></script>
    <script src="../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="index" title="Index" href="../../genindex/" />
    <link rel="search" title="Search" href="../../search/" />
    <link rel="next" title="Importing data" href="../importing/" />
    <link rel="prev" title="Parkinson’s Mouse Tutorial" href="../pd-mice/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="differential-abundance-analysis-with-gneiss">
<h1>Differential abundance analysis with gneiss<a class="headerlink" href="#differential-abundance-analysis-with-gneiss" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a> and completed the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial</span></a>.</p>
</div>
<p>In this tutorial you will learn how to perform differential abundance analysis using balances in gneiss.  The main problem that we will focus on is how to identify differentially abundant taxa in a compositionally coherent way.</p>
<p>Compositionality refers to the issue of dealing with proportions.  To account for differences in sequencing depth, microbial abundances are typically interpreted as proportions (e.g. relative abundance).  Because of this, it becomes challenging to infer exactly which microbes are changing – since proportions add to one, the change of a single microbe will also change the proportions of the remaining microbes.</p>
<p>Consider the following example:</p>
<img alt="../../_images/gneiss-bars.jpg" src="../../_images/gneiss-bars.jpg" />
<p>On the left, we have the true abundances of ten species, and the first species doubles between Time point 1 and Time point 2.  When we normalize these to proportions, it appears as if all of the species have changed between the two time points.  Looking at proportions alone, we would never realize this problem, and we actually cannot exactly determine which species are changing based on proportions alone.</p>
<p>While we cannot exactly solve the problem of identifying differentially abundant species, we can relax this problem and ask which partitions of microbes are changing.  In the case above, if we compute the ratio between the first species and the second species, that ratio will be 1:1 at Time point 1, and 2:1 at Time point 2 for both the original abundances and the proportions.  This is the type of question that balances try to solve.
Rather than focusing on individual taxa, we can focus on the ratio between taxa (or groups of taxa), since these ratios are consist between the true abundances and the observed proportions of the species observed. We typically log transform these ratios for improved visualization (‘log ratios’). The concept of calculating balances (or ratios) for multiple species can be extended to trees as shown in the following example.</p>
<img alt="../../_images/gneiss-balances.jpg" src="../../_images/gneiss-balances.jpg" />
<p>On the left, we define a tree, where each of the tips corresponds to a taxon, and underneath are the proportions of each taxon in the first sample.  The internal nodes (i.e. balances) define the log ratio between the taxa underneath.  On the right is the same tree, and underneath are the proportions of each taxa in a different sample. Only one of the taxa abundances changes.  As we have observed before, the proportions of all of the taxa will change, but looking at the balances, only the balances containing the purple taxa will change.  In this case, balance <span class="math notranslate nohighlight">\(b_3\)</span> won’t change, since it only considers the ratio between the red and taxa.  By looking at balances instead proportions, we can eliminate some of the variance by restricting observations to only focus on the taxa within a given balance.</p>
<p>The outstanding question here is, how do we construct a balance tree to control for the variation, and identify interesting differentially abundant partitions of taxa?  In gneiss, there are three main ways that this can be done:</p>
<ol class="arabic simple">
<li><p>Correlation clustering.  If we don’t have relevant prior information about how to cluster together organisms, we can group together organisms based on how often they co-occur with each other. This is available in the <code class="docutils literal notranslate"><span class="pre">correlation-clustering</span></code> command and creates tree input for <code class="docutils literal notranslate"><span class="pre">ilr-hierarchical</span></code>.</p></li>
<li><p>Gradient clustering.  Use a metadata category to cluster taxa found in similar sample types. For example, if we want to evaluate if pH is a driving factor, we can cluster according to the pH that the taxa are observed in, and observe whether the ratios of low-pH organisms to high-pH organisms change as the pH changes.  This is available in the <code class="docutils literal notranslate"><span class="pre">gradient-clustering</span></code> command and creates tree input for <code class="docutils literal notranslate"><span class="pre">ilr-hierarchical</span></code>.</p></li>
<li><p>Phylogenetic analysis. A phylogenetic tree (e.g. <code class="docutils literal notranslate"><span class="pre">rooted-tree.qza</span></code>) created outside of gneiss can also be used. In this case you can use your phylogenetic tree as input for <code class="docutils literal notranslate"><span class="pre">ilr-phylogenetic</span></code>.</p></li>
</ol>
<p>Once we have a tree, we can calculate balances using the following equation:</p>
<div class="math notranslate nohighlight">
\[b_i = \sqrt{\frac{rs}{r+s}} \log \frac{g(x_r)}{g(x_s)}\]</div>
<p>where <span class="math notranslate nohighlight">\(i\)</span> represents the <span class="math notranslate nohighlight">\(i^{th}\)</span> internal node in the tree, <span class="math notranslate nohighlight">\(g(x)\)</span> represents the geometric mean within set <span class="math notranslate nohighlight">\(x\)</span>, and <span class="math notranslate nohighlight">\(x_r\)</span> represents the set of taxa abundances in the numerator of the balance, <span class="math notranslate nohighlight">\(x_s\)</span> represents the set of taxa abundances in the denominator of the balance, and <span class="math notranslate nohighlight">\(r\)</span> and <span class="math notranslate nohighlight">\(s\)</span> represents the number of taxa within <span class="math notranslate nohighlight">\(x_r\)</span> and <span class="math notranslate nohighlight">\(x_s\)</span> respectively.</p>
<p>After the balances are calculated, standard statistical procedures such as ANOVA and linear regression can be performed.  We will demonstrate running these procedures using the Chronic Fatigue Syndrome dataset.</p>
<div class="section" id="creating-balances">
<h2>Creating balances<a class="headerlink" href="#creating-balances" title="Permalink to this headline">¶</a></h2>
<p>In the Chronic Fatigue Syndrome dataset published in <a class="reference external" href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-016-0171-4">Giloteaux et al (2016)</a>, there are 87 individuals with 48 diseased patients and 39 healthy controls. The data used in this tutorial were sequenced on an Illumina MiSeq using the <a class="reference external" href="http://earthmicrobiome.org/">Earth Microbiome Project</a> hypervariable region 4 (V4) 16S rRNA sequencing protocol.</p>
<p>Before beginning this tutorial, create a new directory and change to that directory.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir qiime2-chronic-fatigue-syndrome-tutorial
<span class="nb">cd</span> qiime2-chronic-fatigue-syndrome-tutorial
</pre></div>
</div>
<p>The datasets required for this tutorial can be found below (to learn how these would be generated, see the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial</span></a>).</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2020.2/tutorials/gneiss/sample-metadata.tsv">
            https://data.qiime2.org/2020.2/tutorials/gneiss/sample-metadata.tsv
          </a>
        </p>
        <p>
          <strong>Save as:</strong> sample-metadata.tsv
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "sample-metadata.tsv" \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/sample-metadata.tsv"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/sample-metadata.tsv" > \
  "sample-metadata.tsv"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2020.2/tutorials/gneiss/table.qza">
            https://data.qiime2.org/2020.2/tutorials/gneiss/table.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> table.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "table.qza" \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/table.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/table.qza" > \
  "table.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-2" aria-controls="browser-2" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-2" aria-controls="wget-2" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-2" aria-controls="messages-2" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-2">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2020.2/tutorials/gneiss/taxa.qza">
            https://data.qiime2.org/2020.2/tutorials/gneiss/taxa.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> taxa.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-2">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "taxa.qza" \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/taxa.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-2">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2020.2/tutorials/gneiss/taxa.qza" > \
  "taxa.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>First, we will define partitions of microbes for which we want to construct balances. Again, there are multiple possible ways to construct a tree (i.e. hierarchy) which defines the partition of microbes (balances) for which we want to construct balances. We will show examples of both <code class="docutils literal notranslate"><span class="pre">correlation-clustering</span></code> and <code class="docutils literal notranslate"><span class="pre">gradient-clustering</span></code> on this dataset.</p>
<p>Note that the differential abundance techniques that we will be running will utilize log ratio transforms. Since it is not possible to take the logarithm of zero, both clustering methods below include a default pseudocount parameter. This replaces all zeroes in the table with a 1, so that we can apply logarithms on this transformed table.</p>
<p>The input table is the raw count table (FeatureTable[Frequency]).</p>
</div>
<div class="section" id="option-1-correlation-clustering">
<h2>Option 1: Correlation-clustering<a class="headerlink" href="#option-1-correlation-clustering" title="Permalink to this headline">¶</a></h2>
<p>This option should be your default option. We will employ unsupervised clustering via Ward’s hierarchical clustering to obtain Principal Balances. In essence, this will define the partitions of microbes that commonly co-occur with each other using Ward hierarchical clustering, which is defined by the following metric.</p>
<div class="math notranslate nohighlight">
\[d(x, y) = V [ \ln \frac{x}{y} ]\]</div>
<p>Where <span class="math notranslate nohighlight">\(x\)</span> and <span class="math notranslate nohighlight">\(y\)</span> represent the proportions of two microbes across all of the samples.   If two microbes are highly correlated, then this quantity will shrink close to zero.  Ward hierarchical cluster will then use this distance metric to iteratively cluster together groups of microbes that are correlated with each other.  In the end, the tree that we obtain will highlight the high level structure and identify any blocks within in the data.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime gneiss correlation-clustering <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-clustering hierarchy.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fgneiss%2Ftable.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/gneiss/table.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">taxa.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fgneiss%2Ftaxa.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/gneiss/taxa.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">hierarchy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fgneiss%2Fhierarchy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/gneiss/hierarchy.qza">download</a></p></li>
</ul>
</div>
</div>
<div class="section" id="option-2-gradient-clustering">
<h2>Option 2: Gradient-clustering<a class="headerlink" href="#option-2-gradient-clustering" title="Permalink to this headline">¶</a></h2>
<p>An alternative to <code class="docutils literal notranslate"><span class="pre">correlation-clustering</span></code> is to create a tree based on a numeric metadata category. With <code class="docutils literal notranslate"><span class="pre">gradient-clustering</span></code>, we can group taxa that occur in similar ranges of a metadata category. In this example, we will create a tree (hierarchy) using the metadata category Age. Note that the metadata category can have no missing variables, and must be numeric.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime gneiss gradient-clustering <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --m-gradient-file sample-metadata.tsv <span class="se">\</span>
  --m-gradient-column Age <span class="se">\</span>
  --o-clustering gradient-hierarchy.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">gradient-hierarchy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fgneiss%2Fgradient-hierarchy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/gneiss/gradient-hierarchy.qza">download</a></p></li>
</ul>
</div>
<p>An important consideration for downstream analyses is the problem of overfitting. When using <code class="docutils literal notranslate"><span class="pre">gradient-clustering</span></code>, you are creating a tree to best highlight compositional differences along the metadata category of your choice, and it is possible to get false positives. Use <code class="docutils literal notranslate"><span class="pre">gradient-clustering</span></code> with caution.</p>
<p>We will visualize the hierarchy (Option 1, above) on a heatmap to see which groups of taxa they represent.  By default, the values within the feature table are log-scaled, with the sample means centered around zero.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime gneiss dendrogram-heatmap <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-tree hierarchy.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-column Subject <span class="se">\</span>
  --p-color-map seismic <span class="se">\</span>
  --o-visualization heatmap.qzv
</pre></div>
</div>
<div class="admonition">
<p><strong>Output visualizations:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">heatmap.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fgneiss%2Fheatmap.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/gneiss/heatmap.qzv">download</a></p></li>
</ul>
</div>
<p>As noted in the legend, the numerators for each balance are highlighted in light red, while the denominators are highlighted in dark red. From here, we can see that the denominator of <span class="math notranslate nohighlight">\(y0\)</span> has few OTUs compared to the numerator of <span class="math notranslate nohighlight">\(y0\)</span>.  These taxa in the denominator could be interesting, so let’s investigate the taxonomies making up this balance with <code class="docutils literal notranslate"><span class="pre">balance_taxonomy</span></code>.</p>
</div>
<div class="section" id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this headline">¶</a></h2>
<p>Remember, based on the toy examples given in the beginning of this tutorial, it is not possible to infer absolute changes of microbes in a given sample.  Balances will not be able to provide this sort of answer, but it can limit the number of possible scenarios.  Specifically, one of the five following scenarios could have happened.</p>
<ol class="arabic simple">
<li><p>The taxa in the <span class="math notranslate nohighlight">\(y0_{numerator}\)</span> on average have increased between patient group and the healthy control.</p></li>
<li><p>The taxa in the <span class="math notranslate nohighlight">\(y0_{denominator}\)</span> on average have decreased between patient group and the healthy control.</p></li>
<li><p>A combination of the above occurred</p></li>
<li><p>Taxa abundances in both <span class="math notranslate nohighlight">\(y0_{numerator}\)</span> and <span class="math notranslate nohighlight">\(y0_{denominator}\)</span> both increased, but taxa abundances in <span class="math notranslate nohighlight">\(y0_{numerator}\)</span> increased more compared to <span class="math notranslate nohighlight">\(y0_{denominator}\)</span></p></li>
<li><p>Taxa abundances in both <span class="math notranslate nohighlight">\(y0_{numerator}\)</span> and <span class="math notranslate nohighlight">\(y0_{denominator}\)</span> both decreased, but taxa abundances in <span class="math notranslate nohighlight">\(y0_{denominator}\)</span> increased more compared to <span class="math notranslate nohighlight">\(y0_{numerator}\)</span></p></li>
</ol>
<p>To further narrow down these hypothesis,  biological prior knowledge or experimental validation will be required.</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="../qiime2-for-experienced-microbiome-researchers/">QIIME 2 for Experienced Microbiome Researchers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">“Moving Pictures” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">“Atacama soil microbiome” tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pd-mice/">Parkinson’s Mouse Tutorial</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../otu-clustering/">Clustering sequences into OTUs using q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../utilities/">Utilities in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../phylogeny/">Phylogenetic inference with q2-phylogeny</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>