<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Phylogenetic inference with q2-phylogeny &#8212; QIIME 2 2020.2.0 documentation</title>
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/language_data.js"></script>
    <script src="../../_static/external-links.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="index" title="Index" href="../../genindex/" />
    <link rel="search" title="Search" href="../../search/" />
    <link rel="next" title="Interfaces" href="../../interfaces/" />
    <link rel="prev" title="Utilities in QIIME 2" href="../utilities/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="phylogenetic-inference-with-q2-phylogeny">
<h1><a class="toc-backref" href="#id4">Phylogenetic inference with q2-phylogeny</a><a class="headerlink" href="#phylogenetic-inference-with-q2-phylogeny" title="Permalink to this headline">¬∂</a></h1>
<div class="contents topic" id="id1">
<p class="topic-title">Phylogenetic inference with q2-phylogeny</p>
<ul class="simple">
<li><p><a class="reference internal" href="#phylogenetic-inference-with-q2-phylogeny" id="id4">Phylogenetic inference with q2-phylogeny</a></p>
<ul>
<li><p><a class="reference internal" href="#inferring-phylogenies" id="id5">Inferring phylogenies</a></p>
<ul>
<li><p><a class="reference internal" href="#sequence-alignment" id="id6">Sequence Alignment</a></p></li>
<li><p><a class="reference internal" href="#reducing-alignment-ambiguity-masking-and-reference-alignments" id="id7">Reducing alignment ambiguity: masking and reference alignments</a></p></li>
<li><p><a class="reference internal" href="#construct-a-phylogeny" id="id8">Construct a phylogeny</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#methods" id="id9">Methods</a></p>
<ul>
<li><p><a class="reference internal" href="#fasttree" id="id10">fasttree</a></p></li>
<li><p><a class="reference internal" href="#raxml" id="id11">raxml</a></p>
<ul>
<li><p><a class="reference internal" href="#perform-multiple-searches-using-raxml" id="id12">Perform multiple searches using raxml</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#raxml-rapid-bootstrap" id="id13">raxml-rapid-bootstrap</a></p></li>
<li><p><a class="reference internal" href="#iqtree" id="id14">iqtree</a></p>
<ul>
<li><p><a class="reference internal" href="#specifying-a-substitution-model" id="id15">Specifying a substitution model</a></p></li>
<li><p><a class="reference internal" href="#single-branch-tests" id="id16">Single branch tests</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#iqtree-ultrafast-bootstrap" id="id17">iqtree-ultrafast-bootstrap</a></p>
<ul>
<li><p><a class="reference internal" href="#perform-single-branch-tests-alongside-ufboot" id="id18">Perform single branch tests alongside ufboot</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#root-the-phylogeny" id="id19">Root the phylogeny</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#pipelines" id="id20">Pipelines</a></p></li>
</ul>
</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tutorial assumes, you‚Äôve read through the <a class="reference internal" href="../overview/"><span class="doc">QIIME 2 Overview</span></a> documentation and have at least worked through some of the other
<a class="reference internal" href="../"><span class="doc">Tutorials</span></a>.</p>
</div>
<div class="section" id="inferring-phylogenies">
<h2><a class="toc-backref" href="#id5">Inferring phylogenies</a><a class="headerlink" href="#inferring-phylogenies" title="Permalink to this headline">¬∂</a></h2>
<p>Several downstream diversity metrics, available within QIIME 2, require that a
phylogenetic tree be constructed using the Operational Taxonomic Units
(<a class="reference external" href="https://en.wikipedia.org/wiki/Operational_taxonomic_unit">OTUs</a>) or Exact Sequence Variants (<a class="reference external" href="https://doi.org/10.1038/ismej.2019.119">ESVs</a>) being investigated.</p>
<p><em>But how do we proceed to construct a phylogeny from our sequence data?</em></p>
<p>Well, there are two phylogeny-based approaches we can use. Deciding upon which
to use is largely dependent on your study questions:</p>
<p>1. A reference-based <a class="reference external" href="https://doi.org/10.1128/mSystems.00021-18">fragment insertion</a> approach. Which, is likely the
ideal choice. Especially, if your reference phylogeny (and associated
representative sequences) encompass neighboring relatives of which your
sequences can be reliably inserted. Any sequences that do not match well enough
to the reference are not inserted. For example, this approach may not work well
if your data contain sequences that are not well represented within your
reference phylogeny (e.g. missing clades, etc.). For more information, check
out these great <a class="reference external" href="https://library.qiime2.org/plugins/q2-fragment-insertion/16/">fragment insertion examples</a>.</p>
<p>2. A <em>de novo</em> approach. Marker genes that can be globally aligned across
divergent taxa, are usually amenable to sequence alignment and phylogenetic
investigation through this approach. Be mindful of the length of your
sequences when constructing a <em>de novo</em> phylogeny, short reads many not have
enough phylogenetic information to capture a meaningful phylogeny. This
community tutorial will focus on the <em>de novo</em> approaches.</p>
<p>Here, you will learn how to make use of <em>de novo</em> phylogenetic approaches
to:</p>
<ol class="arabic simple">
<li><p>generate a sequence alignment within QIIME 2</p></li>
<li><p>mask the alignment if needed</p></li>
<li><p>construct a phylogenetic tree</p></li>
<li><p>root the phylogenetic tree</p></li>
</ol>
<p>If you would like to substitute any of the steps outlined here by making use of
tools external to QIIME 2, please see the <a class="reference internal" href="../importing/"><span class="doc">import</span></a>,
<a class="reference internal" href="../exporting/"><span class="doc">export</span></a>, and <a class="reference internal" href="../filtering/"><span class="doc">filtering</span></a> documentation
where appropriate.</p>
<div class="section" id="sequence-alignment">
<h3><a class="toc-backref" href="#id6">Sequence Alignment</a><a class="headerlink" href="#sequence-alignment" title="Permalink to this headline">¬∂</a></h3>
<p>Prior to constructing a <a class="reference external" href="https://simple.wikipedia.org/wiki/Phylogeny">phylogeny</a> we must generate a multiple sequence
alignment (<a class="reference external" href="https://en.wikipedia.org/wiki/Multiple_sequence_alignment">MSA</a>). When constructing a MSA we are making a statement about the
<a class="reference external" href="http://doi.org/10.1006/mpev.2000.0785">putative homology</a> of the aligned residues (columns of the MSA) by virtue of
their <a class="reference external" href="http://doi.org/10.1002/0471250953.bi0301s42">sequence similarity</a>.</p>
<p>The number of algorithms to construct a MSA are legion. We will make use of
<a class="reference external" href="https://en.wikipedia.org/wiki/MAFFT">MAFFT</a> (Multiple Alignment using Fast Fourier Transform)) via the
<a class="reference internal" href="../../plugins/available/alignment/"><span class="doc">q2-alignment</span></a> plugin. For more
information checkout the <a class="reference external" href="http://doi.org/10.1093/molbev/mst010">MAFFT paper</a>.</p>
<p>Let‚Äôs start by creating a directory to work in:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>mkdir qiime2-phylogeny-tutorial
<span class="nb">cd</span> qiime2-phylogeny-tutorial
</pre></div>
</div>
<p>Next, download the data:</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2020.2/tutorials/phylogeny/rep-seqs.qza">
            https://data.qiime2.org/2020.2/tutorials/phylogeny/rep-seqs.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> rep-seqs.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "rep-seqs.qza" \
  "https://data.qiime2.org/2020.2/tutorials/phylogeny/rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2020.2/tutorials/phylogeny/rep-seqs.qza" > \
  "rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p><strong>Run MAFFT</strong></p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime alignment mafft <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --o-alignment aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/rep-seqs.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Faligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/aligned-rep-seqs.qza">download</a></p></li>
</ul>
</div>
</div>
<div class="section" id="reducing-alignment-ambiguity-masking-and-reference-alignments">
<h3><a class="toc-backref" href="#id7">Reducing alignment ambiguity: masking and reference alignments</a><a class="headerlink" href="#reducing-alignment-ambiguity-masking-and-reference-alignments" title="Permalink to this headline">¬∂</a></h3>
<p><em>Why mask an alignment?</em></p>
<p>Masking helps to eliminate alignment columns that are phylogenetically
uninformative or misleading before phylogenetic analysis. Much of the time
alignment errors can introduce noise and confound phylogenetic inference. It is
common practice to mask (remove) these ambiguously aligned regions prior to
performing phylogenetic inference. In particular, <a class="reference external" href="http://www.worldcat.org/title/nucleic-acid-techniques-in-bacterial-systematics/oclc/22310197">David Lane‚Äôs (1991)</a>
chapter <a class="reference external" href="http://catdir.loc.gov/catdir/toc/onix05/90012998.html">16S/23S rRNA sequencing</a> proposed masking SSU data prior to
phylogenetic analysis. However, knowing how to deal with ambiguously aligned
regions and when to apply masks largely depends on the marker genes being
analyzed and the question being asked of the data.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Keep in mind that this is still an active area of discussion, as
highlighted by the following non-exhaustive list of articles: <a class="reference external" href="https://doi.org/10.1371/journal.pone.0030288">Wu <em>et al</em>. 2012</a>, <a class="reference external" href="https://doi.org/10.1093/sysbio/syy036">Ashkenazy <em>et al</em>. 2018</a>, <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1000844">Schloss 2010</a>, <a class="reference external" href="https://doi.org/10.1093/sysbio/syv033">Tan <em>et al</em>. 2015</a>,
<a class="reference external" href="https://doi.org/10.1093/molbev/mss264">Rajan 2015</a>.</p>
</div>
<p><em>How to mask alignment.</em></p>
<p>For our purposes, we‚Äôll assume that we have ambiguously aligned columns in the
MAFFT alignment we produced above. The default settings for the
<code class="docutils literal notranslate"><span class="pre">--p-min-conservation</span></code> of the
<a class="reference internal" href="../../plugins/available/alignment/mask/"><span class="doc">alignment mask</span></a> approximates the
Lane mask filtering of QIIME 1. Keep an eye out for updates to the alignment
plugin.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime alignment mask <span class="se">\</span>
  --i-alignment aligned-rep-seqs.qza <span class="se">\</span>
  --o-masked-alignment masked-aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">masked-aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmasked-aligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/masked-aligned-rep-seqs.qza">download</a></p></li>
</ul>
</div>
<p><em>Reference based alignments</em></p>
<p>There are a variety of tools such as <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btp636">PyNAST</a>) (using <a class="reference external" href="https://doi.org/10.1093/nar/gkl244">NAST</a>), <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btt509">Infernal</a>,
and <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts252">SINA</a>, etc., that attempt to reduce the amount of ambiguously aligned
regions by using curated reference alignments (e.g. <a class="reference external" href="https://www.arb-silva.de/">SILVA</a>. Reference
alignments are particularly powerful for rRNA gene sequence data, as knowledge
of secondary structure is incorporated into the curation process, thus
increasing alignment quality. For a more in-depth and eloquent overview of
reference-based alignment approaches, check out the great <a class="reference external" href="https://forum.qiime2.org/t/q2-alignment-reference-based-alignment-using-sina/6220">SINA community
tutorial</a>).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Alignments constructed using reference based alignment approaches can
be masked too, just like the above MAFFT example. Also, the reference
alignment approach we are discussing here is distinct from the reference
phylogeny approach (i.e.
<a class="reference internal" href="../../plugins/available/fragment-insertion/"><span class="doc">q2-fragment-insertion</span></a>)
we mentioned earlier. That is, we are not inserting our data into an
existing tree, but simply trying to create a more robust alignment for
making a better <em>de novo</em> phylogeny.</p>
</div>
</div>
<div class="section" id="construct-a-phylogeny">
<h3><a class="toc-backref" href="#id8">Construct a phylogeny</a><a class="headerlink" href="#construct-a-phylogeny" title="Permalink to this headline">¬∂</a></h3>
<p>As with MSA algorithms, phylogenetic inference tools are also legion.
Fortunately, there are many great resources to learn about phylogentics. Below
are just a few introductory resources to get you started:</p>
<ol class="arabic simple">
<li><p><a class="reference external" href="http://doi.org/10.1016/S0168-9525(03)00112-4">Phylogeny for the faint of heart - a tutorial</a></p></li>
<li><p><a class="reference external" href="http://dx.doi.org/10.1038/nrg3186">Molecular phylogenetics - principles and practice</a></p></li>
<li><p><a class="reference external" href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics">Phylogenetics - An Introduction</a></p></li>
</ol>
<p>There are several methods / pipelines available through the
<a class="reference internal" href="../../plugins/available/phylogeny/"><span class="doc">q2-phylogeny</span></a> plugin of :qiime2:.
These are based on the following tools:</p>
<ol class="arabic simple">
<li><p><a class="reference external" href="https://doi.org/10.1371/journal.pone.0009490">FastTree</a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu033">RAxML</a></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/molbev/msu300">IQ-TREE</a></p></li>
</ol>
</div>
</div>
<div class="section" id="methods">
<h2><a class="toc-backref" href="#id9">Methods</a><a class="headerlink" href="#methods" title="Permalink to this headline">¬∂</a></h2>
<div class="section" id="fasttree">
<h3><a class="toc-backref" href="#id10">fasttree</a><a class="headerlink" href="#fasttree" title="Permalink to this headline">¬∂</a></h3>
<p>FastTree is able to construct phylogenies from large sequence alignments quite
rapidly. It does this by using the using a <a class="reference external" href="https://doi.org/10.1109/IPDPS.2006.1639535">CAT-like</a> rate category
approximation, which is also available through RAxML (discussed below). Check
out the <a class="reference external" href="http://www.microbesonline.org/fasttree/">FastTree online manual</a> for more information.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny fasttree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree fasttree-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">fasttree-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Ffasttree-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/fasttree-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For an easy and direct way to view your <code class="docutils literal notranslate"><span class="pre">tree.qza</span></code> files, upload
them to <a class="reference external" href="https://itol.embl.de/">iTOL</a>. Here, you can interactively view and manipulate your
phylogeny. Even better, while viewing the tree topology in ‚ÄúNormal mode‚Äù,
you can drag and drop your associated <code class="docutils literal notranslate"><span class="pre">alignment.qza</span></code> (the one you used to
build the phylogeny) or a relevent <code class="docutils literal notranslate"><span class="pre">taxonomy.qza</span></code> file onto the iTOL tree
visualization. This will allow you to directly view the sequence alignment
or taxonomy alongside the phylogeny. üï∂Ô∏è</p>
</div>
</div>
<div class="section" id="raxml">
<h3><a class="toc-backref" href="#id11">raxml</a><a class="headerlink" href="#raxml" title="Permalink to this headline">¬∂</a></h3>
<p>Like <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>, <code class="docutils literal notranslate"><span class="pre">raxml</span></code> will perform a single phylogentic inference and
return a tree. Note, the default model for <code class="docutils literal notranslate"><span class="pre">raxml</span></code> is
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span> <span class="pre">GTRGAMMA</span></code>. If you‚Äôd like to construct a tree using
the CAT model like <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>, simply replace <code class="docutils literal notranslate"><span class="pre">GTRGAMMA</span></code> with <code class="docutils literal notranslate"><span class="pre">GTRCAT</span></code> as
shown below:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --o-tree raxml-cat-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid hill-climbing mode

Using 1 distinct models/data partitions with joint branch length optimization


Executing 1 inferences on the original alignment using 1 distinct randomized MP trees

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -m GTRCAT -p 6415 -N 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-wm3gc8d3/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp4dzvt6iy -n q2 


Partition: 0 with name: No Name Provided
Base frequencies: 0.243 0.182 0.319 0.256 

Inference[0]: Time 0.651053 CAT-based likelihood -1242.579881, best rearrangement setting 5


Conducting final model optimizations on all 1 trees under GAMMA-based models ....

Inference[0] final GAMMA-based Likelihood: -1388.335399 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp4dzvt6iy/RAxML_result.q2


Starting final GAMMA-based thorough Optimization on tree 0 likelihood -1388.335399 .... 

Final GAMMA-based Score of best tree -1387.278976

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp4dzvt6iy/RAxML_info.q2
Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp4dzvt6iy/RAxML_bestTree.q2

Overall execution time: 1.321623 secs or 0.000367 hours or 0.000015 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -m GTRCAT -p 6415 -N 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-wm3gc8d3/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmp4dzvt6iy -n q2

Saved Phylogeny[Unrooted] to: raxml-cat-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="perform-multiple-searches-using-raxml">
<h4><a class="toc-backref" href="#id12">Perform multiple searches using raxml</a><a class="headerlink" href="#perform-multiple-searches-using-raxml" title="Permalink to this headline">¬∂</a></h4>
<p>If you‚Äôd like to perform a more thorough search of ‚Äútree space‚Äù you can
instruct <code class="docutils literal notranslate"><span class="pre">raxml</span></code> to perform multiple independent searches on the full
alignment by using <code class="docutils literal notranslate"><span class="pre">--p-n-searches</span> <span class="pre">5</span></code>. Once these 5 independent searches are
completed, only the single best scoring tree will be returned. <em>Note, we are
not bootstrapping here, we‚Äôll do that in a later example.</em> Let‚Äôs set
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span> <span class="pre">GTRCAT</span></code>. Finally, let‚Äôs also manually set a seed via
<code class="docutils literal notranslate"><span class="pre">--p-seed</span></code>. By setting our seed, we allow other users the ability to
reproduce our phylogeny. That is, anyone using the same sequence alignment and
substitution model, will generate the same tree as long as they set the same
seed value. Although, <code class="docutils literal notranslate"><span class="pre">--p-seed</span></code> is not a required argument, it is generally
a good idea to set this value.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --p-seed <span class="m">1723</span> <span class="se">\</span>
  --p-n-searches <span class="m">5</span> <span class="se">\</span>
  --o-tree raxml-cat-searches-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid hill-climbing mode

Using 1 distinct models/data partitions with joint branch length optimization


Executing 5 inferences on the original alignment using 5 distinct randomized MP trees

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -m GTRCAT -p 1723 -N 5 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-k2e04py8/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu -n q2 


Partition: 0 with name: No Name Provided
Base frequencies: 0.243 0.182 0.319 0.256 

Inference[0]: Time 0.623663 CAT-based likelihood -1238.242991, best rearrangement setting 5
Inference[1]: Time 0.509429 CAT-based likelihood -1249.502284, best rearrangement setting 5
Inference[2]: Time 0.524310 CAT-based likelihood -1242.978035, best rearrangement setting 5
Inference[3]: Time 0.666095 CAT-based likelihood -1243.159855, best rearrangement setting 5
Inference[4]: Time 0.519867 CAT-based likelihood -1261.321621, best rearrangement setting 5


Conducting final model optimizations on all 5 trees under GAMMA-based models ....

Inference[0] final GAMMA-based Likelihood: -1388.324037 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_result.q2.RUN.0
Inference[1] final GAMMA-based Likelihood: -1392.813982 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_result.q2.RUN.1
Inference[2] final GAMMA-based Likelihood: -1388.073642 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_result.q2.RUN.2
Inference[3] final GAMMA-based Likelihood: -1387.945266 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_result.q2.RUN.3
Inference[4] final GAMMA-based Likelihood: -1387.557031 tree written to file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_result.q2.RUN.4


Starting final GAMMA-based thorough Optimization on tree 4 likelihood -1387.557031 .... 

Final GAMMA-based Score of best tree -1387.385075

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_info.q2
Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu/RAxML_bestTree.q2

Overall execution time: 3.624212 secs or 0.001007 hours or 0.000042 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -m GTRCAT -p 1723 -N 5 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-k2e04py8/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdiga50gu -n q2

Saved Phylogeny[Unrooted] to: raxml-cat-searches-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-searches-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-searches-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-searches-tree.qza">download</a></p></li>
</ul>
</div>
</div>
</div>
<div class="section" id="raxml-rapid-bootstrap">
<h3><a class="toc-backref" href="#id13">raxml-rapid-bootstrap</a><a class="headerlink" href="#raxml-rapid-bootstrap" title="Permalink to this headline">¬∂</a></h3>
<p>In phylogenetics, it is good practice to check how well the <a class="reference external" href="https://en.wikipedia.org/wiki/Split_(phylogenetics)">splits /
bipartitions</a> in your phylogeny are supported. Often one is interested in
which clades are robustly separated from other clades in the phylogeny. One
way, of doing this is via bootstrapping (See the <em>Bootstrapping</em> section of the
first introductory link above). In QIIME 2, we‚Äôve provided access to the RAxML
<a class="reference external" href="http://dx.doi.org/10.1080/10635150802429642">rapid bootstrap</a> feature. The only difference between this command and the
previous are the additional flags <code class="docutils literal notranslate"><span class="pre">--p-bootstrap-replicates</span></code> and
<code class="docutils literal notranslate"><span class="pre">--p-rapid-bootstrap-seed</span></code>. It is quite common to perform anywhere from 100 -
1000 bootstrap replicates. The <code class="docutils literal notranslate"><span class="pre">--p-rapid-bootstrap-seed</span></code> works very much
like the <code class="docutils literal notranslate"><span class="pre">--p-seed</span></code> argument from above except that it allows anyone to
reproduce the bootstrapping process and the associated supports for your
splits.</p>
<p>As per the <a class="reference external" href="https://sco.h-its.org/exelixis/web/software/raxml/hands_on.html">RAxML online documentation</a> and the <a class="reference external" href="https://sco.h-its.org/exelixis/resource/download/NewManual.pdf">RAxML manual</a>, the rapid
bootstrapping command that we will execute below will do the following:</p>
<ol class="arabic simple">
<li><p>Bootstrap the input alignment 100 times and perform a Maximum Likelihood
(ML) search on each.</p></li>
<li><p>Find best scoring ML tree through multiple independent searches using the
original input alignment. The number of independent searches is determined
by the number of bootstrap replicates set in the 1st step. That is, your
search becomes more thorough with increasing bootstrap replicates. The ML
optimization of RAxML uses every 5th bootstrap tree as the starting tree for
an ML search on the original alignment.</p></li>
<li><p>Map the bipartitions (bootstrap supports, 1st step) onto the best scoring ML
tree (2nd step).</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny raxml-rapid-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-seed <span class="m">1723</span> <span class="se">\</span>
  --p-rapid-bootstrap-seed <span class="m">9384</span> <span class="se">\</span>
  --p-bootstrap-replicates <span class="m">100</span> <span class="se">\</span>
  --p-substitution-model GTRCAT <span class="se">\</span>
  --o-tree raxml-cat-bootstrap-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Warning, you specified a working directory via &quot;-w&quot;
Keep in mind that RAxML only accepts absolute path names, not relative ones!

RAxML can&#39;t, parse the alignment file as phylip file 
it will now try to parse it as FASTA file



Using BFGS method to optimize GTR rate parameters, to disable this specify &quot;--no-bfgs&quot; 



This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.

With greatly appreciated code contributions by:
Andre Aberer      (HITS)
Simon Berger      (HITS)
Alexey Kozlov     (HITS)
Kassian Kobert    (HITS)
David Dao         (KIT and HITS)
Sarah Lutteropp   (KIT and HITS)
Nick Pattengale   (Sandia)
Wayne Pfeiffer    (SDSC)
Akifumi S. Tanabe (NRIFS)
Charlie Taylor    (UF)


Alignment has 157 distinct alignment patterns

Proportion of gaps and completely undetermined characters in this alignment: 39.77%

RAxML rapid bootstrapping and subsequent ML search

Using 1 distinct models/data partitions with joint branch length optimization



Executing 100 rapid bootstrap inferences and thereafter a thorough ML search 

All free model parameters will be estimated by RAxML
ML estimate of 25 per site rate categories

Likelihood of final tree will be evaluated and optimized under GAMMA

GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units

Partition: 0
Alignment Patterns: 157
Name: No Name Provided
DataType: DNA
Substitution Matrix: GTR




RAxML was called as follows:

raxmlHPC -f a -m GTRCAT -p 1723 -x 9384 -N 100 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-ayhruhog/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f -n q2bootstrap 



Time for BS model parameter optimization 0.052080
Bootstrap[0]: Time 0.181618 seconds, bootstrap likelihood -1199.758796, best rearrangement setting 12
Bootstrap[1]: Time 0.120677 seconds, bootstrap likelihood -1344.229251, best rearrangement setting 6
Bootstrap[2]: Time 0.108588 seconds, bootstrap likelihood -1295.343000, best rearrangement setting 8
Bootstrap[3]: Time 0.093386 seconds, bootstrap likelihood -1273.768320, best rearrangement setting 8
Bootstrap[4]: Time 0.108861 seconds, bootstrap likelihood -1253.402952, best rearrangement setting 6
Bootstrap[5]: Time 0.126696 seconds, bootstrap likelihood -1260.866113, best rearrangement setting 10
Bootstrap[6]: Time 0.112777 seconds, bootstrap likelihood -1293.636299, best rearrangement setting 14
Bootstrap[7]: Time 0.106289 seconds, bootstrap likelihood -1227.178693, best rearrangement setting 6
Bootstrap[8]: Time 0.110966 seconds, bootstrap likelihood -1321.820787, best rearrangement setting 13
Bootstrap[9]: Time 0.121619 seconds, bootstrap likelihood -1147.233446, best rearrangement setting 6
Bootstrap[10]: Time 0.084652 seconds, bootstrap likelihood -1220.766493, best rearrangement setting 13
Bootstrap[11]: Time 0.120440 seconds, bootstrap likelihood -1200.006355, best rearrangement setting 8
Bootstrap[12]: Time 0.129327 seconds, bootstrap likelihood -1346.392834, best rearrangement setting 14
Bootstrap[13]: Time 0.111474 seconds, bootstrap likelihood -1301.111096, best rearrangement setting 14
Bootstrap[14]: Time 0.118893 seconds, bootstrap likelihood -1262.253559, best rearrangement setting 11
Bootstrap[15]: Time 0.116589 seconds, bootstrap likelihood -1215.017551, best rearrangement setting 14
Bootstrap[16]: Time 0.110033 seconds, bootstrap likelihood -1238.832009, best rearrangement setting 7
Bootstrap[17]: Time 0.096895 seconds, bootstrap likelihood -1393.989732, best rearrangement setting 12
Bootstrap[18]: Time 0.108248 seconds, bootstrap likelihood -1173.921002, best rearrangement setting 15
Bootstrap[19]: Time 0.108892 seconds, bootstrap likelihood -1185.726976, best rearrangement setting 11
Bootstrap[20]: Time 0.100850 seconds, bootstrap likelihood -1158.491940, best rearrangement setting 6
Bootstrap[21]: Time 0.093500 seconds, bootstrap likelihood -1154.664272, best rearrangement setting 11
Bootstrap[22]: Time 0.104642 seconds, bootstrap likelihood -1244.159837, best rearrangement setting 10
Bootstrap[23]: Time 0.127629 seconds, bootstrap likelihood -1211.171036, best rearrangement setting 15
Bootstrap[24]: Time 0.101407 seconds, bootstrap likelihood -1261.440677, best rearrangement setting 12
Bootstrap[25]: Time 0.107081 seconds, bootstrap likelihood -1331.836715, best rearrangement setting 15
Bootstrap[26]: Time 0.110434 seconds, bootstrap likelihood -1129.144509, best rearrangement setting 5
Bootstrap[27]: Time 0.133349 seconds, bootstrap likelihood -1226.624056, best rearrangement setting 7
Bootstrap[28]: Time 0.135498 seconds, bootstrap likelihood -1221.046176, best rearrangement setting 12
Bootstrap[29]: Time 0.091955 seconds, bootstrap likelihood -1211.791204, best rearrangement setting 14
Bootstrap[30]: Time 0.114907 seconds, bootstrap likelihood -1389.442380, best rearrangement setting 5
Bootstrap[31]: Time 0.112914 seconds, bootstrap likelihood -1303.638592, best rearrangement setting 12
Bootstrap[32]: Time 0.124271 seconds, bootstrap likelihood -1172.859456, best rearrangement setting 12
Bootstrap[33]: Time 0.106678 seconds, bootstrap likelihood -1244.617135, best rearrangement setting 9
Bootstrap[34]: Time 0.101719 seconds, bootstrap likelihood -1211.871717, best rearrangement setting 15
Bootstrap[35]: Time 0.123494 seconds, bootstrap likelihood -1299.862912, best rearrangement setting 5
Bootstrap[36]: Time 0.098445 seconds, bootstrap likelihood -1141.967505, best rearrangement setting 5
Bootstrap[37]: Time 0.121983 seconds, bootstrap likelihood -1283.923198, best rearrangement setting 12
Bootstrap[38]: Time 0.102102 seconds, bootstrap likelihood -1304.250946, best rearrangement setting 5
Bootstrap[39]: Time 0.093806 seconds, bootstrap likelihood -1407.084376, best rearrangement setting 15
Bootstrap[40]: Time 0.114389 seconds, bootstrap likelihood -1277.946299, best rearrangement setting 13
Bootstrap[41]: Time 0.113969 seconds, bootstrap likelihood -1279.006200, best rearrangement setting 7
Bootstrap[42]: Time 0.105849 seconds, bootstrap likelihood -1160.274606, best rearrangement setting 6
Bootstrap[43]: Time 0.128212 seconds, bootstrap likelihood -1216.079259, best rearrangement setting 14
Bootstrap[44]: Time 0.098106 seconds, bootstrap likelihood -1382.278311, best rearrangement setting 8
Bootstrap[45]: Time 0.115715 seconds, bootstrap likelihood -1099.004439, best rearrangement setting 11
Bootstrap[46]: Time 0.093864 seconds, bootstrap likelihood -1296.527478, best rearrangement setting 8
Bootstrap[47]: Time 0.131588 seconds, bootstrap likelihood -1291.322658, best rearrangement setting 9
Bootstrap[48]: Time 0.088062 seconds, bootstrap likelihood -1161.908080, best rearrangement setting 6
Bootstrap[49]: Time 0.121469 seconds, bootstrap likelihood -1257.348428, best rearrangement setting 13
Bootstrap[50]: Time 0.143257 seconds, bootstrap likelihood -1309.422533, best rearrangement setting 13
Bootstrap[51]: Time 0.098795 seconds, bootstrap likelihood -1197.633097, best rearrangement setting 11
Bootstrap[52]: Time 0.112454 seconds, bootstrap likelihood -1347.123005, best rearrangement setting 8
Bootstrap[53]: Time 0.103337 seconds, bootstrap likelihood -1234.934890, best rearrangement setting 14
Bootstrap[54]: Time 0.115260 seconds, bootstrap likelihood -1227.092434, best rearrangement setting 6
Bootstrap[55]: Time 0.118732 seconds, bootstrap likelihood -1280.635747, best rearrangement setting 7
Bootstrap[56]: Time 0.099194 seconds, bootstrap likelihood -1225.911449, best rearrangement setting 6
Bootstrap[57]: Time 0.101892 seconds, bootstrap likelihood -1236.213347, best rearrangement setting 11
Bootstrap[58]: Time 0.138348 seconds, bootstrap likelihood -1393.245723, best rearrangement setting 14
Bootstrap[59]: Time 0.126303 seconds, bootstrap likelihood -1212.039371, best rearrangement setting 6
Bootstrap[60]: Time 0.097597 seconds, bootstrap likelihood -1248.692011, best rearrangement setting 10
Bootstrap[61]: Time 0.113407 seconds, bootstrap likelihood -1172.820979, best rearrangement setting 13
Bootstrap[62]: Time 0.125327 seconds, bootstrap likelihood -1126.745788, best rearrangement setting 14
Bootstrap[63]: Time 0.103499 seconds, bootstrap likelihood -1267.434444, best rearrangement setting 12
Bootstrap[64]: Time 0.095928 seconds, bootstrap likelihood -1340.680748, best rearrangement setting 5
Bootstrap[65]: Time 0.101131 seconds, bootstrap likelihood -1072.671059, best rearrangement setting 5
Bootstrap[66]: Time 0.123986 seconds, bootstrap likelihood -1234.294838, best rearrangement setting 8
Bootstrap[67]: Time 0.123783 seconds, bootstrap likelihood -1109.249439, best rearrangement setting 15
Bootstrap[68]: Time 0.095088 seconds, bootstrap likelihood -1314.493588, best rearrangement setting 8
Bootstrap[69]: Time 0.098507 seconds, bootstrap likelihood -1173.850035, best rearrangement setting 13
Bootstrap[70]: Time 0.123258 seconds, bootstrap likelihood -1231.066465, best rearrangement setting 10
Bootstrap[71]: Time 0.103306 seconds, bootstrap likelihood -1146.861379, best rearrangement setting 9
Bootstrap[72]: Time 0.091666 seconds, bootstrap likelihood -1148.753369, best rearrangement setting 8
Bootstrap[73]: Time 0.105392 seconds, bootstrap likelihood -1333.374056, best rearrangement setting 9
Bootstrap[74]: Time 0.089935 seconds, bootstrap likelihood -1259.382378, best rearrangement setting 5
Bootstrap[75]: Time 0.097522 seconds, bootstrap likelihood -1319.944496, best rearrangement setting 6
Bootstrap[76]: Time 0.110489 seconds, bootstrap likelihood -1309.042165, best rearrangement setting 14
Bootstrap[77]: Time 0.133879 seconds, bootstrap likelihood -1232.061289, best rearrangement setting 8
Bootstrap[78]: Time 0.110360 seconds, bootstrap likelihood -1261.333984, best rearrangement setting 9
Bootstrap[79]: Time 0.113974 seconds, bootstrap likelihood -1194.644341, best rearrangement setting 13
Bootstrap[80]: Time 0.100860 seconds, bootstrap likelihood -1214.037389, best rearrangement setting 9
Bootstrap[81]: Time 0.117326 seconds, bootstrap likelihood -1224.527657, best rearrangement setting 8
Bootstrap[82]: Time 0.135541 seconds, bootstrap likelihood -1241.464826, best rearrangement setting 11
Bootstrap[83]: Time 0.097520 seconds, bootstrap likelihood -1230.730558, best rearrangement setting 6
Bootstrap[84]: Time 0.108422 seconds, bootstrap likelihood -1219.034592, best rearrangement setting 10
Bootstrap[85]: Time 0.110529 seconds, bootstrap likelihood -1280.071994, best rearrangement setting 8
Bootstrap[86]: Time 0.098919 seconds, bootstrap likelihood -1444.747777, best rearrangement setting 9
Bootstrap[87]: Time 0.096856 seconds, bootstrap likelihood -1245.890035, best rearrangement setting 14
Bootstrap[88]: Time 0.110396 seconds, bootstrap likelihood -1287.832766, best rearrangement setting 7
Bootstrap[89]: Time 0.105908 seconds, bootstrap likelihood -1325.245976, best rearrangement setting 5
Bootstrap[90]: Time 0.120951 seconds, bootstrap likelihood -1227.883697, best rearrangement setting 5
Bootstrap[91]: Time 0.111035 seconds, bootstrap likelihood -1273.489392, best rearrangement setting 8
Bootstrap[92]: Time 0.046177 seconds, bootstrap likelihood -1234.725870, best rearrangement setting 7
Bootstrap[93]: Time 0.118856 seconds, bootstrap likelihood -1235.733064, best rearrangement setting 11
Bootstrap[94]: Time 0.097637 seconds, bootstrap likelihood -1204.319488, best rearrangement setting 15
Bootstrap[95]: Time 0.095236 seconds, bootstrap likelihood -1183.328582, best rearrangement setting 11
Bootstrap[96]: Time 0.110232 seconds, bootstrap likelihood -1196.298898, best rearrangement setting 13
Bootstrap[97]: Time 0.116684 seconds, bootstrap likelihood -1339.251746, best rearrangement setting 12
Bootstrap[98]: Time 0.045519 seconds, bootstrap likelihood -1404.363552, best rearrangement setting 7
Bootstrap[99]: Time 0.059553 seconds, bootstrap likelihood -1270.157811, best rearrangement setting 7


Overall Time for 100 Rapid Bootstraps 10.947528 seconds
Average Time per Rapid Bootstrap 0.109475 seconds

Starting ML Search ...

Fast ML optimization finished

Fast ML search Time: 4.379153 seconds

Slow ML Search 0 Likelihood: -1387.994678
Slow ML Search 1 Likelihood: -1387.994678
Slow ML Search 2 Likelihood: -1387.994676
Slow ML Search 3 Likelihood: -1387.994650
Slow ML Search 4 Likelihood: -1387.994685
Slow ML Search 5 Likelihood: -1388.092954
Slow ML Search 6 Likelihood: -1388.182551
Slow ML Search 7 Likelihood: -1388.182563
Slow ML Search 8 Likelihood: -1388.182547
Slow ML Search 9 Likelihood: -1387.994723
Slow ML optimization finished

Slow ML search Time: 2.217905 seconds
Thorough ML search Time: 0.593935 seconds

Final ML Optimization Likelihood: -1387.204993

Model Information:

Model Parameters of Partition 0, Name: No Name Provided, Type of Data: DNA
alpha: 1.227800
Tree-Length: 7.823400
rate A &lt;-&gt; C: 0.332564
rate A &lt;-&gt; G: 2.312784
rate A &lt;-&gt; T: 2.215466
rate C &lt;-&gt; G: 1.243321
rate C &lt;-&gt; T: 3.278770
rate G &lt;-&gt; T: 1.000000

freq pi(A): 0.243216
freq pi(C): 0.181967
freq pi(G): 0.319196
freq pi(T): 0.255621


ML search took 7.197641 secs or 0.001999 hours

Combined Bootstrap and ML search took 18.145492 secs or 0.005040 hours

Drawing Bootstrap Support Values on best-scoring ML tree ...



Found 1 tree in File /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bestTree.q2bootstrap



Found 1 tree in File /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bestTree.q2bootstrap

Program execution info written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_info.q2bootstrap
All 100 bootstrapped trees written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bootstrap.q2bootstrap

Best-scoring ML tree written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bestTree.q2bootstrap

Best-scoring ML tree with support values written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bipartitions.q2bootstrap

Best-scoring ML tree with support values as branch labels written to: /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f/RAxML_bipartitionsBranchLabels.q2bootstrap

Overall execution time for full ML analysis: 18.158612 secs or 0.005044 hours or 0.000210 days

Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: raxmlHPC -f a -m GTRCAT -p 1723 -x 9384 -N 100 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-ayhruhog/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -w /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpcgeh77_f -n q2bootstrap

Saved Phylogeny[Unrooted] to: raxml-cat-bootstrap-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">raxml-cat-bootstrap-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fraxml-cat-bootstrap-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/raxml-cat-bootstrap-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>Optimizing RAxML Run Time.</strong>
You may gave noticed that we haven‚Äôt added the flag <code class="docutils literal notranslate"><span class="pre">--p-raxml-version</span></code> to
the RAxML methods. This parameter provides a means to access versions of
RAxML that have optimized vector instructions for various modern x86
processor architectures. Paraphrased from the RAxML manual and help
documentation: Firstly, most recent processors will support SSE3 vector
instructions (i.e. will likely support the faster AVX2 vector instructions).
Secondly, these instructions will substantially accelerate the likelihood
and parsimony computations. In general, SSE3 versions will run approximately
40% faster than the standard version. The AVX2 version will run 10-30%
faster than the SSE3 version. Additionally, keep in mind that using more
cores / threads will not necessarily decrease run time. The RAxML manual
suggests using 1 core per ~500 DNA alignment patterns. Alignment pattern
information is usually visible on screen, when the <code class="docutils literal notranslate"><span class="pre">--verbose</span></code> option is
used. Additionally, try using a rate category (CAT model; via
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span></code>), which results in equally good trees as the
GAMMA models and is approximately 4 times faster. See the <a class="reference external" href="https://doi.org/10.1109/IPDPS.2006.1639535">CAT paper</a>. The
CAT approximation is also Ideal for alignments containing <a class="reference external" href="https://doi.org/10.1186/1471-2105-12-470">10,000 or more
taxa</a>, and is very much similar the <a class="reference external" href="https://doi.org/10.1371/journal.pone.0009490">CAT-like model of FastTree2</a>.</p>
</div>
</div>
<div class="section" id="iqtree">
<h3><a class="toc-backref" href="#id14">iqtree</a><a class="headerlink" href="#iqtree" title="Permalink to this headline">¬∂</a></h3>
<p>Similar to the <code class="docutils literal notranslate"><span class="pre">raxml</span></code> and <code class="docutils literal notranslate"><span class="pre">raxml-rapid-bootstrap</span></code> methods above, we
provide similar functionality for <a class="reference external" href="https://doi.org/10.1093/molbev/msu300">IQ-TREE</a>: <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> and
<code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code>. IQ-TREE is unique compared to the <code class="docutils literal notranslate"><span class="pre">fastree</span></code>
and <code class="docutils literal notranslate"><span class="pre">raxml</span></code> options, in that it provides access to 286 <a class="reference external" href="https://doi.org/10.1016/j.dci.2004.07.007">models of nucleotide
substitution</a>! IQ-TREE can also determine which of these models best fits your
dataset prior to constructing your tree via its built-in <a class="reference external" href="https://doi.org/10.1038/nmeth.4285">ModelFinder</a>
algorithm. This is the default in QIIME 2, but do not worry, you can set any
one of the 286 models of nucleotide substitution via the
<code class="docutils literal notranslate"><span class="pre">--p-substitution-model</span></code> flag, e.g. you can set the model as <code class="docutils literal notranslate"><span class="pre">HKY+I+G</span></code>
instead of the default <code class="docutils literal notranslate"><span class="pre">MFP</span></code> (a basic short-hand for: ‚Äúbuild a phylogeny
after determining the best fit model as determined by ModelFinder‚Äù). Keep in
mind the additional computational time required for model testing via
ModelFinder.</p>
<p>The simplest way to run the
<a class="reference internal" href="../../plugins/available/phylogeny/iqtree/"><span class="doc">iqtree command</span></a> with default
settings and automatic model selection (<code class="docutils literal notranslate"><span class="pre">MFP</span></code>) is like so:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree iqt-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-00veqgb9/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree -nt 1
Seed:    733653 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:27:37 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-00veqgb9/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1425.502     37  2925.003     2940.981     3049.544
  2  JC+I          1424.337     38  2924.673     2941.610     3052.580
  3  JC+G4         1416.625     38  2909.251     2926.188     3037.158
  4  JC+I+G4       1416.981     39  2911.962     2929.893     3043.235
  5  JC+R2         1412.375     39  2902.751     2920.682     3034.024
  6  JC+R3         1412.307     41  2906.614     2926.637     3044.619
 14  F81+F         1422.505     40  2925.010     2943.969     3059.649
 15  F81+F+I       1421.274     41  2924.547     2944.571     3062.552
 16  F81+F+G4      1412.595     41  2907.190     2927.214     3045.195
 17  F81+F+I+G4    1412.902     42  2909.803     2930.926     3051.174
 18  F81+F+R2      1407.531     42  2899.061     2920.184     3040.432
 19  F81+F+R3      1407.512     44  2903.024     2926.456     3051.127
 27  K2P           1417.182     38  2910.364     2927.301     3038.271
 28  K2P+I         1416.201     39  2910.403     2928.334     3041.676
 29  K2P+G4        1407.307     39  2892.614     2910.545     3023.887
 30  K2P+I+G4      1407.582     40  2895.163     2914.123     3029.802
 31  K2P+R2        1401.827     40  2883.655     2902.614     3018.294
 32  K2P+R3        1401.822     42  2887.643     2908.766     3029.014
 40  HKY+F         1414.412     41  2910.825     2930.848     3048.830
 41  HKY+F+I       1413.220     42  2910.440     2931.563     3051.811
 42  HKY+F+G4      1403.354     42  2890.708     2911.831     3032.079
 43  HKY+F+I+G4    1403.525     43  2893.051     2915.310     3037.788
 44  HKY+F+R2      1397.226     43  2880.452     2902.711     3025.189
 45  HKY+F+R3      1397.223     45  2884.447     2909.090     3035.916
 53  TNe           1417.037     39  2912.074     2930.005     3043.347
 54  TNe+I         1415.909     40  2911.819     2930.778     3046.458
 55  TNe+G4        1407.247     40  2894.495     2913.454     3029.134
 56  TNe+I+G4      1407.396     41  2896.791     2916.815     3034.796
 57  TNe+R2        1401.817     41  2885.635     2905.658     3023.640
 58  TNe+R3        1401.816     43  2889.631     2911.890     3034.368
 66  TN+F          1414.128     42  2912.256     2933.379     3053.627
 67  TN+F+I        1413.043     43  2912.085     2934.344     3056.822
 68  TN+F+G4       1402.541     43  2891.082     2913.341     3035.819
 69  TN+F+I+G4     1402.676     44  2893.352     2916.784     3041.455
 70  TN+F+R2       1396.165     44  2880.329     2903.761     3028.432
 71  TN+F+R3       1396.161     46  2884.322     2910.214     3039.157
 79  K3P           1415.605     39  2909.210     2927.141     3040.483
 80  K3P+I         1414.605     40  2909.209     2928.169     3043.848
 81  K3P+G4        1404.683     40  2889.367     2908.326     3024.006
 82  K3P+I+G4      1404.792     41  2891.584     2911.607     3029.589
 83  K3P+R2        1399.310     41  2880.619     2900.642     3018.624
 84  K3P+R3        1399.307     43  2884.615     2906.873     3029.352
 92  K3Pu+F        1412.462     42  2908.923     2930.046     3050.294
 93  K3Pu+F+I      1411.147     43  2908.295     2930.554     3053.032
 94  K3Pu+F+G4     1399.910     43  2885.820     2908.078     3030.557
 95  K3Pu+F+I+G4   1399.980     44  2887.960     2911.392     3036.063
 96  K3Pu+F+R2     1394.028     44  2876.056     2899.488     3024.159
 97  K3Pu+F+R3     1394.028     46  2880.056     2905.948     3034.890
105  TPM2+F        1414.354     42  2912.708     2933.831     3054.079
106  TPM2+F+I      1413.146     43  2912.291     2934.550     3057.028
107  TPM2+F+G4     1403.131     43  2892.263     2914.522     3037.000
108  TPM2+F+I+G4   1403.186     44  2894.372     2917.804     3042.475
109  TPM2+F+R2     1397.005     44  2882.009     2905.441     3030.112
110  TPM2+F+R3     1397.000     46  2886.001     2911.893     3040.836
118  TPM2u+F       1414.354     42  2912.708     2933.831     3054.079
119  TPM2u+F+I     1413.146     43  2912.291     2934.550     3057.028
120  TPM2u+F+G4    1403.133     43  2892.266     2914.525     3037.003
121  TPM2u+F+I+G4  1403.175     44  2894.349     2917.781     3042.452
122  TPM2u+F+R2    1397.003     44  2882.006     2905.438     3030.109
123  TPM2u+F+R3    1397.000     46  2885.999     2911.891     3040.834
131  TPM3+F        1410.743     42  2905.487     2926.610     3046.858
132  TPM3+F+I      1409.845     43  2905.689     2927.948     3050.426
133  TPM3+F+G4     1399.833     43  2885.666     2907.925     3030.403
134  TPM3+F+I+G4   1399.847     44  2887.693     2911.125     3035.796
135  TPM3+F+R2     1394.231     44  2876.462     2899.894     3024.565
136  TPM3+F+R3     1394.230     46  2880.461     2906.353     3035.295
144  TPM3u+F       1410.756     42  2905.513     2926.636     3046.884
145  TPM3u+F+I     1409.836     43  2905.672     2927.931     3050.409
146  TPM3u+F+G4    1399.838     43  2885.677     2907.936     3030.414
147  TPM3u+F+I+G4  1399.840     44  2887.681     2911.113     3035.784
148  TPM3u+F+R2    1394.231     44  2876.462     2899.894     3024.565
149  TPM3u+F+R3    1394.230     46  2880.459     2906.351     3035.294
157  TIMe          1415.481     40  2910.963     2929.922     3045.602
158  TIMe+I        1414.324     41  2910.649     2930.672     3048.654
159  TIMe+G4       1404.645     41  2891.290     2911.313     3029.295
160  TIMe+I+G4     1404.665     42  2893.330     2914.453     3034.701
161  TIMe+R2       1399.310     42  2882.621     2903.744     3023.992
162  TIMe+R3       1399.308     44  2886.616     2910.048     3034.719
170  TIM+F         1412.152     43  2910.303     2932.562     3055.040
171  TIM+F+I       1410.947     44  2909.894     2933.326     3057.997
172  TIM+F+G4      1398.983     44  2885.966     2909.398     3034.069
173  TIM+F+I+G4    1398.990     45  2887.980     2912.623     3039.449
174  TIM+F+R2      1392.900     45  2875.800     2900.443     3027.269
175  TIM+F+R3      1392.852     47  2879.703     2906.884     3037.904
183  TIM2e         1416.979     40  2913.958     2932.917     3048.597
184  TIM2e+I       1415.836     41  2913.673     2933.696     3051.678
185  TIM2e+G4      1407.201     41  2896.402     2916.425     3034.407
186  TIM2e+I+G4    1407.232     42  2898.464     2919.586     3039.835
187  TIM2e+R2      1401.807     42  2887.615     2908.738     3028.986
188  TIM2e+R3      1401.786     44  2891.573     2915.005     3039.675
196  TIM2+F        1414.067     43  2914.134     2936.393     3058.871
197  TIM2+F+I      1412.964     44  2913.929     2937.361     3062.032
198  TIM2+F+G4     1402.379     44  2892.759     2916.191     3040.862
199  TIM2+F+I+G4   1402.357     45  2894.713     2919.356     3046.182
200  TIM2+F+R2     1395.929     45  2881.858     2906.501     3033.327
201  TIM2+F+R3     1395.882     47  2885.763     2912.944     3043.964
209  TIM3e         1409.644     40  2899.289     2918.248     3033.928
210  TIM3e+I       1408.785     41  2899.570     2919.593     3037.575
211  TIM3e+G4      1399.522     41  2881.044     2901.067     3019.049
212  TIM3e+I+G4    1399.529     42  2883.058     2904.181     3024.429
213  TIM3e+R2      1394.889     42  2873.778     2894.901     3015.149
214  TIM3e+R3      1394.879     44  2877.758     2901.190     3025.861
222  TIM3+F        1410.443     43  2906.886     2929.145     3051.623
223  TIM3+F+I      1409.608     44  2907.216     2930.648     3055.319
224  TIM3+F+G4     1399.007     44  2886.014     2909.446     3034.117
225  TIM3+F+I+G4   1398.988     45  2887.977     2912.620     3039.446
226  TIM3+F+R2     1392.985     45  2875.969     2900.612     3027.438
227  TIM3+F+R3     1392.946     47  2879.892     2907.073     3038.093
235  TVMe          1406.753     41  2895.505     2915.528     3033.510
236  TVMe+I        1405.965     42  2895.931     2917.054     3037.302
237  TVMe+G4       1395.545     42  2875.091     2896.214     3016.462
238  TVMe+I+G4     1395.536     43  2877.072     2899.331     3021.809
239  TVMe+R2       1391.166     43  2868.332     2890.591     3013.069
240  TVMe+R3       1391.161     45  2872.322     2896.965     3023.791
248  TVM+F         1407.676     44  2903.352     2926.784     3051.455
249  TVM+F+I       1406.667     45  2903.334     2927.977     3054.803
250  TVM+F+G4      1395.290     45  2880.580     2905.223     3032.049
251  TVM+F+I+G4    1395.281     46  2882.561     2908.453     3037.396
252  TVM+F+R2      1390.099     46  2872.198     2898.090     3027.032
253  TVM+F+R3      1390.062     48  2876.123     2904.632     3037.690
261  SYM           1406.659     42  2897.318     2918.441     3038.689
262  SYM+I         1405.774     43  2897.548     2919.807     3042.285
263  SYM+G4        1395.513     43  2877.027     2899.285     3021.764
264  SYM+I+G4      1395.507     44  2879.014     2902.446     3027.117
265  SYM+R2        1391.161     44  2870.322     2893.754     3018.424
266  SYM+R3        1391.150     46  2874.299     2900.192     3029.134
274  GTR+F         1407.271     45  2904.542     2929.185     3056.011
275  GTR+F+I       1406.431     46  2904.863     2930.755     3059.698
276  GTR+F+G4      1394.460     46  2880.921     2906.813     3035.756
277  GTR+F+I+G4    1394.443     47  2882.886     2910.067     3041.087
278  GTR+F+R2      1388.927     47  2871.855     2899.036     3030.056
279  GTR+F+R3      1388.886     49  2875.773     2905.651     3040.706
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree.model.gz
CPU time for ModelFinder: 0.969 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.975 seconds (0h:0m:0s)

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1418.019
2. Current log-likelihood: -1393.889
3. Current log-likelihood: -1392.403
4. Current log-likelihood: -1391.726
5. Current log-likelihood: -1391.446
6. Current log-likelihood: -1391.309
Optimal log-likelihood: -1391.208
Rate parameters:  A-C: 0.13750  A-G: 1.82207  A-T: 1.46648  C-G: 0.74131  C-T: 1.82207  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.681,0.380) (0.319,2.325)
Parameters optimization took 6 rounds (0.036 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.219
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.088 second
Computing log-likelihood of 98 initial trees ... 0.143 seconds
Current best score: -1389.219

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.347
Iteration 10 / LogL: -1382.348 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.348 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.347 / CPU time: 0.469
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1382.350 / Time: 0h:0m:0s (0h:0m:1s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 38: -1382.104
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 39: -1382.022
Iteration 40 / LogL: -1382.095 / Time: 0h:0m:0s (0h:0m:2s left)
Iteration 50 / LogL: -1403.480 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 60 / LogL: -1382.082 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 70 / LogL: -1391.218 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 80 / LogL: -1382.099 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 90 / LogL: -1382.096 / Time: 0h:0m:1s (0h:0m:0s left)
Iteration 100 / LogL: -1382.307 / Time: 0h:0m:1s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.020
Iteration 110 / LogL: -1382.105 / Time: 0h:0m:1s (0h:0m:0s left)
Iteration 120 / LogL: -1382.036 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 130 / LogL: -1382.866 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 140 / LogL: -1382.031 / Time: 0h:0m:2s (0h:0m:-1s left)
TREE SEARCH COMPLETED AFTER 140 ITERATIONS / Time: 0h:0m:2s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.020
Optimal log-likelihood: -1382.014
Rate parameters:  A-C: 0.19304  A-G: 1.84742  A-T: 1.54181  C-G: 0.77458  C-T: 1.84742  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.720,0.413) (0.280,2.513)
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1382.014
Total tree length: 7.031

Total number of iterations: 140
CPU time used for tree search: 2.336 sec (0h:0m:2s)
Wall-clock time used for tree search: 2.345 sec (0h:0m:2s)
Total CPU time used: 2.390 sec (0h:0m:2s)
Total wall-clock time used: 2.402 sec (0h:0m:2s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree.log

Date and Time: Fri Feb 28 11:27:41 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-00veqgb9/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpb4ro6vaj/q2iqtree -nt 1

Saved Phylogeny[Unrooted] to: iqt-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="specifying-a-substitution-model">
<h4><a class="toc-backref" href="#id15">Specifying a substitution model</a><a class="headerlink" href="#specifying-a-substitution-model" title="Permalink to this headline">¬∂</a></h4>
<p>We can also set a substitution model of our choosing. You may have noticed
while watching the onscreen output of the previous command that the best
fitting model selected by ModelFinder is noted. For the sake of argument, let‚Äôs
say the best selected model was shown as <code class="docutils literal notranslate"><span class="pre">GTR+F+I+G4</span></code>. The <code class="docutils literal notranslate"><span class="pre">F</span></code> is only a
notation to let us know that <em>if</em> a given model supports <em>unequal base
frequencies</em>, then the <em>empirical base frequencies</em> will be used by default.
Using empirical base frequencies (<code class="docutils literal notranslate"><span class="pre">F</span></code>), rather than estimating them, greatly
reduces computational time. The <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> plugin will not accept <code class="docutils literal notranslate"><span class="pre">F</span></code> within
the model notation supplied at the command line, as this will always be implied
automatically for the appropriate model. Also, the <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> plugin only
accepts <code class="docutils literal notranslate"><span class="pre">G</span></code> <em>not</em> <code class="docutils literal notranslate"><span class="pre">G4</span></code> to be specified within the model notation. The <code class="docutils literal notranslate"><span class="pre">4</span></code>
is simply another explicit notation to remind us that four rate categories are
being assumed by default. The notation approach used by the plugin simply helps
to retain simplicity and familiarity when supplying model notations on the
command line. So, in brief, we only have to type <code class="docutils literal notranslate"><span class="pre">GTR+I+G</span></code> as our input
model:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-gtrig-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-scxhh8t5/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree -nt 1
Seed:    340352 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:27:45 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-scxhh8t5/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.228 / LogL: -1394.470
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.008, 1.302 / LogL: -1394.729
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.008, 1.305 / LogL: -1394.727
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.008, 1.301 / LogL: -1394.728
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.008, 1.305 / LogL: -1394.760
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.306 / LogL: -1394.786
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.008, 1.303 / LogL: -1394.730
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.304 / LogL: -1394.743
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.008, 1.305 / LogL: -1394.754
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.008, 1.311 / LogL: -1394.758
Optimal pinv,alpha: 0.000, 1.228 / LogL: -1394.470

Parameters optimization took 0.480 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.888
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.088 second
Computing log-likelihood of 98 initial trees ... 0.198 seconds
Current best score: -1392.888

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.267
Iteration 10 / LogL: -1387.384 / Time: 0h:0m:0s
Iteration 20 / LogL: -1387.280 / Time: 0h:0m:1s
Finish initializing candidate tree set (2)
Current best tree score: -1387.267 / CPU time: 0.653
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1387.348 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 40 / LogL: -1387.342 / Time: 0h:0m:1s (0h:0m:2s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.258
Iteration 50 / LogL: -1387.346 / Time: 0h:0m:1s (0h:0m:2s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.258
Iteration 60 / LogL: -1387.258 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 70 / LogL: -1387.386 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 80 / LogL: -1387.377 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 90 / LogL: -1387.347 / Time: 0h:0m:3s (0h:0m:0s left)
Iteration 100 / LogL: -1387.347 / Time: 0h:0m:3s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 102 ITERATIONS / Time: 0h:0m:3s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.258
Optimal log-likelihood: -1387.254
Rate parameters:  A-C: 0.32866  A-G: 2.26506  A-T: 2.14271  C-G: 1.17415  C-T: 3.29496  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.315
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1387.254
Total tree length: 6.769

Total number of iterations: 102
CPU time used for tree search: 3.050 sec (0h:0m:3s)
Wall-clock time used for tree search: 3.062 sec (0h:0m:3s)
Total CPU time used: 3.554 sec (0h:0m:3s)
Total wall-clock time used: 3.569 sec (0h:0m:3s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree.log

Date and Time: Fri Feb 28 11:27:48 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-scxhh8t5/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpaz5m_db_/q2iqtree -nt 1

Saved Phylogeny[Unrooted] to: iqt-gtrig-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-gtrig-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-gtrig-tree.qza">download</a></p></li>
</ul>
</div>
<p>Let‚Äôs rerun the command above and add the <code class="docutils literal notranslate"><span class="pre">--p-fast</span></code> option. This option,
only compatible with the <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> method, resembles the fast search performed
by <code class="docutils literal notranslate"><span class="pre">fasttree</span></code>. üèéÔ∏è Secondly, let‚Äôs also perform multiple tree searches and
keep the best of those trees (as we did earlier with the
<code class="docutils literal notranslate"><span class="pre">raxml</span> <span class="pre">--p-n-searches</span> <span class="pre">...</span></code> command):</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --p-fast <span class="se">\</span>
  --p-n-runs <span class="m">10</span> <span class="se">\</span>
  --o-tree iqt-gtrig-fast-ms-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 10 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-s_1xjkve/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree -nt 1 -fast
Seed:    47350 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:27:53 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-s_1xjkve/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

---&gt; START RUN NUMBER 1 (seed: 47350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.240 / LogL: -1394.551
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.019, 1.306 / LogL: -1395.300
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.023, 1.366 / LogL: -1395.504
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.023, 1.367 / LogL: -1395.485
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.026, 1.365 / LogL: -1395.669
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.028, 1.365 / LogL: -1395.797
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.023, 1.365 / LogL: -1395.543
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.024, 1.367 / LogL: -1395.601
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.025, 1.367 / LogL: -1395.650
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.026, 1.367 / LogL: -1395.697
Optimal pinv,alpha: 0.000, 1.240 / LogL: -1394.551

Parameters optimization took 0.322 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.871
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.282
UPDATE BEST LOG-LIKELIHOOD: -1387.274
Finish initializing candidate tree set (3)
Current best tree score: -1387.274 / CPU time: 0.051
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.274
Optimal log-likelihood: -1387.259
Rate parameters:  A-C: 0.33569  A-G: 2.26227  A-T: 2.13669  C-G: 1.17589  C-T: 3.29103  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.304
Parameters optimization took 1 rounds (0.005 sec)
BEST SCORE FOUND : -1387.259
Total tree length: 6.801

Total number of iterations: 2
CPU time used for tree search: 0.050 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.051 sec (0h:0m:0s)
Total CPU time used: 0.397 sec (0h:0m:0s)
Total wall-clock time used: 0.401 sec (0h:0m:0s)

---&gt; START RUN NUMBER 2 (seed: 48350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.894
2. Current log-likelihood: -1402.077
3. Current log-likelihood: -1396.805
4. Current log-likelihood: -1395.377
5. Current log-likelihood: -1394.645
Optimal log-likelihood: -1394.074
Rate parameters:  A-C: 0.27529  A-G: 2.39359  A-T: 2.12128  C-G: 1.20984  C-T: 3.30250  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.394
Parameters optimization took 5 rounds (0.040 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.804
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.215
BETTER TREE FOUND at iteration 2: -1388.181
Finish initializing candidate tree set (4)
Current best tree score: -1388.181 / CPU time: 0.064
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.974
3. Current log-likelihood: -1387.831
4. Current log-likelihood: -1387.726
5. Current log-likelihood: -1387.646
6. Current log-likelihood: -1387.585
Optimal log-likelihood: -1387.535
Rate parameters:  A-C: 0.36871  A-G: 2.32155  A-T: 2.12849  C-G: 1.22859  C-T: 3.29561  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.006
Gamma shape alpha: 1.331
Parameters optimization took 6 rounds (0.026 sec)
BEST SCORE FOUND : -1387.535
Total tree length: 7.507

Total number of iterations: 2
CPU time used for tree search: 0.063 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.064 sec (0h:0m:0s)
Total CPU time used: 0.146 sec (0h:0m:0s)
Total wall-clock time used: 0.149 sec (0h:0m:0s)

---&gt; START RUN NUMBER 3 (seed: 49350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1480.521
2. Current log-likelihood: -1403.150
3. Current log-likelihood: -1398.384
4. Current log-likelihood: -1396.968
5. Current log-likelihood: -1396.253
Optimal log-likelihood: -1395.745
Rate parameters:  A-C: 0.23809  A-G: 2.08501  A-T: 1.96789  C-G: 1.08192  C-T: 2.78800  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.364
Parameters optimization took 5 rounds (0.040 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.993
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1387.951
Finish initializing candidate tree set (4)
Current best tree score: -1387.951 / CPU time: 0.080
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1387.951
2. Current log-likelihood: -1387.792
3. Current log-likelihood: -1387.673
4. Current log-likelihood: -1387.583
5. Current log-likelihood: -1387.515
6. Current log-likelihood: -1387.462
Optimal log-likelihood: -1387.420
Rate parameters:  A-C: 0.33363  A-G: 2.24758  A-T: 2.12142  C-G: 1.16644  C-T: 3.24795  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.358
Parameters optimization took 6 rounds (0.023 sec)
BEST SCORE FOUND : -1387.420
Total tree length: 6.700

Total number of iterations: 2
CPU time used for tree search: 0.079 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.080 sec (0h:0m:0s)
Total CPU time used: 0.161 sec (0h:0m:0s)
Total wall-clock time used: 0.165 sec (0h:0m:0s)

---&gt; START RUN NUMBER 4 (seed: 50350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1479.269
2. Current log-likelihood: -1404.566
3. Current log-likelihood: -1399.236
4. Current log-likelihood: -1397.830
5. Current log-likelihood: -1397.061
Optimal log-likelihood: -1396.470
Rate parameters:  A-C: 0.24622  A-G: 2.08129  A-T: 1.99444  C-G: 1.06443  C-T: 2.85655  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.437
Parameters optimization took 5 rounds (0.038 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.961
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.181
Finish initializing candidate tree set (3)
Current best tree score: -1388.181 / CPU time: 0.055
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.955
3. Current log-likelihood: -1387.793
4. Current log-likelihood: -1387.674
5. Current log-likelihood: -1387.584
6. Current log-likelihood: -1387.516
7. Current log-likelihood: -1387.463
Optimal log-likelihood: -1387.421
Rate parameters:  A-C: 0.33227  A-G: 2.23486  A-T: 2.10851  C-G: 1.15950  C-T: 3.22710  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.360
Parameters optimization took 7 rounds (0.024 sec)
BEST SCORE FOUND : -1387.421
Total tree length: 6.696

Total number of iterations: 2
CPU time used for tree search: 0.055 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.056 sec (0h:0m:0s)
Total CPU time used: 0.136 sec (0h:0m:0s)
Total wall-clock time used: 0.139 sec (0h:0m:0s)

---&gt; START RUN NUMBER 5 (seed: 51350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1481.520
2. Current log-likelihood: -1401.794
3. Current log-likelihood: -1396.546
4. Current log-likelihood: -1395.117
5. Current log-likelihood: -1394.384
Optimal log-likelihood: -1393.812
Rate parameters:  A-C: 0.27489  A-G: 2.41323  A-T: 2.17672  C-G: 1.25266  C-T: 3.29513  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.421
Parameters optimization took 5 rounds (0.038 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.791
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.211
Finish initializing candidate tree set (3)
Current best tree score: -1388.211 / CPU time: 0.048
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.211
2. Current log-likelihood: -1388.012
3. Current log-likelihood: -1387.866
4. Current log-likelihood: -1387.758
5. Current log-likelihood: -1387.675
6. Current log-likelihood: -1387.610
7. Current log-likelihood: -1387.559
Optimal log-likelihood: -1387.518
Rate parameters:  A-C: 0.35581  A-G: 2.35552  A-T: 2.14265  C-G: 1.20478  C-T: 3.37542  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.361
Parameters optimization took 7 rounds (0.027 sec)
BEST SCORE FOUND : -1387.518
Total tree length: 6.815

Total number of iterations: 2
CPU time used for tree search: 0.048 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.048 sec (0h:0m:0s)
Total CPU time used: 0.133 sec (0h:0m:0s)
Total wall-clock time used: 0.136 sec (0h:0m:0s)

---&gt; START RUN NUMBER 6 (seed: 52350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1486.149
2. Current log-likelihood: -1402.098
3. Current log-likelihood: -1396.806
4. Current log-likelihood: -1395.378
5. Current log-likelihood: -1394.645
Optimal log-likelihood: -1394.074
Rate parameters:  A-C: 0.27529  A-G: 2.39363  A-T: 2.12129  C-G: 1.20989  C-T: 3.30258  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.394
Parameters optimization took 5 rounds (0.039 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.804
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.216
BETTER TREE FOUND at iteration 2: -1388.181
Finish initializing candidate tree set (4)
Current best tree score: -1388.181 / CPU time: 0.059
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.974
3. Current log-likelihood: -1387.831
4. Current log-likelihood: -1387.726
5. Current log-likelihood: -1387.646
6. Current log-likelihood: -1387.585
Optimal log-likelihood: -1387.535
Rate parameters:  A-C: 0.36871  A-G: 2.32156  A-T: 2.12850  C-G: 1.22859  C-T: 3.29563  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.006
Gamma shape alpha: 1.331
Parameters optimization took 6 rounds (0.026 sec)
BEST SCORE FOUND : -1387.535
Total tree length: 7.507

Total number of iterations: 2
CPU time used for tree search: 0.058 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.059 sec (0h:0m:0s)
Total CPU time used: 0.140 sec (0h:0m:0s)
Total wall-clock time used: 0.143 sec (0h:0m:0s)

---&gt; START RUN NUMBER 7 (seed: 53350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.954
2. Current log-likelihood: -1402.081
3. Current log-likelihood: -1396.805
4. Current log-likelihood: -1395.378
5. Current log-likelihood: -1394.645
Optimal log-likelihood: -1394.074
Rate parameters:  A-C: 0.27529  A-G: 2.39359  A-T: 2.12128  C-G: 1.20985  C-T: 3.30251  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.394
Parameters optimization took 5 rounds (0.040 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.804
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.216
BETTER TREE FOUND at iteration 2: -1388.181
Finish initializing candidate tree set (4)
Current best tree score: -1388.181 / CPU time: 0.063
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.974
3. Current log-likelihood: -1387.831
4. Current log-likelihood: -1387.726
5. Current log-likelihood: -1387.646
6. Current log-likelihood: -1387.585
Optimal log-likelihood: -1387.535
Rate parameters:  A-C: 0.36871  A-G: 2.32155  A-T: 2.12849  C-G: 1.22859  C-T: 3.29561  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.006
Gamma shape alpha: 1.331
Parameters optimization took 6 rounds (0.024 sec)
BEST SCORE FOUND : -1387.535
Total tree length: 7.507

Total number of iterations: 2
CPU time used for tree search: 0.063 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.063 sec (0h:0m:0s)
Total CPU time used: 0.147 sec (0h:0m:0s)
Total wall-clock time used: 0.150 sec (0h:0m:0s)

---&gt; START RUN NUMBER 8 (seed: 54350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.403
2. Current log-likelihood: -1404.572
3. Current log-likelihood: -1399.236
4. Current log-likelihood: -1397.831
5. Current log-likelihood: -1397.061
Optimal log-likelihood: -1396.470
Rate parameters:  A-C: 0.24622  A-G: 2.08130  A-T: 1.99445  C-G: 1.06444  C-T: 2.85652  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.437
Parameters optimization took 5 rounds (0.041 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.961
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.181
Finish initializing candidate tree set (3)
Current best tree score: -1388.181 / CPU time: 0.056
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.181
2. Current log-likelihood: -1387.955
3. Current log-likelihood: -1387.793
4. Current log-likelihood: -1387.674
5. Current log-likelihood: -1387.584
6. Current log-likelihood: -1387.516
7. Current log-likelihood: -1387.463
Optimal log-likelihood: -1387.421
Rate parameters:  A-C: 0.33228  A-G: 2.23486  A-T: 2.10851  C-G: 1.15950  C-T: 3.22710  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.360
Parameters optimization took 7 rounds (0.024 sec)
BEST SCORE FOUND : -1387.421
Total tree length: 6.696

Total number of iterations: 2
CPU time used for tree search: 0.055 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.056 sec (0h:0m:0s)
Total CPU time used: 0.141 sec (0h:0m:0s)
Total wall-clock time used: 0.144 sec (0h:0m:0s)

---&gt; START RUN NUMBER 9 (seed: 55350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1482.404
2. Current log-likelihood: -1403.041
3. Current log-likelihood: -1398.273
4. Current log-likelihood: -1396.900
5. Current log-likelihood: -1396.172
Optimal log-likelihood: -1395.670
Rate parameters:  A-C: 0.23522  A-G: 2.07026  A-T: 1.96673  C-G: 1.07739  C-T: 2.84705  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.026
Gamma shape alpha: 1.338
Parameters optimization took 5 rounds (0.037 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
WARNING: Some pairwise ML distances are too long (saturated)
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.979
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.172
Finish initializing candidate tree set (4)
Current best tree score: -1388.172 / CPU time: 0.042
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.172
2. Current log-likelihood: -1387.952
3. Current log-likelihood: -1387.788
4. Current log-likelihood: -1387.670
5. Current log-likelihood: -1387.581
6. Current log-likelihood: -1387.513
7. Current log-likelihood: -1387.460
Optimal log-likelihood: -1387.418
Rate parameters:  A-C: 0.33567  A-G: 2.25588  A-T: 2.13013  C-G: 1.17112  C-T: 3.26236  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.356
Parameters optimization took 7 rounds (0.025 sec)
BEST SCORE FOUND : -1387.418
Total tree length: 6.706

Total number of iterations: 2
CPU time used for tree search: 0.042 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.042 sec (0h:0m:0s)
Total CPU time used: 0.123 sec (0h:0m:0s)
Total wall-clock time used: 0.126 sec (0h:0m:0s)

---&gt; START RUN NUMBER 10 (seed: 56350)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.500)
1. Initial log-likelihood: -1481.468
2. Current log-likelihood: -1401.790
3. Current log-likelihood: -1396.546
4. Current log-likelihood: -1395.117
5. Current log-likelihood: -1394.384
Optimal log-likelihood: -1393.812
Rate parameters:  A-C: 0.27488  A-G: 2.41325  A-T: 2.17675  C-G: 1.25265  C-T: 3.29501  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.027
Gamma shape alpha: 1.421
Parameters optimization took 5 rounds (0.035 sec)
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1393.791
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.500)
BETTER TREE FOUND at iteration 1: -1388.211
Finish initializing candidate tree set (3)
Current best tree score: -1388.211 / CPU time: 0.049
Number of iterations: 2
TREE SEARCH COMPLETED AFTER 2 ITERATIONS / Time: 0h:0m:0s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.050)
1. Initial log-likelihood: -1388.211
2. Current log-likelihood: -1388.012
3. Current log-likelihood: -1387.866
4. Current log-likelihood: -1387.758
5. Current log-likelihood: -1387.675
6. Current log-likelihood: -1387.610
7. Current log-likelihood: -1387.559
Optimal log-likelihood: -1387.518
Rate parameters:  A-C: 0.35581  A-G: 2.35550  A-T: 2.14264  C-G: 1.20478  C-T: 3.37540  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.004
Gamma shape alpha: 1.361
Parameters optimization took 7 rounds (0.027 sec)
BEST SCORE FOUND : -1387.518
Total tree length: 6.815

Total number of iterations: 2
CPU time used for tree search: 0.049 sec (0h:0m:0s)
Wall-clock time used for tree search: 0.049 sec (0h:0m:0s)
Total CPU time used: 0.131 sec (0h:0m:0s)
Total wall-clock time used: 0.134 sec (0h:0m:0s)

---&gt; SUMMARIZE RESULTS FROM 10 RUNS

Run 1 gave best log-likelihood: -1387.259
Total CPU time for 10 runs: 1.671 seconds.
Total wall-clock time for 10 runs: 1.709 seconds.


Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree.treefile
  Trees from independent runs:   /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree.runtrees
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree.log

Date and Time: Fri Feb 28 11:27:54 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 10 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-s_1xjkve/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpjtozjjyf/q2iqtree -nt 1 -fast

Saved Phylogeny[Unrooted] to: iqt-gtrig-fast-ms-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-gtrig-fast-ms-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-gtrig-fast-ms-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-gtrig-fast-ms-tree.qza">download</a></p></li>
</ul>
</div>
</div>
<div class="section" id="single-branch-tests">
<h4><a class="toc-backref" href="#id16">Single branch tests</a><a class="headerlink" href="#single-branch-tests" title="Permalink to this headline">¬∂</a></h4>
<p>IQ-TREE provides access to a few <a class="reference external" href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with-single-branch-tests">single branch testing methods</a></p>
<ol class="arabic simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/sysbio/syq010">SH-aLRT</a> via <code class="docutils literal notranslate"><span class="pre">--p-alrt</span> <span class="pre">[INT</span> <span class="pre">&gt;=</span> <span class="pre">1000]</span></code></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/sysbio/syr041">aBayes</a> via <code class="docutils literal notranslate"><span class="pre">--p-abayes</span> <span class="pre">[TRUE</span> <span class="pre">|</span> <span class="pre">FALSE]</span></code></p></li>
<li><p><a class="reference external" href="https://doi.org/10.1007/BF0249864">local bootstrap test</a> via <code class="docutils literal notranslate"><span class="pre">--p-lbp</span> <span class="pre">[INT</span> <span class="pre">&gt;=</span> <span class="pre">1000]</span></code></p></li>
</ol>
<p>Single branch tests are commonly used as an alternative to the bootstrapping
approach we‚Äôve discussed above, as they are substantially faster and <a class="reference external" href="http://www.iqtree.org/doc/Command-Reference#single-branch-tests">often
recommended</a> when constructing large phylogenies (e.g. &gt;10,000 taxa). All
three of these methods can be applied simultaneously and viewed within <a class="reference external" href="https://itol.embl.de/">iTOL</a>
as separate bootstrap support values. These values are always in listed in the
following order of <em>alrt / lbp / abayes</em>. We‚Äôll go ahead and apply all of the
branch tests in our next command, while specifying the same substitution model
as above. Feel free to combine this with the <code class="docutils literal notranslate"><span class="pre">--p-fast</span></code> option. üòâ</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-alrt <span class="m">1000</span> <span class="se">\</span>
  --p-abayes <span class="se">\</span>
  --p-lbp <span class="m">1000</span> <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-sbt-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g3kizhqg/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree -nt 1 -alrt 1000 -abayes -lbp 1000
Seed:    933482 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:27:58 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g3kizhqg/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.281 / LogL: -1392.558
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.008, 1.381 / LogL: -1392.827
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.009, 1.392 / LogL: -1392.900
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.009, 1.386 / LogL: -1392.892
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.008, 1.381 / LogL: -1392.857
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.386 / LogL: -1392.886
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.008, 1.377 / LogL: -1392.831
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.379 / LogL: -1392.845
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.008, 1.381 / LogL: -1392.850
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.860
Optimal pinv,alpha: 0.000, 1.281 / LogL: -1392.558

Parameters optimization took 0.485 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.710
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.091 second
Computing log-likelihood of 98 initial trees ... 0.207 seconds
Current best score: -1392.558

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.256
Iteration 10 / LogL: -1387.269 / Time: 0h:0m:0s
Iteration 20 / LogL: -1387.289 / Time: 0h:0m:1s
Finish initializing candidate tree set (2)
Current best tree score: -1387.256 / CPU time: 0.615
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Iteration 30 / LogL: -1387.374 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 40 / LogL: -1387.374 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 50 / LogL: -1387.375 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 60 / LogL: -1387.352 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 70 / LogL: -1387.731 / Time: 0h:0m:2s (0h:0m:1s left)
Iteration 80 / LogL: -1387.381 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 90 / LogL: -1388.542 / Time: 0h:0m:3s (0h:0m:0s left)
Iteration 100 / LogL: -1387.370 / Time: 0h:0m:3s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 102 ITERATIONS / Time: 0h:0m:3s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.256
Optimal log-likelihood: -1387.253
Rate parameters:  A-C: 0.33248  A-G: 2.27708  A-T: 2.15579  C-G: 1.18365  C-T: 3.30920  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.323
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1387.253

Testing tree branches by SH-like aLRT with 1000 replicates...
Testing tree branches by local-BP test with 1000 replicates...
Testing tree branches by aBayes parametric test...
0.057 sec.
Total tree length: 6.714

Total number of iterations: 102
CPU time used for tree search: 2.986 sec (0h:0m:2s)
Wall-clock time used for tree search: 2.989 sec (0h:0m:2s)
Total CPU time used: 3.551 sec (0h:0m:3s)
Total wall-clock time used: 3.557 sec (0h:0m:3s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree.log

Date and Time: Fri Feb 28 11:28:02 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-g3kizhqg/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpfk73zti6/q2iqtree -nt 1 -alrt 1000 -abayes -lbp 1000

Saved Phylogeny[Unrooted] to: iqt-sbt-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-sbt-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-sbt-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-sbt-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>IQ-TREE search settings</strong>.
There are quite a few adjustable parameters available for <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> that
can be modified improve searches through ‚Äútree space‚Äù and prevent the search
algorithms from getting stuck in local optima. One particular <a class="reference external" href="https://groups.google.com/forum/#!searchin/iqtree/iterations|sort:date/iqtree/0mwGhDokNns/vlBryIwXHAAJ">best
practice</a> to aid in this regard, is to adjust the following parameters:
<code class="docutils literal notranslate"><span class="pre">--p-perturb-nni-strength</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-stop-iter</span></code> (each respectively maps
to the <code class="docutils literal notranslate"><span class="pre">-pers</span></code> and <code class="docutils literal notranslate"><span class="pre">-nstop</span></code> flags of <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> ). In brief, the larger
the value for NNI (nearest-neighbor interchange) perturbation, the larger
the jumps in ‚Äútree space‚Äù. This value should be set high enough to allow the
search algorithm to avoid being trapped in local optima, but not to high
that the search is haphazardly jumping around ‚Äútree space‚Äù. That is, like
Goldilocks and the three üêªs you need to find a setting that is ‚Äújust
right‚Äù, or at least within a set of reasonable bounds. One way of assessing
this, is to do a few short trial runs using the <code class="docutils literal notranslate"><span class="pre">--verbose</span></code> flag. If you
see that the likelihood values are jumping around to much, then lowering the
value for <code class="docutils literal notranslate"><span class="pre">--p-perturb-nni-strength</span></code> may be warranted. As for the stopping
criteria, i.e. <code class="docutils literal notranslate"><span class="pre">--p-stop-iter</span></code>, the higher this value, the more thorough
your search in ‚Äútree space‚Äù. Be aware, increasing this value may also
increase the run time. That is, the search will continue until it has
sampled a number of trees, say 100 (default), without finding a better
scoring tree. If a better tree is found, then the counter resets, and the
search continues. These two parameters deserve special consideration when a
given data set contains many short sequences, quite common for microbiome
survey data. We can modify our original command to include these extra
parameters with the recommended modifications for short sequences, i.e. a
lower value for perturbation strength (shorter reads do not contain as much
phylogenetic information, thus we should limit how far we jump around in
‚Äútree space‚Äù) and a larger number of stop iterations. See the <a class="reference external" href="http://www.iqtree.org/doc/Command-Reference">IQ-TREE
command reference</a> for more details about default parameter settings.
Finally, we‚Äôll let <code class="docutils literal notranslate"><span class="pre">iqtree</span></code> perform the model testing, and automatically
determine the optimal number of CPU cores to use.</p>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --o-tree iqt-nnisi-fast-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-fx13358_/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree -nt 1 -nstop 200 -pers 0.200000
Seed:    660724 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:28:06 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-fx13358_/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1424.993     37  2923.986     2939.964     3048.528
  2  JC+I          1423.695     38  2923.391     2940.328     3051.298
  3  JC+G4         1415.170     38  2906.339     2923.277     3034.246
  4  JC+I+G4       1415.529     39  2909.058     2926.989     3040.331
  5  JC+R2         1411.535     39  2901.071     2919.002     3032.344
  6  JC+R3         1411.462     41  2904.923     2924.946     3042.928
 14  F81+F         1423.131     40  2926.262     2945.221     3060.901
 15  F81+F+I       1421.685     41  2925.370     2945.393     3063.375
 16  F81+F+G4      1412.577     41  2907.154     2927.177     3045.159
 17  F81+F+I+G4    1412.881     42  2909.761     2930.884     3051.132
 18  F81+F+R2      1408.249     42  2900.498     2921.621     3041.869
 19  F81+F+R3      1408.230     44  2904.461     2927.892     3052.563
 27  K2P           1414.902     38  2905.805     2922.742     3033.712
 28  K2P+I         1413.837     39  2905.674     2923.605     3036.947
 29  K2P+G4        1404.066     39  2886.132     2904.063     3017.405
 30  K2P+I+G4      1404.352     40  2888.704     2907.664     3023.343
 31  K2P+R2        1399.045     40  2878.091     2897.050     3012.730
 32  K2P+R3        1399.042     42  2882.084     2903.207     3023.455
 40  HKY+F         1413.262     41  2908.523     2928.547     3046.528
 41  HKY+F+I       1411.921     42  2907.841     2928.964     3049.212
 42  HKY+F+G4      1401.469     42  2886.938     2908.061     3028.309
 43  HKY+F+I+G4    1401.702     43  2889.403     2911.662     3034.140
 44  HKY+F+R2      1395.787     43  2877.573     2899.832     3022.310
 45  HKY+F+R3      1395.785     45  2881.571     2906.214     3033.040
 53  TNe           1414.871     39  2907.743     2925.674     3039.016
 54  TNe+I         1413.730     40  2907.460     2926.419     3042.099
 55  TNe+G4        1404.041     40  2888.081     2907.041     3022.720
 56  TNe+I+G4      1404.223     41  2890.447     2910.470     3028.452
 57  TNe+R2        1399.033     41  2880.065     2900.088     3018.070
 58  TNe+R3        1399.031     43  2884.063     2906.322     3028.800
 66  TN+F          1412.704     42  2909.408     2930.531     3050.779
 67  TN+F+I        1411.527     43  2909.054     2931.313     3053.791
 68  TN+F+G4       1400.564     43  2887.128     2909.387     3031.865
 69  TN+F+I+G4     1400.724     44  2889.449     2912.881     3037.552
 70  TN+F+R2       1394.489     44  2876.979     2900.411     3025.082
 71  TN+F+R3       1394.487     46  2880.974     2906.866     3035.808
 79  K3P           1412.734     39  2903.469     2921.400     3034.742
 80  K3P+I         1411.661     40  2903.322     2922.281     3037.961
 81  K3P+G4        1400.853     40  2881.706     2900.666     3016.345
 82  K3P+I+G4      1400.986     41  2883.971     2903.995     3021.976
 83  K3P+R2        1395.851     41  2873.702     2893.725     3011.707
 84  K3P+R3        1395.850     43  2877.700     2899.959     3022.437
 92  K3Pu+F        1410.542     42  2905.084     2926.207     3046.455
 93  K3Pu+F+I      1409.073     43  2904.145     2926.404     3048.882
 94  K3Pu+F+G4     1397.230     43  2880.460     2902.719     3025.197
 95  K3Pu+F+I+G4   1397.352     44  2882.704     2906.136     3030.807
 96  K3Pu+F+R2     1391.701     44  2871.401     2894.833     3019.504
 97  K3Pu+F+R3     1391.669     46  2875.338     2901.230     3030.173
105  TPM2+F        1413.181     42  2910.362     2931.484     3051.733
106  TPM2+F+I      1411.826     43  2909.652     2931.911     3054.389
107  TPM2+F+G4     1401.155     43  2888.310     2910.569     3033.047
108  TPM2+F+I+G4   1401.242     44  2890.485     2913.917     3038.588
109  TPM2+F+R2     1395.447     44  2878.894     2902.326     3026.997
110  TPM2+F+R3     1395.415     46  2882.830     2908.722     3037.665
118  TPM2u+F       1413.181     42  2910.363     2931.486     3051.734
119  TPM2u+F+I     1411.826     43  2909.652     2931.911     3054.389
120  TPM2u+F+G4    1401.160     43  2888.320     2910.579     3033.057
121  TPM2u+F+I+G4  1401.228     44  2890.457     2913.889     3038.560
122  TPM2u+F+R2    1395.447     44  2878.894     2902.326     3026.997
123  TPM2u+F+R3    1395.415     46  2882.830     2908.722     3037.665
131  TPM3+F        1409.772     42  2903.545     2924.668     3044.916
132  TPM3+F+I      1408.714     43  2903.427     2925.686     3048.164
133  TPM3+F+G4     1398.034     43  2882.068     2904.327     3026.805
134  TPM3+F+I+G4   1398.089     44  2884.177     2907.609     3032.280
135  TPM3+F+R2     1392.917     44  2873.834     2897.266     3021.937
136  TPM3+F+R3     1392.894     46  2877.788     2903.681     3032.623
144  TPM3u+F       1409.810     42  2903.619     2924.742     3044.990
145  TPM3u+F+I     1408.704     43  2903.408     2925.667     3048.145
146  TPM3u+F+G4    1398.037     43  2882.073     2904.332     3026.810
147  TPM3u+F+I+G4  1398.077     44  2884.154     2907.586     3032.257
148  TPM3u+F+R2    1392.916     44  2873.833     2897.265     3021.936
149  TPM3u+F+R3    1392.893     46  2877.785     2903.678     3032.620
157  TIMe          1412.731     40  2905.462     2924.422     3040.101
158  TIMe+I        1411.552     41  2905.105     2925.128     3043.110
159  TIMe+G4       1400.835     41  2883.670     2903.693     3021.675
160  TIMe+I+G4     1400.867     42  2885.734     2906.856     3027.105
161  TIMe+R2       1395.821     42  2875.642     2896.765     3017.013
162  TIMe+R3       1395.799     44  2879.597     2903.029     3027.700
170  TIM+F         1409.958     43  2905.916     2928.174     3050.653
171  TIM+F+I       1408.653     44  2905.307     2928.739     3053.410
172  TIM+F+G4      1396.232     44  2880.464     2903.896     3028.567
173  TIM+F+I+G4    1396.282     45  2882.563     2907.206     3034.032
174  TIM+F+R2      1390.319     45  2870.639     2895.282     3022.108
175  TIM+F+R3      1390.271     47  2874.543     2901.723     3032.744
183  TIM2e         1414.830     40  2909.660     2928.619     3044.299
184  TIM2e+I       1413.678     41  2909.356     2929.379     3047.361
185  TIM2e+G4      1404.027     41  2890.055     2910.078     3028.060
186  TIM2e+I+G4    1404.057     42  2892.114     2913.237     3033.485
187  TIM2e+R2      1398.983     42  2881.966     2903.089     3023.337
188  TIM2e+R3      1398.957     44  2885.914     2909.346     3034.017
196  TIM2+F        1412.622     43  2911.245     2933.504     3055.982
197  TIM2+F+I      1411.425     44  2910.851     2934.283     3058.954
198  TIM2+F+G4     1400.276     44  2888.553     2911.985     3036.656
199  TIM2+F+I+G4   1400.286     45  2890.572     2915.215     3042.041
200  TIM2+F+R2     1394.100     45  2878.200     2902.843     3029.669
201  TIM2+F+R3     1394.051     47  2882.102     2909.283     3040.303
209  TIM3e         1407.728     40  2895.455     2914.415     3030.094
210  TIM3e+I       1406.829     41  2895.658     2915.681     3033.663
211  TIM3e+G4      1396.518     41  2875.035     2895.059     3013.040
212  TIM3e+I+G4    1396.536     42  2877.072     2898.195     3018.443
213  TIM3e+R2      1392.306     42  2868.612     2889.735     3009.983
214  TIM3e+R3      1392.302     44  2872.604     2896.036     3020.707
222  TIM3+F        1409.151     43  2904.302     2926.561     3049.039
223  TIM3+F+I      1408.234     44  2904.467     2927.899     3052.570
224  TIM3+F+G4     1397.147     44  2882.293     2905.725     3030.396
225  TIM3+F+I+G4   1397.143     45  2884.285     2908.928     3035.754
226  TIM3+F+R2     1391.454     45  2872.908     2897.551     3024.377
227  TIM3+F+R3     1391.416     47  2876.833     2904.014     3035.034
235  TVMe          1404.546     41  2891.091     2911.114     3029.096
236  TVMe+I        1403.650     42  2891.299     2912.422     3032.670
237  TVMe+G4       1392.865     42  2869.730     2890.853     3011.101
238  TVMe+I+G4     1392.877     43  2871.754     2894.013     3016.491
239  TVMe+R2       1388.927     43  2863.853     2886.112     3008.590
240  TVMe+R3       1388.922     45  2867.844     2892.487     3019.313
248  TVM+F         1406.321     44  2900.642     2924.074     3048.745
249  TVM+F+I       1405.053     45  2900.106     2924.749     3051.575
250  TVM+F+G4      1393.365     45  2876.729     2901.372     3028.198
251  TVM+F+I+G4    1393.378     46  2878.756     2904.648     3033.591
252  TVM+F+R2      1388.517     46  2869.033     2894.926     3023.868
253  TVM+F+R3      1388.485     48  2872.969     2901.478     3034.536
261  SYM           1404.527     42  2893.053     2914.176     3034.424
262  SYM+I         1403.612     43  2893.225     2915.484     3037.962
263  SYM+G4        1392.845     43  2871.689     2893.948     3016.426
264  SYM+I+G4      1392.852     44  2873.704     2897.136     3021.807
265  SYM+R2        1388.921     44  2865.842     2889.274     3013.945
266  SYM+R3        1388.914     46  2869.829     2895.721     3024.664
274  GTR+F         1405.640     45  2901.279     2925.922     3052.748
275  GTR+F+I       1404.620     46  2901.240     2927.133     3056.075
276  GTR+F+G4      1392.562     46  2877.125     2903.017     3031.960
277  GTR+F+I+G4    1392.555     47  2879.110     2906.291     3037.311
278  GTR+F+R2      1387.241     47  2868.482     2895.662     3026.683
279  GTR+F+R3      1387.205     49  2872.410     2902.288     3037.343
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree.model.gz
CPU time for ModelFinder: 0.912 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.918 seconds (0h:0m:0s)

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1416.636
2. Current log-likelihood: -1395.960
3. Current log-likelihood: -1390.000
4. Current log-likelihood: -1389.337
5. Current log-likelihood: -1389.140
6. Current log-likelihood: -1389.022
Optimal log-likelihood: -1388.947
Rate parameters:  A-C: 0.15999  A-G: 2.04528  A-T: 1.56960  C-G: 0.79365  C-T: 2.04528  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.668,0.376) (0.332,2.254)
Parameters optimization took 6 rounds (0.035 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.067
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.088 second
Computing log-likelihood of 97 initial trees ... 0.136 seconds
Current best score: -1388.947

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.381
Iteration 10 / LogL: -1382.387 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.389 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.381 / CPU time: 0.451
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
BETTER TREE FOUND at iteration 21: -1382.380
Iteration 30 / LogL: -1382.922 / Time: 0h:0m:0s (0h:0m:4s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 37: -1382.036
Iteration 40 / LogL: -1382.871 / Time: 0h:0m:0s (0h:0m:3s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.034
Iteration 50 / LogL: -1388.652 / Time: 0h:0m:0s (0h:0m:3s left)
Iteration 60 / LogL: -1382.045 / Time: 0h:0m:0s (0h:0m:2s left)
Iteration 70 / LogL: -1382.438 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 80 / LogL: -1382.437 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 90 / LogL: -1382.439 / Time: 0h:0m:1s (0h:0m:2s left)
Iteration 100 / LogL: -1382.438 / Time: 0h:0m:1s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.034
Iteration 110 / LogL: -1382.437 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 120 / LogL: -1382.437 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 130 / LogL: -1382.438 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 140 / LogL: -1382.105 / Time: 0h:0m:1s (0h:0m:1s left)
Iteration 150 / LogL: -1382.436 / Time: 0h:0m:1s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.034
Iteration 160 / LogL: -1382.104 / Time: 0h:0m:1s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.034
Iteration 170 / LogL: -1382.104 / Time: 0h:0m:2s (0h:0m:0s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.034
Iteration 180 / LogL: -1382.185 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 190 / LogL: -1382.437 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 200 / LogL: -1382.225 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 210 / LogL: -1382.108 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 220 / LogL: -1382.106 / Time: 0h:0m:2s (0h:0m:0s left)
Iteration 230 / LogL: -1382.104 / Time: 0h:0m:2s (0h:0m:0s left)
TREE SEARCH COMPLETED AFTER 238 ITERATIONS / Time: 0h:0m:2s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.034
Optimal log-likelihood: -1382.026
Rate parameters:  A-C: 0.18964  A-G: 1.83455  A-T: 1.53204  C-G: 0.76868  C-T: 1.83455  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.718,0.416) (0.282,2.491)
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1382.026
Total tree length: 6.982

Total number of iterations: 238
CPU time used for tree search: 2.705 sec (0h:0m:2s)
Wall-clock time used for tree search: 2.711 sec (0h:0m:2s)
Total CPU time used: 2.759 sec (0h:0m:2s)
Total wall-clock time used: 2.767 sec (0h:0m:2s)

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree.mldist
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree.log

Date and Time: Fri Feb 28 11:28:10 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-fx13358_/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpdqj32h9n/q2iqtree -nt 1 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-fast-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-fast-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-fast-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-fast-tree.qza">download</a></p></li>
</ul>
</div>
</div>
</div>
<div class="section" id="iqtree-ultrafast-bootstrap">
<h3><a class="toc-backref" href="#id17">iqtree-ultrafast-bootstrap</a><a class="headerlink" href="#iqtree-ultrafast-bootstrap" title="Permalink to this headline">¬∂</a></h3>
<p>As per our discussion in the <code class="docutils literal notranslate"><span class="pre">raxml-rapid-bootstrap</span></code> section above, we can
also use IQ-TREE to evaluate how well our splits / bipartitions are supported
within our phylogeny via the <a class="reference external" href="https://doi.org/10.1093/molbev/msx281">ultrafast bootstrap algorithm</a>. Below, we‚Äôll
apply the plugin‚Äôs
<a class="reference internal" href="../../plugins/available/phylogeny/iqtree-ultrafast-bootstrap/"><span class="doc">ultrafast bootstrap command</span></a>:
automatic model selection (<code class="docutils literal notranslate"><span class="pre">MFP</span></code>), perform <code class="docutils literal notranslate"><span class="pre">1000</span></code> bootstrap replicates
(minimum required), set the same generally suggested parameters for
constructing a phylogeny from short sequences, and automatically determine the
optimal number of CPU cores to use:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree-ultrafast-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --o-tree iqt-nnisi-bootstrap-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-zlvwycxz/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot -nt 1 -nstop 200 -pers 0.200000
Seed:    580991 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:28:14 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-zlvwycxz/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)


Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
NOTE: ModelFinder requires 1 MB RAM!
ModelFinder will test 286 DNA models (sample size: 214) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  JC            1425.567     37  2925.133     2941.110     3049.674
  2  JC+I          1424.230     38  2924.459     2941.396     3052.366
  3  JC+G4         1415.401     38  2906.802     2923.739     3034.709
  4  JC+I+G4       1415.767     39  2909.534     2927.465     3040.807
  5  JC+R2         1411.638     39  2901.276     2919.207     3032.549
  6  JC+R3         1411.576     41  2905.151     2925.174     3043.156
 14  F81+F         1424.052     40  2928.104     2947.063     3062.743
 15  F81+F+I       1422.562     41  2927.125     2947.148     3065.130
 16  F81+F+G4      1413.077     41  2908.155     2928.178     3046.160
 17  F81+F+I+G4    1413.383     42  2910.766     2931.889     3052.137
 18  F81+F+R2      1408.573     42  2901.145     2922.268     3042.516
 19  F81+F+R3      1408.558     44  2905.115     2928.547     3053.218
 27  K2P           1415.022     38  2906.044     2922.981     3033.951
 28  K2P+I         1413.934     39  2905.868     2923.799     3037.141
 29  K2P+G4        1404.053     39  2886.107     2904.038     3017.380
 30  K2P+I+G4      1404.343     40  2888.686     2907.646     3023.325
 31  K2P+R2        1399.029     40  2878.059     2897.018     3012.698
 32  K2P+R3        1399.027     42  2882.054     2903.176     3023.425
 40  HKY+F         1413.605     41  2909.210     2929.233     3047.215
 41  HKY+F+I       1412.216     42  2908.432     2929.554     3049.803
 42  HKY+F+G4      1401.517     42  2887.033     2908.156     3028.404
 43  HKY+F+I+G4    1401.751     43  2889.503     2911.762     3034.240
 44  HKY+F+R2      1395.775     43  2877.550     2899.809     3022.287
 45  HKY+F+R3      1395.775     45  2881.550     2906.193     3033.019
 53  TNe           1414.995     39  2907.990     2925.921     3039.263
 54  TNe+I         1413.835     40  2907.669     2926.629     3042.308
 55  TNe+G4        1404.030     40  2888.061     2907.020     3022.700
 56  TNe+I+G4      1404.221     41  2890.441     2910.464     3028.446
 57  TNe+R2        1399.014     41  2880.027     2900.051     3018.032
 58  TNe+R3        1399.013     43  2884.025     2906.284     3028.762
 66  TN+F          1413.020     42  2910.040     2931.163     3051.411
 67  TN+F+I        1411.801     43  2909.602     2931.861     3054.339
 68  TN+F+G4       1400.568     43  2887.137     2909.396     3031.874
 69  TN+F+I+G4     1400.736     44  2889.472     2912.903     3037.574
 70  TN+F+R2       1394.454     44  2876.908     2900.340     3025.011
 71  TN+F+R3       1394.454     46  2880.907     2906.799     3035.742
 79  K3P           1413.059     39  2904.118     2922.049     3035.391
 80  K3P+I         1411.960     40  2903.921     2922.880     3038.560
 81  K3P+G4        1401.013     40  2882.026     2900.985     3016.665
 82  K3P+I+G4      1401.147     41  2884.294     2904.317     3022.299
 83  K3P+R2        1395.921     41  2873.841     2893.864     3011.846
 84  K3P+R3        1395.919     43  2877.839     2900.098     3022.576
 92  K3Pu+F        1411.190     42  2906.380     2927.503     3047.751
 93  K3Pu+F+I      1409.682     43  2905.363     2927.622     3050.100
 94  K3Pu+F+G4     1397.628     43  2881.256     2903.515     3025.993
 95  K3Pu+F+I+G4   1397.752     44  2883.504     2906.936     3031.607
 96  K3Pu+F+R2     1391.987     44  2871.973     2895.405     3020.076
 97  K3Pu+F+R3     1391.928     46  2875.856     2901.749     3030.691
105  TPM2+F        1413.534     42  2911.068     2932.191     3052.439
106  TPM2+F+I      1412.133     43  2910.265     2932.524     3055.002
107  TPM2+F+G4     1401.209     43  2888.417     2910.676     3033.154
108  TPM2+F+I+G4   1401.298     44  2890.597     2914.029     3038.700
109  TPM2+F+R2     1395.472     44  2878.944     2902.376     3027.047
110  TPM2+F+R3     1395.418     46  2882.836     2908.728     3037.670
118  TPM2u+F       1413.535     42  2911.070     2932.193     3052.441
119  TPM2u+F+I     1412.133     43  2910.265     2932.524     3055.002
120  TPM2u+F+G4    1401.212     43  2888.424     2910.683     3033.161
121  TPM2u+F+I+G4  1401.284     44  2890.569     2914.001     3038.672
122  TPM2u+F+R2    1395.472     44  2878.943     2902.375     3027.046
123  TPM2u+F+R3    1395.418     46  2882.836     2908.728     3037.670
131  TPM3+F        1409.968     42  2903.935     2925.058     3045.306
132  TPM3+F+I      1408.875     43  2903.750     2926.009     3048.487
133  TPM3+F+G4     1398.029     43  2882.059     2904.318     3026.796
134  TPM3+F+I+G4   1398.092     44  2884.185     2907.617     3032.288
135  TPM3+F+R2     1392.918     44  2873.836     2897.267     3021.938
136  TPM3+F+R3     1392.880     46  2877.760     2903.652     3032.594
144  TPM3u+F       1410.004     42  2904.009     2925.132     3045.380
145  TPM3u+F+I     1408.865     43  2903.731     2925.990     3048.468
146  TPM3u+F+G4    1398.031     43  2882.063     2904.321     3026.800
147  TPM3u+F+I+G4  1398.080     44  2884.159     2907.591     3032.262
148  TPM3u+F+R2    1392.918     44  2873.835     2897.267     3021.938
149  TPM3u+F+R3    1392.879     46  2877.759     2903.651     3032.593
157  TIMe          1413.059     40  2906.119     2925.078     3040.758
158  TIMe+I        1411.862     41  2905.725     2925.748     3043.730
159  TIMe+G4       1401.003     41  2884.006     2904.029     3022.011
160  TIMe+I+G4     1401.035     42  2886.071     2907.194     3027.442
161  TIMe+R2       1395.912     42  2875.824     2896.947     3017.195
162  TIMe+R3       1395.872     44  2879.744     2903.176     3027.847
170  TIM+F         1410.576     43  2907.153     2929.412     3051.890
171  TIM+F+I       1409.238     44  2906.476     2929.908     3054.579
172  TIM+F+G4      1396.545     44  2881.091     2904.523     3029.194
173  TIM+F+I+G4    1396.596     45  2883.192     2907.835     3034.661
174  TIM+F+R2      1390.521     45  2871.043     2895.686     3022.512
175  TIM+F+R3      1390.475     47  2874.951     2902.131     3033.151
183  TIM2e         1414.950     40  2909.900     2928.860     3044.539
184  TIM2e+I       1413.779     41  2909.558     2929.581     3047.563
185  TIM2e+G4      1404.022     41  2890.045     2910.068     3028.050
186  TIM2e+I+G4    1404.060     42  2892.120     2913.243     3033.491
187  TIM2e+R2      1398.987     42  2881.975     2903.097     3023.346
188  TIM2e+R3      1398.965     44  2885.929     2909.361     3034.032
196  TIM2+F        1412.949     43  2911.898     2934.157     3056.635
197  TIM2+F+I      1411.711     44  2911.421     2934.853     3059.524
198  TIM2+F+G4     1400.286     44  2888.573     2912.005     3036.676
199  TIM2+F+I+G4   1400.302     45  2890.605     2915.247     3042.073
200  TIM2+F+R2     1394.103     45  2878.205     2902.848     3029.674
201  TIM2+F+R3     1394.061     47  2882.121     2909.302     3040.322
209  TIM3e         1407.784     40  2895.569     2914.528     3030.208
210  TIM3e+I       1406.875     41  2895.749     2915.772     3033.754
211  TIM3e+G4      1396.516     41  2875.032     2895.055     3013.037
212  TIM3e+I+G4    1396.538     42  2877.077     2898.200     3018.448
213  TIM3e+R2      1392.307     42  2868.613     2889.736     3009.984
214  TIM3e+R3      1392.303     44  2872.605     2896.037     3020.708
222  TIM3+F        1409.328     43  2904.655     2926.914     3049.392
223  TIM3+F+I      1408.384     44  2904.767     2928.199     3052.870
224  TIM3+F+G4     1397.129     44  2882.258     2905.690     3030.361
225  TIM3+F+I+G4   1397.137     45  2884.274     2908.917     3035.743
226  TIM3+F+R2     1391.457     45  2872.914     2897.557     3024.383
227  TIM3+F+R3     1391.424     47  2876.849     2904.029     3035.050
235  TVMe          1404.746     41  2891.492     2911.515     3029.497
236  TVMe+I        1403.834     42  2891.668     2912.791     3033.039
237  TVMe+G4       1392.944     42  2869.887     2891.010     3011.258
238  TVMe+I+G4     1392.955     43  2871.911     2894.170     3016.648
239  TVMe+R2       1388.955     43  2863.909     2886.168     3008.646
240  TVMe+R3       1388.950     45  2867.901     2892.544     3019.370
248  TVM+F         1406.790     44  2901.581     2925.013     3049.684
249  TVM+F+I       1405.500     45  2901.001     2925.644     3052.470
250  TVM+F+G4      1393.696     45  2877.392     2902.034     3028.860
251  TVM+F+I+G4    1393.700     46  2879.400     2905.292     3034.235
252  TVM+F+R2      1388.726     46  2869.452     2895.345     3024.287
253  TVM+F+R3      1388.694     48  2873.388     2901.897     3034.955
261  SYM           1404.730     42  2893.460     2914.583     3034.831
262  SYM+I         1403.801     43  2893.601     2915.860     3038.338
263  SYM+G4        1392.928     43  2871.856     2894.115     3016.593
264  SYM+I+G4      1392.935     44  2873.869     2897.301     3021.972
265  SYM+R2        1388.946     44  2865.892     2889.324     3013.995
266  SYM+R3        1388.940     46  2869.879     2895.771     3024.714
274  GTR+F         1406.089     45  2902.179     2926.821     3053.648
275  GTR+F+I       1405.055     46  2902.110     2928.003     3056.945
276  GTR+F+G4      1392.836     46  2877.672     2903.564     3032.507
277  GTR+F+I+G4    1392.818     47  2879.636     2906.816     3037.836
278  GTR+F+R2      1387.393     47  2868.785     2895.966     3026.986
279  GTR+F+R3      1387.357     49  2872.715     2902.593     3037.648
Akaike Information Criterion:           TVMe+R2
Corrected Akaike Information Criterion: TVMe+R2
Bayesian Information Criterion:         TVMe+R2
Best-fit model: TVMe+R2 chosen according to BIC

All model information printed to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.model.gz
CPU time for ModelFinder: 0.934 seconds (0h:0m:0s)
Wall-clock time for ModelFinder: 0.940 seconds (0h:0m:0s)
Generating 1000 samples for ultrafast bootstrap (seed: 580991)...

NOTE: 0 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
1. Initial log-likelihood: -1416.738
2. Current log-likelihood: -1396.507
3. Current log-likelihood: -1389.944
4. Current log-likelihood: -1389.313
5. Current log-likelihood: -1389.127
6. Current log-likelihood: -1389.021
Optimal log-likelihood: -1388.956
Rate parameters:  A-C: 0.16337  A-G: 2.05004  A-T: 1.55714  C-G: 0.78643  C-T: 2.05004  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.668,0.373) (0.332,2.263)
Parameters optimization took 6 rounds (0.036 sec)
Computing ML distances based on estimated model parameters... 0.002 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1389.065
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.087 second
Computing log-likelihood of 98 initial trees ... 0.133 seconds
Current best score: -1388.851

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1382.373
UPDATE BEST LOG-LIKELIHOOD: -1382.372
Iteration 10 / LogL: -1382.373 / Time: 0h:0m:0s
Iteration 20 / LogL: -1382.381 / Time: 0h:0m:0s
Finish initializing candidate tree set (1)
Current best tree score: -1382.372 / CPU time: 0.618
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Estimate model parameters (epsilon = 0.100)
UPDATE BEST LOG-LIKELIHOOD: -1382.104
Iteration 30 / LogL: -1382.113 / Time: 0h:0m:0s (0h:0m:4s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.103
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 34: -1382.024
Iteration 40 / LogL: -1382.434 / Time: 0h:0m:1s (0h:0m:5s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.023
Iteration 50 / LogL: -1382.040 / Time: 0h:0m:1s (0h:0m:4s left)
Log-likelihood cutoff on original alignment: -1400.814
Iteration 60 / LogL: -1382.437 / Time: 0h:0m:1s (0h:0m:4s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.022
Iteration 70 / LogL: -1382.025 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 80 / LogL: -1382.434 / Time: 0h:0m:1s (0h:0m:3s left)
Iteration 90 / LogL: -1382.118 / Time: 0h:0m:2s (0h:0m:3s left)
Iteration 100 / LogL: -1382.758 / Time: 0h:0m:2s (0h:0m:3s left)
Log-likelihood cutoff on original alignment: -1399.570
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.997
Iteration 110 / LogL: -1382.105 / Time: 0h:0m:2s (0h:0m:2s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.022
Iteration 120 / LogL: -1382.097 / Time: 0h:0m:2s (0h:0m:2s left)
Iteration 130 / LogL: -1382.434 / Time: 0h:0m:2s (0h:0m:2s left)
Iteration 140 / LogL: -1382.024 / Time: 0h:0m:3s (0h:0m:2s left)
Iteration 150 / LogL: -1382.059 / Time: 0h:0m:3s (0h:0m:1s left)
Log-likelihood cutoff on original alignment: -1401.179
Iteration 160 / LogL: -1382.097 / Time: 0h:0m:3s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1382.022
Iteration 170 / LogL: -1382.025 / Time: 0h:0m:3s (0h:0m:1s left)
Iteration 180 / LogL: -1382.147 / Time: 0h:0m:3s (0h:0m:1s left)
Iteration 190 / LogL: -1382.309 / Time: 0h:0m:4s (0h:0m:0s left)
Iteration 200 / LogL: -1382.032 / Time: 0h:0m:4s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -1402.144
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.998
UPDATE BEST LOG-LIKELIHOOD: -1382.022
Iteration 210 / LogL: -1382.455 / Time: 0h:0m:4s (0h:0m:1s left)
Iteration 220 / LogL: -1382.062 / Time: 0h:0m:4s (0h:0m:1s left)
Iteration 230 / LogL: -1382.033 / Time: 0h:0m:4s (0h:0m:1s left)
TREE SEARCH COMPLETED AFTER 235 ITERATIONS / Time: 0h:0m:5s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1382.022
Optimal log-likelihood: -1382.016
Rate parameters:  A-C: 0.19114  A-G: 1.83092  A-T: 1.52678  C-G: 0.76544  C-T: 1.83092  G-T: 1.00000
Base frequencies:  A: 0.250  C: 0.250  G: 0.250  T: 0.250
Site proportion and rates:  (0.720,0.414) (0.280,2.505)
Parameters optimization took 1 rounds (0.004 sec)
BEST SCORE FOUND : -1382.016
Creating bootstrap support values...
Split supports printed to NEXUS file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.splits.nex
Total tree length: 7.011

Total number of iterations: 235
CPU time used for tree search: 4.922 sec (0h:0m:4s)
Wall-clock time used for tree search: 4.954 sec (0h:0m:4s)
Total CPU time used: 5.055 sec (0h:0m:5s)
Total wall-clock time used: 5.093 sec (0h:0m:5s)

Computing bootstrap consensus tree...
Reading input file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.splits.nex...
20 taxa and 145 splits.
Consensus tree written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.contree
Reading input trees file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.contree
Log-likelihood of consensus tree: -1382.016

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.splits.nex
  Consensus tree:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.contree
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot.log

Date and Time: Fri Feb 28 11:28:20 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-zlvwycxz/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m MFP -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpi9rhim1o/q2iqtreeufboot -nt 1 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-bootstrap-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-tree.qza">download</a></p></li>
</ul>
</div>
<div class="section" id="perform-single-branch-tests-alongside-ufboot">
<h4><a class="toc-backref" href="#id18">Perform single branch tests alongside ufboot</a><a class="headerlink" href="#perform-single-branch-tests-alongside-ufboot" title="Permalink to this headline">¬∂</a></h4>
<p>We can also apply single branch test methods concurrently with ultrafast
bootstrapping. The support values will always be represented in the following
order: <em>alrt / lbp / abayes / ufboot</em>. Again, these values can be seen as
separately listed bootstrap values in <a class="reference external" href="https://itol.embl.de/">iTOL</a>. We‚Äôll also specify a model as we
did earlier.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny iqtree-ultrafast-bootstrap <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --p-perturb-nni-strength <span class="m">0</span>.2 <span class="se">\</span>
  --p-stop-iter <span class="m">200</span> <span class="se">\</span>
  --p-n-cores <span class="m">1</span> <span class="se">\</span>
  --p-alrt <span class="m">1000</span> <span class="se">\</span>
  --p-abayes <span class="se">\</span>
  --p-lbp <span class="m">1000</span> <span class="se">\</span>
  --p-substitution-model <span class="s1">&#39;GTR+I+G&#39;</span> <span class="se">\</span>
  --o-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza <span class="se">\</span>
  --verbose
</pre></div>
</div>
<p class="section-subtitle">stdout:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>IQ-TREE multicore version 1.6.12 for Mac OS X 64-bit built Aug 15 2019
Developed by Bui Quang Minh, Nguyen Lam Tung, Olga Chernomor,
Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    ghost.mggen.nau.edu (AVX, 16 GB RAM)
Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-jzcm2io3/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot -nt 1 -alrt 1000 -abayes -lbp 1000 -nstop 200 -pers 0.200000
Seed:    818587 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Fri Feb 28 11:28:24 2020
Kernel:  AVX - 1 threads (8 CPU cores detected)

HINT: Use -nt option to specify number of threads because your CPU has 8 cores!
HINT: -nt AUTO will automatically determine the best number of threads to use.

Reading alignment file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-jzcm2io3/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 20 sequences with 214 columns, 157 distinct patterns
104 parsimony-informative, 33 singleton sites, 77 constant sites
                                          Gap/Ambiguity  Composition  p-value
   1  e84fcf85a6a4065231dcf343bb862f1cb32abae6   40.65%    passed     90.91%
   2  5525fb6dab7b6577960147574465990c6df070ad   42.99%    passed     99.80%
   3  eb3564a35320b53cef22a77288838c7446357327   42.99%    passed     25.49%
   4  418f1d469f08c99976b313028cf6d3f18f61dd55   43.93%    passed     71.86%
   5  2e3b2c075901640c4de739473f9246385430b1ed   31.31%    passed     90.76%
   6  0469f8d819bd45c7638d1c8b0895270a05f34267   38.79%    passed     92.82%
   7  d162ed685007f5adede58f14aece31dfa1b60c18   40.65%    passed     97.17%
   8  1d45b2bce36cd995c5dcb755babf512e612ce8b9   41.59%    passed     39.04%
   9  5aba6bd9debc23ded7041ffdcfe5d68a427e8ce8   31.31%    passed     87.21%
  10  206656bec2abdbc4aee37a661ef5f4a62b5dd6ae   42.99%    passed     85.00%
  11  606c23e79bb730ad74e3c6efd72004c36674c17a   47.20%    passed     87.78%
  12  682e91d7e510ab134d0625234ad224f647c14eb0   41.59%    passed     31.01%
  13  6a36152105590b1eb095b9503e8f1f226fc73e43   39.25%    passed     86.29%
  14  6ca685c39a33bfbcb3123129e7af88d573df7d6f   42.06%    failed      0.02%
  15  8a1c44eb462ed58b21f3fdd72dd22bb657db2980   31.78%    passed     54.40%
  16  9b220cae8d375ea38b8b481cb95949cda8722fcb   36.92%    passed     88.78%
  17  aa4698d2e2b1fa71d08e2934a923aad7374a18f6   37.85%    passed     90.52%
  18  b31aa3f04bc9d5e2498d45cf1983dfaf09faa258   31.78%    passed     72.69%
  19  d44b129a6181f052198bda3813f0802a91612441   41.59%    passed     41.69%
  20  ed1acad8a98e8579a44370733533ad7d3fed8006   48.13%    passed     58.15%
****  TOTAL                                      39.77%  1 sequences failed composition chi2 test (p-value&lt;5%; df=3)

Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.001 seconds
Generating 1000 samples for ultrafast bootstrap (seed: 818587)...

NOTE: 1 MB RAM (0 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 1.000 / Estimate: 0.000, 1.281 / LogL: -1392.558
Init pinv, alpha: 0.040, 1.000 / Estimate: 0.008, 1.381 / LogL: -1392.827
Init pinv, alpha: 0.080, 1.000 / Estimate: 0.008, 1.382 / LogL: -1392.826
Init pinv, alpha: 0.120, 1.000 / Estimate: 0.009, 1.386 / LogL: -1392.893
Init pinv, alpha: 0.160, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.858
Init pinv, alpha: 0.200, 1.000 / Estimate: 0.009, 1.387 / LogL: -1392.886
Init pinv, alpha: 0.240, 1.000 / Estimate: 0.008, 1.378 / LogL: -1392.831
Init pinv, alpha: 0.280, 1.000 / Estimate: 0.008, 1.378 / LogL: -1392.845
Init pinv, alpha: 0.320, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.851
Init pinv, alpha: 0.360, 1.000 / Estimate: 0.008, 1.383 / LogL: -1392.860
Optimal pinv,alpha: 0.000, 1.281 / LogL: -1392.558

Parameters optimization took 0.478 sec
Computing ML distances based on estimated model parameters... 0.003 sec
Computing BIONJ tree...
0.001 seconds
Log-likelihood of BIONJ tree: -1392.710
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Generating 98 parsimony trees... 0.084 second
Computing log-likelihood of 98 initial trees ... 0.210 seconds
Current best score: -1389.881

Do NNI search on 20 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -1387.263
Iteration 10 / LogL: -1387.284 / Time: 0h:0m:1s
Iteration 20 / LogL: -1387.275 / Time: 0h:0m:1s
Finish initializing candidate tree set (1)
Current best tree score: -1387.263 / CPU time: 0.749
Number of iterations: 20
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
BETTER TREE FOUND at iteration 27: -1387.260
BETTER TREE FOUND at iteration 29: -1387.260
Iteration 30 / LogL: -1387.266 / Time: 0h:0m:1s (0h:0m:10s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.260
Iteration 40 / LogL: -1387.345 / Time: 0h:0m:1s (0h:0m:8s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.259
Iteration 50 / LogL: -1387.563 / Time: 0h:0m:2s (0h:0m:7s left)
Log-likelihood cutoff on original alignment: -1406.007
Iteration 60 / LogL: -1387.283 / Time: 0h:0m:2s (0h:0m:6s left)
Iteration 70 / LogL: -1387.360 / Time: 0h:0m:2s (0h:0m:6s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.259
Iteration 80 / LogL: -1387.259 / Time: 0h:0m:2s (0h:0m:5s left)
Iteration 90 / LogL: -1387.333 / Time: 0h:0m:3s (0h:0m:4s left)
Iteration 100 / LogL: -1387.359 / Time: 0h:0m:3s (0h:0m:4s left)
Log-likelihood cutoff on original alignment: -1407.608
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.995
Iteration 110 / LogL: -1387.299 / Time: 0h:0m:3s (0h:0m:4s left)
Iteration 120 / LogL: -1387.359 / Time: 0h:0m:4s (0h:0m:3s left)
Iteration 130 / LogL: -1387.307 / Time: 0h:0m:4s (0h:0m:3s left)
Iteration 140 / LogL: -1387.360 / Time: 0h:0m:4s (0h:0m:3s left)
Iteration 150 / LogL: -1387.352 / Time: 0h:0m:5s (0h:0m:2s left)
Log-likelihood cutoff on original alignment: -1408.422
Iteration 160 / LogL: -1387.370 / Time: 0h:0m:5s (0h:0m:2s left)
Iteration 170 / LogL: -1387.527 / Time: 0h:0m:5s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.259
Iteration 180 / LogL: -1387.278 / Time: 0h:0m:5s (0h:0m:1s left)
Iteration 190 / LogL: -1387.363 / Time: 0h:0m:6s (0h:0m:1s left)
UPDATE BEST LOG-LIKELIHOOD: -1387.259
Iteration 200 / LogL: -1387.527 / Time: 0h:0m:6s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -1408.422
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.999
Iteration 210 / LogL: -1387.324 / Time: 0h:0m:6s (0h:0m:2s left)
Iteration 220 / LogL: -1387.361 / Time: 0h:0m:7s (0h:0m:2s left)
Iteration 230 / LogL: -1388.510 / Time: 0h:0m:7s (0h:0m:2s left)
TREE SEARCH COMPLETED AFTER 230 ITERATIONS / Time: 0h:0m:7s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -1387.259
Optimal log-likelihood: -1387.254
Rate parameters:  A-C: 0.33716  A-G: 2.26442  A-T: 2.13773  C-G: 1.17713  C-T: 3.28833  G-T: 1.00000
Base frequencies:  A: 0.243  C: 0.182  G: 0.319  T: 0.256
Proportion of invariable sites: 0.000
Gamma shape alpha: 1.312
Parameters optimization took 1 rounds (0.006 sec)
BEST SCORE FOUND : -1387.254

Testing tree branches by SH-like aLRT with 1000 replicates...
Testing tree branches by local-BP test with 1000 replicates...
Testing tree branches by aBayes parametric test...
0.061 sec.
Creating bootstrap support values...
Split supports printed to NEXUS file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.splits.nex
Total tree length: 6.763

Total number of iterations: 230
CPU time used for tree search: 7.017 sec (0h:0m:7s)
Wall-clock time used for tree search: 7.043 sec (0h:0m:7s)
Total CPU time used: 7.661 sec (0h:0m:7s)
Total wall-clock time used: 7.692 sec (0h:0m:7s)

Computing bootstrap consensus tree...
Reading input file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.splits.nex...
20 taxa and 151 splits.
Consensus tree written to /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.contree
Reading input trees file /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.contree
Log-likelihood of consensus tree: -1387.254

Analysis results written to: 
  IQ-TREE report:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.iqtree
  Maximum-likelihood tree:       /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.treefile
  Likelihood distances:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.splits.nex
  Consensus tree:                /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.contree
  Screen log file:               /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot.log

Date and Time: Fri Feb 28 11:28:32 2020
Running external command line application. This may print messages to stdout and/or stderr.
The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.

Command: iqtree -bb 1000 -st DNA --runs 1 -s /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/qiime2-archive-jzcm2io3/c67b6543-8e61-435d-b61e-36a863284e6e/data/aligned-dna-sequences.fasta -m GTR+I+G -pre /var/folders/9h/268zfwl56h37jwt5qv866jcr0000gp/T/tmpsv18dn9z/q2iqtreeufboot -nt 1 -alrt 1000 -abayes -lbp 1000 -nstop 200 -pers 0.200000

Saved Phylogeny[Unrooted] to: iqt-nnisi-bootstrap-sbt-gtrig-tree.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-sbt-gtrig-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If there is a need to reduce the impact of <a class="reference external" href="http://www.iqtree.org/doc/Tutorial#reducing-impact-of-severe-model-violations-with-ufboot">potential model
violations</a> that occur during a <a class="reference external" href="https://doi.org/10.1093/molbev/msx281">UFBoot search</a>, and / or would simply
like to be more rigorous, we can add the <code class="docutils literal notranslate"><span class="pre">--p-bnni</span></code> option to any of the
<code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code> commands above.</p>
</div>
</div>
</div>
<div class="section" id="root-the-phylogeny">
<h3><a class="toc-backref" href="#id19">Root the phylogeny</a><a class="headerlink" href="#root-the-phylogeny" title="Permalink to this headline">¬∂</a></h3>
<p>In order to make proper use of diversity metrics such as UniFrac, the phylogeny
must be <a class="reference external" href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics/what-phylogeny/aspects-phylogenies/nodes/root">rooted</a>. Typically an <a class="reference external" href="http://phylobotanist.blogspot.com/2015/01/how-to-root-phylogenetic-tree-outgroup.html">outgroup</a> is chosen when rooting a tree. In
general, phylogenetic inference tools using Maximum Likelihood often return an
unrooted tree by default.</p>
<p>QIIME 2 provides a way to
<a class="reference internal" href="../../plugins/available/phylogeny/midpoint-root/"><span class="doc">mid-point root</span></a> our
phylogeny. Other rooting options may be available in the future. For now, we‚Äôll
root our bootstrap tree from <code class="docutils literal notranslate"><span class="pre">iqtree-ultrafast-bootstrap</span></code> like so:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny midpoint-root <span class="se">\</span>
  --i-tree iqt-nnisi-bootstrap-sbt-gtrig-tree.qza <span class="se">\</span>
  --o-rooted-tree iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fiqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza">download</a></p></li>
</ul>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p><strong>iTOL viewing Reminder</strong>. We can view our tree and its associated
alignment via <a class="reference external" href="https://itol.embl.de/">iTOL</a>. All you need to do is upload the
<cite>iqt-nnisi-bootstrap-sbt-gtrig-tree-rooted.qza</cite> tree file. Display the tree
in <cite>Normal</cite> mode. Then drag and drop the <cite>masked-aligned-rep-seqs.qza</cite> file
onto the visualization. Now you can view the phylogeny alongside the
alignment.</p>
</div>
</div>
</div>
<div class="section" id="pipelines">
<h2><a class="toc-backref" href="#id20">Pipelines</a><a class="headerlink" href="#pipelines" title="Permalink to this headline">¬∂</a></h2>
<p>Here we will outline the use of the phylogeny pipeline
<a class="reference internal" href="../../plugins/available/phylogeny/align-to-tree-mafft-fasttree/"><span class="doc">align-to-tree-mafft-fasttree</span></a></p>
<p>One advantage of pipelines is that they combine ordered sets of commonly used
commands, into one condensed simple command. To keep these ‚Äúconvenience‚Äù
pipelines easy to use, it is quite common to only expose a few options to the
user. That is, most of the commands executed via pipelines are often configured
to use default option settings. However, options that are deemed important
enough for the user to consider setting, are made available. The options
exposed via a given pipeline will largely depend upon what it is doing.
Pipelines are also a great way for new users to get started, as it helps to lay
a foundation of good practices in setting up standard operating procedures.</p>
<p>Rather than run one or more of the following QIIME 2 commands listed below:</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">alignment</span> <span class="pre">mafft</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">alignment</span> <span class="pre">mask</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">phylogeny</span> <span class="pre">fasttree</span> <span class="pre">...</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">phylogeny</span> <span class="pre">midpoint-root</span> <span class="pre">...</span></code></p></li>
</ol>
<p>We can make use of the pipeline
<a class="reference internal" href="../../plugins/available/phylogeny/align-to-tree-mafft-fasttree/"><span class="doc">align-to-tree-mafft-fasttree</span></a>
to automate the above four steps in one go. Here is the description taken from
the pipeline help doc:</p>
<blockquote>
<div><p>This pipeline will start by creating a sequence alignment using MAFFT,
after which any alignment columns that are phylogenetically uninformative
or ambiguously aligned will be removed (masked). The resulting masked
alignment will be used to infer a phylogenetic tree and then subsequently
rooted at its midpoint. Output files from each step of the pipeline will be
saved. This includes both the unmasked and masked MAFFT alignment from
q2-alignment methods, and both the rooted and unrooted phylogenies from
q2-phylogeny methods.</p>
</div></blockquote>
<p>This can all be accomplished by simply running the following:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime phylogeny align-to-tree-mafft-fasttree <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --output-dir mafft-fasttree-output
</pre></div>
</div>
<div class="admonition">
<p><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/masked_alignment.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Fmasked_alignment.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/masked_alignment.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/rooted_tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Frooted_tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/rooted_tree.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/alignment.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Falignment.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/alignment.qza">download</a></p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">mafft-fasttree-output/tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2020.2%2Fdata%2Ftutorials%2Fphylogeny%2Fmafft-fasttree-output%2Ftree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2020.2/data/tutorials/phylogeny/mafft-fasttree-output/tree.qza">download</a></p></li>
</ul>
</div>
<p><strong>Congratulations! You now know how to construct a phylogeny in QIIME 2!</strong></p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="../qiime2-for-experienced-microbiome-researchers/">QIIME 2 for Experienced Microbiome Researchers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">‚ÄúMoving Pictures‚Äù tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">‚ÄúAtacama soil microbiome‚Äù tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pd-mice/">Parkinson‚Äôs Mouse Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../otu-clustering/">Clustering sequences into OTUs using q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../utilities/">Utilities in QIIME 2</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Phylogenetic inference with q2-phylogeny</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">User Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search/" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2020, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.4.3.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>