<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Training feature classifiers with q2-feature-classifier &#8212; QIIME 2 2018.8.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2018.8.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2018.8.0 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Evaluating and controlling data quality with q2-quality-control" href="../quality-control/" />
    <link rel="prev" title="Filtering data" href="../filtering/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="training-feature-classifiers-with-q2-feature-classifier">
<h1>Training feature classifiers with q2-feature-classifier<a class="headerlink" href="#training-feature-classifiers-with-q2-feature-classifier" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a>.</p>
</div>
<p>This tutorial will demonstrate how to train <code class="docutils literal"><span class="pre">q2-feature-classifier</span></code> for a particular dataset. We will train the <a class="reference external" href="http://scikit-learn.org/stable/modules/naive_bayes.html#multinomial-naive-bayes">Naive Bayes</a> classifier using <a class="reference external" href="http://qiime.org/home_static/dataFiles.html">Greengenes</a> reference sequences and classify the representative sequences from the <a class="reference external" href="../moving-pictures/index.html">Moving Pictures</a> dataset.</p>
<p>Note that several pre-trained classifiers are provided in the QIIME 2 <a class="reference internal" href="../../data-resources/"><span class="doc">data resources</span></a>. These can be used for some common marker-gene targets (e.g., 16S rRNA genes). Pre-trained classifiers for other marker genes are also available on the <a class="reference external" href="https://forum.qiime2.org/c/community-contributions/data-resources">QIIME 2 forum</a>.</p>
<p>We will download and create several files, so first create a working directory.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir training-feature-classifiers
<span class="nb">cd</span> training-feature-classifiers
</pre></div>
</div>
<div class="section" id="obtaining-and-importing-reference-data-sets">
<h2>Obtaining and importing reference data sets<a class="headerlink" href="#obtaining-and-importing-reference-data-sets" title="Permalink to this headline">¶</a></h2>
<p>Two elements are required for training the classifier: the reference sequences and the corresponding taxonomic classifications. To reduce computation time for this tutorial we will use the relatively small <a class="reference external" href="http://qiime.org/home_static/dataFiles.html">Greengenes</a> 13_8 85% OTU data set. <strong>Do not use the 85% OTU data set used in this tutorial for classification of real experimental data</strong>. We recommend using more information-rich sequences, e.g., reference sequences clustered at 99% sequence similarity, for classification of real data. See the QIIME 2 <a class="reference internal" href="../../data-resources/"><span class="doc">data resources page</span></a> for links to complete QIIME-compatible reference datasets.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">All sequence IDs present in the reference sequences must also be present in the reference taxonomy. If using reference sequence databases that have been clustered into OTUs, make sure that the corresponding reference taxonomy is used. For example, if using the Greengenes 99% OTU sequences, the 99% OTU taxonomy must also be used.</p>
</div>
<p>We will also download the representative sequences from the <a class="reference external" href="../moving-pictures/index.html">Moving Pictures</a> tutorial to test our classifier.</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otus.fasta">
            https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otus.fasta
          </a>
        </p>
        <p>
          <strong>Save as:</strong> 85_otus.fasta
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "85_otus.fasta" \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otus.fasta"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otus.fasta" > \
  "85_otus.fasta"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otu_taxonomy.txt">
            https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otu_taxonomy.txt
          </a>
        </p>
        <p>
          <strong>Save as:</strong> 85_otu_taxonomy.txt
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "85_otu_taxonomy.txt" \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otu_taxonomy.txt"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/85_otu_taxonomy.txt" > \
  "85_otu_taxonomy.txt"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-2" aria-controls="browser-2" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-2" aria-controls="wget-2" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-2" aria-controls="messages-2" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-2">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/rep-seqs.qza">
            https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/rep-seqs.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> rep-seqs.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-2">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "rep-seqs.qza" \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-2">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2018.8/tutorials/training-feature-classifiers/rep-seqs.qza" > \
  "rep-seqs.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>Next we import these data into QIIME 2 Artifacts. Since the Greengenes reference taxonomy file (<code class="file docutils literal"><span class="pre">85_otu_taxonomy.txt</span></code>) is a tab-separated (TSV) file without a header, we must specify <code class="docutils literal"><span class="pre">HeaderlessTSVTaxonomyFormat</span></code> as the <em>source format</em> since the default <em>source format</em> requires a header.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --type <span class="s1">&#39;FeatureData[Sequence]&#39;</span> <span class="se">\</span>
  --input-path 85_otus.fasta <span class="se">\</span>
  --output-path 85_otus.qza

qiime tools import <span class="se">\</span>
  --type <span class="s1">&#39;FeatureData[Taxonomy]&#39;</span> <span class="se">\</span>
  --input-format HeaderlessTSVTaxonomyFormat <span class="se">\</span>
  --input-path 85_otu_taxonomy.txt <span class="se">\</span>
  --output-path ref-taxonomy.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">85_otus.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2F85_otus.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/85_otus.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/rep-seqs.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">ref-taxonomy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Fref-taxonomy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/ref-taxonomy.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="extract-reference-reads">
<h2>Extract reference reads<a class="headerlink" href="#extract-reference-reads" title="Permalink to this headline">¶</a></h2>
<p>It has been shown that taxonomic classification accuracy of 16S rRNA gene sequences improves when a Naive Bayes classifier is trained on only the region of the target sequences that was sequenced <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/21716311">(Werner et al., 2012)</a>. This may not necessarily generalize to other marker genes (see note on fungal ITS classification below). We know from the <a class="reference external" href="../moving-pictures/index.html">Moving Pictures</a> tutorial that the sequence reads that we&#8217;re trying to classify are 120-base single-end reads that were amplified with the 515F/806R primer pair for 16S rRNA gene sequences. We optimize for that here by extracting reads from the reference database based on matches to this primer pair, and then slicing the result to 120 bases.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-classifier extract-reads <span class="se">\</span>
  --i-sequences 85_otus.qza <span class="se">\</span>
  --p-f-primer GTGCCAGCMGCCGCGGTAA <span class="se">\</span>
  --p-r-primer GGACTACHVGGGTWTCTAAT <span class="se">\</span>
  --p-trunc-len <span class="m">120</span> <span class="se">\</span>
  --o-reads ref-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">ref-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Fref-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/ref-seqs.qza">download</a></li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <code class="docutils literal"><span class="pre">--p-trunc-len</span></code> parameter should only be used to trim reference sequences if query sequences are trimmed to this same length or shorter. Paired-end sequences that successfully join will typically be variable in length. Single-end reads that are not truncated at a specific length may also be variable in length. For classification of paired-end reads and untrimmed single-end reads, we recommend training a classifier on sequences that have been extracted at the appropriate primer sites, but are not trimmed.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The primer sequences used for extracting reads should be the actual DNA-binding (i.e., biological) sequence contained within a primer construct. It should NOT contain any non-biological, non-binding sequence, e.g., adapter, linker, or barcode sequences. If you are not sure what section of your primer sequences are actual DNA-binding, you should consult whoever constructed your sequencing library, your sequencing center, or the original source literature on these primers. If your primer sequences are &gt; 30 nt long, they most likely contain some non-biological sequence.</p>
</div>
</div>
<div class="section" id="train-the-classifier">
<h2>Train the classifier<a class="headerlink" href="#train-the-classifier" title="Permalink to this headline">¶</a></h2>
<p>We can now train a <a class="reference external" href="http://scikit-learn.org/stable/modules/naive_bayes.html#multinomial-naive-bayes">Naive Bayes</a> classifier as follows, using the reference reads and taxonomy that we just created.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-classifier fit-classifier-naive-bayes <span class="se">\</span>
  --i-reference-reads ref-seqs.qza <span class="se">\</span>
  --i-reference-taxonomy ref-taxonomy.qza <span class="se">\</span>
  --o-classifier classifier.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">classifier.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Fclassifier.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/classifier.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="test-the-classifier">
<h2>Test the classifier<a class="headerlink" href="#test-the-classifier" title="Permalink to this headline">¶</a></h2>
<p>Finally, we verify that the classifier works by classifying the representative sequences from the <a class="reference external" href="../moving-pictures/index.html">Moving Pictures</a> tutorial and visualizing the resulting taxonomic assignments.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-classifier classify-sklearn <span class="se">\</span>
  --i-classifier classifier.qza <span class="se">\</span>
  --i-reads rep-seqs.qza <span class="se">\</span>
  --o-classification taxonomy.qza

qiime metadata tabulate <span class="se">\</span>
  --m-input-file taxonomy.qza <span class="se">\</span>
  --o-visualization taxonomy.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">taxonomy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Ftaxonomy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/taxonomy.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">taxonomy.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Ffeature-classifier%2Ftaxonomy.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/feature-classifier/taxonomy.qzv">download</a></li>
</ul>
</div>
</div>
<div class="section" id="classification-of-fungal-its-sequences">
<h2>Classification of fungal ITS sequences<a class="headerlink" href="#classification-of-fungal-its-sequences" title="Permalink to this headline">¶</a></h2>
<p>In our experience, fungal ITS classifiers trained on the <a class="reference external" href="https://unite.ut.ee/repository.php">UNITE reference database</a> do NOT benefit from extracting/trimming reads to primer sites. We recommend training UNITE classifiers on the full reference sequences. Furthermore, we recommend the &#8220;developer&#8221; sequences (located within the QIIME-compatible release download) because the standard versions of the sequences have already been trimmed to the ITS region (excluding portions of flanking rRNA genes that may be present in amplicons generated with standard ITS primers).</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">&#8220;Atacama soil microbiome&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../otu-clustering/">Clustering sequences into OTUs using q2-vsearch</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2018, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>