<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>raxml-rapid-bootstrap: Construct a phylogenetic tree with bootstrap supports using RAxML. &#8212; QIIME 2 2018.8.0 documentation</title>
    
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../../',
        VERSION:     '2018.8.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="top" title="QIIME 2 2018.8.0 documentation" href="../../../../" />
    <link rel="up" title="phylogeny" href="../" />
    <link rel="next" title="quality-control" href="../../quality-control/" />
    <link rel="prev" title="raxml: Construct a phylogenetic tree with RAxML." href="../raxml/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="raxml-rapid-bootstrap-construct-a-phylogenetic-tree-with-bootstrap-supports-using-raxml">
<h1>raxml-rapid-bootstrap: Construct a phylogenetic tree with bootstrap supports using RAxML.<a class="headerlink" href="#raxml-rapid-bootstrap-construct-a-phylogenetic-tree-with-bootstrap-supports-using-raxml" title="Permalink to this headline">¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><p id="bibtex-bibliography-plugins/available/phylogeny/raxml-rapid-bootstrap-0"><table class="docutils citation" frame="void" id="phylogeny-raxml-rapid-bootstrap-key0" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:raxml-rapid-bootstrap:Sta14]</td><td>Alexandros Stamatakis. Raxml version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. <em>Bioinformatics</em>, 30(9):1312–1313, 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btu033">http://dx.doi.org/10.1093/bioinformatics/btu033</a>, <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu033">doi:10.1093/bioinformatics/btu033</a>.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="phylogeny-raxml-rapid-bootstrap-key1" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[phylogeny:raxml-rapid-bootstrap:SHR08]</td><td>Alexandros Stamatakis, Paul Hoover, and Jacques Rougemont. A rapid bootstrap algorithm for the raxml web servers. <em>Systematic Biology</em>, 57(5):758–771, 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1080/10635150802429642">http://dx.doi.org/10.1080/10635150802429642</a>, <a class="reference external" href="https://doi.org/10.1080/10635150802429642">doi:10.1080/10635150802429642</a>.</td></tr>
</tbody>
</table>
</p>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime phylogeny raxml-rapid-bootstrap [OPTIONS]

  Construct a phylogenetic tree with RAxML with the addition of rapid
  bootstrapping support values. See:
  https://sco.h-its.org/exelixis/web/software/raxml/

Options:
  --i-alignment ARTIFACT PATH FeatureData[AlignedSequence]
                                  Aligned sequences to be used for
                                  phylogenetic reconstruction.  [required]
  --p-seed INTEGER                Random number seed for the parsimony
                                  starting tree. This allows you to reproduce
                                  tree results. If not supplied then one will
                                  be randomly chosen.  [optional]
  --p-rapid-bootstrap-seed INTEGER
                                  Specify a random seed for
                                  rapid bootstrapping. This allows you to
                                  reproduce rapid bootstrap results. If not
                                  supplied then one will be randomly chosen.
                                  [optional]
  --p-bootstrap-replicates INTEGER RANGE
                                  The number of bootstrap searches to perform.
                                  [default: 100]
  --p-n-threads INTEGER RANGE     The number of threads to use for
                                  multithreaded processing. Using more than
                                  one thread will enable the PTHREADS version
                                  of RAxML.  [default: 1]
  --p-raxml-version [SSE3|AVX2|Standard]
                                  Select a specific CPU optimization of RAxML
                                  to use. The SSE3 versions will run
                                  approximately 40% faster than the standard
                                  version. The AVX2 version will run 10-30%
                                  faster than the SSE3 version.  [default:
                                  Standard]
  --p-substitution-model [GTRGAMMA|GTRCATI|GTRCAT|GTRGAMMAI]
                                  Model of Nucleotide Substitution  [default:
                                  GTRGAMMA]
  --o-tree ARTIFACT PATH Phylogeny[Unrooted]
                                  The resulting phylogenetic tree.  [required
                                  if not passing --output-dir]
  --output-dir DIRECTORY          Output unspecified results to a directory
  --cmd-config PATH               Use config file for command options
  --verbose                       Display verbose output to stdout and/or
                                  stderr during execution of this action.
                                  [default: False]
  --quiet                         Silence output if execution is successful
                                  (silence is golden).  [default: False]
  --citations                     Show citations and exit.
  --help                          Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell"><div class="highlight"><pre><span></span>from qiime2.plugins.phylogeny.methods import raxml_rapid_bootstrap
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Construct a phylogenetic tree with bootstrap supports using RAxML.

Construct a phylogenetic tree with RAxML with the addition of rapid
bootstrapping support values. See:
https://sco.h-its.org/exelixis/web/software/raxml/

Parameters
----------
alignment : FeatureData[AlignedSequence]
    Aligned sequences to be used for phylogenetic reconstruction.
seed : Int, optional
    Random number seed for the parsimony starting tree. This allows you to
    reproduce tree results. If not supplied then one will be randomly
    chosen.
rapid_bootstrap_seed : Int, optional
    Specify a random seed for rapid bootstrapping. This allows you to
    reproduce rapid bootstrap results. If not supplied then one will be
    randomly chosen.
bootstrap_replicates : Int % Range(10, None), optional
    The number of bootstrap searches to perform.
n_threads : Int % Range(1, None), optional
    The number of threads to use for multithreaded processing. Using more
    than one thread will enable the PTHREADS version of RAxML.
raxml_version : Str % Choices({'AVX2', 'SSE3', 'Standard'}), optional
    Select a specific CPU optimization of RAxML to use. The SSE3 versions
    will run approximately 40% faster than the standard version. The AVX2
    version will run 10-30% faster than the SSE3 version.
substitution_model : Str % Choices({'GTRCAT', 'GTRCATI', 'GTRGAMMA', 'GTRGAMMAI'}), optional
    Model of Nucleotide Substitution

Returns
-------
tree : Phylogeny[Unrooted]
    The resulting phylogenetic tree.</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2018, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>