<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Predicting sample metadata values with q2-sample-classifier &#8212; QIIME 2 2017.9.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2017.9.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2017.9.0 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Performing longitudinal and paired sample comparisons with q2-longitudinal" href="../longitudinal/" />
    <link rel="prev" title="Training feature classifiers with q2-feature-classifier" href="../feature-classifier/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="predicting-sample-metadata-values-with-q2-sample-classifier">
<h1>Predicting sample metadata values with q2-sample-classifier<a class="headerlink" href="#predicting-sample-metadata-values-with-q2-sample-classifier" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a> and completed the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial</span></a>.</p>
</div>
<p>This tutorial will demonstrate how to use <code class="docutils literal"><span class="pre">q2-sample-classifier</span></code> to predict sample metadata values. Supervised learning methods predict sample data (e.g., metadata values) as a function of other sample data (e.g., microbiota composition). The predicted targets may be discrete sample classes (for classification problems) or continuous values (for regression problems). Any other data may be used as predictive features, but for the purposes of q2-sample-classifier this will most commonly be microbial sequence variant, operational taxonomic unit (OTU), or taxonomic composition data. However, any features contained in a feature table may be used —&nbsp;for non-microbial data, just <a class="reference external" href="http://biom-format.org/documentation/biom_conversion.html">convert your observation tables to biom format</a> and <a class="reference internal" href="../importing/"><span class="doc">import the feature table data into qiime2</span></a>.</p>
<p>We will download and create several files, so first create a working directory.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir sample-classifier-tutorial
<span class="nb">cd</span> sample-classifier-tutorial
</pre></div>
</div>
<div class="section" id="predicting-categorical-sample-data">
<h2>Predicting categorical sample data<a class="headerlink" href="#predicting-categorical-sample-data" title="Permalink to this headline">¶</a></h2>
<p>Supervised learning classifiers predict the categorical metadata classes of unlabeled samples by learning the composition of labeled training samples. For example, we may use a classifier to diagnose or predict disease susceptibility based on stool microbiome composition, or predict sample type as a function of the sequence variants, microbial taxa, or metabolites detected in a sample. In this tutorial, we will use the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial data</span></a> to train a classifier that predicts the body site from which a sample was collected. Download the feature table and sample metadata with the following links:</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.9/tutorials/moving-pictures/sample_metadata.tsv">
            https://data.qiime2.org/2017.9/tutorials/moving-pictures/sample_metadata.tsv
          </a>
        </p>
        <p>
          <strong>Save as:</strong> moving-pictures-sample-metadata.tsv
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "moving-pictures-sample-metadata.tsv" "https://data.qiime2.org/2017.9/tutorials/moving-pictures/sample_metadata.tsv"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.9/tutorials/moving-pictures/sample_metadata.tsv" > "moving-pictures-sample-metadata.tsv"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.9/tutorials/sample-classifier/moving-pictures-table.qza">
            https://data.qiime2.org/2017.9/tutorials/sample-classifier/moving-pictures-table.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> moving-pictures-table.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "moving-pictures-table.qza" "https://data.qiime2.org/2017.9/tutorials/sample-classifier/moving-pictures-table.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.9/tutorials/sample-classifier/moving-pictures-table.qza" > "moving-pictures-table.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>Next, we will attempt to predict which body site each sample originated from based on its microbial composition.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime sample-classifier classify-samples <span class="se">\</span>
  --i-table moving-pictures-table.qza <span class="se">\</span>
  --m-metadata-file moving-pictures-sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category BodySite <span class="se">\</span>
  --p-optimize-feature-selection <span class="se">\</span>
  --p-parameter-tuning <span class="se">\</span>
  --p-estimator RandomForestClassifier <span class="se">\</span>
  --p-n-estimators <span class="m">100</span> <span class="se">\</span>
  --o-visualization moving-pictures-BodySite.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">moving-pictures-table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.9%2Fdata%2Ftutorials%2Fsample-classifier%2Fmoving-pictures-table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.9/data/tutorials/sample-classifier/moving-pictures-table.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">moving-pictures-BodySite.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.9%2Fdata%2Ftutorials%2Fsample-classifier%2Fmoving-pictures-BodySite.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.9/data/tutorials/sample-classifier/moving-pictures-BodySite.qzv">download</a></li>
</ul>
</div>
<p>The visualization produced by this command presents classification accuracy results in the form of a confusion matrix. This matrix indicates how frequently a sample is classified with to the correct class vs. all other classes. The confusion matrix is displayed at the top of the visualization in the form of a heatmap, and below that as a table containing overall accuracy (the fraction of times that test samples are assigned the correct class).</p>
<div class="question admonition" id="question-0">
<p class="first admonition-title">Question</p>
<p class="last">What other metadata can we predict with <code class="docutils literal"><span class="pre">classify-samples</span></code>? Take a look at the metadata categories in the <code class="docutils literal"><span class="pre">sample-metadata</span></code> and try some other categories. Not all metadata can be easily learned by the classifier!</p>
</div>
<p>If <code class="docutils literal"><span class="pre">--p-optimize-feature-selection</span></code> is enabled, the visualization will also display a recursive feature extraction plot, which illustrates how model accuracy changes as a function of feature count. The combination of features that maximize accuracy are automatically selected for the final model, which is used for sample prediction results that are displayed in the visualization. A list of the features chosen, and their relative importances, will be displayed at the bottom of the visualization. Features with higher importance scores are more important for distinguishing each class.</p>
<div class="question admonition" id="question-1">
<p class="first admonition-title">Question</p>
<p class="last">What happens when feature optimization is disabled with the option <code class="docutils literal"><span class="pre">--p-no-optimize-feature-selection</span></code>? How does this impact classification accuracy?</p>
</div>
<p>K-fold cross-validation is performed during automatic feature selection and parameter optimization steps. Five-fold cross-validation is performed by default, and this value can be adjusted using the <code class="docutils literal"><span class="pre">--p-cv</span></code> parameter. A separate portion of samples is removed from the data set prior to model training and optimization, and used as a test set to determine model accuracy. The fraction of test samples to remove is adjusted with the <code class="docutils literal"><span class="pre">--p-test-size</span></code> parameter.</p>
<div class="question admonition" id="question-2">
<p class="first admonition-title">Question</p>
<p class="last">Try to figure out what the <code class="docutils literal"><span class="pre">--p-parameter-tuning</span></code> parameter does. What happens when it is disabled with the option <code class="docutils literal"><span class="pre">--p-no-parameter-tuning</span></code>? How does this impact classification accuracy?</p>
</div>
<div class="question admonition" id="question-3">
<p class="first admonition-title">Question</p>
<p class="last">Many different classifiers can be trained via the <code class="docutils literal"><span class="pre">--p-estimator</span></code> parameter in <code class="docutils literal"><span class="pre">classify-samples</span></code>. Try some of the other classifiers. How do these methods compare?</p>
</div>
<div class="question admonition" id="question-4">
<p class="first admonition-title">Question</p>
<p class="last">Sequence variants are not the only feature data that can be used to train a classifier or regressor. Taxonomic composition is another feature type that can be easily created using the tutorial data provided in QIIME2. Try to figure out how this works (hint: you will need to assign taxonomy, as described in the <a class="reference internal" href="../moving-pictures/"><span class="doc">moving pictures tutorial</span></a>, and <a class="reference internal" href="../../plugins/available/taxa/collapse/"><span class="doc">collapse taxonomy</span></a> to create a new feature table). Try using feature tables collapsed to different taxonomic levels. How does taxonomic specificity (e.g., species-level is more specific than phylum-level) impact classifier performance?</p>
</div>
<div class="question admonition" id="question-5">
<p class="first admonition-title">Question</p>
<p class="last">The <code class="docutils literal"><span class="pre">--p-n-estimators</span></code> parameter adjusts the number of trees grown by ensemble estimators, such as random forest classifiers (this parameter will have no effect on non-ensemble methods), which increases classifier accuracy up to a certain point, but at the cost of increased computation time. Try the same command above with different numbers of estimators, e.g., 10, 50, 100, 250, and 500 estimators. How does this impact the overall accuracy of predictions? Are more trees worth the time?</p>
</div>
</div>
<div class="section" id="predicting-continuous-i-e-numerical-sample-data">
<h2>Predicting continuous (i.e., numerical) sample data<a class="headerlink" href="#predicting-continuous-i-e-numerical-sample-data" title="Permalink to this headline">¶</a></h2>
<p>Supervised learning regressors predict continuous metadata values of unlabeled samples by learning the composition of labeled training samples. For example, we may use a regressor to predict the abundance of a metabolite that will be producted by a microbial community, or a sample&#8217;s pH,  temperature, or altitude as a function of the sequence variants, microbial taxa, or metabolites detected in a sample. In this tutorial, we will use the <a class="reference external" href="https://doi.org/10.1126/scitranslmed.aad7121">ECAM study</a>, a longitudinal cohort study of microbiome development in U.S. infants. Download the feature table and sample metadata with the following links:</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-2" aria-controls="browser-2" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-2" aria-controls="wget-2" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-2" aria-controls="messages-2" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-2">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.9/tutorials/longitudinal/sample_metadata.tsv">
            https://data.qiime2.org/2017.9/tutorials/longitudinal/sample_metadata.tsv
          </a>
        </p>
        <p>
          <strong>Save as:</strong> ecam-metadata.tsv
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-2">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "ecam-metadata.tsv" "https://data.qiime2.org/2017.9/tutorials/longitudinal/sample_metadata.tsv"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-2">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.9/tutorials/longitudinal/sample_metadata.tsv" > "ecam-metadata.tsv"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-3" aria-controls="browser-3" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-3" aria-controls="wget-3" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-3" aria-controls="messages-3" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-3">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.9/tutorials/longitudinal/ecam_table_maturity.qza">
            https://data.qiime2.org/2017.9/tutorials/longitudinal/ecam_table_maturity.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> ecam-table.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-3">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "ecam-table.qza" "https://data.qiime2.org/2017.9/tutorials/longitudinal/ecam_table_maturity.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-3">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.9/tutorials/longitudinal/ecam_table_maturity.qza" > "ecam-table.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>Next, we will train a regressor to predict an infant&#8217;s age based on its microbiota composition.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime sample-classifier regress-samples <span class="se">\</span>
  --i-table ecam-table.qza <span class="se">\</span>
  --m-metadata-file ecam-metadata.tsv <span class="se">\</span>
  --m-metadata-category month <span class="se">\</span>
  --p-optimize-feature-selection <span class="se">\</span>
  --p-parameter-tuning <span class="se">\</span>
  --p-estimator RandomForestRegressor <span class="se">\</span>
  --p-n-estimators <span class="m">100</span> <span class="se">\</span>
  --o-visualization ecam-month.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">ecam-table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.9%2Fdata%2Ftutorials%2Fsample-classifier%2Fecam-table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.9/data/tutorials/sample-classifier/ecam-table.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">ecam-month.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.9%2Fdata%2Ftutorials%2Fsample-classifier%2Fecam-month.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.9/data/tutorials/sample-classifier/ecam-month.qzv">download</a></li>
</ul>
</div>
<p>The visualization produced by this command presents classification accuracy results in the form of a scatter plot showing predicted vs. true values for each test sample, accompanied by a linear regression line fitted to the data with 95% confidence intervals (grey shading). The true 1:1 ratio between predicted and true values is represented by a dotted line for comparison. Below this, model accuracy is quantified in a table displaying mean square error and the R value, P value, standard error of the estimated gradient, slope, and intercept of the linear regression fit. The remainder of the visualization shows optional feature selection data, as described above for <code class="docutils literal"><span class="pre">classify-samples</span></code>.</p>
<div class="question admonition" id="question-6">
<p class="first admonition-title">Question</p>
<p class="last">What other metadata can we predict with <code class="docutils literal"><span class="pre">regress-samples</span></code>? Take a look at the metadata categories in the <code class="docutils literal"><span class="pre">sample-metadata</span></code> and try some other values. Not all metadata can be easily learned by the regressor!</p>
</div>
<div class="question admonition" id="question-7">
<p class="first admonition-title">Question</p>
<p class="last">Many different regressors can be trained via the <code class="docutils literal"><span class="pre">--p-estimator</span></code> parameter in <code class="docutils literal"><span class="pre">regress-samples</span></code>. Try some of the other regressors. How do these methods compare?</p>
</div>
</div>
<div class="section" id="maturity-index-prediction">
<h2>&#8220;Maturity Index&#8221; prediction<a class="headerlink" href="#maturity-index-prediction" title="Permalink to this headline">¶</a></h2>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This analysis currently works best for comparing groups that are sampled fairly evenly across time (the category used for regression). Datasets that contain groups sampled sporadically at different times are not supported, and users should either filter out those samples or “bin” them with other groups prior to using this visualizer.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This analysis will only work on data sets with a large sample size, particularly in the &#8220;control&#8221; group, and with sufficient biological replication at each time point.</p>
</div>
<p>This method calculates a &#8220;microbial maturity&#8221; index from a regression model trained on feature data to predict a given continuous metadata category, e.g., to predict a subject&#8217;s age as a function of microbiota composition. This method is different from standard supervised regression because it quantifies the relative rate of change over time in two or more groups. The model is trained on a subset of control group samples, then predicts the category value for all samples. This visualization computes maturity index z-scores (MAZ) to compare relative &#8220;maturity&#8221; between each group, as described in <a class="reference external" href="https://doi.org/10.1038/nature13421">Sathish et al. 2014</a>. This method was designed to predict between-group differences in intestinal microbiome development by age, so <code class="docutils literal"><span class="pre">category</span></code> should typically be a measure of time. Other types of continuous metadata gradients might be testable, as long as two or more different &#8220;treatment&#8221; groups are being compared <em>with a large number of biological replicates</em> in the &#8220;control&#8221; group and treatment groups are sampled at the same &#8220;states&#8221; (time or position on gradient) for comparison. However, we do not necessarily recommend <em>or offer technical support</em> for unusual approaches.</p>
<p>Here we will compare microbial maturity between vaginally born and cesarean-delivered infants as a function of age in the ECAM dataset.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime sample-classifier maturity-index <span class="se">\</span>
  --i-table ecam-table.qza <span class="se">\</span>
  --m-metadata-file ecam-metadata.tsv <span class="se">\</span>
  --p-category month <span class="se">\</span>
  --p-group-by delivery <span class="se">\</span>
  --p-control Vaginal <span class="se">\</span>
  --p-test-size <span class="m">0</span>.4 <span class="se">\</span>
  --o-visualization maturity.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">maturity.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.9%2Fdata%2Ftutorials%2Fsample-classifier%2Fmaturity.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.9/data/tutorials/sample-classifier/maturity.qzv">download</a></li>
</ul>
</div>
<p>The visualizer produces a linear regression plot of predicted vs. expected values on the control test samples (as described above for regression models). Predicted vs. expected values are also shown for all samples in both control and test sets.</p>
<p>MAZ scores are calculated based on these predictions, statistically compared across each value &#8220;bin&#8221; (e.g., month of life) using ANOVA and paired t-tests, and shown as boxplots of MAZ distributions for each group in each value &#8220;bin&#8221;. A link within the visualizers allows download of the MAZ scores for each sample, facilitating customized follow-up testing, e.g., in R, or use as metadata, e.g., for constructing PCoA plots. Want to take this analysis to the next level? Download the raw MAZ scores from within the visualization and feed these scores into <a class="reference internal" href="../longitudinal/"><span class="doc">linear mixed effects models</span></a></p>
<p>The average abundances of features used for training maturity models are viewed as heatmaps within the visualization. Feature abundance is averaged across all samples within each value bin (e.g., month of life) and within each individual sample group (e.g., vaginal controls vs. cesarean), demonstrating how different patterns of feature abundance (e.g., trajectories of development in the case of age or time-based models) may affect model predictions and MAZ scores.</p>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">&#8220;Atacama soil microbiome&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2017, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>