<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>sepp: Insert fragment sequences using SEPP into reference phylogenies like Greengenes 13_8 &#8212; QIIME 2 2019.4.0 documentation</title>
    <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../../../_static/language_data.js"></script>
    <script type="text/javascript" src="../../../../_static/external-links.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../../../about/" />
    <link rel="index" title="Index" href="../../../../genindex/" />
    <link rel="search" title="Search" href="../../../../search/" />
    <link rel="next" title="gneiss" href="../../gneiss/" />
    <link rel="prev" title="filter-features: Filter fragments in tree from table." href="../filter-features/" />
  <script src="../../../../_static/bootstrap.min.js"></script>
  <script src="../../../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head><body>
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="sepp-insert-fragment-sequences-using-sepp-into-reference-phylogenies-like-greengenes-13-8">
<h1>sepp: Insert fragment sequences using SEPP into reference phylogenies like Greengenes 13_8<a class="headerlink" href="#sepp-insert-fragment-sequences-using-sepp-into-reference-phylogenies-like-greengenes-13-8" title="Permalink to this headline">¶</a></h1>
<table class="table action-info">
  <tbody>
    <tr>
      <th scope="row"><p></p>Citations</th>
      <td><ul class="simple" id="bibtex-bibliography-plugins/available/fragment-insertion/sepp-0">
<li><p>Stefan Janssen, Daniel McDonald, Antonio Gonzalez, Jose A. Navas-Molina, Lingjing Jiang, Zhenjiang Zech Xu, Kevin Winker, Deborah M. Kado, Eric Orwoll, Mark Manary, Siavash Mirarab, and Rob Knight. Phylogenetic placement of exact amplicon sequences improves associations with clinical information. <em>mSystems</em>, 2018. <a class="reference external" href="https://doi.org/10.1128/mSystems.00021-18">doi:10.1128/mSystems.00021-18</a>.</p></li>
</ul>
      </td>
    </tr>
  </tbody>
</table><div class="tabbed">
  <ul class="nav nav-tabs">
    <li class="active"><a data-toggle="tab" href="#cli">Command line interface</a></li>
    <li><a data-toggle="tab" href="#api">Artifact API</a></li>
  </ul>
  <div class="tab-content">
    <div id="cli" class="tab-pane fade in active">
      <h4>Docstring:</h4>
      <pre>Usage: qiime fragment-insertion sepp [OPTIONS]

  Perform fragment insertion of 16S sequences using the SEPP algorithm
  against the Greengenes 13_8 99% tree.

Inputs:
  --i-representative-sequences ARTIFACT FeatureData[Sequence]
                       The sequences to insert                      [required]
  --i-reference-alignment ARTIFACT FeatureData[AlignedSequence]
                       The reference multiple nucleotide alignment used to
                       construct the reference phylogeny.           [optional]
  --i-reference-phylogeny ARTIFACT
    Phylogeny[Rooted]  The rooted reference phylogeny. Must be in sync with
                       reference-alignment, i.e. each tip name must have
                       exactly one corresponding record.            [optional]
Parameters:
  --p-threads INTEGER  The number of threads to use               [default: 1]
  --p-alignment-subset-size INTEGER
                       Each placement subset is further broken into subsets
                       of at most these many sequences and a separate HMM is
                       trained on each subset. The default alignment subset
                       size is set to balance the exhaustiveness of the
                       alignment step with the running time.   [default: 1000]
  --p-placement-subset-size INTEGER
                       The tree is divided into subsets such that each subset
                       includes at most these many subsets. The placement step
                       places the fragment on only one subset, determined
                       based on alignment scores. The default placement subset
                       is set to make sure the memory requirement of the
                       pplacer step does not become prohibitively
                       large.
Further reading:
                       https://github.com/smirarab/sepp/blob/master/tutorial/s
                       epp-tutorial.md#sample-datasets-default-parameters
                                                               [default: 5000]
  --p-debug / --p-no-debug
                       Print additional run information to STDOUT for
                       debugging. Run together with --verbose to actually see
                       the information on STDOUT. Temporary directories will
                       not be removed if run fails.           [default: False]
Outputs:
  --o-tree ARTIFACT    The tree with inserted feature data
    Phylogeny[Rooted]                                               [required]
  --o-placements ARTIFACT
    Placements                                                      [required]
Miscellaneous:
  --output-dir PATH    Output unspecified results to a directory
  --verbose / --quiet  Display verbose output to stdout and/or stderr during
                       execution of this action. Or silence output if
                       execution is successful (silence is golden).
  --citations          Show citations and exit.
  --help               Show this message and exit.</pre>
    </div>
    <div id="api" class="tab-pane fade">
    <h4>Import:</h4><div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>from qiime2.plugins.fragment_insertion.methods import sepp
</pre></div>
</div>
    <h4>Docstring:</h4>
      <pre>Insert fragment sequences using SEPP into reference phylogenies like
Greengenes 13_8

Perform fragment insertion of 16S sequences using the SEPP algorithm
against the Greengenes 13_8 99% tree.

Parameters
----------
representative_sequences : FeatureData[Sequence]
    The sequences to insert
threads : Int, optional
    The number of threads to use
alignment_subset_size : Int, optional
    Each placement subset is further broken into subsets of at most these
    many sequences and a separate HMM is trained on each subset. The
    default alignment subset size is set to balance the exhaustiveness of
    the alignment step with the running time.
placement_subset_size : Int, optional
    The tree is divided into subsets such that each subset includes at most
    these many subsets. The placement step places the fragment on only one
    subset, determined based on alignment scores. The default placement
    subset is set to make sure the memory requirement of the pplacer step
    does not become prohibitively large. Further reading:
    https://github.com/smirarab/sepp/blob/master/tutorial/sepp-
    tutorial.md#sample-datasets-default-parameters
reference_alignment : FeatureData[AlignedSequence], optional
    The reference multiple nucleotide alignment used to construct the
    reference phylogeny.
reference_phylogeny : Phylogeny[Rooted], optional
    The rooted reference phylogeny. Must be in sync with reference-
    alignment, i.e. each tip name must have exactly one corresponding
    record.
debug : Bool, optional
    Print additional run information to STDOUT for debugging. Run together
    with --verbose to actually see the information on STDOUT. Temporary
    directories will not be removed if run fails.

Returns
-------
tree : Phylogeny[Rooted]
    The tree with inserted feature data
placements : Placements</pre>
    </div>
  </div>
</div></div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../../../">Table of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../tutorials/">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../interfaces/">Interfaces</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../../">Plugins</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../">Available plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../future/">Future plugins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../../developing/">Developing a QIIME 2 plugin</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../glossary/">Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../citation/">Citing QIIME 2</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../../../search/" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2019, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.0.1.
    </div>
  </div>
  <script src="../../../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>