<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>“Moving Pictures” tutorial &#8212; QIIME 2 2017.12.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2017.12.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2017.12.0 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Fecal microbiota transplant (FMT) study: an exercise" href="../fmt/" />
    <link rel="prev" title="Tutorials" href="../" />
  <script src="../../_static/bootstrap.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="moving-pictures-tutorial">
<h1>&#8220;Moving Pictures&#8221; tutorial<a class="headerlink" href="#moving-pictures-tutorial" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This guide assumes you have installed QIIME 2 using one of the procedures in the <a class="reference internal" href="../../install/"><span class="doc">install documents</span></a>.</p>
</div>
<p>In this tutorial you&#8217;ll use QIIME 2 to perform an analysis of human microbiome samples from two individuals at four body sites at five timepoints, the first of which immediately followed antibiotic usage. A study based on these samples was originally published in <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/21624126">Caporaso et al. (2011)</a>. The data used in this tutorial were sequenced on an Illumina HiSeq using the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> hypervariable region 4 (V4) 16S rRNA sequencing protocol.</p>
<div class="qiime1 admonition" id="qiime1users-0">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">These are the same data that are used in the QIIME 1 <a class="reference external" href="http://nbviewer.jupyter.org/github/biocore/qiime/blob/1.9.1/examples/ipynb/illumina_overview_tutorial.ipynb">Illumina Overview Tutorial</a>.</p>
</div>
<p>Before beginning this tutorial, create a new directory and change to that directory.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir qiime2-moving-pictures-tutorial
<span class="nb">cd</span> qiime2-moving-pictures-tutorial
</pre></div>
</div>
<div class="section" id="sample-metadata">
<h2>Sample metadata<a class="headerlink" href="#sample-metadata" title="Permalink to this headline">¶</a></h2>
<p>Before starting the analysis, explore the sample metadata to familiarize yourself with the samples used in this study. The <a class="reference external" href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata">sample metadata</a> is available as a Google Sheet. You can download this file as tab-separated text by selecting <code class="docutils literal"><span class="pre">File</span></code> &gt; <code class="docutils literal"><span class="pre">Download</span> <span class="pre">as</span></code> &gt; <code class="docutils literal"><span class="pre">Tab-separated</span> <span class="pre">values</span></code>. Alternatively, the following command will download the sample metadata as tab-separated text and save it in the file <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code>. This <code class="docutils literal"><span class="pre">sample-metadata.tsv</span></code> file is used throughout the rest of the tutorial.</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata.tsv">
            https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata.tsv
          </a>
        </p>
        <p>
          <strong>Save as:</strong> sample-metadata.tsv
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "sample-metadata.tsv" "https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata.tsv"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata.tsv" > "sample-metadata.tsv"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last"><a class="reference external" href="http://keemei.qiime.org">Keemei</a> is a Google Sheets add-on for validating sample metadata. Validation of sample metadata is important before beginning any analysis. Try installing Keemei following the instructions on its website, and then validate the sample metadata spreadsheet linked above. The spreadsheet also includes a sheet with some invalid data to try out with Keemei.</p>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">To learn more about metadata, check out <a class="reference internal" href="../metadata/"><span class="doc">the metadata tutorial</span></a>.</p>
</div>
</div>
<div class="section" id="obtaining-and-importing-data">
<h2>Obtaining and importing data<a class="headerlink" href="#obtaining-and-importing-data" title="Permalink to this headline">¶</a></h2>
<p>Download the sequence reads that we&#8217;ll use in this analysis. In this tutorial we&#8217;ll work with a small subset of the complete sequence data so that the commands will run quickly.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir emp-single-end-sequences
</pre></div>
</div>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/barcodes.fastq.gz">
            https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/barcodes.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-single-end-sequences/barcodes.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-single-end-sequences/barcodes.fastq.gz" "https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/barcodes.fastq.gz" > "emp-single-end-sequences/barcodes.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-2" aria-controls="browser-2" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-2" aria-controls="wget-2" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-2" aria-controls="messages-2" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-2">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/sequences.fastq.gz">
            https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/sequences.fastq.gz
          </a>
        </p>
        <p>
          <strong>Save as:</strong> emp-single-end-sequences/sequences.fastq.gz
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-2">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "emp-single-end-sequences/sequences.fastq.gz" "https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/sequences.fastq.gz"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-2">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.12/tutorials/moving-pictures/emp-single-end-sequences/sequences.fastq.gz" > "emp-single-end-sequences/sequences.fastq.gz"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><p>All data that is used as input to QIIME 2 is in form of QIIME 2 artifacts, which contain information about the type of data and the source of the data. So, the first thing we need to do is import these sequence data files into a QIIME 2 artifact.</p>
<p>The semantic type of this QIIME 2 artifact is <code class="docutils literal"><span class="pre">EMPSingleEndSequences</span></code>. <code class="docutils literal"><span class="pre">EMPSingleEndSequences</span></code> QIIME 2 artifacts contain sequences that are multiplexed, meaning that the sequences have not yet been assigned to samples (hence the inclusion of both <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code> and <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> files, where the <code class="docutils literal"><span class="pre">barcodes.fastq.gz</span></code> contains the barcode read associated with each sequence in <code class="docutils literal"><span class="pre">sequences.fastq.gz</span></code>.) To learn about how to import sequence data in other formats, see the <a class="reference internal" href="../importing/"><span class="doc">importing data tutorial</span></a>.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --type EMPSingleEndSequences <span class="se">\</span>
  --input-path emp-single-end-sequences <span class="se">\</span>
  --output-path emp-single-end-sequences.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">emp-single-end-sequences.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Femp-single-end-sequences.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/emp-single-end-sequences.qza">download</a></li>
</ul>
</div>
<div class="admonition tip">
<p class="first admonition-title">Tip</p>
<p class="last">Links are included to view and download precomputed QIIME 2 artifacts and visualizations created by commands in the documentation. For example, the command above created a single <code class="docutils literal"><span class="pre">emp-single-end-sequences.qza</span></code> file, and a corresponding precomputed file is linked above. You can view precomputed QIIME 2 artifacts and visualizations without needing to install additional software (e.g. QIIME 2).</p>
</div>
<div class="qiime1 admonition" id="qiime1users-1">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">In QIIME 1, we generally suggested performing demultiplexing through QIIME (e.g., with <code class="docutils literal"><span class="pre">split_libraries.py</span></code> or <code class="docutils literal"><span class="pre">split_libraries_fastq.py</span></code>) as this step also performed quality control of sequences. We now separate the demultiplexing and quality control steps, so you can begin QIIME 2 with either multiplexed sequences (as we&#8217;re doing here) or demultiplexed sequences.</p>
</div>
</div>
<div class="section" id="demultiplexing-sequences">
<h2>Demultiplexing sequences<a class="headerlink" href="#demultiplexing-sequences" title="Permalink to this headline">¶</a></h2>
<p>To demultiplex sequences we need to know which barcode sequence is associated with each sample. This information is contained in the <a class="reference external" href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata">sample metadata</a> file. You can run the following commands to demultiplex the sequences (the <code class="docutils literal"><span class="pre">demux</span> <span class="pre">emp-single</span></code> command refers to the fact that these sequences are barcoded according to the <a class="reference external" href="http://earthmicrobiome.org">Earth Microbiome Project</a> protocol, and are single-end reads). The <code class="docutils literal"><span class="pre">demux.qza</span></code> QIIME 2 artifact will contain the demultiplexed sequences.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime demux emp-single <span class="se">\</span>
  --i-seqs emp-single-end-sequences.qza <span class="se">\</span>
  --m-barcodes-file sample-metadata.tsv <span class="se">\</span>
  --m-barcodes-category BarcodeSequence <span class="se">\</span>
  --o-per-sample-sequences demux.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/demux.qza">download</a></li>
</ul>
</div>
<p>After demultiplexing, it&#8217;s useful to generate a summary of the demultiplexing results. This allows you to determine how many sequences were obtained per sample, and also to get a summary of the distribution of sequence qualities at each position in your sequence data.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime demux summarize <span class="se">\</span>
  --i-data demux.qza <span class="se">\</span>
  --o-visualization demux.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/demux.qzv">download</a></li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>All QIIME 2 visualizers (i.e., commands that take a <code class="docutils literal"><span class="pre">--o-visualization</span></code> parameter) will generate a <code class="docutils literal"><span class="pre">.qzv</span></code> file. You can view these files with <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code>. We provide the command to view this first visualization, but for the remainder of this tutorial we&#8217;ll tell you to <em>view the resulting visualization</em> after running a visualizer, which means that you should run <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">view</span></code> on the .qzv file that was generated.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools view demux.qzv
</pre></div>
</div>
<p class="last">Alternatively, you can view QIIME 2 artifacts and visualizations at <a class="reference external" href="https://view.qiime2.org">view.qiime2.org</a> by uploading files or providing URLs. There are also precomputed results that can be viewed or downloaded after each step in the tutorial. These can be used if you&#8217;re reading the tutorial, but not running the commands yourself.</p>
</div>
</div>
<div class="section" id="sequence-quality-control-and-feature-table-construction">
<h2>Sequence quality control and feature table construction<a class="headerlink" href="#sequence-quality-control-and-feature-table-construction" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2 plugins are available for several quality control methods, including <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a>, <a class="reference external" href="http://msystems.asm.org/content/2/2/e00191-16">Deblur</a>, and <a class="reference external" href="http://www.nature.com/nmeth/journal/v10/n1/abs/nmeth.2276.html">basic quality-score-based filtering</a>. In this tutorial we present this step using <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a> and <a class="reference external" href="http://msystems.asm.org/content/2/2/e00191-16">Deblur</a>. These steps are interchangeable, so you can use whichever of these you prefer. The result of both of these methods will be a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> QIIME 2 artifact, which contains counts (frequencies) of each unique sequence in each sample in the dataset, and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> QIIME 2 artifact, which maps feature identifiers in the <code class="docutils literal"><span class="pre">FeatureTable</span></code> to the sequences they represent.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">As you work through one or both of the options in this section, you&#8217;ll create artifacts with filenames that are specific to the method that you&#8217;re running (e.g., the feature table that you generate with <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise-single</span></code> will be called <code class="docutils literal"><span class="pre">table-dada2.qza</span></code>). After creating these artifacts you&#8217;ll rename the artifacts from one of the two options to more generic filenames (e.g., <code class="docutils literal"><span class="pre">table.qza</span></code>). This process of creating a specific name for an artifact and then renaming it is only done to allow you to choose which of the two options you&#8217;d like to use for this step, and then complete the tutorial without paying attention to that choice again. It&#8217;s important to note that in this step, or any step in QIIME 2, the filenames that you&#8217;re giving to artifacts or visualizations are not important.</p>
</div>
<div class="qiime1 admonition" id="qiime1users-2">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">The <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> QIIME 2 artifact is the equivalent of the QIIME 1 OTU or BIOM table, and the <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> QIIME 2 artifact is the equivalent of the QIIME 1 <em>representative sequences</em> file. Because the &#8220;OTUs&#8221; resulting from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a> and <a class="reference external" href="http://msystems.asm.org/content/2/2/e00191-16">Deblur</a> are created by grouping unique sequences, these are the equivalent of 100% OTUs from QIIME 1, and are generally referred to as <em>sequence variants</em>. In QIIME 2, these OTUs are higher resolution than the QIIME 1 default of 97% OTUs, and they&#8217;re higher quality since these quality control steps are better than those implemented in QIIME 1. This should therefore result in more accurate estimates of diversity and taxonomic composition of samples than was achieved with QIIME 1.</p>
</div>
<div class="section" id="option-1-dada2">
<h3>Option 1: DADA2<a class="headerlink" href="#option-1-dada2" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27214047">DADA2</a> is a pipeline for detecting and correcting (where possible) Illumina amplicon sequence data. As implemented in the <code class="docutils literal"><span class="pre">q2-dada2</span></code> plugin, this quality control process will additionally filter any phiX reads (commonly present in marker gene Illumina sequence data) that are identified in the sequencing data, and will filter chimeric sequences.</p>
<p>The <code class="docutils literal"><span class="pre">dada2</span> <span class="pre">denoise-single</span></code> method requires two parameters that are used in quality filtering: <code class="docutils literal"><span class="pre">--p-trim-left</span> <span class="pre">m</span></code>, which trims off the first <code class="docutils literal"><span class="pre">m</span></code> bases of each sequence, and <code class="docutils literal"><span class="pre">--p-trunc-len</span> <span class="pre">n</span></code> which truncates each sequence at position <code class="docutils literal"><span class="pre">n</span></code>. This allows the user to remove low quality regions of the sequences. To determine what values to pass for these two parameters, you should review the <em>Interactive Quality Plot</em> tab in the <code class="docutils literal"><span class="pre">demux.qzv</span></code> file that was generated by <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">demux</span> <span class="pre">summarize</span></code> above.</p>
<div class="question admonition" id="question-0">
<p class="first admonition-title">Question</p>
<p class="last">Based on the plots you see in <code class="docutils literal"><span class="pre">demux.qzv</span></code>, what values would you choose for <code class="docutils literal"><span class="pre">--p-trunc-len</span></code> and <code class="docutils literal"><span class="pre">--p-trim-left</span></code> in this case?</p>
</div>
<p>In the <code class="docutils literal"><span class="pre">demux.qzv</span></code> quality plots, we see that the quality of the initial bases seems to be high, so we won&#8217;t trim any bases from the beginning of the sequences. The quality seems to drop off around position 120, so we&#8217;ll truncate our sequences at 120 bases. This next command may take up to 10 minutes to run, and is the slowest step in this tutorial.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime dada2 denoise-single <span class="se">\</span>
  --i-demultiplexed-seqs demux.qza <span class="se">\</span>
  --p-trim-left <span class="m">0</span> <span class="se">\</span>
  --p-trunc-len <span class="m">120</span> <span class="se">\</span>
  --o-representative-sequences rep-seqs-dada2.qza <span class="se">\</span>
  --o-table table-dada2.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs-dada2.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs-dada2.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/rep-seqs-dada2.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table-dada2.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable-dada2.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/table-dada2.qza">download</a></li>
</ul>
</div>
<p>If you&#8217;d like to continue the tutorial using this FeatureTable (opposed to the Deblur feature table generated in <em>Option 2</em>), run the following commands.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mv rep-seqs-dada2.qza rep-seqs.qza
mv table-dada2.qza table.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/rep-seqs.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/table.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="option-2-deblur">
<h3>Option 2: Deblur<a class="headerlink" href="#option-2-deblur" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://msystems.asm.org/content/2/2/e00191-16">Deblur</a> uses sequence error profiles to associate erroneous sequence reads with the true biological sequence from which they are derived, resulting in high quality sequence variant data. This is applied in two steps. First, an initial quality filtering process based on quality scores is applied. This method is an implementation of the quality filtering approach described by <a class="reference external" href="http://www.nature.com/nmeth/journal/v10/n1/abs/nmeth.2276.html">Bokulich et al. (2013)</a>.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime quality-filter q-score <span class="se">\</span>
 --i-demux demux.qza <span class="se">\</span>
 --o-filtered-sequences demux-filtered.qza <span class="se">\</span>
 --o-filter-stats demux-filter-stats.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">demux-filter-stats.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux-filter-stats.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/demux-filter-stats.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">demux-filtered.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux-filtered.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/demux-filtered.qza">download</a></li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In the <a class="reference external" href="http://msystems.asm.org/content/2/2/e00191-16">Deblur</a> paper, the authors used different quality-filtering parameters than what <a class="reference external" href="https://qiita.ucsd.edu/static/doc/html/deblur_quality.html">they currently recommend after additional analysis</a>. The parameters used here are based on those more recent recommendations.</p>
</div>
<p>Next, the Deblur workflow is applied using the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">deblur</span> <span class="pre">denoise-16S</span></code> method. This method requires one parameter that is used in quality filtering, <code class="docutils literal"><span class="pre">--p-trim-length</span> <span class="pre">n</span></code> which truncates the sequences at position <code class="docutils literal"><span class="pre">n</span></code>. In general, the Deblur developers recommend setting this value to a length where the median quality score begins to drop too low. On these data, the quality plots (prior to quality filtering) suggest a reasonable choice is in the 115 to 130 sequence position range. This is a subjective assessment. One situation where you might deviate from that recommendation is when performing a meta-analysis across multiple sequencing runs. In this type of meta-analysis, it is critical that the read lengths be the same for all of the sequencing runs being compared to avoid introducing a study-specific bias. Since we already using a trim length of 120 for <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">dada2</span> <span class="pre">denoise-single</span></code>, and since 120 is reasonable given the quality plots, we&#8217;ll pass <code class="docutils literal"><span class="pre">--p-trim-length</span> <span class="pre">120</span></code>. This next command may take up to 10 minutes to run.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime deblur denoise-16S <span class="se">\</span>
  --i-demultiplexed-seqs demux-filtered.qza <span class="se">\</span>
  --p-trim-length <span class="m">120</span> <span class="se">\</span>
  --o-representative-sequences rep-seqs-deblur.qza <span class="se">\</span>
  --o-table table-deblur.qza <span class="se">\</span>
  --p-sample-stats <span class="se">\</span>
  --o-stats deblur-stats.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">deblur-stats.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdeblur-stats.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/deblur-stats.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs-deblur.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs-deblur.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/rep-seqs-deblur.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table-deblur.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable-deblur.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/table-deblur.qza">download</a></li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The two commands used in this section generate QIIME 2 artifacts containing summary statistics. To view those summary statistics, you can visualize them using <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">metadata</span> <span class="pre">tabulate</span></code> and <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">deblur</span> <span class="pre">visualize-stats</span></code>, respectively:</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime metadata tabulate <span class="se">\</span>
  --m-input-file demux-filter-stats.qza <span class="se">\</span>
  --o-visualization demux-filter-stats.qzv
qiime deblur visualize-stats <span class="se">\</span>
  --i-deblur-stats deblur-stats.qza <span class="se">\</span>
  --o-visualization deblur-stats.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">deblur-stats.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdeblur-stats.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/deblur-stats.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">demux-filter-stats.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fdemux-filter-stats.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/demux-filter-stats.qzv">download</a></li>
</ul>
</div>
<p>If you&#8217;d like to continue the tutorial using this FeatureTable (opposed to the DADA2 feature table generated in <em>Option 1</em>), run the following commands.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mv rep-seqs-deblur.qza rep-seqs.qza
mv table-deblur.qza table.qza
</pre></div>
</div>
</div>
</div>
<div class="section" id="featuretable-and-featuredata-summaries">
<h2>FeatureTable and FeatureData summaries<a class="headerlink" href="#featuretable-and-featuredata-summaries" title="Permalink to this headline">¶</a></h2>
<p>After the quality filtering step completes, you&#8217;ll want to explore the resulting data. You can do this using the following two commands, which will create visual summaries of the data. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">summarize</span></code> command will give you information on how many sequences are associated with each sample and with each feature, histograms of those distributions, and some related summary statistics. The <code class="docutils literal"><span class="pre">feature-table</span> <span class="pre">tabulate-seqs</span></code> command will provide a mapping of feature IDs to sequences, and provide links to easily BLAST each sequence against the NCBI nt database. The latter visualization will be very useful later in the tutorial, when you want to learn more about specific features that are important in the data set.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-table summarize <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --o-visualization table.qzv <span class="se">\</span>
  --m-sample-metadata-file sample-metadata.tsv
qiime feature-table tabulate-seqs <span class="se">\</span>
  --i-data rep-seqs.qza <span class="se">\</span>
  --o-visualization rep-seqs.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Frep-seqs.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/rep-seqs.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftable.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/table.qzv">download</a></li>
</ul>
</div>
</div>
<div class="section" id="generate-a-tree-for-phylogenetic-diversity-analyses">
<h2>Generate a tree for phylogenetic diversity analyses<a class="headerlink" href="#generate-a-tree-for-phylogenetic-diversity-analyses" title="Permalink to this headline">¶</a></h2>
<p>QIIME supports several phylogenetic diversity metrics, including Faith&#8217;s Phylogenetic Diversity and weighted and unweighted UniFrac. In addition to counts of features per sample (i.e., the data in the <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> QIIME 2 artifact), these metrics require a rooted phylogenetic tree relating the features to one another. This information will be stored in a <code class="docutils literal"><span class="pre">Phylogeny[Rooted]</span></code> QIIME 2 artifact. The following steps will generate this QIIME 2 artifact.</p>
<p>First, we perform a multiple sequence alignment of the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> to create a <code class="docutils literal"><span class="pre">FeatureData[AlignedSequence]</span></code> QIIME 2 artifact. Here we do this with the <code class="docutils literal"><span class="pre">mafft</span></code> program.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mafft <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --o-alignment aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Faligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/aligned-rep-seqs.qza">download</a></li>
</ul>
</div>
<p>Next, we mask (or filter) the alignment to remove positions that are highly variable. These positions are generally considered to add noise to a resulting phylogenetic tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime alignment mask <span class="se">\</span>
  --i-alignment aligned-rep-seqs.qza <span class="se">\</span>
  --o-masked-alignment masked-aligned-rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">masked-aligned-rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fmasked-aligned-rep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/masked-aligned-rep-seqs.qza">download</a></li>
</ul>
</div>
<p>Next, we&#8217;ll apply FastTree to generate a phylogenetic tree from the masked alignment.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny fasttree <span class="se">\</span>
  --i-alignment masked-aligned-rep-seqs.qza <span class="se">\</span>
  --o-tree unrooted-tree.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">unrooted-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Funrooted-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/unrooted-tree.qza">download</a></li>
</ul>
</div>
<p>The FastTree program creates an unrooted tree, so in the final step in this section we apply midpoint rooting to place the root of the tree at the midpoint of the longest tip-to-tip distance in the unrooted tree.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime phylogeny midpoint-root <span class="se">\</span>
  --i-tree unrooted-tree.qza <span class="se">\</span>
  --o-rooted-tree rooted-tree.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rooted-tree.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Frooted-tree.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/rooted-tree.qza">download</a></li>
</ul>
</div>
</div>
<div class="section" id="alpha-and-beta-diversity-analysis">
<h2>Alpha and beta diversity analysis<a class="headerlink" href="#alpha-and-beta-diversity-analysis" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2&#8217;s diversity analyses are available through the <code class="docutils literal"><span class="pre">q2-diversity</span></code> plugin, which supports computing alpha and beta diversity metrics, applying related statistical tests, and generating interactive visualizations. We&#8217;ll first apply the <code class="docutils literal"><span class="pre">core-metrics-phylogenetic</span></code> method, which rarefies a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> to a user-specified depth, computes several alpha and beta diversity metrics, and generates principle coordinates analysis (PCoA) plots using Emperor for each of the beta diversity metrics. The metrics computed by default are:</p>
<ul class="simple">
<li>Alpha diversity<ul>
<li>Shannon&#8217;s diversity index (a quantitative measure of community richness)</li>
<li>Observed OTUs (a qualitative measure of community richness)</li>
<li>Faith&#8217;s Phylogenetic Diversity (a qualitiative measure of community richness that incorporates phylogenetic relationships between the features)</li>
<li>Evenness (or Pielou&#8217;s Evenness; a measure of community evenness)</li>
</ul>
</li>
<li>Beta diversity<ul>
<li>Jaccard distance (a qualitative measure of community dissimilarity)</li>
<li>Bray-Curtis distance (a quantitative measure of community dissimilarity)</li>
<li>unweighted UniFrac distance (a qualitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
<li>weighted UniFrac distance (a quantitative measure of community dissimilarity that incorporates phylogenetic relationships between the features)</li>
</ul>
</li>
</ul>
<p>An important parameter that needs to be provided to this script is <code class="docutils literal"><span class="pre">--p-sampling-depth</span></code>, which is the even sampling (i.e. rarefaction) depth. Because most diversity metrics are sensitive to different sampling depths across different samples, this script will randomly subsample the counts from each sample to the value provided for this parameter. For example, if you provide <code class="docutils literal"><span class="pre">--p-sampling-depth</span> <span class="pre">500</span></code>, this step will subsample the counts in each sample without replacement so that each sample in the resulting table has a total count of 500. If the total count for any sample(s) are smaller than this value, those samples will be dropped from the diversity analysis. Choosing this value is tricky. We recommend making your choice by reviewing the information presented in the <code class="docutils literal"><span class="pre">table.qzv</span></code> file that was created above and choosing a value that is as high as possible (so you retain more sequences per sample) while excluding as few samples as possible.</p>
<div class="question admonition" id="question-1">
<p class="first admonition-title">Question</p>
<p class="last">View the <code class="docutils literal"><span class="pre">table.qzv</span></code> QIIME 2 artifact, and in particular the <em>Interactive Sample Detail</em> tab in that visualization. What value would you choose to pass for <code class="docutils literal"><span class="pre">--p-sampling-depth</span></code>? How many samples will be excluded from your analysis based on this choice? How many total sequences will you be analyzing in the <code class="docutils literal"><span class="pre">core-metrics-phylogenetic</span></code> command?</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity core-metrics-phylogenetic <span class="se">\</span>
  --i-phylogeny rooted-tree.qza <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --p-sampling-depth <span class="m">1109</span> <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --output-dir core-metrics-results
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">core-metrics-results/bray_curtis_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fbray_curtis_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/bray_curtis_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/bray_curtis_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fbray_curtis_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/bray_curtis_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/evenness_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fevenness_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/evenness_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/faith_pd_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Ffaith_pd_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/faith_pd_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/jaccard_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fjaccard_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/jaccard_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/jaccard_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fjaccard_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/jaccard_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/observed_otus_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fobserved_otus_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/observed_otus_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/rarefied_table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Frarefied_table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/rarefied_table.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/shannon_vector.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fshannon_vector.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/shannon_vector.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted_unifrac_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted_unifrac_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted_unifrac_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted_unifrac_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted_unifrac_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted_unifrac_pcoa_results.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/weighted_unifrac_distance_matrix.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fweighted_unifrac_distance_matrix.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/weighted_unifrac_distance_matrix.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/weighted_unifrac_pcoa_results.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fweighted_unifrac_pcoa_results.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/weighted_unifrac_pcoa_results.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">core-metrics-results/bray_curtis_emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fbray_curtis_emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/bray_curtis_emperor.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/jaccard_emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fjaccard_emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/jaccard_emperor.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted_unifrac_emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted_unifrac_emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted_unifrac_emperor.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/weighted_unifrac_emperor.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fweighted_unifrac_emperor.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/weighted_unifrac_emperor.qzv">download</a></li>
</ul>
</div>
<p>Here we set the <code class="docutils literal"><span class="pre">--p-sampling-depth</span></code> parameter to 1109. This value was chosen based on the number of sequences in the <code class="docutils literal"><span class="pre">L3S341</span></code> sample because it&#8217;s close to the number of sequences in the next few samples that have higher sequence counts, and because it is considerably higher (relatively) than the number of sequences in the one sample that has fewer sequences. This will allow us to retain most of our samples. The one sample that has fewer sequences will be dropped from the <code class="docutils literal"><span class="pre">core-metrics-phylogenetic</span></code> analyses and anything that uses these results.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The sampling depth of 1109 was chosen based on the DADA2 feature table summary. If you are using a Deblur feature table rather than a DADA2 feature table, you might want to choose a different even sampling depth. Apply the logic from the previous paragraph to help you choose an even sampling depth.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In many Illumina runs you&#8217;ll observe a few samples that have very low sequence counts. You will typically want to exclude those from the analysis by choosing a larger value for the sampling depth at this stage.</p>
</div>
<p>After computing diversity metrics, we can begin to explore the microbial composition of the samples in the context of the sample metadata. This information is present in the <a class="reference external" href="https://data.qiime2.org/2017.12/tutorials/moving-pictures/sample_metadata">sample metadata</a> file that was downloaded earlier.</p>
<p>We&#8217;ll first test for associations between discrete metadata categories and alpha diversity data. We&#8217;ll do that here for the Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-group-significance <span class="se">\</span>
  --i-alpha-diversity core-metrics-results/faith_pd_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization core-metrics-results/faith-pd-group-significance.qzv

qiime diversity alpha-group-significance <span class="se">\</span>
  --i-alpha-diversity core-metrics-results/evenness_vector.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization core-metrics-results/evenness-group-significance.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">core-metrics-results/evenness-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fevenness-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/evenness-group-significance.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/faith-pd-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Ffaith-pd-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/faith-pd-group-significance.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-2">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>richness</strong>? Are these differences statistically significant?</p>
</div>
<div class="question admonition" id="question-3">
<p class="first admonition-title">Question</p>
<p class="last">What discrete sample metadata categories are most strongly associated with the differences in microbial community <strong>evenness</strong>? Are these differences statistically significant?</p>
</div>
<p>In this data set, no continuous sample metadata categories (e.g., <code class="docutils literal"><span class="pre">DaysSinceExperimentStart</span></code>) are correlated with alpha diversity, so we won&#8217;t test for those associations here. If you&#8217;re interested in performing those tests (for this data set, or for others), you can use the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">alpha-correlation</span></code> command.</p>
<p>Next we&#8217;ll analyze sample composition in the context of discrete metadata using PERMANOVA (first described in <a class="reference external" href="http://onlinelibrary.wiley.com/doi/10.1111/j.1442-9993.2001.01070.pp.x/full">Anderson (2001)</a>) using the <code class="docutils literal"><span class="pre">beta-group-significance</span></code> command. The following commands will test whether distances between samples within a group, such as samples from the same body site (e.g., gut), are more similar to each other then they are to samples from the other groups (e.g., tongue, left palm, and right palm). If you call this command with the <code class="docutils literal"><span class="pre">--p-pairwise</span></code> parameter, as we&#8217;ll do here, it will also perform pairwise tests that will allow you to determine which specific pairs of groups (e.g., tongue and gut) differ from one another, if any. This command can be slow to run, especially when passing <code class="docutils literal"><span class="pre">--p-pairwise</span></code>, since it is based on permutation tests. So, unlike the previous commands, we&#8217;ll run this on specific categories of metadata that we&#8217;re interested in exploring, rather than all metadata categories that it&#8217;s applicable to. Here we&#8217;ll apply this to our unweighted UniFrac distances, using two sample metadata categories, as follows.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity beta-group-significance <span class="se">\</span>
  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category BodySite <span class="se">\</span>
  --o-visualization core-metrics-results/unweighted-unifrac-body-site-significance.qzv <span class="se">\</span>
  --p-pairwise

qiime diversity beta-group-significance <span class="se">\</span>
  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category Subject <span class="se">\</span>
  --o-visualization core-metrics-results/unweighted-unifrac-subject-group-significance.qzv <span class="se">\</span>
  --p-pairwise
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted-unifrac-body-site-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted-unifrac-body-site-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted-unifrac-body-site-significance.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted-unifrac-subject-group-significance.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted-unifrac-subject-group-significance.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted-unifrac-subject-group-significance.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-4">
<p class="first admonition-title">Question</p>
<p class="last">Are the associations between subjects and differences in microbial composition statistically significant? How about body sites? What specific pairs of body sites are significantly different from each other?</p>
</div>
<p>Again, none of the continuous sample metadata that we have for this data set are correlated with sample composition, so we won&#8217;t test for those associations here. If you&#8217;re interested in performing those tests, you can use the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">metadata</span> <span class="pre">distance-matrix</span></code> in combination with <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">mantel</span></code> and <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">bioenv</span></code> commands.</p>
<p>Finally, ordination is a popular approach for exploring microbial community composition in the context of sample metadata. We can use the <a class="reference external" href="http://emperor.microbio.me">Emperor</a> tool to explore principal coordinates (PCoA) plots in the context of sample metadata. While our <code class="docutils literal"><span class="pre">core-metrics-phylogenetic</span></code> command did already generate some Emperor plots, we want to pass an optional parameter, <code class="docutils literal"><span class="pre">--p-custom-axis</span></code>, which is very useful for exploring time series data. The PCoA results that were used in <code class="docutils literal"><span class="pre">core-metrics-phylogeny</span></code> are also available, making it easy to generate new visualizations with Emperor. We will generate Emperor plots for unweighted UniFrac and Bray-Curtis so that the resulting plot will contain axes for principal coordinate 1, principal coordinate 2, and days since the experiment start. We will use that last axis to explore how these samples changed over time.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime emperor plot <span class="se">\</span>
  --i-pcoa core-metrics-results/unweighted_unifrac_pcoa_results.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --p-custom-axis DaysSinceExperimentStart <span class="se">\</span>
  --o-visualization core-metrics-results/unweighted-unifrac-emperor-DaysSinceExperimentStart.qzv

qiime emperor plot <span class="se">\</span>
  --i-pcoa core-metrics-results/bray_curtis_pcoa_results.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --p-custom-axis DaysSinceExperimentStart <span class="se">\</span>
  --o-visualization core-metrics-results/bray-curtis-emperor-DaysSinceExperimentStart.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">core-metrics-results/bray-curtis-emperor-DaysSinceExperimentStart.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Fbray-curtis-emperor-DaysSinceExperimentStart.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/bray-curtis-emperor-DaysSinceExperimentStart.qzv">download</a></li>
<li><code class="file docutils literal"><span class="pre">core-metrics-results/unweighted-unifrac-emperor-DaysSinceExperimentStart.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted-unifrac-emperor-DaysSinceExperimentStart.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/core-metrics-results/unweighted-unifrac-emperor-DaysSinceExperimentStart.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-5">
<p class="first admonition-title">Question</p>
<p class="last">Do the Emperor plots support the other beta diversity analyses we&#8217;ve performed here? (Hint: Experiment with coloring points by different metadata.)</p>
</div>
<div class="question admonition" id="question-6">
<p class="first admonition-title">Question</p>
<p class="last">What differences do you observe between the unweighted UniFrac and Bray-Curtis PCoA plots?</p>
</div>
</div>
<div class="section" id="alpha-rarefaction-plotting">
<h2>Alpha rarefaction plotting<a class="headerlink" href="#alpha-rarefaction-plotting" title="Permalink to this headline">¶</a></h2>
<p>In this section we&#8217;ll explore alpha diversity as a function of sampling depth using the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">diversity</span> <span class="pre">alpha-rarefaction</span></code> visualizer. This visualizer computes one or more alpha diversity metrics at multiple sampling depths, in steps between 1 (optionally controlled with <code class="docutils literal"><span class="pre">--p-min-depth</span></code>) and the value provided as <code class="docutils literal"><span class="pre">--p-max-depth</span></code>. At each sampling depth step, 10 rarefied tables will be generated, and the diversity metrics will be computed for all samples in the tables. The number of iterations (rarefied tables computed at each sampling depth) can be controlled with <code class="docutils literal"><span class="pre">--p-iterations</span></code>. Average diversity values will be plotted for each sample at each even sampling depth, and samples can be grouped based on metadata in the resulting visualization if sample metadata is provided with the <code class="docutils literal"><span class="pre">--m-metadata-file</span></code> parameter.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime diversity alpha-rarefaction <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-phylogeny rooted-tree.qza <span class="se">\</span>
  --p-max-depth <span class="m">4000</span> <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization alpha-rarefaction.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">alpha-rarefaction.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Falpha-rarefaction.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/alpha-rarefaction.qzv">download</a></li>
</ul>
</div>
<p>The visualization will have two plots. The top plot is an alpha rarefaction plot, and is primarily used to determine if the richness of the samples has been fully observed or sequenced. If the lines in the plot appear to &#8220;level out&#8221; (i.e., approach a slope of zero) at some sampling depth along the x-axis, that suggests that collecting additional sequences beyond that sampling depth would not be likely to result in the observation of additional features. If the lines in a plot don&#8217;t level out, this may be because the richness of the samples hasn&#8217;t been fully observed yet (because too few sequences were collected), or it could be an indicator that a lot of sequencing error remains in the data (which is being mistaken for novel diversity).</p>
<p>The bottom plot in this visualization is important when grouping samples by metadata. It illustrates the number of samples that remain in each group when the feature table is rarefied to each sampling depth. If a given sampling depth <code class="docutils literal"><span class="pre">d</span></code> is larger than the total frequency of a sample <code class="docutils literal"><span class="pre">s</span></code> (i.e., the number of sequences that were obtained for sample <code class="docutils literal"><span class="pre">s</span></code>), it is not possible to compute the diversity metric for sample <code class="docutils literal"><span class="pre">s</span></code> at sampling depth <code class="docutils literal"><span class="pre">d</span></code>. If many of the samples in a group have lower total frequencies than <code class="docutils literal"><span class="pre">d</span></code>, the average diversity presented for that group at <code class="docutils literal"><span class="pre">d</span></code> in the top plot will be unreliable because it will have been computed on relatively few samples. When grouping samples by metadata, it is therefore essential to look at the bottom plot to ensure that the data presented in the top plot is reliable.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The value that you provide for <code class="docutils literal"><span class="pre">--p-max-depth</span></code> should be determined by reviewing the &#8220;Frequency per sample&#8221; information presented in the <code class="docutils literal"><span class="pre">table.qzv</span></code> file that was created above. In general, choosing a value that is somewhere around the median frequency seems to work well, but you may want to increase that value if the lines in the resulting rarefaction plot don&#8217;t appear to be leveling out, or decrease that value if you seem to be losing many of your samples due to low total frequencies closer to the minimum sampling depth than the maximum sampling depth.</p>
</div>
<div class="question admonition" id="question-7">
<p class="first admonition-title">Question</p>
<p class="last">When grouping samples by &#8220;BodySite&#8221; and viewing the alpha rarefaction plot for the &#8220;observed_otus&#8221; metric, which body sites (if any) appear to exhibit sufficient diversity coverage (i.e., their rarefaction curves level off)? How many sequence variants appear to be present in those body sites?</p>
</div>
<div class="question admonition" id="question-8">
<p class="first admonition-title">Question</p>
<p class="last">When grouping samples by &#8220;BodySite&#8221; and viewing the alpha rarefaction plot for the &#8220;observed_otus&#8221; metric, the line for the &#8220;right palm&#8221; samples appears to level out at about 40, but then jumps to about 140. What do you think is happening here? (Hint: be sure to look at both the top and bottom plots.)</p>
</div>
</div>
<div class="section" id="taxonomic-analysis">
<h2>Taxonomic analysis<a class="headerlink" href="#taxonomic-analysis" title="Permalink to this headline">¶</a></h2>
<p>In the next sections we&#8217;ll begin to explore the taxonomic composition of the samples, and again relate that to sample metadata. The first step in this process is to assign taxonomy to the sequences in our <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> QIIME 2 artifact. We&#8217;ll do that using a pre-trained Naive Bayes classifier and the <code class="docutils literal"><span class="pre">q2-feature-classifier</span></code> plugin. This classifier was trained on the Greengenes 13_8 99% OTUs, where the sequences have been trimmed to only include 250 bases from the region of the 16S that was sequenced in this analysis (the V4 region, bound by the 515F/806R primer pair). We&#8217;ll apply this classifier to our sequences, and we can generate a visualization of the resulting mapping from sequence to taxonomy.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Taxonomic classifiers perform best when they are trained based on your specific sample preparation and sequencing parameters, including the primers that were used for amplification and the length of your sequence reads. Therefore in general you should follow the instructions in <a class="reference internal" href="../feature-classifier/"><span class="doc">Training feature classifiers with q2-feature-classifier</span></a> to train your own taxonomic classifiers. We provide some common classifiers on our <a class="reference internal" href="../../data-resources/"><span class="doc">data resources page</span></a>, including Silva-based 16S classifiers, though in the future we may stop providing these in favor of having users train their own classifiers which will be most relevant to their sequence data.</p>
</div>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-3" aria-controls="browser-3" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-3" aria-controls="wget-3" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-3" aria-controls="messages-3" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-3">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2017.12/common/gg-13-8-99-515-806-nb-classifier.qza">
            https://data.qiime2.org/2017.12/common/gg-13-8-99-515-806-nb-classifier.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> gg-13-8-99-515-806-nb-classifier.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-3">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget -O "gg-13-8-99-515-806-nb-classifier.qza" "https://data.qiime2.org/2017.12/common/gg-13-8-99-515-806-nb-classifier.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-3">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL "https://data.qiime2.org/2017.12/common/gg-13-8-99-515-806-nb-classifier.qza" > "gg-13-8-99-515-806-nb-classifier.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-classifier classify-sklearn <span class="se">\</span>
  --i-classifier gg-13-8-99-515-806-nb-classifier.qza <span class="se">\</span>
  --i-reads rep-seqs.qza <span class="se">\</span>
  --o-classification taxonomy.qza

qiime metadata tabulate <span class="se">\</span>
  --m-input-file taxonomy.qza <span class="se">\</span>
  --o-visualization taxonomy.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">gg-13-8-99-515-806-nb-classifier.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fgg-13-8-99-515-806-nb-classifier.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/gg-13-8-99-515-806-nb-classifier.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">taxonomy.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxonomy.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/taxonomy.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">taxonomy.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxonomy.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/taxonomy.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-9">
<p class="first admonition-title">Question</p>
<p class="last">Recall that our <code class="docutils literal"><span class="pre">rep-seqs.qzv</span></code> visualization allows you to easily BLAST the sequence associated with each feature against the NCBI nt database. Using that visualization and the <code class="docutils literal"><span class="pre">taxonomy.qzv</span></code> visualization created here, compare the taxonomic assignments with the taxonomy of the best BLAST hit for a few features. How similar are the assignments? If they&#8217;re dissimilar, at what <em>taxonomic level</em> do they begin to differ (e.g., species, genus, family, ...)?</p>
</div>
<p>Next, we can view the taxonomic composition of our samples with interactive bar plots. Generate those plots with the following command and then open the visualization.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa barplot <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-taxonomy taxonomy.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --o-visualization taxa-bar-plots.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">taxa-bar-plots.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Ftaxa-bar-plots.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/taxa-bar-plots.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-10">
<p class="first admonition-title">Question</p>
<p class="last">Visualize the samples at <em>Level 2</em> (which corresponds to the phylum level in this analysis), and then sort the samples by BodySite, then by Subject, and then by DaysSinceExperimentStart. What are the dominant phyla in each in BodySite? Do you observe any consistent change across the two subjects between DaysSinceExperimentStart <code class="docutils literal"><span class="pre">0</span></code> and the later timepoints?</p>
</div>
</div>
<div class="section" id="differential-abundance-testing-with-ancom">
<h2>Differential abundance testing with ANCOM<a class="headerlink" href="#differential-abundance-testing-with-ancom" title="Permalink to this headline">¶</a></h2>
<p>ANCOM can be applied to identify features that are differentially abundant (i.e. present in different abundances) across sample groups. As with any bioinformatics method, you should be aware of the assumptions and limitations of ANCOM before using it. We recommend reviewing the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/26028277">ANCOM paper</a> before using this method.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Differential abundance testing in microbiome analysis is an active area of research. There are two QIIME 2 plugins that can be used for this: <code class="docutils literal"><span class="pre">q2-gneiss</span></code> and <code class="docutils literal"><span class="pre">q2-composition</span></code>. This section uses <code class="docutils literal"><span class="pre">q2-composition</span></code>, but there is <a class="reference internal" href="../gneiss/"><span class="doc">another tutorial which uses gneiss</span></a> on a different dataset if you are interested in learning more.</p>
</div>
<p>ANCOM is implemented in the <code class="docutils literal"><span class="pre">q2-composition</span></code> plugin. ANCOM assumes that few (less than about 25%) of the features are changing between groups. If you expect that more features are changing between your groups, you should not use ANCOM as it will be more error-prone (an increase in both Type I and Type II errors is possible). Because we expect a lot of features to change in abundance across body sites, in this tutorial we&#8217;ll filter our full feature table to only contain gut samples. We&#8217;ll then apply ANCOM to determine which, if any, sequence variants and genera are differentially abundant across the gut samples of our two subjects.</p>
<p>We&#8217;ll start by creating a feature table that contains only the gut samples. (To learn more about filtering, see the <a class="reference internal" href="../filtering/"><span class="doc">Filtering Data</span></a> tutorial.)</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime feature-table filter-samples <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --p-where <span class="s2">&quot;BodySite=&#39;gut&#39;&quot;</span> <span class="se">\</span>
  --o-filtered-table gut-table.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">gut-table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fgut-table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/gut-table.qza">download</a></li>
</ul>
</div>
<p>ANCOM operates on a <code class="docutils literal"><span class="pre">FeatureTable[Composition]</span></code> QIIME 2 artifact, which is based on frequencies of features on a per-sample basis, but cannot tolerate frequencies of zero. To build the composition artifact, a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code>  artifact must be provided to <code class="docutils literal"><span class="pre">add-pseudocount</span></code> (an imputation method), which will produce the <code class="docutils literal"><span class="pre">FeatureTable[Composition]</span></code> artifact.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime composition add-pseudocount <span class="se">\</span>
  --i-table gut-table.qza <span class="se">\</span>
  --o-composition-table comp-gut-table.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">comp-gut-table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcomp-gut-table.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/comp-gut-table.qza">download</a></li>
</ul>
</div>
<p>We can then run ANCOM on the <code class="docutils literal"><span class="pre">Subject</span></code> category to determine what features differ in abundance across the gut samples of the two subjects.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime composition ancom <span class="se">\</span>
  --i-table comp-gut-table.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category Subject <span class="se">\</span>
  --o-visualization ancom-Subject.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">ancom-Subject.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fancom-Subject.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/ancom-Subject.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-11">
<p class="first admonition-title">Question</p>
<p class="last">Which sequence variants differ in abundance across Subject? In which subject is each sequence variant more abundant? What are the taxonomies of some of these sequence variants? (To answer the last question you&#8217;ll need to refer to another visualization that was generated in this tutorial.)</p>
</div>
<p>We&#8217;re also often interested in performing a differential abundance test at a specific taxonomic level. To do this, we can collapse the features in our <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> at the taxonomic level of interest, and then re-run the above steps. In this tutorial, we collapse our feature table at the genus level (i.e. level 6 of the Greengenes taxonomy).</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime taxa collapse <span class="se">\</span>
  --i-table gut-table.qza <span class="se">\</span>
  --i-taxonomy taxonomy.qza <span class="se">\</span>
  --p-level <span class="m">6</span> <span class="se">\</span>
  --o-collapsed-table gut-table-l6.qza

qiime composition add-pseudocount <span class="se">\</span>
  --i-table gut-table-l6.qza <span class="se">\</span>
  --o-composition-table comp-gut-table-l6.qza

qiime composition ancom <span class="se">\</span>
  --i-table comp-gut-table-l6.qza <span class="se">\</span>
  --m-metadata-file sample-metadata.tsv <span class="se">\</span>
  --m-metadata-category Subject <span class="se">\</span>
  --o-visualization l6-ancom-Subject.qzv
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="simple">
<li><code class="file docutils literal"><span class="pre">comp-gut-table-l6.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcomp-gut-table-l6.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/comp-gut-table-l6.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">gut-table-l6.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fgut-table-l6.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/gut-table-l6.qza">download</a></li>
</ul>
<p><strong>Output visualizations:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">l6-ancom-Subject.qzv</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2017.12%2Fdata%2Ftutorials%2Fmoving-pictures%2Fl6-ancom-Subject.qzv">view</a> | <a class="reference external" href="https://docs.qiime2.org/2017.12/data/tutorials/moving-pictures/l6-ancom-Subject.qzv">download</a></li>
</ul>
</div>
<div class="question admonition" id="question-12">
<p class="first admonition-title">Question</p>
<p class="last">Which genera differ in abundance across Subject? In which subject is each genus more abundant?</p>
</div>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">&#8220;Atacama soil microbiome&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2017, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>