<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Clustering sequences into OTUs using q2-vsearch &#8212; QIIME 2 2018.8.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '2018.8.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/external-links.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="author" title="About these documents" href="../../about/" />
    <link rel="top" title="QIIME 2 2018.8.0 documentation" href="../../" />
    <link rel="up" title="Tutorials" href="../" />
    <link rel="next" title="Interfaces" href="../../interfaces/" />
    <link rel="prev" title="Alternative methods of read-joining in QIIME 2" href="../read-joining/" />
  <script src="../../_static/bootstrap.min.js"></script>
  <script src="../../_static/clipboard.min.js"></script>
  <link rel="icon" href="../../_static/favicon.ico" type="image/png" />
  <link rel="stylesheet" href="../../_static/css/bootstrap.min.css" media="screen" />
  <link rel="stylesheet" href="../../_static/css/normalize.css" media="screen">
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-86671044-2', 'auto');
    ga('send', 'pageview');
  </script>
  <meta name="viewport" content="width=device-width, initial-scale=1">

  </head>
  <body role="document">
<a href="https://github.com/qiime2/qiime2" id="github-banner">
  <img
    src="../../_static/github_banner.png"
    alt="Fork me on GitHub"
  />
</a>

  <div id="wrapper" class="container clearfix">
    <div id="header">
      <a href="/">
        <img
          class="img-responsive"
          src="../../_static/q2docs.png" alt="QIIME 2 Docs"
        />
      </a>
    </div>
    <div id="content">
      
      <div id="body" class="col-lg-10 col-md-10 col-sm-12 col-xs-12 main pull-right">
        
  <div class="section" id="clustering-sequences-into-otus-using-q2-vsearch">
<h1>Clustering sequences into OTUs using q2-vsearch<a class="headerlink" href="#clustering-sequences-into-otus-using-q2-vsearch" title="Permalink to this headline">¶</a></h1>
<p><em>De novo</em>, closed-reference, and open-reference clustering are currently
supported in QIIME 2.</p>
<p>Clustering of sequences or features into OTUs using <a class="reference external" href="https://peerj.com/articles/2584/">vsearch</a> is currently
possible from demultiplexed, quality-controlled sequence data (i.e., a
<code class="docutils literal"><span class="pre">SampleData[Sequences]</span></code> artifact), or from dereplicated, quality-controlled
data in feature table and feature representative sequences (i.e., the
<code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> and <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifacts, which
could be generated using the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">dada2</span> <span class="pre">denoise-*</span></code> or <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">deblur</span>
<span class="pre">denoise-*</span></code> commands). The first option is currently performed in two steps
(but will likely be accessible through a single command in the future for
convenience). The second option is performed in one step.</p>
<div class="qiime1 admonition" id="qiime1users-0">
<p class="first admonition-title">QIIME 1 Users</p>
<p class="last">Demultiplexed, quality-filtered sequence data is synonymous with the
<code class="docutils literal"><span class="pre">seqs.fna</span></code> file, generated by the QIIME 1 <code class="docutils literal"><span class="pre">split_libraries*.py</span></code>
commands.</p>
</div>
<p>After working through this tutorial, you will know how to run <em>de novo</em>,
closed-reference, and open-reference clustering. This will be illustrated
beginning with a QIIME 1 <code class="docutils literal"><span class="pre">seqs.fna</span></code> file that will be read into an
<code class="docutils literal"><span class="pre">SampleData[Sequences]</span></code> artifact.  If you already have
<code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> and <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifacts that you&#8217;d
like to cluster, you can skip ahead to the <a class="reference internal" href="#cluster-table-and-seqs"><span class="std std-ref">Clustering of  FeatureTable[Frequency] and FeatureData[Sequence]</span></a>
section of this tutorial.</p>
<div class="section" id="obtain-the-data">
<h2>Obtain the data<a class="headerlink" href="#obtain-the-data" title="Permalink to this headline">¶</a></h2>
<p>Start by creating a directory to work in.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>mkdir qiime2-otu-clustering-tutorial
<span class="nb">cd</span> qiime2-otu-clustering-tutorial
</pre></div>
</div>
<p>Next, download the necessary files:</p>
<div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-0" aria-controls="browser-0" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-0" aria-controls="wget-0" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-0" aria-controls="messages-0" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-0">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2018.8/tutorials/otu-clustering/seqs.fna">
            https://data.qiime2.org/2018.8/tutorials/otu-clustering/seqs.fna
          </a>
        </p>
        <p>
          <strong>Save as:</strong> seqs.fna
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-0">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "seqs.fna" \
  "https://data.qiime2.org/2018.8/tutorials/otu-clustering/seqs.fna"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-0">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2018.8/tutorials/otu-clustering/seqs.fna" > \
  "seqs.fna"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div><div class="tabbed panel panel-default">
  <div class="panel-heading white-bg">
    Please select a download option that is most appropriate for your environment:
  </div>

  <div class="panel-body">
    <ul class="nav nav-tabs gray-bg" role="tablist">
      <li role="presentation" class="active">
        <a href="#browser-1" aria-controls="browser-1" role="tab" data-toggle="tab">
          Browser
        </a>
      </li>
      <li role="presentation">
        <a href="#wget-1" aria-controls="wget-1" role="tab" data-toggle="tab">
          wget
        </a>
      </li>
      <li role="presentation">
        <a href="#curl-1" aria-controls="messages-1" role="tab" data-toggle="tab">
          curl
        </a>
      </li>
    </ul>

    <div class="tab-content">
      <div role="tabpanel" class="tab-pane fade in active" id="browser-1">
        <p>
          <strong>Download URL:</strong> <a href="https://data.qiime2.org/2018.8/tutorials/otu-clustering/85_otus.qza">
            https://data.qiime2.org/2018.8/tutorials/otu-clustering/85_otus.qza
          </a>
        </p>
        <p>
          <strong>Save as:</strong> 85_otus.qza
        </p>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="wget-1">
        <div class="clearfix"></div>
        <div class="highlight-shell">
          <div class="highlight">
            <pre>wget \
  -O "85_otus.qza" \
  "https://data.qiime2.org/2018.8/tutorials/otu-clustering/85_otus.qza"</pre>
          </div>
        </div>
      </div>
      <div role="tabpanel" class="tab-pane fade" id="curl-1">
        <div class="highlight-shell">
          <div class="highlight">
            <pre>curl -sL \
  "https://data.qiime2.org/2018.8/tutorials/otu-clustering/85_otus.qza" > \
  "85_otus.qza"</pre>
          </div>
        </div>
      </div>
    </div>
  </div>
</div></div>
<div class="section" id="dereplicating-a-sampledata-sequences-artifact">
<h2>Dereplicating a <code class="docutils literal"><span class="pre">SampleData[Sequences]</span></code> artifact<a class="headerlink" href="#dereplicating-a-sampledata-sequences-artifact" title="Permalink to this headline">¶</a></h2>
<p>If you are beginning your analysis with demultiplexed, quality controlled
sequences, such as those in a <a class="reference external" href="http://qiime.org/documentation/file_formats.html#post-split-libraries-fasta-file-overview">QIIME 1 <code class="docutils literal"><span class="pre">seqs.fna</span></code> file</a> your first step
is to import that data into a QIIME 1 artifact. The semantic type used here is
<code class="docutils literal"><span class="pre">SampleData[Sequences]</span></code>, indicating that the data represents collections of
sequences associated with one or more samples.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime tools import <span class="se">\</span>
  --input-path seqs.fna <span class="se">\</span>
  --output-path seqs.qza <span class="se">\</span>
  --type <span class="s1">&#39;SampleData[Sequences]&#39;</span>
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">85_otus.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2F85_otus.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/85_otus.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Fseqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/seqs.qza">download</a></li>
</ul>
</div>
<p>After importing data, you can dereplicate it with the <code class="docutils literal"><span class="pre">dereplicate-sequences</span></code>
command.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime vsearch dereplicate-sequences <span class="se">\</span>
  --i-sequences seqs.qza <span class="se">\</span>
  --o-dereplicated-table table.qza <span class="se">\</span>
  --o-dereplicated-sequences rep-seqs.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">rep-seqs.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Frep-seqs.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/rep-seqs.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Ftable.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/table.qza">download</a></li>
</ul>
</div>
<p>The outputs from <code class="docutils literal"><span class="pre">dereplicate-sequences</span></code> are a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code>
artifact and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact. The
<code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact is the feature table indicating the number
of times each amplicon sequence variant (ASV) is observed in each of your
samples. The <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> contains the mapping of each feature
identifier to the sequence variant that defines that feature. These files are
analogous to those generated by <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">dada2</span> <span class="pre">denoise-*</span></code> and <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">deblur</span>
<span class="pre">denoise-*</span></code>, except that no denoising, chimera removal, or other quality
control has been applied in the dereplication process. (In this example, the
only quality control of these data is what was applied outside of QIIME 2,
before the <code class="docutils literal"><span class="pre">import</span></code> step.)</p>
</div>
<div class="section" id="clustering-of-featuretable-frequency-and-featuredata-sequence">
<span id="cluster-table-and-seqs"></span><h2>Clustering of  <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> and <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code><a class="headerlink" href="#clustering-of-featuretable-frequency-and-featuredata-sequence" title="Permalink to this headline">¶</a></h2>
<p>OTU clustering in QIIME 2 is currently applied to a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code>
artifact and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact. These artifacts can come
from a variety of analysis pipelines, including <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">vsearch</span>
<span class="pre">dereplicate-sequences</span></code> (illustrated above), <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">dada2</span> <span class="pre">denoise-*</span></code>, <code class="docutils literal"><span class="pre">qiime</span>
<span class="pre">deblur</span> <span class="pre">denoise-*</span></code>, or one of the clustering processes illustrated below (for
example, to recluster data at a lower percent identity).</p>
<p>The sequences in the <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact are clustered against
one another (in <em>de novo</em> clustering) or a reference database (in
closed-reference clustering), and then features in the <code class="docutils literal"><span class="pre">FeatureTable</span></code> are
collapsed, resulting in new features that are clusters of the input features.</p>
<div class="section" id="de-novo-clustering">
<h3><em>De novo</em> clustering<a class="headerlink" href="#de-novo-clustering" title="Permalink to this headline">¶</a></h3>
<p><em>De novo</em> clustering of a feature table can be performed as follows. In this
example, clustering is performed at 99% identity to create 99% OTUs.</p>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime vsearch cluster-features-de-novo <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --p-perc-identity <span class="m">0</span>.99 <span class="se">\</span>
  --o-clustered-table table-dn-99.qza <span class="se">\</span>
  --o-clustered-sequences rep-seqs-dn-99.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">table-dn-99.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Ftable-dn-99.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/table-dn-99.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs-dn-99.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Frep-seqs-dn-99.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/rep-seqs-dn-99.qza">download</a></li>
</ul>
</div>
<p>The outputs from this process are a <code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact and a
<code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact. The <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact will
contain the centroid sequence defining each OTU cluster.</p>
</div>
<div class="section" id="closed-reference-clustering">
<h3>Closed-reference clustering<a class="headerlink" href="#closed-reference-clustering" title="Permalink to this headline">¶</a></h3>
<p>Closed-reference clustering of a feature table can be performed as follows. In
this example, clustering is performed at 85% identity against the Greengenes
13_8 85% OTUs reference database. The reference database is provided as a
<code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Closed-reference OTU clustering is generally performed at a higher percent
identity, but 85% is used here so users of this tutorial don&#8217;t have to
download a larger reference database. Typically clustering at some percent
identity is performed against a reference database clustered at the same
percent identity, but this has not been properly benchmarked to determine if
it is the optimal way to perform closed-reference clustering.</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime vsearch cluster-features-closed-reference <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --i-reference-sequences 85_otus.qza <span class="se">\</span>
  --p-perc-identity <span class="m">0</span>.85 <span class="se">\</span>
  --o-clustered-table table-cr-85.qza <span class="se">\</span>
  --o-clustered-sequences rep-seqs-cr-85.qza <span class="se">\</span>
  --o-unmatched-sequences unmatched-cr-85.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">table-cr-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Ftable-cr-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/table-cr-85.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">unmatched-cr-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Funmatched-cr-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/unmatched-cr-85.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs-cr-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Frep-seqs-cr-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/rep-seqs-cr-85.qza">download</a></li>
</ul>
</div>
<p>The outputs from <code class="docutils literal"><span class="pre">cluster-features-closed-reference</span></code> are a
<code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact and a <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact.
The <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifact in this case is <strong>not</strong> the sequences
defining the features in the <code class="docutils literal"><span class="pre">FeatureTable</span></code>, but rather the collection of
feature ids and their sequences that didn&#8217;t match the reference database at 85%
identity. The reference sequences provided as input should be used as sequences
defining the features in the <code class="docutils literal"><span class="pre">FeatureTable</span></code> in closed-reference OTU picking.</p>
</div>
<div class="section" id="open-reference-clustering">
<h3>Open-reference clustering<a class="headerlink" href="#open-reference-clustering" title="Permalink to this headline">¶</a></h3>
<p>Like the closed-reference clustering example above, open-reference clustering
can be performed using the <code class="docutils literal"><span class="pre">qiime</span> <span class="pre">vsearch</span> <span class="pre">cluster-features-open-reference</span></code>
command.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Open-reference OTU clustering is generally performed at a higher percent
identity, but 85% is used here so users of this tutorial don&#8217;t have to
download a larger reference database. Typically clustering at some percent
identity is performed against a reference database clustered at the same
percent identity, but this has not been properly benchmarked to determine if
it is the optimal way to perform open-reference clustering.</p>
</div>
<div class="highlight-shell"><div class="highlight"><pre><span></span>qiime vsearch cluster-features-open-reference <span class="se">\</span>
  --i-table table.qza <span class="se">\</span>
  --i-sequences rep-seqs.qza <span class="se">\</span>
  --i-reference-sequences 85_otus.qza <span class="se">\</span>
  --p-perc-identity <span class="m">0</span>.85 <span class="se">\</span>
  --o-clustered-table table-or-85.qza <span class="se">\</span>
  --o-clustered-sequences rep-seqs-or-85.qza <span class="se">\</span>
  --o-new-reference-sequences new-ref-seqs-or-85.qza
</pre></div>
</div>
<div class="admonition">
<p class="first"><strong>Output artifacts:</strong></p>
<ul class="last simple">
<li><code class="file docutils literal"><span class="pre">new-ref-seqs-or-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Fnew-ref-seqs-or-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/new-ref-seqs-or-85.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">rep-seqs-or-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Frep-seqs-or-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/rep-seqs-or-85.qza">download</a></li>
<li><code class="file docutils literal"><span class="pre">table-or-85.qza</span></code>: <a class="reference external" href="https://view.qiime2.org?src=https%3A%2F%2Fdocs.qiime2.org%2F2018.8%2Fdata%2Ftutorials%2Fotu-clustering%2Ftable-or-85.qza">view</a> | <a class="reference external" href="https://docs.qiime2.org/2018.8/data/tutorials/otu-clustering/table-or-85.qza">download</a></li>
</ul>
</div>
<p>The outputs from <code class="docutils literal"><span class="pre">cluster-features-open-reference</span></code> are a
<code class="docutils literal"><span class="pre">FeatureTable[Frequency]</span></code> artifact and two <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code>
artifacts. One of the <code class="docutils literal"><span class="pre">FeatureData[Sequence]</span></code> artifacts represents the
clustered sequences, while the other artifact represents the new reference
sequences, composed of the reference sequences used for input, as well as the
sequences clustered as part of the internal <em>de novo</em> clustering step.</p>
</div>
</div>
</div>


      </div>
      

      
      <div id="sidebar" class="col-lg-2 col-md-2 col-sm-12 col-xs-12 side pull-left">
        <div id="version-div" class="btn-group">
          <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"></button>
          <ul id="version-list" class="dropdown-menu"></ul>
        </div>
        
<h3><a href="../../">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../getting-started/">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../about/">What is QIIME 2?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../concepts/">Core concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../install/">Installing QIIME 2</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../">Tutorials</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../overview/">Overview of QIIME 2 Plugin Workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="../moving-pictures/">&#8220;Moving Pictures&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../fmt/">Fecal microbiota transplant (FMT) study: an exercise</a></li>
<li class="toctree-l2"><a class="reference internal" href="../atacama-soils/">&#8220;Atacama soil microbiome&#8221; tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../gneiss/">Differential abundance analysis with gneiss</a></li>
<li class="toctree-l2"><a class="reference internal" href="../importing/">Importing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../exporting/">Exporting data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../metadata/">Metadata in QIIME 2</a></li>
<li class="toctree-l2"><a class="reference internal" href="../filtering/">Filtering data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../feature-classifier/">Training feature classifiers with q2-feature-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../quality-control/">Evaluating and controlling data quality with q2-quality-control</a></li>
<li class="toctree-l2"><a class="reference internal" href="../sample-classifier/">Predicting sample metadata values with q2-sample-classifier</a></li>
<li class="toctree-l2"><a class="reference internal" href="../longitudinal/">Performing longitudinal and paired sample comparisons with q2-longitudinal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../chimera/">Identifying and filtering chimeric feature sequences with q2-vsearch</a></li>
<li class="toctree-l2"><a class="reference internal" href="../read-joining/">Alternative methods of read-joining in QIIME 2</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Clustering sequences into OTUs using q2-vsearch</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../interfaces/">Interfaces</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../plugins/">Plugins</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../semantic-types/">Semantic types</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../community/">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data-resources/">Data resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../supplementary-resources/">Supplementary resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary/">Glossary</a></li>
</ul>

<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="../../search/" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
      </div>
      
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2016-2018, QIIME 2 development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.6.
    </div>
  </div>
  <script src="../../_static/bootstrap-dropdown.js" charset="utf-8"></script>
  <script src="../../_static/clipboard-driver.js"></script>
  <script src="/versions.js" charset="utf-8"></script>
  </body>
</html>